{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Set up libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "\n",
    "# Bring in Pandas for Dataframe functionality\n",
    "import pandas as pd\n",
    "\n",
    "# numpy for basics\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Import seaborn for plots\n",
    "import seaborn as sns\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.core.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#!pip install rpy2\n",
    "#Install rpy2 if not already installed - uncomment if running for the first time"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "import rpy2.rinterface"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Set up billing project and data path variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "manuela-norway\n",
      "gs://amp-pd-data/releases/2021_v2-5release_0510/clinical\n",
      "gs://amp-pd-genomics/releases/2021_v2-5release_0510/wgs/plink\n",
      "gs://amp-pd-genomics/releases/2021_v2-5release_0510/wgs/gatk\n"
     ]
    }
   ],
   "source": [
    "BILLING_PROJECT_ID = os.environ['GOOGLE_PROJECT']\n",
    "WORKSPACE_NAMESPACE = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE_NAME = os.environ['WORKSPACE_NAME']\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "\n",
    "WORKSPACE_ATTRIBUTES = fapi.get_workspace(WORKSPACE_NAMESPACE, WORKSPACE_NAME).json().get('workspace',{}).get('attributes',{})\n",
    "\n",
    "GS_RELEASE_PATH = 'gs://amp-pd-data/releases/2021_v2-5release_0510'\n",
    "GS_CLINICAL_RELEASE_PATH = f'{GS_RELEASE_PATH}/clinical'\n",
    "\n",
    "GS_WGS_RELEASE_PATH = 'gs://amp-pd-genomics/releases/2021_v2-5release_0510/wgs'\n",
    "GS_WGS_RELEASE_PLINK_PATH = os.path.join(GS_WGS_RELEASE_PATH, 'plink')\n",
    "GS_WGS_RELEASE_GATK_PATH = os.path.join(GS_WGS_RELEASE_PATH, 'gatk')\n",
    "\n",
    "BQ_RELEASE_DATASET = 'amp-pd-research.2021_v2_5release_0510'\n",
    "\n",
    "print(BILLING_PROJECT_ID)\n",
    "print(GS_CLINICAL_RELEASE_PATH)\n",
    "print(GS_WGS_RELEASE_PLINK_PATH)\n",
    "print(GS_WGS_RELEASE_GATK_PATH)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Set up functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}')\n",
    "    !$command\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "    \n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DataFrame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for display a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "    </p>\n",
    "    '''\n",
    "\n",
    "    display(HTML(html))\n",
    "    \n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"gs://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "\n",
    "    display_html_link(description, link_text, url)\n",
    "    \n",
    "# Get the data from a query\n",
    "def bq_query(query):\n",
    "    \"\"\"Return the contents of a query against BigQuery\"\"\"\n",
    "    return pd.read_gbq(\n",
    "        query,\n",
    "        project_id=BILLING_PROJECT_ID,\n",
    "        dialect='standard')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Copy pathway lists to workspace"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "allPD_HapMap_geneticPCs.tab\n",
      "AMP_forEuroPCA.bed\n",
      "AMP_forEuroPCA.bim\n",
      "AMP_forEuroPCA.fam\n",
      "AMP_forEuroPCA.log\n",
      "AMP_forEuroPCA.pruned.bed\n",
      "AMP_forEuroPCA.pruned.bim\n",
      "AMP_forEuroPCA.pruned.fam\n",
      "AMP_forEuroPCA.pruned.log\n",
      "AMP_forEuroPCA.pruning.log\n",
      "AMP_forEuroPCA.pruning.prune.in\n",
      "AMP_forEuroPCA.pruning.prune.out\n",
      "AMP_HapMap_merged.filtered.bed\n",
      "AMP_HapMap_merged.filtered.bim\n",
      "AMP_HapMap_merged.filtered.fam\n",
      "AMP_HapMap_merged.filtered.log\n",
      "AMP_HapMap_merged.filtered.pruned.bed\n",
      "AMP_HapMap_merged.filtered.pruned.bim\n",
      "AMP_HapMap_merged.filtered.pruned.fam\n",
      "AMP_HapMap_merged.filtered.pruned.log\n",
      "AMP.QCed_chrbp.txt\n",
      "AMP.QCed.updated_names.bed\n",
      "AMP.QCed.updated_names.bim\n",
      "AMP.QCed.updated_names.fam\n",
      "AMP.QCed.updated_names.log\n",
      "AMP.QCed.updated_names.no_dupli.bed\n",
      "AMP.QCed.updated_names.no_dupli.bim\n",
      "AMP.QCed.updated_names.no_dupli.fam\n",
      "AMP.QCed.updated_names.no_dupli.log\n",
      "AMP_toMergeHapMap.bed\n",
      "AMP_toMergeHapMap.bim\n",
      "AMP_toMergeHapMap.fam\n",
      "AMP_toMergeHapMap.log\n",
      "APOE.log\n",
      "APOE.raw\n",
      "BioFIND.pca.eigenval\n",
      "BioFIND.pca.eigenvec\n",
      "BioFIND.pca.log\n",
      "BioFIND.pruned.bed\n",
      "BioFIND.pruned.bim\n",
      "BioFIND.pruned.fam\n",
      "BioFIND.pruned.log\n",
      "BioFIND.pruning.log\n",
      "BioFIND.pruning.prune.in\n",
      "BioFIND.pruning.prune.out\n",
      "euro_pca.eigenval\n",
      "euro_pca.eigenvec\n",
      "euro_pca.log\n",
      "Europeans_PC1vsPC2.png\n",
      "Europeans_PC1vsPC3.png\n",
      "exclusion_regions_hg38.txt\n",
      "FINAL_pca.eigenval\n",
      "FINAL_pca.eigenvec\n",
      "FINAL_pca.log\n",
      "GBA.bed\n",
      "GBA.bim\n",
      "GBA.fam\n",
      "GBA.freq.frq\n",
      "GBA.freq.log\n",
      "GBA.log\n",
      "GBA.recodeA.log\n",
      "GBA.recodeA.raw\n",
      "GBA_vars_annotated.xlsx\n",
      "GENDER_FAILURES.txt\n",
      "genetic_ancestry_ADMIX.txt\n",
      "genetic_ancestry_AFRICA.txt\n",
      "genetic_ancestry_all_pca.csv\n",
      "genetic_ancestry_ASIA.txt\n",
      "genetic_ancestry_EUROPE.txt\n",
      "GRM_matrix_0125.grm.bin\n",
      "GRM_matrix_0125.grm.id\n",
      "GRM_matrix_0125.grm.N.bin\n",
      "GRM_matrix_0125.log\n",
      "GRM_matrix.grm.bin\n",
      "GRM_matrix.grm.id\n",
      "GRM_matrix.grm.N.bin\n",
      "GRM_matrix.log\n",
      "HAPLOTYPE_TEST_MISSING_SNPS.txt\n",
      "HAPMAP_GRCh38.bed\n",
      "HAPMAP_GRCh38.bim\n",
      "HAPMAP_GRCh38_extract.txt\n",
      "HAPMAP_GRCh38.fam\n",
      "HAPMAP_GRCh38.updated_names.bed\n",
      "HAPMAP_GRCh38.updated_names.bim\n",
      "HAPMAP_GRCh38.updated_names.fam\n",
      "HAPMAP_GRCh38.updated_names.log\n",
      "HAPMAP_GRCh38.updated_names.palindromes_excluded.bed\n",
      "HAPMAP_GRCh38.updated_names.palindromes_excluded.bim\n",
      "HAPMAP_GRCh38.updated_names.palindromes_excluded.excluded_snps.bed\n",
      "HAPMAP_GRCh38.updated_names.palindromes_excluded.excluded_snps.bim\n",
      "HAPMAP_GRCh38.updated_names.palindromes_excluded.excluded_snps.fam\n",
      "HAPMAP_GRCh38.updated_names.palindromes_excluded.excluded_snps.log\n",
      "HAPMAP_GRCh38.updated_names.palindromes_excluded.fam\n",
      "HAPMAP_GRCh38.updated_names.palindromes_excluded.log\n",
      "HAPMAP_GRCh38.updated_names.palindrome.txt\n",
      "HBS.pca.eigenval\n",
      "HBS.pca.eigenvec\n",
      "HBS.pca.log\n",
      "HBS.pruned.bed\n",
      "HBS.pruned.bim\n",
      "HBS.pruned.fam\n",
      "HBS.pruned.log\n",
      "HBS.pruning.log\n",
      "HBS.pruning.prune.in\n",
      "HBS.pruning.prune.out\n",
      "HETEROZYGOSITY_DATA.txt\n",
      "HETEROZYGOSITY_OUTLIERS.txt\n",
      "king_all_chr.kin0\n",
      "king_all_chr.log\n",
      "LBD.pruning.log\n",
      "LCC.pca.eigenval\n",
      "LCC.pca.eigenvec\n",
      "LCC.pca.log\n",
      "LCC.pruned.bed\n",
      "LCC.pruned.bim\n",
      "LCC.pruned.fam\n",
      "LCC.pruned.log\n",
      "LCC.pruning.log\n",
      "LCC.pruning.prune.in\n",
      "LCC.pruning.prune.out\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.bed\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.bim\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.fam\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.log\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.bed\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.bim\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.fam\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.hh\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.log\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nomutations.no_dupli.bed\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nomutations.no_dupli.bim\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nomutations.no_dupli.fam\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nomutations.no_dupli.log\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.bed\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.bim\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.bed\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.bim\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.fam\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.geno01_hwe0001.bed\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.geno01_hwe0001.bim\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.geno01_hwe0001.fam\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.geno01_hwe0001.log\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.log\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.bed\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.bim\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.fam\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.bed\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.bim\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.fam\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.log\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.bed\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.bim\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.fam\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.bed\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.bim\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.fam\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.log\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bed\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.log\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.log\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.log\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.fam\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.log\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.bed\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.bim\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.fam\n",
      "merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.log\n",
      "merged.PDonly.updated_sex.mind_0.05.het.log\n",
      "merged.PDonly.updated_sex.mind_0.05.het.pgen\n",
      "merged.PDonly.updated_sex.mind_0.05.het.psam\n",
      "merged.PDonly.updated_sex.mind_0.05.het.pvar\n",
      "mild_GBA.recodeA.log\n",
      "mild_GBA.recodeA.raw\n",
      "mild_GBA.tab\n",
      "missing_hap.log\n",
      "missing_hap.missing.hap\n",
      "missing_haps_1E4_final.txt\n",
      "missing_haps_1E4.txt\n",
      "pca.eigenval\n",
      "pca.eigenvec\n",
      "pca.log\n",
      "PDBP.pca.eigenval\n",
      "PDBP.pca.eigenvec\n",
      "PDBP.pca.log\n",
      "PDBP.pruned.bed\n",
      "PDBP.pruned.bim\n",
      "PDBP.pruned.fam\n",
      "PDBP.pruned.log\n",
      "PDBP.pruning.log\n",
      "PDBP.pruning.prune.in\n",
      "PDBP.pruning.prune.out\n",
      "PD_EURO-AMPv2-5-PC1and2.png\n",
      "PD_EURO-AMPv2-5-PC1and3.png\n",
      "PopulationPlotAll.png\n",
      "PPMI.pca.eigenval\n",
      "PPMI.pca.eigenvec\n",
      "PPMI.pca.log\n",
      "PPMI.pruned.bed\n",
      "PPMI.pruned.bim\n",
      "PPMI.pruned.fam\n",
      "PPMI.pruned.log\n",
      "PPMI.pruning.log\n",
      "PPMI.pruning.prune.in\n",
      "PPMI.pruning.prune.out\n",
      "pruned.bed\n",
      "pruned.bim\n",
      "pruned_data.bed\n",
      "pruned_data.bim\n",
      "pruned_data.fam\n",
      "pruned_data.log\n",
      "pruned.fam\n",
      "prunedHet.het\n",
      "prunedHet.log\n",
      "pruned.log\n",
      "pruning.log\n",
      "pruning.prune.in\n",
      "pruning.prune.out\n",
      "samples_to_remove.txt\n",
      "Scree-AMPv2-5-AllPop.png\n",
      "Scree-AMPv2-5-EURO.png\n",
      "Scree-AMPv2-AllPop.png\n",
      "severe_GBA.bed\n",
      "severe_GBA.bim\n",
      "severe_GBA.fam\n",
      "severe_GBA.log\n",
      "severe_GBA.recodeA.log\n",
      "severe_GBA.recodeA.raw\n",
      "severe_GBA.tab\n",
      "Steady.pca.eigenval\n",
      "Steady.pca.eigenvec\n",
      "Steady.pca.log\n",
      "Steady.pruned.bed\n",
      "Steady.pruned.bim\n",
      "Steady.pruned.fam\n",
      "Steady.pruned.log\n",
      "Steady.pruning.log\n",
      "Steady.pruning.prune.in\n",
      "Steady.pruning.prune.out\n",
      "Sure.pca.eigenval\n",
      "Sure.pca.eigenvec\n",
      "Sure.pca.log\n",
      "Sure.pruned.bed\n",
      "Sure.pruned.bim\n",
      "Sure.pruned.fam\n",
      "Sure.pruned.log\n",
      "Sure.pruning.log\n",
      "Sure.pruning.prune.in\n",
      "Sure.pruning.prune.out\n",
      "test_snps.freq.frq\n",
      "test_snps.freq.log\n",
      "test_snps.recodeA.log\n",
      "test_snps.recodeA.raw\n",
      "trymerge.bed\n",
      "trymerge.bim\n",
      "trymerge.fam\n",
      "trymerge.log\n",
      "undetermined_GBA.recodeA.log\n",
      "undetermined_GBA.recodeA.raw\n",
      "undetermined_GBA.tab\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/genetic_data_v2-5"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Make new directory for pathway lists"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "mkdir -p /home/jupyter/pathways"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -mu manuela-norway cp gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181/*_extract_hg38.txt /home/jupyter/pathways/\n",
      "Copying gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181/adaptive_immune_extract_hg38.txt...\n",
      "Copying gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181/alpha_synuclein_extract_hg38.txt...\n",
      "Copying gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181/microglia_extract_hg38.txt...\n",
      "Copying gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181/endocytosis_extract_hg38.txt...\n",
      "Copying gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181/mitochondria_extract_hg38.txt...\n",
      "Copying gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181/monocytes_extract_hg38.txt...\n",
      "Copying gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181/innate_immune_extract_hg38.txt...\n",
      "Copying gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181/lysosome_extract_hg38.txt...\n",
      "/ [8/8 files][  2.1 MiB/  2.1 MiB] 100% Done                                    \n",
      "Operation completed over 8 objects/2.1 MiB.                                      \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp {WORKSPACE_BUCKET}/*_extract_hg38.txt /home/jupyter/pathways/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "adaptive_immune.bed\n",
      "adaptive_immune.bim\n",
      "adaptive_immune_chrbp.txt\n",
      "adaptive_immune_extract_hg38.txt\n",
      "adaptive_immune.fam\n",
      "adaptive_immune.log\n",
      "adaptive_immune_PD_PRS_Chang23andMe.all_score\n",
      "adaptive_immune_PD_PRS_Chang23andMe.log\n",
      "adaptive_immune_PD_PRS_Chang23andMe.mismatch\n",
      "adaptive_immune_PD_PRS_Chang23andMe.prsice\n",
      "adaptive_immune.updated_names.bed\n",
      "adaptive_immune.updated_names.bim\n",
      "adaptive_immune.updated_names.fam\n",
      "adaptive_immune.updated_names.log\n",
      "adaptive_immune.updated_names.no_dupli.bed\n",
      "adaptive_immune.updated_names.no_dupli.bim\n",
      "adaptive_immune.updated_names.no_dupli.fam\n",
      "adaptive_immune.updated_names.no_dupli.log\n",
      "alpha_synuclein.bed\n",
      "alpha_synuclein.bim\n",
      "alpha_synuclein_chrbp.txt\n",
      "alpha_synuclein_extract_hg38.txt\n",
      "alpha_synuclein.fam\n",
      "alpha_synuclein.log\n",
      "alpha_synuclein_PD_PRS_Chang23andMe.all_score\n",
      "alpha_synuclein_PD_PRS_Chang23andMe.log\n",
      "alpha_synuclein_PD_PRS_Chang23andMe.prsice\n",
      "alpha_synuclein.updated_names.bed\n",
      "alpha_synuclein.updated_names.bim\n",
      "alpha_synuclein.updated_names.fam\n",
      "alpha_synuclein.updated_names.log\n",
      "alpha_synuclein.updated_names.no_dupli.bed\n",
      "alpha_synuclein.updated_names.no_dupli.bim\n",
      "alpha_synuclein.updated_names.no_dupli.fam\n",
      "alpha_synuclein.updated_names.no_dupli.log\n",
      "BioFIND.adaptive_immune_PRS_final.txt\n",
      "BioFIND.alpha_synuclein_PRS_final.txt\n",
      "BioFIND.endocytosis_PRS_final.txt\n",
      "BioFIND.innate_immune_PRS_final.txt\n",
      "BioFIND.lysosome_PRS_final.txt\n",
      "BioFIND.microglia_PRS_final.txt\n",
      "BioFIND.mitochondria_PRS_final.txt\n",
      "BioFIND.monocytes_PRS_final.txt\n",
      "endocytosis.bed\n",
      "endocytosis.bim\n",
      "endocytosis_chrbp.txt\n",
      "endocytosis_extract_hg38.txt\n",
      "endocytosis.fam\n",
      "endocytosis.log\n",
      "endocytosis_PD_PRS_Chang23andMe.all_score\n",
      "endocytosis_PD_PRS_Chang23andMe.log\n",
      "endocytosis_PD_PRS_Chang23andMe.mismatch\n",
      "endocytosis_PD_PRS_Chang23andMe.prsice\n",
      "endocytosis.updated_names.bed\n",
      "endocytosis.updated_names.bim\n",
      "endocytosis.updated_names.fam\n",
      "endocytosis.updated_names.log\n",
      "endocytosis.updated_names.no_dupli.bed\n",
      "endocytosis.updated_names.no_dupli.bim\n",
      "endocytosis.updated_names.no_dupli.fam\n",
      "endocytosis.updated_names.no_dupli.log\n",
      "HBS.adaptive_immune_PRS_final.txt\n",
      "HBS.alpha_synuclein_PRS_final.txt\n",
      "HBS.endocytosis_PRS_final.txt\n",
      "HBS.innate_immune_PRS_final.txt\n",
      "HBS.lysosome_PRS_final.txt\n",
      "HBS.microglia_PRS_final.txt\n",
      "HBS.mitochondria_PRS_final.txt\n",
      "HBS.monocytes_PRS_final.txt\n",
      "innate_immune.bed\n",
      "innate_immune.bim\n",
      "innate_immune_chrbp.txt\n",
      "innate_immune_extract_hg38.txt\n",
      "innate_immune.fam\n",
      "innate_immune.log\n",
      "innate_immune_PD_PRS_Chang23andMe.all_score\n",
      "innate_immune_PD_PRS_Chang23andMe.log\n",
      "innate_immune_PD_PRS_Chang23andMe.mismatch\n",
      "innate_immune_PD_PRS_Chang23andMe.prsice\n",
      "innate_immune.updated_names.bed\n",
      "innate_immune.updated_names.bim\n",
      "innate_immune.updated_names.fam\n",
      "innate_immune.updated_names.log\n",
      "innate_immune.updated_names.no_dupli.bed\n",
      "innate_immune.updated_names.no_dupli.bim\n",
      "innate_immune.updated_names.no_dupli.fam\n",
      "innate_immune.updated_names.no_dupli.log\n",
      "LCC.adaptive_immune_PRS_final.txt\n",
      "LCC.alpha_synuclein_PRS_final.txt\n",
      "LCC.endocytosis_PRS_final.txt\n",
      "LCC.innate_immune_PRS_final.txt\n",
      "LCC.lysosome_PRS_final.txt\n",
      "LCC.microglia_PRS_final.txt\n",
      "LCC.mitochondria_PRS_final.txt\n",
      "LCC.monocytes_PRS_final.txt\n",
      "lysosome.bed\n",
      "lysosome.bim\n",
      "lysosome_chrbp.txt\n",
      "lysosome_extract_hg38.txt\n",
      "lysosome.fam\n",
      "lysosome.log\n",
      "lysosome_PD_PRS_Chang23andMe.all_score\n",
      "lysosome_PD_PRS_Chang23andMe.log\n",
      "lysosome_PD_PRS_Chang23andMe.mismatch\n",
      "lysosome_PD_PRS_Chang23andMe.prsice\n",
      "lysosome.updated_names.bed\n",
      "lysosome.updated_names.bim\n",
      "lysosome.updated_names.fam\n",
      "lysosome.updated_names.log\n",
      "lysosome.updated_names.no_dupli.bed\n",
      "lysosome.updated_names.no_dupli.bim\n",
      "lysosome.updated_names.no_dupli.fam\n",
      "lysosome.updated_names.no_dupli.log\n",
      "microglia.bed\n",
      "microglia.bim\n",
      "microglia_chrbp.txt\n",
      "microglia_extract_hg38.txt\n",
      "microglia.fam\n",
      "microglia.log\n",
      "microglia_PD_PRS_Chang23andMe.all_score\n",
      "microglia_PD_PRS_Chang23andMe.log\n",
      "microglia_PD_PRS_Chang23andMe.mismatch\n",
      "microglia_PD_PRS_Chang23andMe.prsice\n",
      "microglia.updated_names.bed\n",
      "microglia.updated_names.bim\n",
      "microglia.updated_names.fam\n",
      "microglia.updated_names.log\n",
      "microglia.updated_names.no_dupli.bed\n",
      "microglia.updated_names.no_dupli.bim\n",
      "microglia.updated_names.no_dupli.fam\n",
      "microglia.updated_names.no_dupli.log\n",
      "mitochondria.bed\n",
      "mitochondria.bim\n",
      "mitochondria_chrbp.txt\n",
      "mitochondria_extract_hg38.txt\n",
      "mitochondria.fam\n",
      "mitochondria.log\n",
      "mitochondria_PD_PRS_Chang23andMe.all_score\n",
      "mitochondria_PD_PRS_Chang23andMe.log\n",
      "mitochondria_PD_PRS_Chang23andMe.mismatch\n",
      "mitochondria_PD_PRS_Chang23andMe.prsice\n",
      "mitochondria.updated_names.bed\n",
      "mitochondria.updated_names.bim\n",
      "mitochondria.updated_names.fam\n",
      "mitochondria.updated_names.log\n",
      "mitochondria.updated_names.no_dupli.bed\n",
      "mitochondria.updated_names.no_dupli.bim\n",
      "mitochondria.updated_names.no_dupli.fam\n",
      "mitochondria.updated_names.no_dupli.log\n",
      "monocytes.bed\n",
      "monocytes.bim\n",
      "monocytes_chrbp.txt\n",
      "monocytes_extract_hg38.txt\n",
      "monocytes.fam\n",
      "monocytes.log\n",
      "monocytes_PD_PRS_Chang23andMe.all_score\n",
      "monocytes_PD_PRS_Chang23andMe.log\n",
      "monocytes_PD_PRS_Chang23andMe.mismatch\n",
      "monocytes_PD_PRS_Chang23andMe.prsice\n",
      "monocytes.updated_names.bed\n",
      "monocytes.updated_names.bim\n",
      "monocytes.updated_names.fam\n",
      "monocytes.updated_names.log\n",
      "monocytes.updated_names.no_dupli.bed\n",
      "monocytes.updated_names.no_dupli.bim\n",
      "monocytes.updated_names.no_dupli.fam\n",
      "monocytes.updated_names.no_dupli.log\n",
      "n_snps_in_pathwayPRS_summary.tab\n",
      "pathways_agediagnosis_results.txt\n",
      "pathways_dementia_results.txt\n",
      "pathways_HY3_results.txt\n",
      "pathways_mdsupdrs2_results.txt\n",
      "pathways_mdsupdrs3_results.txt\n",
      "pathways_moca_results.txt\n",
      "pathways_pdq8_results.txt\n",
      "pathways_rbd_results.txt\n",
      "PDBP.adaptive_immune_PRS_final.txt\n",
      "PDBP.alpha_synuclein_PRS_final.txt\n",
      "PDBP.endocytosis_PRS_final.txt\n",
      "PDBP.innate_immune_PRS_final.txt\n",
      "PDBP.lysosome_PRS_final.txt\n",
      "PDBP.microglia_PRS_final.txt\n",
      "PDBP.mitochondria_PRS_final.txt\n",
      "PDBP.monocytes_PRS_final.txt\n",
      "PPMI.adaptive_immune_PRS_final.txt\n",
      "PPMI.alpha_synuclein_PRS_final.txt\n",
      "PPMI.endocytosis_PRS_final.txt\n",
      "PPMI.innate_immune_PRS_final.txt\n",
      "PPMI.lysosome_PRS_final.txt\n",
      "PPMI.microglia_PRS_final.txt\n",
      "PPMI.mitochondria_PRS_final.txt\n",
      "PPMI.monocytes_PRS_final.txt\n",
      "Steady.adaptive_immune_PRS_final.txt\n",
      "Steady.alpha_synuclein_PRS_final.txt\n",
      "Steady.endocytosis_PRS_final.txt\n",
      "Steady.innate_immune_PRS_final.txt\n",
      "Steady.lysosome_PRS_final.txt\n",
      "Steady.microglia_PRS_final.txt\n",
      "Steady.mitochondria_PRS_final.txt\n",
      "Steady.monocytes_PRS_final.txt\n",
      "Sure.adaptive_immune_PRS_final.txt\n",
      "Sure.alpha_synuclein_PRS_final.txt\n",
      "Sure.endocytosis_PRS_final.txt\n",
      "Sure.innate_immune_PRS_final.txt\n",
      "Sure.lysosome_PRS_final.txt\n",
      "Sure.microglia_PRS_final.txt\n",
      "Sure.mitochondria_PRS_final.txt\n",
      "Sure.monocytes_PRS_final.txt\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/pathways"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# For each pathway, extract regions from AMP-PD QC'ed genetic data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "This is done on the full AMP-PD genetic data and not separated by cohorts yet. As the cohorts were joint called and the genetic data is in a merged file, the same variants are present across all cohorts. Later we will analyse the PRSs separately in each cohort."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/adaptive_immune.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --exclude range /home/jupyter/genetic_data_v2-5/exclusion_regions_hg38.txt\n",
      "  --extract range /home/jupyter/pathways/adaptive_immune_extract_hg38.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/adaptive_immune\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun  9 13:20:36 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract bed1: 7623779 variants excluded.\n",
      "--exclude bed1: 5687 variants excluded.\n",
      "108520 variants remaining after main filters.\n",
      "Writing /home/jupyter/pathways/adaptive_immune.fam ... done.\n",
      "Writing /home/jupyter/pathways/adaptive_immune.bim ... done.\n",
      "Writing /home/jupyter/pathways/adaptive_immune.bed ... 0%\b\b2%\b\b3%\b\b4%\b\b7%\b\b8%\b\b9%\b\b11%\b\b\b12%\b\b\b13%\b\b\b15%\b\b\b17%\b\b\b19%\b\b\b20%\b\b\b22%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b46%\b\b\b48%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b62%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b89%\b\b\b90%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "End time: Thu Jun  9 13:20:56 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/alpha_synuclein.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --exclude range /home/jupyter/genetic_data_v2-5/exclusion_regions_hg38.txt\n",
      "  --extract range /home/jupyter/pathways/alpha_synuclein_extract_hg38.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/alpha_synuclein\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun  9 13:20:56 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract bed1: 7727452 variants excluded.\n",
      "--exclude bed1: 302 variants excluded.\n",
      "10232 variants remaining after main filters.\n",
      "Writing /home/jupyter/pathways/alpha_synuclein.fam ... done.\n",
      "Writing /home/jupyter/pathways/alpha_synuclein.bim ... done.\n",
      "Writing /home/jupyter/pathways/alpha_synuclein.bed ... 0%\b\b1%\b\b2%\b\b7%\b\b8%\b\b12%\b\b\b14%\b\b\b15%\b\b\b20%\b\b\b68%\b\b\b72%\b\b\b76%\b\b\b80%\b\b\b81%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b92%\b\b\b94%\b\b\b96%\b\b\bdone.\n",
      "End time: Thu Jun  9 13:20:57 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/innate_immune.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --exclude range /home/jupyter/genetic_data_v2-5/exclusion_regions_hg38.txt\n",
      "  --extract range /home/jupyter/pathways/innate_immune_extract_hg38.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/innate_immune\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun  9 13:20:57 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract bed1: 7602870 variants excluded.\n",
      "--exclude bed1: 2276 variants excluded.\n",
      "132840 variants remaining after main filters.\n",
      "Writing /home/jupyter/pathways/innate_immune.fam ... done.\n",
      "Writing /home/jupyter/pathways/innate_immune.bim ... done.\n",
      "Writing /home/jupyter/pathways/innate_immune.bed ... 0%\b\b1%\b\b3%\b\b5%\b\b7%\b\b8%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b40%\b\b\b42%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b60%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b84%\b\b\b85%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b91%\b\b\b92%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "End time: Thu Jun  9 13:21:15 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/lysosome.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --exclude range /home/jupyter/genetic_data_v2-5/exclusion_regions_hg38.txt\n",
      "  --extract range /home/jupyter/pathways/lysosome_extract_hg38.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/lysosome\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun  9 13:21:15 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract bed1: 7720896 variants excluded.\n",
      "--exclude bed1: 449 variants excluded.\n",
      "16641 variants remaining after main filters.\n",
      "Writing /home/jupyter/pathways/lysosome.fam ... done.\n",
      "Writing /home/jupyter/pathways/lysosome.bim ... done.\n",
      "Writing /home/jupyter/pathways/lysosome.bed ... 0%\b\b1%\b\b2%\b\b3%\b\b5%\b\b23%\b\b\b25%\b\b\b26%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b45%\b\b\b47%\b\b\b48%\b\b\b50%\b\b\b51%\b\b\b53%\b\b\b54%\b\b\b56%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b67%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b73%\b\b\b78%\b\b\b79%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b86%\b\b\b87%\b\b\b91%\b\b\b92%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "End time: Thu Jun  9 13:21:17 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/endocytosis.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --exclude range /home/jupyter/genetic_data_v2-5/exclusion_regions_hg38.txt\n",
      "  --extract range /home/jupyter/pathways/endocytosis_extract_hg38.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/endocytosis\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun  9 13:21:17 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract bed1: 7690212 variants excluded.\n",
      "--exclude bed1: 995 variants excluded.\n",
      "46779 variants remaining after main filters.\n",
      "Writing /home/jupyter/pathways/endocytosis.fam ... done.\n",
      "Writing /home/jupyter/pathways/endocytosis.bim ... done.\n",
      "Writing /home/jupyter/pathways/endocytosis.bed ... 0%\b\b1%\b\b2%\b\b3%\b\b6%\b\b7%\b\b8%\b\b9%\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b19%\b\b\b20%\b\b\b22%\b\b\b23%\b\b\b25%\b\b\b26%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b32%\b\b\b33%\b\b\b37%\b\b\b38%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b45%\b\b\b48%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b58%\b\b\b62%\b\b\b63%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b76%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b84%\b\b\b87%\b\b\b88%\b\b\b90%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b97%\b\b\b98%\b\b\bdone.\n",
      "End time: Thu Jun  9 13:21:23 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/mitochondria.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --exclude range /home/jupyter/genetic_data_v2-5/exclusion_regions_hg38.txt\n",
      "  --extract range /home/jupyter/pathways/mitochondria_extract_hg38.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/mitochondria\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun  9 13:21:23 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract bed1: 7482158 variants excluded.\n",
      "--exclude bed1: 1808 variants excluded.\n",
      "254020 variants remaining after main filters.\n",
      "Writing /home/jupyter/pathways/mitochondria.fam ... done.\n",
      "Writing /home/jupyter/pathways/mitochondria.bim ... done.\n",
      "Writing /home/jupyter/pathways/mitochondria.bed ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b7%\b\b8%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b16%\b\b\b17%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b40%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b47%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b56%\b\b\b57%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b80%\b\b\b82%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b89%\b\b\b90%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "End time: Thu Jun  9 13:21:49 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/microglia.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --exclude range /home/jupyter/genetic_data_v2-5/exclusion_regions_hg38.txt\n",
      "  --extract range /home/jupyter/pathways/microglia_extract_hg38.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/microglia\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun  9 13:21:49 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract bed1: 7668266 variants excluded.\n",
      "--exclude bed1: 2060 variants excluded.\n",
      "67660 variants remaining after main filters.\n",
      "Writing /home/jupyter/pathways/microglia.fam ... done.\n",
      "Writing /home/jupyter/pathways/microglia.bim ... done.\n",
      "Writing /home/jupyter/pathways/microglia.bed ... 0%\b\b2%\b\b3%\b\b4%\b\b5%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b21%\b\b\b22%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b91%\b\b\b92%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "End time: Thu Jun  9 13:23:17 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/monocytes.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --exclude range /home/jupyter/genetic_data_v2-5/exclusion_regions_hg38.txt\n",
      "  --extract range /home/jupyter/pathways/monocytes_extract_hg38.txt\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/monocytes\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun  9 13:23:17 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract bed1: 7676522 variants excluded.\n",
      "--exclude bed1: 1846 variants excluded.\n",
      "59618 variants remaining after main filters.\n",
      "Writing /home/jupyter/pathways/monocytes.fam ... done.\n",
      "Writing /home/jupyter/pathways/monocytes.bim ... done.\n",
      "Writing /home/jupyter/pathways/monocytes.bed ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b89%\b\b\b91%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "End time: Thu Jun  9 13:24:08 2022\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "for PATHWAY in adaptive_immune alpha_synuclein innate_immune lysosome endocytosis mitochondria microglia monocytes\n",
    "do\n",
    "/home/jupyter/plink2 --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf \\\n",
    "            --extract range /home/jupyter/pathways/\"$PATHWAY\"_extract_hg38.txt \\\n",
    "            --exclude range /home/jupyter/genetic_data_v2-5/exclusion_regions_hg38.txt \\\n",
    "            --make-bed \\\n",
    "            --out /home/jupyter/pathways/\"$PATHWAY\" \\\n",
    "            --threads 16\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Install PRSice2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PRSice2 is already installed in /home/jupyter/\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "if test -e /home/jupyter/PRSice_linux; then\n",
    "\n",
    "echo \"PRSice2 is already installed in /home/jupyter/\"\n",
    "else\n",
    "echo \"PRSice2 is not installed\"\n",
    "cd /home/jupyter/\n",
    "\n",
    "wget https://github.com/choishingwan/PRSice/releases/download/2.3.3/PRSice_linux.zip\n",
    "\n",
    "unzip -o PRSice_linux.zip\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AMP-PD_Pathway_PRS\n",
      "clinical_data_v2-5\n",
      "entrypoint.out\n",
      "gcta_1.93.2beta\n",
      "gcta_1.93.2beta.zip\n",
      "genetic_data_v2-5\n",
      "HY3_results\n",
      "jupyter.log\n",
      "LDpred2\n",
      "LICENSE\n",
      "lost+found\n",
      "notebooks\n",
      "packages\n",
      "pathways\n",
      "plink\n",
      "plink2\n",
      "plink2_linux_avx2_20210826.zip\n",
      "plink_linux_x86_64_20190304.zip\n",
      "prettify\n",
      "PRSice_linux\n",
      "PRSice_linux.zip\n",
      "PRSice.R\n",
      "R_packages\n",
      "SNCA\n",
      "sumstats\n",
      "TOY_BASE_GWAS.assoc\n",
      "toy.map\n",
      "toy.ped\n",
      "TOY_TARGET_DATA.bed\n",
      "TOY_TARGET_DATA.bim\n",
      "TOY_TARGET_DATA.fam\n",
      "TOY_TARGET_DATA.pheno\n",
      "welder.log\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Copy Chang 2017 summary stats to workspace and tidy"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Using Chang 2017 GWAS discovery to create PRSs: http://research-pub.gene.com/chang_et_al_2017/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "INCLUDING 23andMe data (need contract with 23andMe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "mkdir -p /home/jupyter/sumstats\n",
    "cd /home/jupyter/sumstats\n",
    "mkdir -p Chang2017_no23andMe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "cd /home/jupyter/sumstats\n",
    "mkdir -p Chang2017_with23andMe\n",
    "mkdir -p Chang2017_with23andMe_Sara"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chang2017_no23andMe\n",
      "Chang2017_with23andMe\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/sumstats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -mu manuela-norway cp gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181/PD_all_xmeta.dat  /home/jupyter/sumstats/Chang2017_with23andMe\n",
      "Copying gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181/PD_all_xmeta.dat...\n",
      "\\ [1/1 files][  1.5 GiB/  1.5 GiB] 100% Done  55.3 MiB/s ETA 00:00:00           \n",
      "Operation completed over 1 objects/1.5 GiB.                                      \n",
      "Executing: gsutil -mu manuela-norway cp gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181/all_snp_info.txt  /home/jupyter/sumstats/Chang2017_with23andMe\n",
      "Copying gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181/all_snp_info.txt...\n",
      "| [1/1 files][  7.1 GiB/  7.1 GiB] 100% Done  50.6 MiB/s ETA 00:00:00           \n",
      "Operation completed over 1 objects/7.1 GiB.                                      \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp {WORKSPACE_BUCKET}/PD_all_xmeta.dat  /home/jupyter/sumstats/Chang2017_with23andMe')\n",
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp {WORKSPACE_BUCKET}/all_snp_info.txt  /home/jupyter/sumstats/Chang2017_with23andMe')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "all_snp_info.txt\n",
      "chang_37_for_liftOver.bed\n",
      "chang_37_forPRSice.txt\n",
      "Chang_hg38_forPRSice.tab\n",
      "Chang_rsids.txt\n",
      "liftOver_output.bed\n",
      "PD_all_xmeta.dat\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/sumstats/Chang2017_with23andMe"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "There are currently issues with the 23andMe sumstats that 23andMe sent, as they are truncated at chr4 and 23andMe have not replied to any emails. Sara has kindly sent over another version of the Chang2017 sumstats including 23andMe data. I have lifted this over to hg38 and formatted for PRSice."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -mu manuela-norway cp gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181/META4_hg38.tab  /home/jupyter/sumstats/Chang2017_with23andMe_Sara\n",
      "Copying gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181/META4_hg38.tab...\n",
      "/ [1/1 files][314.4 MiB/314.4 MiB] 100% Done                                    \n",
      "Operation completed over 1 objects/314.4 MiB.                                    \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp {WORKSPACE_BUCKET}/META4_hg38.tab  /home/jupyter/sumstats/Chang2017_with23andMe_Sara')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Format Chang2017 summary stats"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "The data is currently in two files\n",
    "* PD_all_xmeta.dat has the GWAS summary stats information with a unique number identifier for each SNP\n",
    "* all_snp_info.txt has the SNP information, chr, position, alleles etc. in GRCh37\n",
    "\n",
    "This data is for European ancestry GWAS only.\n",
    "\n",
    "Need to merge these files."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#First read in both files\n",
    "Chang = pd.read_csv(f'/home/jupyter/sumstats/Chang2017_with23andMe/PD_all_xmeta.dat',\n",
    "                   delimiter = \"\\t\")\n",
    "\n",
    "snps = pd.read_csv(f'/home/jupyter/sumstats/Chang2017_with23andMe/all_snp_info.txt',\n",
    "                  delimiter = \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Merge files by all.data.id (unique identifier for each SNP), keeping only SNPs in PD_all_xmeta.dat\n",
    "Chang_snps = pd.merge(Chang, snps, left_on = 'all.data.id', right_on = 'all.data.id', how = 'left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 15635593 entries, 0 to 15635592\n",
      "Data columns (total 16 columns):\n",
      " #   Column       Dtype  \n",
      "---  ------       -----  \n",
      " 0   all.data.id  int64  \n",
      " 1   src          object \n",
      " 2   pvalue       float64\n",
      " 3   effect       float64\n",
      " 4   stderr       float64\n",
      " 5   pass         object \n",
      " 6   im.num.0     float64\n",
      " 7   dose.b.0     float64\n",
      " 8   im.num.1     float64\n",
      " 9   dose.b.1     float64\n",
      " 10  AA.0         float64\n",
      " 11  AB.0         float64\n",
      " 12  BB.0         float64\n",
      " 13  AA.1         float64\n",
      " 14  AB.1         float64\n",
      " 15  BB.1         float64\n",
      "dtypes: float64(13), int64(1), object(2)\n",
      "memory usage: 1.9+ GB\n"
     ]
    }
   ],
   "source": [
    "Chang.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Int64Index: 15635593 entries, 0 to 15635592\n",
      "Data columns (total 33 columns):\n",
      " #   Column        Dtype  \n",
      "---  ------        -----  \n",
      " 0   all.data.id   int64  \n",
      " 1   src           object \n",
      " 2   pvalue        float64\n",
      " 3   effect        float64\n",
      " 4   stderr        float64\n",
      " 5   pass          object \n",
      " 6   im.num.0      float64\n",
      " 7   dose.b.0      float64\n",
      " 8   im.num.1      float64\n",
      " 9   dose.b.1      float64\n",
      " 10  AA.0          float64\n",
      " 11  AB.0          float64\n",
      " 12  BB.0          float64\n",
      " 13  AA.1          float64\n",
      " 14  AB.1          float64\n",
      " 15  BB.1          float64\n",
      " 16  gt.data.id    float64\n",
      " 17  im.data.id    float64\n",
      " 18  assay.name    object \n",
      " 19  scaffold      object \n",
      " 20  position      int64  \n",
      " 21  alleles       object \n",
      " 22  ploidy        object \n",
      " 23  cytoband      object \n",
      " 24  gene.context  object \n",
      " 25  is.v1         bool   \n",
      " 26  is.v2         bool   \n",
      " 27  is.v3         bool   \n",
      " 28  is.v4         bool   \n",
      " 29  is.v5         bool   \n",
      " 30  h550          bool   \n",
      " 31  omni          bool   \n",
      " 32  strand        object \n",
      "dtypes: bool(7), float64(15), int64(2), object(9)\n",
      "memory usage: 3.2+ GB\n"
     ]
    }
   ],
   "source": [
    "Chang_snps.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Remove SNPs that have not passed QC\n",
    "Chang_snps_pass = Chang_snps.copy()\n",
    "Chang_snps_pass = Chang_snps_pass[(Chang_snps_pass['pass']== \"Y\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all.data.id</th>\n",
       "      <th>src</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>effect</th>\n",
       "      <th>stderr</th>\n",
       "      <th>pass</th>\n",
       "      <th>im.num.0</th>\n",
       "      <th>dose.b.0</th>\n",
       "      <th>im.num.1</th>\n",
       "      <th>dose.b.1</th>\n",
       "      <th>...</th>\n",
       "      <th>cytoband</th>\n",
       "      <th>gene.context</th>\n",
       "      <th>is.v1</th>\n",
       "      <th>is.v2</th>\n",
       "      <th>is.v3</th>\n",
       "      <th>is.v4</th>\n",
       "      <th>is.v5</th>\n",
       "      <th>h550</th>\n",
       "      <th>omni</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>I</td>\n",
       "      <td>0.771412</td>\n",
       "      <td>-0.009939</td>\n",
       "      <td>0.034239</td>\n",
       "      <td>Y</td>\n",
       "      <td>159956.0</td>\n",
       "      <td>0.847624</td>\n",
       "      <td>4947.0</td>\n",
       "      <td>0.848227</td>\n",
       "      <td>...</td>\n",
       "      <td>1p36.33</td>\n",
       "      <td>[]--OR4F5</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>I</td>\n",
       "      <td>0.788298</td>\n",
       "      <td>0.007978</td>\n",
       "      <td>0.029746</td>\n",
       "      <td>Y</td>\n",
       "      <td>159956.0</td>\n",
       "      <td>0.177142</td>\n",
       "      <td>4947.0</td>\n",
       "      <td>0.176896</td>\n",
       "      <td>...</td>\n",
       "      <td>1p36.33</td>\n",
       "      <td>[]--OR4F5</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>8</td>\n",
       "      <td>I</td>\n",
       "      <td>0.917844</td>\n",
       "      <td>-0.002636</td>\n",
       "      <td>0.025539</td>\n",
       "      <td>Y</td>\n",
       "      <td>302034.0</td>\n",
       "      <td>0.823991</td>\n",
       "      <td>6476.0</td>\n",
       "      <td>0.823804</td>\n",
       "      <td>...</td>\n",
       "      <td>1p36.33</td>\n",
       "      <td>[]--OR4F5</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>I</td>\n",
       "      <td>0.787689</td>\n",
       "      <td>0.008319</td>\n",
       "      <td>0.030927</td>\n",
       "      <td>Y</td>\n",
       "      <td>302034.0</td>\n",
       "      <td>0.173884</td>\n",
       "      <td>6476.0</td>\n",
       "      <td>0.173288</td>\n",
       "      <td>...</td>\n",
       "      <td>1p36.33</td>\n",
       "      <td>[]--OR4F5</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>11</td>\n",
       "      <td>I</td>\n",
       "      <td>0.820159</td>\n",
       "      <td>0.007291</td>\n",
       "      <td>0.032136</td>\n",
       "      <td>Y</td>\n",
       "      <td>159956.0</td>\n",
       "      <td>0.161230</td>\n",
       "      <td>4947.0</td>\n",
       "      <td>0.160554</td>\n",
       "      <td>...</td>\n",
       "      <td>1p36.33</td>\n",
       "      <td>[]--OR4F5</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    all.data.id src    pvalue    effect    stderr pass  im.num.0  dose.b.0  \\\n",
       "2             3   I  0.771412 -0.009939  0.034239    Y  159956.0  0.847624   \n",
       "5             6   I  0.788298  0.007978  0.029746    Y  159956.0  0.177142   \n",
       "7             8   I  0.917844 -0.002636  0.025539    Y  302034.0  0.823991   \n",
       "9            10   I  0.787689  0.008319  0.030927    Y  302034.0  0.173884   \n",
       "10           11   I  0.820159  0.007291  0.032136    Y  159956.0  0.161230   \n",
       "\n",
       "    im.num.1  dose.b.1  ...  cytoband  gene.context  is.v1  is.v2  is.v3  \\\n",
       "2     4947.0  0.848227  ...   1p36.33     []--OR4F5  False  False  False   \n",
       "5     4947.0  0.176896  ...   1p36.33     []--OR4F5  False  False  False   \n",
       "7     6476.0  0.823804  ...   1p36.33     []--OR4F5  False  False  False   \n",
       "9     6476.0  0.173288  ...   1p36.33     []--OR4F5  False  False  False   \n",
       "10    4947.0  0.160554  ...   1p36.33     []--OR4F5  False  False  False   \n",
       "\n",
       "    is.v4  is.v5   h550   omni strand  \n",
       "2   False  False  False  False      +  \n",
       "5   False  False  False  False      +  \n",
       "7   False  False  False  False      +  \n",
       "9   False  False  False  False      +  \n",
       "10  False  False  False  False      +  \n",
       "\n",
       "[5 rows x 33 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chang_snps_pass.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Select just relevant columns\n",
    "Chang_snps_pass = Chang_snps_pass[['all.data.id', 'pvalue', 'effect', 'stderr', 'assay.name', 'scaffold', 'position', 'alleles']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all.data.id</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>effect</th>\n",
       "      <th>stderr</th>\n",
       "      <th>assay.name</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>position</th>\n",
       "      <th>alleles</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>0.771412</td>\n",
       "      <td>-0.009939</td>\n",
       "      <td>0.034239</td>\n",
       "      <td>rs555500075</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10352</td>\n",
       "      <td>D/I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>0.788298</td>\n",
       "      <td>0.007978</td>\n",
       "      <td>0.029746</td>\n",
       "      <td>rs558604819</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10642</td>\n",
       "      <td>A/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>8</td>\n",
       "      <td>0.917844</td>\n",
       "      <td>-0.002636</td>\n",
       "      <td>0.025539</td>\n",
       "      <td>rs544419019</td>\n",
       "      <td>chr1</td>\n",
       "      <td>11012</td>\n",
       "      <td>C/G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>0.787689</td>\n",
       "      <td>0.008319</td>\n",
       "      <td>0.030927</td>\n",
       "      <td>rs574746232</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13011</td>\n",
       "      <td>G/T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>11</td>\n",
       "      <td>0.820159</td>\n",
       "      <td>0.007291</td>\n",
       "      <td>0.032136</td>\n",
       "      <td>rs540538026</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13110</td>\n",
       "      <td>A/G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    all.data.id    pvalue    effect    stderr   assay.name scaffold  position  \\\n",
       "2             3  0.771412 -0.009939  0.034239  rs555500075     chr1     10352   \n",
       "5             6  0.788298  0.007978  0.029746  rs558604819     chr1     10642   \n",
       "7             8  0.917844 -0.002636  0.025539  rs544419019     chr1     11012   \n",
       "9            10  0.787689  0.008319  0.030927  rs574746232     chr1     13011   \n",
       "10           11  0.820159  0.007291  0.032136  rs540538026     chr1     13110   \n",
       "\n",
       "   alleles  \n",
       "2      D/I  \n",
       "5      A/G  \n",
       "7      C/G  \n",
       "9      G/T  \n",
       "10     A/G  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chang_snps_pass.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all.data.id</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>effect</th>\n",
       "      <th>stderr</th>\n",
       "      <th>assay.name</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>position</th>\n",
       "      <th>alleles</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>0.771412</td>\n",
       "      <td>-0.009939</td>\n",
       "      <td>0.034239</td>\n",
       "      <td>rs555500075</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10352</td>\n",
       "      <td>D/I</td>\n",
       "      <td>D</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6</td>\n",
       "      <td>0.788298</td>\n",
       "      <td>0.007978</td>\n",
       "      <td>0.029746</td>\n",
       "      <td>rs558604819</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10642</td>\n",
       "      <td>A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>0.917844</td>\n",
       "      <td>-0.002636</td>\n",
       "      <td>0.025539</td>\n",
       "      <td>rs544419019</td>\n",
       "      <td>chr1</td>\n",
       "      <td>11012</td>\n",
       "      <td>C/G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>0.787689</td>\n",
       "      <td>0.008319</td>\n",
       "      <td>0.030927</td>\n",
       "      <td>rs574746232</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13011</td>\n",
       "      <td>G/T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>11</td>\n",
       "      <td>0.820159</td>\n",
       "      <td>0.007291</td>\n",
       "      <td>0.032136</td>\n",
       "      <td>rs540538026</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13110</td>\n",
       "      <td>A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   all.data.id    pvalue    effect    stderr   assay.name scaffold  position  \\\n",
       "0            3  0.771412 -0.009939  0.034239  rs555500075     chr1     10352   \n",
       "1            6  0.788298  0.007978  0.029746  rs558604819     chr1     10642   \n",
       "2            8  0.917844 -0.002636  0.025539  rs544419019     chr1     11012   \n",
       "3           10  0.787689  0.008319  0.030927  rs574746232     chr1     13011   \n",
       "4           11  0.820159  0.007291  0.032136  rs540538026     chr1     13110   \n",
       "\n",
       "  alleles alleleA alleleB  \n",
       "0     D/I       D       I  \n",
       "1     A/G       A       G  \n",
       "2     C/G       C       G  \n",
       "3     G/T       G       T  \n",
       "4     A/G       A       G  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Split alleles into two columns. Note that the order of the alleles is just alphabetical so will call alleleA and B.\n",
    "#Note that B is the effect allele in the summary statistics\n",
    "Chang_snps_pass_formatted = Chang_snps_pass.copy()\n",
    "Chang_snps_pass_formatted = Chang_snps_pass_formatted.reset_index(drop = True)\n",
    "Chang_snps_pass_formatted[['alleleA', 'alleleB']] = Chang_snps_pass_formatted['alleles'].str.split('/', expand = True)\n",
    "Chang_snps_pass_formatted.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Make new column for chromosome without 'chr'\n",
    "Chang_snps_pass_formatted['chr'] = Chang_snps_pass_formatted['scaffold'].str.replace(r'chr', '').astype(str)\n",
    "\n",
    "#Make BP str format\n",
    "Chang_snps_pass_formatted[\"position\"] = Chang_snps_pass_formatted[\"position\"].astype(str)\n",
    "\n",
    "#Make combined column chrbp\n",
    "Chang_snps_pass_formatted['chrbp'] = Chang_snps_pass_formatted['chr'] + \":\" + Chang_snps_pass_formatted[\"position\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all.data.id</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>effect</th>\n",
       "      <th>stderr</th>\n",
       "      <th>assay.name</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>position</th>\n",
       "      <th>alleles</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>chr</th>\n",
       "      <th>chrbp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>0.771412</td>\n",
       "      <td>-0.009939</td>\n",
       "      <td>0.034239</td>\n",
       "      <td>rs555500075</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10352</td>\n",
       "      <td>D/I</td>\n",
       "      <td>D</td>\n",
       "      <td>I</td>\n",
       "      <td>1</td>\n",
       "      <td>1:10352</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6</td>\n",
       "      <td>0.788298</td>\n",
       "      <td>0.007978</td>\n",
       "      <td>0.029746</td>\n",
       "      <td>rs558604819</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10642</td>\n",
       "      <td>A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:10642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>0.917844</td>\n",
       "      <td>-0.002636</td>\n",
       "      <td>0.025539</td>\n",
       "      <td>rs544419019</td>\n",
       "      <td>chr1</td>\n",
       "      <td>11012</td>\n",
       "      <td>C/G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:11012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>0.787689</td>\n",
       "      <td>0.008319</td>\n",
       "      <td>0.030927</td>\n",
       "      <td>rs574746232</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13011</td>\n",
       "      <td>G/T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>1:13011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>11</td>\n",
       "      <td>0.820159</td>\n",
       "      <td>0.007291</td>\n",
       "      <td>0.032136</td>\n",
       "      <td>rs540538026</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13110</td>\n",
       "      <td>A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:13110</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   all.data.id    pvalue    effect    stderr   assay.name scaffold position  \\\n",
       "0            3  0.771412 -0.009939  0.034239  rs555500075     chr1    10352   \n",
       "1            6  0.788298  0.007978  0.029746  rs558604819     chr1    10642   \n",
       "2            8  0.917844 -0.002636  0.025539  rs544419019     chr1    11012   \n",
       "3           10  0.787689  0.008319  0.030927  rs574746232     chr1    13011   \n",
       "4           11  0.820159  0.007291  0.032136  rs540538026     chr1    13110   \n",
       "\n",
       "  alleles alleleA alleleB chr    chrbp  \n",
       "0     D/I       D       I   1  1:10352  \n",
       "1     A/G       A       G   1  1:10642  \n",
       "2     C/G       C       G   1  1:11012  \n",
       "3     G/T       G       T   1  1:13011  \n",
       "4     A/G       A       G   1  1:13110  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chang_snps_pass_formatted.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrbp</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>BETA</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>stderr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:10352</td>\n",
       "      <td>I</td>\n",
       "      <td>D</td>\n",
       "      <td>-0.009939</td>\n",
       "      <td>0.771412</td>\n",
       "      <td>0.034239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:10642</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.007978</td>\n",
       "      <td>0.788298</td>\n",
       "      <td>0.029746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:11012</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>-0.002636</td>\n",
       "      <td>0.917844</td>\n",
       "      <td>0.025539</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:13011</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.008319</td>\n",
       "      <td>0.787689</td>\n",
       "      <td>0.030927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:13110</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.007291</td>\n",
       "      <td>0.820159</td>\n",
       "      <td>0.032136</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chrbp alleleB alleleA      BETA    pvalue    stderr\n",
       "0  1:10352       I       D -0.009939  0.771412  0.034239\n",
       "1  1:10642       G       A  0.007978  0.788298  0.029746\n",
       "2  1:11012       G       C -0.002636  0.917844  0.025539\n",
       "3  1:13011       T       G  0.008319  0.787689  0.030927\n",
       "4  1:13110       G       A  0.007291  0.820159  0.032136"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "export_chang_hg37 = Chang_snps_pass_formatted.copy()\n",
    "export_chang_hg37 = export_chang_hg37[['chrbp', 'alleleB', 'alleleA', 'effect', 'pvalue', 'stderr']]\n",
    "\n",
    "#Rename effect column to BETA\n",
    "export_chang_hg37 = export_chang_hg37.rename(columns={'effect': 'BETA'})\n",
    "\n",
    "export_chang_hg37.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Export hg37 data\n",
    "with open('/home/jupyter/sumstats/Chang2017_with23andMe/chang_37_forPRSice.txt', 'w') as f:\n",
    "    f.write(export_chang_hg37.to_csv(index=False, header = True, sep='\\t'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u manuela-norway cp -r /home/jupyter/sumstats/Chang2017_with23andMe/chang_37_forPRSice.txt gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181\n",
      "Copying file:///home/jupyter/sumstats/Chang2017_with23andMe/chang_37_forPRSice.txt [Content-Type=text/plain]...\n",
      "==> NOTE: You are uploading one or more large file(s), which would run          \n",
      "significantly faster if you enable parallel composite uploads. This\n",
      "feature can be enabled by editing the\n",
      "\"parallel_composite_upload_threshold\" value in your .boto\n",
      "configuration file. However, note that if you do this large files will\n",
      "be uploaded as `composite objects\n",
      "<https://cloud.google.com/storage/docs/composite-objects>`_,which\n",
      "means that any user who downloads such objects will need to have a\n",
      "compiled crcmod installed (see \"gsutil help crcmod\"). This is because\n",
      "without a compiled crcmod, computing checksums on composite objects is\n",
      "so slow that gsutil disables downloads of composite objects.\n",
      "\n",
      "- [1 files][550.8 MiB/550.8 MiB]   51.9 MiB/s                                   \n",
      "Operation completed over 1 objects/550.8 MiB.                                    \n"
     ]
    }
   ],
   "source": [
    "#Copy to workspace\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} cp -r /home/jupyter/sumstats/Chang2017_with23andMe/chang_37_forPRSice.txt {WORKSPACE_BUCKET}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Liftover Chang2017 summary stats to hg38"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "I cannot get liftOver working on Terra because 'cannot execute binary file: Exec format error'. So I have run liftOver on my local machine and copied the liftOver output files into the workspace"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "I have checked liftover coordinates against BioMart coordinates and they match perfectly. There are some instances where BioMart is missing coords because the rsID has been merged with another, and one SNP which is not lifted over in liftOver."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Read in Chang 2017 summary stats and format for liftover"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all.data.id</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>effect</th>\n",
       "      <th>stderr</th>\n",
       "      <th>assay.name</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>position</th>\n",
       "      <th>alleles</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>chr</th>\n",
       "      <th>chrbp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>0.771412</td>\n",
       "      <td>-0.009939</td>\n",
       "      <td>0.034239</td>\n",
       "      <td>rs555500075</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10352</td>\n",
       "      <td>D/I</td>\n",
       "      <td>D</td>\n",
       "      <td>I</td>\n",
       "      <td>1</td>\n",
       "      <td>1:10352</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6</td>\n",
       "      <td>0.788298</td>\n",
       "      <td>0.007978</td>\n",
       "      <td>0.029746</td>\n",
       "      <td>rs558604819</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10642</td>\n",
       "      <td>A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:10642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>0.917844</td>\n",
       "      <td>-0.002636</td>\n",
       "      <td>0.025539</td>\n",
       "      <td>rs544419019</td>\n",
       "      <td>chr1</td>\n",
       "      <td>11012</td>\n",
       "      <td>C/G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:11012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>0.787689</td>\n",
       "      <td>0.008319</td>\n",
       "      <td>0.030927</td>\n",
       "      <td>rs574746232</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13011</td>\n",
       "      <td>G/T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>1:13011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>11</td>\n",
       "      <td>0.820159</td>\n",
       "      <td>0.007291</td>\n",
       "      <td>0.032136</td>\n",
       "      <td>rs540538026</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13110</td>\n",
       "      <td>A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:13110</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   all.data.id    pvalue    effect    stderr   assay.name scaffold position  \\\n",
       "0            3  0.771412 -0.009939  0.034239  rs555500075     chr1    10352   \n",
       "1            6  0.788298  0.007978  0.029746  rs558604819     chr1    10642   \n",
       "2            8  0.917844 -0.002636  0.025539  rs544419019     chr1    11012   \n",
       "3           10  0.787689  0.008319  0.030927  rs574746232     chr1    13011   \n",
       "4           11  0.820159  0.007291  0.032136  rs540538026     chr1    13110   \n",
       "\n",
       "  alleles alleleA alleleB chr    chrbp  \n",
       "0     D/I       D       I   1  1:10352  \n",
       "1     A/G       A       G   1  1:10642  \n",
       "2     C/G       C       G   1  1:11012  \n",
       "3     G/T       G       T   1  1:13011  \n",
       "4     A/G       A       G   1  1:13110  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chang_snps_pass_formatted.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Write just rsID list for bioMart to get positions in hg38"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "chang_rsids = Chang_snps_pass_formatted.copy()\n",
    "chang_rsids = chang_rsids[['assay.name']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "with open('/home/jupyter/sumstats/Chang2017_with23andMe/Chang_rsids.txt', 'w') as f:\n",
    "    f.write(chang_rsids.to_csv(index=False, header = False, sep='\\t'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u manuela-norway cp -r /home/jupyter/sumstats/Chang2017_with23andMe/Chang_rsids.txt gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181\n",
      "Copying file:///home/jupyter/sumstats/Chang2017_with23andMe/Chang_rsids.txt [Content-Type=text/plain]...\n",
      "/ [1 files][144.1 MiB/144.1 MiB]                                                \n",
      "Operation completed over 1 objects/144.1 MiB.                                    \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} cp -r /home/jupyter/sumstats/Chang2017_with23andMe/Chang_rsids.txt {WORKSPACE_BUCKET}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "For liftOver you need chr$CHR, start and end bp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all.data.id</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>effect</th>\n",
       "      <th>stderr</th>\n",
       "      <th>assay.name</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>position</th>\n",
       "      <th>alleles</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>chr</th>\n",
       "      <th>chrbp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>0.771412</td>\n",
       "      <td>-0.009939</td>\n",
       "      <td>0.034239</td>\n",
       "      <td>rs555500075</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10352</td>\n",
       "      <td>D/I</td>\n",
       "      <td>D</td>\n",
       "      <td>I</td>\n",
       "      <td>1</td>\n",
       "      <td>1:10352</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6</td>\n",
       "      <td>0.788298</td>\n",
       "      <td>0.007978</td>\n",
       "      <td>0.029746</td>\n",
       "      <td>rs558604819</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10642</td>\n",
       "      <td>A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:10642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>0.917844</td>\n",
       "      <td>-0.002636</td>\n",
       "      <td>0.025539</td>\n",
       "      <td>rs544419019</td>\n",
       "      <td>chr1</td>\n",
       "      <td>11012</td>\n",
       "      <td>C/G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:11012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>0.787689</td>\n",
       "      <td>0.008319</td>\n",
       "      <td>0.030927</td>\n",
       "      <td>rs574746232</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13011</td>\n",
       "      <td>G/T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>1:13011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>11</td>\n",
       "      <td>0.820159</td>\n",
       "      <td>0.007291</td>\n",
       "      <td>0.032136</td>\n",
       "      <td>rs540538026</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13110</td>\n",
       "      <td>A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:13110</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   all.data.id    pvalue    effect    stderr   assay.name scaffold position  \\\n",
       "0            3  0.771412 -0.009939  0.034239  rs555500075     chr1    10352   \n",
       "1            6  0.788298  0.007978  0.029746  rs558604819     chr1    10642   \n",
       "2            8  0.917844 -0.002636  0.025539  rs544419019     chr1    11012   \n",
       "3           10  0.787689  0.008319  0.030927  rs574746232     chr1    13011   \n",
       "4           11  0.820159  0.007291  0.032136  rs540538026     chr1    13110   \n",
       "\n",
       "  alleles alleleA alleleB chr    chrbp  \n",
       "0     D/I       D       I   1  1:10352  \n",
       "1     A/G       A       G   1  1:10642  \n",
       "2     C/G       C       G   1  1:11012  \n",
       "3     G/T       G       T   1  1:13011  \n",
       "4     A/G       A       G   1  1:13110  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chang_snps_pass_formatted.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Add 1 base to create BP_end. Must first convert position column to numeric"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "chang_for_liftover = Chang_snps_pass_formatted.copy()\n",
    "chang_for_liftover['BP_end'] = pd.to_numeric(chang_for_liftover['position']) + 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all.data.id</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>effect</th>\n",
       "      <th>stderr</th>\n",
       "      <th>assay.name</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>position</th>\n",
       "      <th>alleles</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>chr</th>\n",
       "      <th>chrbp</th>\n",
       "      <th>BP_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>0.771412</td>\n",
       "      <td>-0.009939</td>\n",
       "      <td>0.034239</td>\n",
       "      <td>rs555500075</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10352</td>\n",
       "      <td>D/I</td>\n",
       "      <td>D</td>\n",
       "      <td>I</td>\n",
       "      <td>1</td>\n",
       "      <td>1:10352</td>\n",
       "      <td>10353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6</td>\n",
       "      <td>0.788298</td>\n",
       "      <td>0.007978</td>\n",
       "      <td>0.029746</td>\n",
       "      <td>rs558604819</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10642</td>\n",
       "      <td>A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:10642</td>\n",
       "      <td>10643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>0.917844</td>\n",
       "      <td>-0.002636</td>\n",
       "      <td>0.025539</td>\n",
       "      <td>rs544419019</td>\n",
       "      <td>chr1</td>\n",
       "      <td>11012</td>\n",
       "      <td>C/G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:11012</td>\n",
       "      <td>11013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>0.787689</td>\n",
       "      <td>0.008319</td>\n",
       "      <td>0.030927</td>\n",
       "      <td>rs574746232</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13011</td>\n",
       "      <td>G/T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>1:13011</td>\n",
       "      <td>13012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>11</td>\n",
       "      <td>0.820159</td>\n",
       "      <td>0.007291</td>\n",
       "      <td>0.032136</td>\n",
       "      <td>rs540538026</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13110</td>\n",
       "      <td>A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:13110</td>\n",
       "      <td>13111</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   all.data.id    pvalue    effect    stderr   assay.name scaffold position  \\\n",
       "0            3  0.771412 -0.009939  0.034239  rs555500075     chr1    10352   \n",
       "1            6  0.788298  0.007978  0.029746  rs558604819     chr1    10642   \n",
       "2            8  0.917844 -0.002636  0.025539  rs544419019     chr1    11012   \n",
       "3           10  0.787689  0.008319  0.030927  rs574746232     chr1    13011   \n",
       "4           11  0.820159  0.007291  0.032136  rs540538026     chr1    13110   \n",
       "\n",
       "  alleles alleleA alleleB chr    chrbp  BP_end  \n",
       "0     D/I       D       I   1  1:10352   10353  \n",
       "1     A/G       A       G   1  1:10642   10643  \n",
       "2     C/G       C       G   1  1:11012   11013  \n",
       "3     G/T       G       T   1  1:13011   13012  \n",
       "4     A/G       A       G   1  1:13110   13111  "
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chang_for_liftover.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "chang_for_liftover = chang_for_liftover[['scaffold', 'position', 'BP_end', 'assay.name']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>scaffold</th>\n",
       "      <th>position</th>\n",
       "      <th>BP_end</th>\n",
       "      <th>assay.name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10352</td>\n",
       "      <td>10353</td>\n",
       "      <td>rs555500075</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10642</td>\n",
       "      <td>10643</td>\n",
       "      <td>rs558604819</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11012</td>\n",
       "      <td>11013</td>\n",
       "      <td>rs544419019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13011</td>\n",
       "      <td>13012</td>\n",
       "      <td>rs574746232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13110</td>\n",
       "      <td>13111</td>\n",
       "      <td>rs540538026</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  scaffold position  BP_end   assay.name\n",
       "0     chr1    10352   10353  rs555500075\n",
       "1     chr1    10642   10643  rs558604819\n",
       "2     chr1    11012   11013  rs544419019\n",
       "3     chr1    13011   13012  rs574746232\n",
       "4     chr1    13110   13111  rs540538026"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chang_for_liftover.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "with open('/home/jupyter/sumstats/Chang2017_with23andMe/chang_37_for_liftOver.bed', 'w') as f:\n",
    "    f.write(chang_for_liftover.to_csv(index=False, header = False, sep='\\t'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr4\t44604298\t44604299\trs59905107\n",
      "chr4\t44604490\t44604491\trs370922792\n",
      "chr4\t44604516\t44604517\trs193293102\n",
      "chr4\t44604548\t44604549\trs62306189\n",
      "chr4\t44604601\t44604602\trs4473688\n",
      "chr4\t44604765\t44604766\trs535845793\n",
      "chr4\t44604776\t44604777\trs566170966\n",
      "chr4\t44604782\t44604783\trs539975122\n",
      "chr4\t44604876\t44604877\trs556428381\n",
      "chr4\t44604897\t44604898\trs190343291\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "tail /home/jupyter/sumstats/Chang2017_with23andMe/chang_37_for_liftOver.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Copy formatted file to workspace bucket"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u manuela-norway cp -r /home/jupyter/sumstats/Chang2017_with23andMe/chang_37_for_liftOver.bed gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181\n",
      "Copying file:///home/jupyter/sumstats/Chang2017_with23andMe/chang_37_for_liftOver.bed [Content-Type=application/octet-stream]...\n",
      "==> NOTE: You are uploading one or more large file(s), which would run          \n",
      "significantly faster if you enable parallel composite uploads. This\n",
      "feature can be enabled by editing the\n",
      "\"parallel_composite_upload_threshold\" value in your .boto\n",
      "configuration file. However, note that if you do this large files will\n",
      "be uploaded as `composite objects\n",
      "<https://cloud.google.com/storage/docs/composite-objects>`_,which\n",
      "means that any user who downloads such objects will need to have a\n",
      "compiled crcmod installed (see \"gsutil help crcmod\"). This is because\n",
      "without a compiled crcmod, computing checksums on composite objects is\n",
      "so slow that gsutil disables downloads of composite objects.\n",
      "\n",
      "- [1 files][437.4 MiB/437.4 MiB]                                                \n",
      "Operation completed over 1 objects/437.4 MiB.                                    \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} cp -r /home/jupyter/sumstats/Chang2017_with23andMe/chang_37_for_liftOver.bed {WORKSPACE_BUCKET}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Now run liftOver on local machine using command like below (or online)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Executing: /home/jupyter-user/notebooks/liftOver         /home/jupyter-user/notebooks/sumstats/Chang2017_no23andMe/PD_meta_analysis_PDGENE_PDWBS_liftOver.tab         hg19ToHg38.over.chain         /home/jupyter-user/notebooks/sumstats/Chang2017_no23andMe/liftOver_output.bed         /home/jupyter-user/notebooks/sumstats/Chang2017_no23andMe/liftOver_unlifted.bed\n",
      "/bin/sh: 1: /home/jupyter-user/notebooks/liftOver: Exec format error\n"
     ]
    }
   ],
   "source": [
    "#shell_do(f'/home/jupyter-user/notebooks/liftOver \\\n",
    "#        /home/jupyter-user/notebooks/sumstats/Chang2017_no23andMe/PD_meta_analysis_PDGENE_PDWBS_liftOver.tab \\\n",
    "#        hg19ToHg38.over.chain \\\n",
    "#        /home/jupyter-user/notebooks/sumstats/Chang2017_no23andMe/liftOver_output.bed \\\n",
    "#        /home/jupyter-user/notebooks/sumstats/Chang2017_no23andMe/liftOver_unlifted.bed')\n",
    "\n",
    "#NOT RUN - LIFTOVER NOT WORKING ON TERRA"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Copy liftOver output into workspace bucket"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -mu manuela-norway cp gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181/liftOver_output.bed /home/jupyter/sumstats/Chang2017_with23andMe\n",
      "Copying gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181/liftOver_output.bed...\n",
      "| [1/1 files][437.4 MiB/437.4 MiB] 100% Done                                    \n",
      "Operation completed over 1 objects/437.4 MiB.                                    \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -mu {BILLING_PROJECT_ID} cp {WORKSPACE_BUCKET}/liftOver_output.bed /home/jupyter/sumstats/Chang2017_with23andMe')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all.data.id</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>effect</th>\n",
       "      <th>stderr</th>\n",
       "      <th>assay.name</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>position</th>\n",
       "      <th>alleles</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>chr</th>\n",
       "      <th>chrbp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>0.771412</td>\n",
       "      <td>-0.009939</td>\n",
       "      <td>0.034239</td>\n",
       "      <td>rs555500075</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10352</td>\n",
       "      <td>D/I</td>\n",
       "      <td>D</td>\n",
       "      <td>I</td>\n",
       "      <td>1</td>\n",
       "      <td>1:10352</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6</td>\n",
       "      <td>0.788298</td>\n",
       "      <td>0.007978</td>\n",
       "      <td>0.029746</td>\n",
       "      <td>rs558604819</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10642</td>\n",
       "      <td>A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:10642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>0.917844</td>\n",
       "      <td>-0.002636</td>\n",
       "      <td>0.025539</td>\n",
       "      <td>rs544419019</td>\n",
       "      <td>chr1</td>\n",
       "      <td>11012</td>\n",
       "      <td>C/G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:11012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>0.787689</td>\n",
       "      <td>0.008319</td>\n",
       "      <td>0.030927</td>\n",
       "      <td>rs574746232</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13011</td>\n",
       "      <td>G/T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>1:13011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>11</td>\n",
       "      <td>0.820159</td>\n",
       "      <td>0.007291</td>\n",
       "      <td>0.032136</td>\n",
       "      <td>rs540538026</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13110</td>\n",
       "      <td>A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:13110</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   all.data.id    pvalue    effect    stderr   assay.name scaffold position  \\\n",
       "0            3  0.771412 -0.009939  0.034239  rs555500075     chr1    10352   \n",
       "1            6  0.788298  0.007978  0.029746  rs558604819     chr1    10642   \n",
       "2            8  0.917844 -0.002636  0.025539  rs544419019     chr1    11012   \n",
       "3           10  0.787689  0.008319  0.030927  rs574746232     chr1    13011   \n",
       "4           11  0.820159  0.007291  0.032136  rs540538026     chr1    13110   \n",
       "\n",
       "  alleles alleleA alleleB chr    chrbp  \n",
       "0     D/I       D       I   1  1:10352  \n",
       "1     A/G       A       G   1  1:10642  \n",
       "2     C/G       C       G   1  1:11012  \n",
       "3     G/T       G       T   1  1:13011  \n",
       "4     A/G       A       G   1  1:13110  "
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Chang_snps_pass_formatted.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Merge sumstats with liftover "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Read in liftOver output\n",
    "liftover = pd.read_csv(f'/home/jupyter/sumstats/Chang2017_with23andMe/liftOver_output.bed',\n",
    "                   delimiter = \"\\t\", header = None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP_start</th>\n",
       "      <th>BP_end</th>\n",
       "      <th>rsid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10352</td>\n",
       "      <td>10353</td>\n",
       "      <td>rs555500075</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>10642</td>\n",
       "      <td>10643</td>\n",
       "      <td>rs558604819</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11012</td>\n",
       "      <td>11013</td>\n",
       "      <td>rs544419019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13011</td>\n",
       "      <td>13012</td>\n",
       "      <td>rs574746232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>13110</td>\n",
       "      <td>13111</td>\n",
       "      <td>rs540538026</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHR  BP_start  BP_end         rsid\n",
       "0  chr1     10352   10353  rs555500075\n",
       "1  chr1     10642   10643  rs558604819\n",
       "2  chr1     11012   11013  rs544419019\n",
       "3  chr1     13011   13012  rs574746232\n",
       "4  chr1     13110   13111  rs540538026"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "liftover.columns=['CHR', 'BP_start', 'BP_end', 'rsid']\n",
    "liftover.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "IMPORTANT: The BP_start is the correct coordinates for the SNP in hg38"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Merge sumstats with liftover output - keeping only the SNPs that have been successfully lifted over"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all.data.id</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>effect</th>\n",
       "      <th>stderr</th>\n",
       "      <th>assay.name</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>position</th>\n",
       "      <th>alleles</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>chr</th>\n",
       "      <th>chrbp</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP_start</th>\n",
       "      <th>BP_end</th>\n",
       "      <th>rsid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>0.771412</td>\n",
       "      <td>-0.009939</td>\n",
       "      <td>0.034239</td>\n",
       "      <td>rs555500075</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10352</td>\n",
       "      <td>D/I</td>\n",
       "      <td>D</td>\n",
       "      <td>I</td>\n",
       "      <td>1</td>\n",
       "      <td>1:10352</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10352.0</td>\n",
       "      <td>10353.0</td>\n",
       "      <td>rs555500075</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6</td>\n",
       "      <td>0.788298</td>\n",
       "      <td>0.007978</td>\n",
       "      <td>0.029746</td>\n",
       "      <td>rs558604819</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10642</td>\n",
       "      <td>A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:10642</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10642.0</td>\n",
       "      <td>10643.0</td>\n",
       "      <td>rs558604819</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>0.917844</td>\n",
       "      <td>-0.002636</td>\n",
       "      <td>0.025539</td>\n",
       "      <td>rs544419019</td>\n",
       "      <td>chr1</td>\n",
       "      <td>11012</td>\n",
       "      <td>C/G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:11012</td>\n",
       "      <td>chr1</td>\n",
       "      <td>11012.0</td>\n",
       "      <td>11013.0</td>\n",
       "      <td>rs544419019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>0.787689</td>\n",
       "      <td>0.008319</td>\n",
       "      <td>0.030927</td>\n",
       "      <td>rs574746232</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13011</td>\n",
       "      <td>G/T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>1:13011</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13011.0</td>\n",
       "      <td>13012.0</td>\n",
       "      <td>rs574746232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>11</td>\n",
       "      <td>0.820159</td>\n",
       "      <td>0.007291</td>\n",
       "      <td>0.032136</td>\n",
       "      <td>rs540538026</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13110</td>\n",
       "      <td>A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:13110</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13110.0</td>\n",
       "      <td>13111.0</td>\n",
       "      <td>rs540538026</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   all.data.id    pvalue    effect    stderr   assay.name scaffold position  \\\n",
       "0            3  0.771412 -0.009939  0.034239  rs555500075     chr1    10352   \n",
       "1            6  0.788298  0.007978  0.029746  rs558604819     chr1    10642   \n",
       "2            8  0.917844 -0.002636  0.025539  rs544419019     chr1    11012   \n",
       "3           10  0.787689  0.008319  0.030927  rs574746232     chr1    13011   \n",
       "4           11  0.820159  0.007291  0.032136  rs540538026     chr1    13110   \n",
       "\n",
       "  alleles alleleA alleleB chr    chrbp   CHR  BP_start   BP_end         rsid  \n",
       "0     D/I       D       I   1  1:10352  chr1   10352.0  10353.0  rs555500075  \n",
       "1     A/G       A       G   1  1:10642  chr1   10642.0  10643.0  rs558604819  \n",
       "2     C/G       C       G   1  1:11012  chr1   11012.0  11013.0  rs544419019  \n",
       "3     G/T       G       T   1  1:13011  chr1   13011.0  13012.0  rs574746232  \n",
       "4     A/G       A       G   1  1:13110  chr1   13110.0  13111.0  rs540538026  "
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged = pd.merge(Chang_snps_pass_formatted, liftover, left_on='assay.name', right_on='rsid', how = 'left')\n",
    "merged.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Int64Index: 12896220 entries, 0 to 12896219\n",
      "Data columns (total 16 columns):\n",
      " #   Column       Dtype  \n",
      "---  ------       -----  \n",
      " 0   all.data.id  int64  \n",
      " 1   pvalue       float64\n",
      " 2   effect       float64\n",
      " 3   stderr       float64\n",
      " 4   assay.name   object \n",
      " 5   scaffold     object \n",
      " 6   position     object \n",
      " 7   alleles      object \n",
      " 8   alleleA      object \n",
      " 9   alleleB      object \n",
      " 10  chr          object \n",
      " 11  chrbp        object \n",
      " 12  CHR          object \n",
      " 13  BP_start     float64\n",
      " 14  BP_end       float64\n",
      " 15  rsid         object \n",
      "dtypes: float64(5), int64(1), object(10)\n",
      "memory usage: 1.6+ GB\n"
     ]
    }
   ],
   "source": [
    "merged.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all.data.id</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>effect</th>\n",
       "      <th>stderr</th>\n",
       "      <th>assay.name</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>position</th>\n",
       "      <th>alleles</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>chr</th>\n",
       "      <th>chrbp</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP_start</th>\n",
       "      <th>BP_end</th>\n",
       "      <th>rsid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>25085</th>\n",
       "      <td>29157</td>\n",
       "      <td>0.342646</td>\n",
       "      <td>-0.098071</td>\n",
       "      <td>0.102126</td>\n",
       "      <td>rs560097800</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1584097</td>\n",
       "      <td>C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>1:1584097</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25086</th>\n",
       "      <td>29158</td>\n",
       "      <td>0.631880</td>\n",
       "      <td>0.020462</td>\n",
       "      <td>0.042813</td>\n",
       "      <td>rs28525262</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1584102</td>\n",
       "      <td>A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:1584102</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25087</th>\n",
       "      <td>29160</td>\n",
       "      <td>0.390288</td>\n",
       "      <td>0.022722</td>\n",
       "      <td>0.026506</td>\n",
       "      <td>rs4648756</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1584117</td>\n",
       "      <td>C/G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:1584117</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25090</th>\n",
       "      <td>29163</td>\n",
       "      <td>0.266015</td>\n",
       "      <td>0.028806</td>\n",
       "      <td>0.025969</td>\n",
       "      <td>rs4648607</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1584218</td>\n",
       "      <td>C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>1:1584218</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25091</th>\n",
       "      <td>29164</td>\n",
       "      <td>0.178164</td>\n",
       "      <td>0.033290</td>\n",
       "      <td>0.024796</td>\n",
       "      <td>rs4648757</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1584224</td>\n",
       "      <td>C/G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:1584224</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12198266</th>\n",
       "      <td>14781142</td>\n",
       "      <td>0.921972</td>\n",
       "      <td>-0.002974</td>\n",
       "      <td>0.030403</td>\n",
       "      <td>rs34433551</td>\n",
       "      <td>chr4</td>\n",
       "      <td>9556102</td>\n",
       "      <td>C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4</td>\n",
       "      <td>4:9556102</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12198267</th>\n",
       "      <td>14781143</td>\n",
       "      <td>0.948783</td>\n",
       "      <td>0.002074</td>\n",
       "      <td>0.032316</td>\n",
       "      <td>rs192467273</td>\n",
       "      <td>chr4</td>\n",
       "      <td>9556105</td>\n",
       "      <td>G/T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>4</td>\n",
       "      <td>4:9556105</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12198520</th>\n",
       "      <td>14781438</td>\n",
       "      <td>0.980380</td>\n",
       "      <td>-0.003491</td>\n",
       "      <td>0.142193</td>\n",
       "      <td>rs34899372</td>\n",
       "      <td>chr4</td>\n",
       "      <td>9561627</td>\n",
       "      <td>C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4</td>\n",
       "      <td>4:9561627</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12198539</th>\n",
       "      <td>14781458</td>\n",
       "      <td>0.493654</td>\n",
       "      <td>-0.180768</td>\n",
       "      <td>0.269673</td>\n",
       "      <td>rs2061994</td>\n",
       "      <td>chr4</td>\n",
       "      <td>9561927</td>\n",
       "      <td>C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4</td>\n",
       "      <td>4:9561927</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12198743</th>\n",
       "      <td>14781685</td>\n",
       "      <td>0.683686</td>\n",
       "      <td>-0.015648</td>\n",
       "      <td>0.038335</td>\n",
       "      <td>rs59140157</td>\n",
       "      <td>chr4</td>\n",
       "      <td>9565999</td>\n",
       "      <td>D/I</td>\n",
       "      <td>D</td>\n",
       "      <td>I</td>\n",
       "      <td>4</td>\n",
       "      <td>4:9565999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1759 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          all.data.id    pvalue    effect    stderr   assay.name scaffold  \\\n",
       "25085           29157  0.342646 -0.098071  0.102126  rs560097800     chr1   \n",
       "25086           29158  0.631880  0.020462  0.042813   rs28525262     chr1   \n",
       "25087           29160  0.390288  0.022722  0.026506    rs4648756     chr1   \n",
       "25090           29163  0.266015  0.028806  0.025969    rs4648607     chr1   \n",
       "25091           29164  0.178164  0.033290  0.024796    rs4648757     chr1   \n",
       "...               ...       ...       ...       ...          ...      ...   \n",
       "12198266     14781142  0.921972 -0.002974  0.030403   rs34433551     chr4   \n",
       "12198267     14781143  0.948783  0.002074  0.032316  rs192467273     chr4   \n",
       "12198520     14781438  0.980380 -0.003491  0.142193   rs34899372     chr4   \n",
       "12198539     14781458  0.493654 -0.180768  0.269673    rs2061994     chr4   \n",
       "12198743     14781685  0.683686 -0.015648  0.038335   rs59140157     chr4   \n",
       "\n",
       "         position alleles alleleA alleleB chr      chrbp  CHR  BP_start  \\\n",
       "25085     1584097     C/T       C       T   1  1:1584097  NaN       NaN   \n",
       "25086     1584102     A/G       A       G   1  1:1584102  NaN       NaN   \n",
       "25087     1584117     C/G       C       G   1  1:1584117  NaN       NaN   \n",
       "25090     1584218     C/T       C       T   1  1:1584218  NaN       NaN   \n",
       "25091     1584224     C/G       C       G   1  1:1584224  NaN       NaN   \n",
       "...           ...     ...     ...     ...  ..        ...  ...       ...   \n",
       "12198266  9556102     C/T       C       T   4  4:9556102  NaN       NaN   \n",
       "12198267  9556105     G/T       G       T   4  4:9556105  NaN       NaN   \n",
       "12198520  9561627     C/T       C       T   4  4:9561627  NaN       NaN   \n",
       "12198539  9561927     C/T       C       T   4  4:9561927  NaN       NaN   \n",
       "12198743  9565999     D/I       D       I   4  4:9565999  NaN       NaN   \n",
       "\n",
       "          BP_end rsid  \n",
       "25085        NaN  NaN  \n",
       "25086        NaN  NaN  \n",
       "25087        NaN  NaN  \n",
       "25090        NaN  NaN  \n",
       "25091        NaN  NaN  \n",
       "...          ...  ...  \n",
       "12198266     NaN  NaN  \n",
       "12198267     NaN  NaN  \n",
       "12198520     NaN  NaN  \n",
       "12198539     NaN  NaN  \n",
       "12198743     NaN  NaN  \n",
       "\n",
       "[1759 rows x 16 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Check SNPs that were not lifted over\n",
    "merged[pd.isnull(merged['BP_start'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Remove SNPs that were not lifted over (and any rows where there are any NAs)\n",
    "merged_nomiss = merged.copy()\n",
    "merged_nomiss = merged_nomiss.dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Int64Index: 12894461 entries, 0 to 12896219\n",
      "Data columns (total 16 columns):\n",
      " #   Column       Dtype  \n",
      "---  ------       -----  \n",
      " 0   all.data.id  int64  \n",
      " 1   pvalue       float64\n",
      " 2   effect       float64\n",
      " 3   stderr       float64\n",
      " 4   assay.name   object \n",
      " 5   scaffold     object \n",
      " 6   position     object \n",
      " 7   alleles      object \n",
      " 8   alleleA      object \n",
      " 9   alleleB      object \n",
      " 10  chr          object \n",
      " 11  chrbp        object \n",
      " 12  CHR          object \n",
      " 13  BP_start     float64\n",
      " 14  BP_end       float64\n",
      " 15  rsid         object \n",
      "dtypes: float64(5), int64(1), object(10)\n",
      "memory usage: 1.6+ GB\n"
     ]
    }
   ],
   "source": [
    "merged_nomiss.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all.data.id</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>effect</th>\n",
       "      <th>stderr</th>\n",
       "      <th>assay.name</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>position</th>\n",
       "      <th>alleles</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>chr</th>\n",
       "      <th>chrbp</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP_start</th>\n",
       "      <th>BP_end</th>\n",
       "      <th>rsid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>0.771412</td>\n",
       "      <td>-0.009939</td>\n",
       "      <td>0.034239</td>\n",
       "      <td>rs555500075</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10352</td>\n",
       "      <td>D/I</td>\n",
       "      <td>D</td>\n",
       "      <td>I</td>\n",
       "      <td>1</td>\n",
       "      <td>1:10352</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10352.0</td>\n",
       "      <td>10353.0</td>\n",
       "      <td>rs555500075</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6</td>\n",
       "      <td>0.788298</td>\n",
       "      <td>0.007978</td>\n",
       "      <td>0.029746</td>\n",
       "      <td>rs558604819</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10642</td>\n",
       "      <td>A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:10642</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10642.0</td>\n",
       "      <td>10643.0</td>\n",
       "      <td>rs558604819</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8</td>\n",
       "      <td>0.917844</td>\n",
       "      <td>-0.002636</td>\n",
       "      <td>0.025539</td>\n",
       "      <td>rs544419019</td>\n",
       "      <td>chr1</td>\n",
       "      <td>11012</td>\n",
       "      <td>C/G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:11012</td>\n",
       "      <td>chr1</td>\n",
       "      <td>11012.0</td>\n",
       "      <td>11013.0</td>\n",
       "      <td>rs544419019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>0.787689</td>\n",
       "      <td>0.008319</td>\n",
       "      <td>0.030927</td>\n",
       "      <td>rs574746232</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13011</td>\n",
       "      <td>G/T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1</td>\n",
       "      <td>1:13011</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13011.0</td>\n",
       "      <td>13012.0</td>\n",
       "      <td>rs574746232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>11</td>\n",
       "      <td>0.820159</td>\n",
       "      <td>0.007291</td>\n",
       "      <td>0.032136</td>\n",
       "      <td>rs540538026</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13110</td>\n",
       "      <td>A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1:13110</td>\n",
       "      <td>chr1</td>\n",
       "      <td>13110.0</td>\n",
       "      <td>13111.0</td>\n",
       "      <td>rs540538026</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   all.data.id    pvalue    effect    stderr   assay.name scaffold position  \\\n",
       "0            3  0.771412 -0.009939  0.034239  rs555500075     chr1    10352   \n",
       "1            6  0.788298  0.007978  0.029746  rs558604819     chr1    10642   \n",
       "2            8  0.917844 -0.002636  0.025539  rs544419019     chr1    11012   \n",
       "3           10  0.787689  0.008319  0.030927  rs574746232     chr1    13011   \n",
       "4           11  0.820159  0.007291  0.032136  rs540538026     chr1    13110   \n",
       "\n",
       "  alleles alleleA alleleB chr    chrbp   CHR  BP_start   BP_end         rsid  \n",
       "0     D/I       D       I   1  1:10352  chr1   10352.0  10353.0  rs555500075  \n",
       "1     A/G       A       G   1  1:10642  chr1   10642.0  10643.0  rs558604819  \n",
       "2     C/G       C       G   1  1:11012  chr1   11012.0  11013.0  rs544419019  \n",
       "3     G/T       G       T   1  1:13011  chr1   13011.0  13012.0  rs574746232  \n",
       "4     A/G       A       G   1  1:13110  chr1   13110.0  13111.0  rs540538026  "
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_nomiss.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Format Chang SNP names to harmonise between sumstats and target data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "In the Chang2017 sumstats, make a new column for the SNP name which is chr:bp - to match AMP data\n",
    "(We are not naming SNPs with as chr:bp:A2:A1 because the order of the alleles is not the same in the Chang data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "merged_formatted = merged_nomiss.copy()\n",
    "\n",
    "#First make a column for chr which is just the chromosome number. Convert to string format\n",
    "merged_formatted['chr'] = merged_formatted['scaffold'].str.replace(r'chr', '').astype(str)\n",
    "\n",
    "#Round BP_start to remove decimal places\n",
    "merged_formatted[\"BP_start\"] = merged_formatted[\"BP_start\"].round().astype(int)\n",
    "\n",
    "#Make BP str format\n",
    "merged_formatted[\"BP_start\"] = merged_formatted[\"BP_start\"].astype(str)\n",
    "\n",
    "#Make combined column chrbp\n",
    "merged_formatted['chrbp'] = merged_formatted['chr'] + \":\" + merged_formatted[\"BP_start\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all.data.id</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>effect</th>\n",
       "      <th>stderr</th>\n",
       "      <th>assay.name</th>\n",
       "      <th>scaffold</th>\n",
       "      <th>position</th>\n",
       "      <th>alleles</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>chr</th>\n",
       "      <th>chrbp</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP_start</th>\n",
       "      <th>BP_end</th>\n",
       "      <th>rsid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12896215</th>\n",
       "      <td>15630329</td>\n",
       "      <td>0.341476</td>\n",
       "      <td>-0.019517</td>\n",
       "      <td>0.020535</td>\n",
       "      <td>rs535845793</td>\n",
       "      <td>chr4</td>\n",
       "      <td>44604765</td>\n",
       "      <td>A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>4</td>\n",
       "      <td>4:44602748</td>\n",
       "      <td>chr4</td>\n",
       "      <td>44602748</td>\n",
       "      <td>44602749.0</td>\n",
       "      <td>rs535845793</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12896216</th>\n",
       "      <td>15630330</td>\n",
       "      <td>0.419701</td>\n",
       "      <td>-0.021318</td>\n",
       "      <td>0.026467</td>\n",
       "      <td>rs566170966</td>\n",
       "      <td>chr4</td>\n",
       "      <td>44604776</td>\n",
       "      <td>C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4</td>\n",
       "      <td>4:44602759</td>\n",
       "      <td>chr4</td>\n",
       "      <td>44602759</td>\n",
       "      <td>44602760.0</td>\n",
       "      <td>rs566170966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12896217</th>\n",
       "      <td>15630331</td>\n",
       "      <td>0.964053</td>\n",
       "      <td>0.003946</td>\n",
       "      <td>0.087707</td>\n",
       "      <td>rs539975122</td>\n",
       "      <td>chr4</td>\n",
       "      <td>44604782</td>\n",
       "      <td>C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4</td>\n",
       "      <td>4:44602765</td>\n",
       "      <td>chr4</td>\n",
       "      <td>44602765</td>\n",
       "      <td>44602766.0</td>\n",
       "      <td>rs539975122</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12896218</th>\n",
       "      <td>15630335</td>\n",
       "      <td>0.785134</td>\n",
       "      <td>-0.007339</td>\n",
       "      <td>0.026939</td>\n",
       "      <td>rs556428381</td>\n",
       "      <td>chr4</td>\n",
       "      <td>44604876</td>\n",
       "      <td>A/G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>4</td>\n",
       "      <td>4:44602859</td>\n",
       "      <td>chr4</td>\n",
       "      <td>44602859</td>\n",
       "      <td>44602860.0</td>\n",
       "      <td>rs556428381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12896219</th>\n",
       "      <td>15630336</td>\n",
       "      <td>0.329613</td>\n",
       "      <td>0.019403</td>\n",
       "      <td>0.019921</td>\n",
       "      <td>rs190343291</td>\n",
       "      <td>chr4</td>\n",
       "      <td>44604897</td>\n",
       "      <td>C/T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>4</td>\n",
       "      <td>4:44602880</td>\n",
       "      <td>chr4</td>\n",
       "      <td>44602880</td>\n",
       "      <td>44602881.0</td>\n",
       "      <td>rs190343291</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          all.data.id    pvalue    effect    stderr   assay.name scaffold  \\\n",
       "12896215     15630329  0.341476 -0.019517  0.020535  rs535845793     chr4   \n",
       "12896216     15630330  0.419701 -0.021318  0.026467  rs566170966     chr4   \n",
       "12896217     15630331  0.964053  0.003946  0.087707  rs539975122     chr4   \n",
       "12896218     15630335  0.785134 -0.007339  0.026939  rs556428381     chr4   \n",
       "12896219     15630336  0.329613  0.019403  0.019921  rs190343291     chr4   \n",
       "\n",
       "          position alleles alleleA alleleB chr       chrbp   CHR  BP_start  \\\n",
       "12896215  44604765     A/G       A       G   4  4:44602748  chr4  44602748   \n",
       "12896216  44604776     C/T       C       T   4  4:44602759  chr4  44602759   \n",
       "12896217  44604782     C/T       C       T   4  4:44602765  chr4  44602765   \n",
       "12896218  44604876     A/G       A       G   4  4:44602859  chr4  44602859   \n",
       "12896219  44604897     C/T       C       T   4  4:44602880  chr4  44602880   \n",
       "\n",
       "              BP_end         rsid  \n",
       "12896215  44602749.0  rs535845793  \n",
       "12896216  44602760.0  rs566170966  \n",
       "12896217  44602766.0  rs539975122  \n",
       "12896218  44602860.0  rs556428381  \n",
       "12896219  44602881.0  rs190343291  "
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_formatted.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrbp</th>\n",
       "      <th>chr</th>\n",
       "      <th>BP_start</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>effect</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:10352</td>\n",
       "      <td>1</td>\n",
       "      <td>10352</td>\n",
       "      <td>I</td>\n",
       "      <td>D</td>\n",
       "      <td>0.771412</td>\n",
       "      <td>-0.009939</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:10642</td>\n",
       "      <td>1</td>\n",
       "      <td>10642</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.788298</td>\n",
       "      <td>0.007978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:11012</td>\n",
       "      <td>1</td>\n",
       "      <td>11012</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.917844</td>\n",
       "      <td>-0.002636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:13011</td>\n",
       "      <td>1</td>\n",
       "      <td>13011</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.787689</td>\n",
       "      <td>0.008319</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:13110</td>\n",
       "      <td>1</td>\n",
       "      <td>13110</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.820159</td>\n",
       "      <td>0.007291</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chrbp chr BP_start alleleB alleleA    pvalue    effect\n",
       "0  1:10352   1    10352       I       D  0.771412 -0.009939\n",
       "1  1:10642   1    10642       G       A  0.788298  0.007978\n",
       "2  1:11012   1    11012       G       C  0.917844 -0.002636\n",
       "3  1:13011   1    13011       T       G  0.787689  0.008319\n",
       "4  1:13110   1    13110       G       A  0.820159  0.007291"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_formatted_export = merged_formatted.copy()\n",
    "merged_formatted_export = merged_formatted_export[['chrbp','chr', 'BP_start', 'alleleB', 'alleleA', 'pvalue', 'effect']]\n",
    "merged_formatted_export.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrbp</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "      <th>A1</th>\n",
       "      <th>A2</th>\n",
       "      <th>P</th>\n",
       "      <th>beta</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:10352</td>\n",
       "      <td>1</td>\n",
       "      <td>10352</td>\n",
       "      <td>I</td>\n",
       "      <td>D</td>\n",
       "      <td>0.771412</td>\n",
       "      <td>-0.009939</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:10642</td>\n",
       "      <td>1</td>\n",
       "      <td>10642</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.788298</td>\n",
       "      <td>0.007978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:11012</td>\n",
       "      <td>1</td>\n",
       "      <td>11012</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.917844</td>\n",
       "      <td>-0.002636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:13011</td>\n",
       "      <td>1</td>\n",
       "      <td>13011</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.787689</td>\n",
       "      <td>0.008319</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:13110</td>\n",
       "      <td>1</td>\n",
       "      <td>13110</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.820159</td>\n",
       "      <td>0.007291</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     chrbp CHR     BP A1 A2         P      beta\n",
       "0  1:10352   1  10352  I  D  0.771412 -0.009939\n",
       "1  1:10642   1  10642  G  A  0.788298  0.007978\n",
       "2  1:11012   1  11012  G  C  0.917844 -0.002636\n",
       "3  1:13011   1  13011  T  G  0.787689  0.008319\n",
       "4  1:13110   1  13110  G  A  0.820159  0.007291"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Rename columns\n",
    "\n",
    "#Note that in the Chang summary statistics (read the documentation provided by 23andMe, the effect statistic is for the B allele, the alphabetically higher allele)\n",
    "#So alleleB is going to be A1 - in PRSice A1 is the effect allele\n",
    "merged_formatted_export.columns = ['chrbp', 'CHR', 'BP', 'A1', 'A2', 'P', 'beta']\n",
    "merged_formatted_export.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(12894461, 7)"
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_formatted_export.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(12022304, 7)"
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Remove duplicates by chrbp\n",
    "merged_formatted_export_nodupli = merged_formatted_export.copy()\n",
    "\n",
    "#First remove SNPs with D/I alleles\n",
    "merged_formatted_export_nodupli = merged_formatted_export_nodupli[(merged_formatted_export_nodupli['A1'] !=\"I\") & (merged_formatted_export_nodupli['A1']!=\"D\")]\n",
    "\n",
    "#Now remove duplicate SNPs by chrbp\n",
    "merged_formatted_export_nodupli = merged_formatted_export_nodupli.drop_duplicates(subset = ['chrbp'])\n",
    "merged_formatted_export_nodupli.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "with open('/home/jupyter/sumstats/Chang2017_with23andMe/Chang_hg38_forPRSice.tab', 'w') as f:\n",
    "    f.write(merged_formatted_export_nodupli.to_csv(index=False, sep='\\t'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chrbp\tCHR\tBP\tA1\tA2\tP\tbeta\n",
      "1:10642\t1\t10642\tG\tA\t0.788298\t0.00797757\n",
      "1:11012\t1\t11012\tG\tC\t0.917844\t-0.00263639\n",
      "1:13011\t1\t13011\tT\tG\t0.787689\t0.00831877\n",
      "1:13110\t1\t13110\tG\tA\t0.820159\t0.00729131\n",
      "1:13116\t1\t13116\tT\tG\t0.871642\t-0.00423368\n",
      "1:13118\t1\t13118\tG\tA\t0.814349\t-0.007677\n",
      "1:13284\t1\t13284\tG\tA\t0.766499\t-0.00964793\n",
      "1:13289\t1\t13289\tT\tC\t0.768181\t-0.0095832\n",
      "1:13380\t1\t13380\tG\tC\t0.586043\t0.017903\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head /home/jupyter/sumstats/Chang2017_with23andMe/Chang_hg38_forPRSice.tab"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Prepare AMP-PD data. Rename SNPs to chr:bp"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Format AMP-PD files to have SNP names as chr:bp. This is still on the full AMP-PD data (not separating cohorts yet)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/adaptive_immune.updated_names.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/pathways/adaptive_immune\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/adaptive_immune.updated_names\n",
      "  --update-name /home/jupyter/pathways/adaptive_immune_chrbp.txt\n",
      "\n",
      "Start time: Thu Jun  9 13:27:33 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/pathways/adaptive_immune.fam.\n",
      "108520 variants loaded from /home/jupyter/pathways/adaptive_immune.bim.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/pathways/adaptive_immune.updated_names.fam ... done.\n",
      "Writing /home/jupyter/pathways/adaptive_immune.updated_names.bim ... done.\n",
      "Writing /home/jupyter/pathways/adaptive_immune.updated_names.bed ... 0%\b\b60%\b\b\bdone.\n",
      "End time: Thu Jun  9 13:27:33 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/adaptive_immune.updated_names.no_dupli.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/pathways/adaptive_immune.updated_names\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/adaptive_immune.updated_names.no_dupli\n",
      "  --rm-dup force-first\n",
      "\n",
      "Start time: Thu Jun  9 13:27:33 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/pathways/adaptive_immune.updated_names.fam.\n",
      "108520 variants loaded from\n",
      "/home/jupyter/pathways/adaptive_immune.updated_names.bim.\n",
      "Note: No phenotype data present.\n",
      "Note: Skipping --rm-dup since no duplicate IDs are present.\n",
      "Writing /home/jupyter/pathways/adaptive_immune.updated_names.no_dupli.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/pathways/adaptive_immune.updated_names.no_dupli.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/pathways/adaptive_immune.updated_names.no_dupli.bed ...\n",
      "0%\b\b60%\b\b\bdone.\n",
      "End time: Thu Jun  9 13:27:34 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/alpha_synuclein.updated_names.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/pathways/alpha_synuclein\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/alpha_synuclein.updated_names\n",
      "  --update-name /home/jupyter/pathways/alpha_synuclein_chrbp.txt\n",
      "\n",
      "Start time: Thu Jun  9 13:27:34 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/pathways/alpha_synuclein.fam.\n",
      "10232 variants loaded from /home/jupyter/pathways/alpha_synuclein.bim.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/pathways/alpha_synuclein.updated_names.fam ... done.\n",
      "Writing /home/jupyter/pathways/alpha_synuclein.updated_names.bim ... done.\n",
      "Writing /home/jupyter/pathways/alpha_synuclein.updated_names.bed ... 0%\b\bdone.\n",
      "End time: Thu Jun  9 13:27:34 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/alpha_synuclein.updated_names.no_dupli.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/pathways/alpha_synuclein.updated_names\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/alpha_synuclein.updated_names.no_dupli\n",
      "  --rm-dup force-first\n",
      "\n",
      "Start time: Thu Jun  9 13:27:34 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/pathways/alpha_synuclein.updated_names.fam.\n",
      "10232 variants loaded from\n",
      "/home/jupyter/pathways/alpha_synuclein.updated_names.bim.\n",
      "Note: No phenotype data present.\n",
      "Note: Skipping --rm-dup since no duplicate IDs are present.\n",
      "Writing /home/jupyter/pathways/alpha_synuclein.updated_names.no_dupli.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/pathways/alpha_synuclein.updated_names.no_dupli.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/pathways/alpha_synuclein.updated_names.no_dupli.bed ...\n",
      "0%\b\bdone.\n",
      "End time: Thu Jun  9 13:27:34 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/innate_immune.updated_names.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/pathways/innate_immune\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/innate_immune.updated_names\n",
      "  --update-name /home/jupyter/pathways/innate_immune_chrbp.txt\n",
      "\n",
      "Start time: Thu Jun  9 13:27:34 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/pathways/innate_immune.fam.\n",
      "132840 variants loaded from /home/jupyter/pathways/innate_immune.bim.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/pathways/innate_immune.updated_names.fam ... done.\n",
      "Writing /home/jupyter/pathways/innate_immune.updated_names.bim ... done.\n",
      "Writing /home/jupyter/pathways/innate_immune.updated_names.bed ... 0%\b\b49%\b\b\b98%\b\b\bdone.\n",
      "End time: Thu Jun  9 13:27:34 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/innate_immune.updated_names.no_dupli.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/pathways/innate_immune.updated_names\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/innate_immune.updated_names.no_dupli\n",
      "  --rm-dup force-first\n",
      "\n",
      "Start time: Thu Jun  9 13:27:34 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/pathways/innate_immune.updated_names.fam.\n",
      "132840 variants loaded from\n",
      "/home/jupyter/pathways/innate_immune.updated_names.bim.\n",
      "Note: No phenotype data present.\n",
      "Note: Skipping --rm-dup since no duplicate IDs are present.\n",
      "Writing /home/jupyter/pathways/innate_immune.updated_names.no_dupli.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/pathways/innate_immune.updated_names.no_dupli.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/pathways/innate_immune.updated_names.no_dupli.bed ...\n",
      "0%\b\b49%\b\b\b98%\b\b\bdone.\n",
      "End time: Thu Jun  9 13:27:34 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/lysosome.updated_names.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/pathways/lysosome\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/lysosome.updated_names\n",
      "  --update-name /home/jupyter/pathways/lysosome_chrbp.txt\n",
      "\n",
      "Start time: Thu Jun  9 13:27:34 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/pathways/lysosome.fam.\n",
      "16641 variants loaded from /home/jupyter/pathways/lysosome.bim.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/pathways/lysosome.updated_names.fam ... done.\n",
      "Writing /home/jupyter/pathways/lysosome.updated_names.bim ... done.\n",
      "Writing /home/jupyter/pathways/lysosome.updated_names.bed ... 0%\b\bdone.\n",
      "End time: Thu Jun  9 13:27:34 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/lysosome.updated_names.no_dupli.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/pathways/lysosome.updated_names\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/lysosome.updated_names.no_dupli\n",
      "  --rm-dup force-first\n",
      "\n",
      "Start time: Thu Jun  9 13:27:34 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/pathways/lysosome.updated_names.fam.\n",
      "16641 variants loaded from /home/jupyter/pathways/lysosome.updated_names.bim.\n",
      "Note: No phenotype data present.\n",
      "Note: Skipping --rm-dup since no duplicate IDs are present.\n",
      "Writing /home/jupyter/pathways/lysosome.updated_names.no_dupli.fam ... done.\n",
      "Writing /home/jupyter/pathways/lysosome.updated_names.no_dupli.bim ... done.\n",
      "Writing /home/jupyter/pathways/lysosome.updated_names.no_dupli.bed ... 0%\b\bdone.\n",
      "End time: Thu Jun  9 13:27:34 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/endocytosis.updated_names.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/pathways/endocytosis\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/endocytosis.updated_names\n",
      "  --update-name /home/jupyter/pathways/endocytosis_chrbp.txt\n",
      "\n",
      "Start time: Thu Jun  9 13:27:34 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/pathways/endocytosis.fam.\n",
      "46779 variants loaded from /home/jupyter/pathways/endocytosis.bim.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/pathways/endocytosis.updated_names.fam ... done.\n",
      "Writing /home/jupyter/pathways/endocytosis.updated_names.bim ... done.\n",
      "Writing /home/jupyter/pathways/endocytosis.updated_names.bed ... 0%\b\bdone.\n",
      "End time: Thu Jun  9 13:27:34 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/endocytosis.updated_names.no_dupli.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/pathways/endocytosis.updated_names\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/endocytosis.updated_names.no_dupli\n",
      "  --rm-dup force-first\n",
      "\n",
      "Start time: Thu Jun  9 13:27:34 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/pathways/endocytosis.updated_names.fam.\n",
      "46779 variants loaded from\n",
      "/home/jupyter/pathways/endocytosis.updated_names.bim.\n",
      "Note: No phenotype data present.\n",
      "Note: Skipping --rm-dup since no duplicate IDs are present.\n",
      "Writing /home/jupyter/pathways/endocytosis.updated_names.no_dupli.fam ... done.\n",
      "Writing /home/jupyter/pathways/endocytosis.updated_names.no_dupli.bim ... done.\n",
      "Writing /home/jupyter/pathways/endocytosis.updated_names.no_dupli.bed ... 0%\b\bdone.\n",
      "End time: Thu Jun  9 13:27:34 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/mitochondria.updated_names.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/pathways/mitochondria\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/mitochondria.updated_names\n",
      "  --update-name /home/jupyter/pathways/mitochondria_chrbp.txt\n",
      "\n",
      "Start time: Thu Jun  9 13:27:34 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/pathways/mitochondria.fam.\n",
      "254020 variants loaded from /home/jupyter/pathways/mitochondria.bim.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/pathways/mitochondria.updated_names.fam ... done.\n",
      "Writing /home/jupyter/pathways/mitochondria.updated_names.bim ... done.\n",
      "Writing /home/jupyter/pathways/mitochondria.updated_names.bed ... 0%\b\b25%\b\b\b51%\b\b\b77%\b\b\bdone.\n",
      "End time: Thu Jun  9 13:27:35 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/mitochondria.updated_names.no_dupli.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/pathways/mitochondria.updated_names\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/mitochondria.updated_names.no_dupli\n",
      "  --rm-dup force-first\n",
      "\n",
      "Start time: Thu Jun  9 13:27:35 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/pathways/mitochondria.updated_names.fam.\n",
      "254020 variants loaded from\n",
      "/home/jupyter/pathways/mitochondria.updated_names.bim.\n",
      "Note: No phenotype data present.\n",
      "Note: Skipping --rm-dup since no duplicate IDs are present.\n",
      "Writing /home/jupyter/pathways/mitochondria.updated_names.no_dupli.fam ...\n",
      "done.\n",
      "Writing /home/jupyter/pathways/mitochondria.updated_names.no_dupli.bim ...\n",
      "done.\n",
      "Writing /home/jupyter/pathways/mitochondria.updated_names.no_dupli.bed ...\n",
      "0%\b\b25%\b\b\b51%\b\b\b77%\b\b\bdone.\n",
      "End time: Thu Jun  9 13:27:35 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/microglia.updated_names.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/pathways/microglia\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/microglia.updated_names\n",
      "  --update-name /home/jupyter/pathways/microglia_chrbp.txt\n",
      "\n",
      "Start time: Thu Jun  9 13:27:35 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/pathways/microglia.fam.\n",
      "67660 variants loaded from /home/jupyter/pathways/microglia.bim.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/pathways/microglia.updated_names.fam ... done.\n",
      "Writing /home/jupyter/pathways/microglia.updated_names.bim ... done.\n",
      "Writing /home/jupyter/pathways/microglia.updated_names.bed ... 0%\b\b96%\b\b\bdone.\n",
      "End time: Thu Jun  9 13:27:35 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/microglia.updated_names.no_dupli.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/pathways/microglia.updated_names\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/microglia.updated_names.no_dupli\n",
      "  --rm-dup force-first\n",
      "\n",
      "Start time: Thu Jun  9 13:27:35 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/pathways/microglia.updated_names.fam.\n",
      "67660 variants loaded from /home/jupyter/pathways/microglia.updated_names.bim.\n",
      "Note: No phenotype data present.\n",
      "Note: Skipping --rm-dup since no duplicate IDs are present.\n",
      "Writing /home/jupyter/pathways/microglia.updated_names.no_dupli.fam ... done.\n",
      "Writing /home/jupyter/pathways/microglia.updated_names.no_dupli.bim ... done.\n",
      "Writing /home/jupyter/pathways/microglia.updated_names.no_dupli.bed ... 0%\b\b96%\b\b\bdone.\n",
      "End time: Thu Jun  9 13:27:35 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/monocytes.updated_names.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/pathways/monocytes\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/monocytes.updated_names\n",
      "  --update-name /home/jupyter/pathways/monocytes_chrbp.txt\n",
      "\n",
      "Start time: Thu Jun  9 13:27:35 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/pathways/monocytes.fam.\n",
      "59618 variants loaded from /home/jupyter/pathways/monocytes.bim.\n",
      "Note: No phenotype data present.\n",
      "Writing /home/jupyter/pathways/monocytes.updated_names.fam ... done.\n",
      "Writing /home/jupyter/pathways/monocytes.updated_names.bim ... done.\n",
      "Writing /home/jupyter/pathways/monocytes.updated_names.bed ... 0%\b\bdone.\n",
      "End time: Thu Jun  9 13:27:35 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/pathways/monocytes.updated_names.no_dupli.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/pathways/monocytes.updated_names\n",
      "  --make-bed\n",
      "  --out /home/jupyter/pathways/monocytes.updated_names.no_dupli\n",
      "  --rm-dup force-first\n",
      "\n",
      "Start time: Thu Jun  9 13:27:35 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/pathways/monocytes.updated_names.fam.\n",
      "59618 variants loaded from /home/jupyter/pathways/monocytes.updated_names.bim.\n",
      "Note: No phenotype data present.\n",
      "Note: Skipping --rm-dup since no duplicate IDs are present.\n",
      "Writing /home/jupyter/pathways/monocytes.updated_names.no_dupli.fam ... done.\n",
      "Writing /home/jupyter/pathways/monocytes.updated_names.no_dupli.bim ... done.\n",
      "Writing /home/jupyter/pathways/monocytes.updated_names.no_dupli.bed ... 0%\b\bdone.\n",
      "End time: Thu Jun  9 13:27:35 2022\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /home/jupyter/pathways/\n",
    "\n",
    "for PATHWAY in adaptive_immune alpha_synuclein innate_immune lysosome endocytosis mitochondria microglia monocytes\n",
    "do\n",
    "\n",
    "    #Make a new file to rename SNPs in the AMP-PD dataset - rename with just chr:bp\n",
    "    awk '{ print $2\"\\t\"$1\":\"$4}' \"$PATHWAY\".bim > \"$PATHWAY\"_chrbp.txt\n",
    "\n",
    "    #Update names in plink\n",
    "    /home/jupyter/plink2 --bfile /home/jupyter/pathways/\"$PATHWAY\" \\\n",
    "    --update-name /home/jupyter/pathways/\"$PATHWAY\"_chrbp.txt \\\n",
    "    --make-bed \\\n",
    "    --out /home/jupyter/pathways/\"$PATHWAY\".updated_names\n",
    "\n",
    "    #Remove duplicate SNPs in plink\n",
    "    /home/jupyter/plink2 --bfile /home/jupyter/pathways/\"$PATHWAY\".updated_names \\\n",
    "    --rm-dup force-first \\\n",
    "    --make-bed \\\n",
    "    --out /home/jupyter/pathways/\"$PATHWAY\".updated_names.no_dupli\n",
    "    \n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Run PRSice2: Merging by chr:bp. Loop over all pathways"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Run PRSice2 without p-value optimisation, just make PRSs at set thresholds"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "The question is should we be merging the base and target datasets by SNP names including A1 and A2, or just chr:bp? PRSice says that strand flips are automatically detected and accounted for. I have tried both options and there are more SNPs overlapping between base and target datasets when using chr:bp"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "This is run in the full AMP-PD dataset so there are enough samples for clumping (recommended N > 500 - see https://www.prsice.info/command_detail/#clumping). If we separated out the cohorts before this stage, some cohorts would not be large enough for estimating LD for clumping internally. The PRSs are calculated independently for each sample and there are the same SNPs in the genetic data in each cohort. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "**Now using updated Chang2017 sumstats including 23andMe from Sara**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {
    "hidden": true,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n",
      "Begin plotting\n",
      "Current Rscript version = 2.3.3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-04-25 12:31:29\n",
      "/home/jupyter/PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base /home/jupyter/sumstats/Chang2017_with23andMe_Sara/META4_hg38.tab \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --bp BP \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --interval 5e-08 \\\n",
      "    --lower 5e-08 \\\n",
      "    --no-regress  \\\n",
      "    --num-auto 22 \\\n",
      "    --out /home/jupyter/pathways/adaptive_immune_PD_PRS_Chang23andMe \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --seed 1709566672 \\\n",
      "    --snp chrbp \\\n",
      "    --stat BETA \\\n",
      "    --target /home/jupyter/pathways/adaptive_immune.updated_names.no_dupli \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.05\n",
      "\n",
      "Initializing Genotype file: \n",
      "/home/jupyter/pathways/adaptive_immune.updated_names.no_dupli \n",
      "(bed) \n",
      "\n",
      "Start processing META4_hg38 \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "/home/jupyter/sumstats/Chang2017_with23andMe_Sara/META4_hg38.tab \n",
      "Header of file is: \n",
      "chrbp\tCHR\tBP\tA1\tA2\tP\tBETA \n",
      "\n",
      "Reading 100.00%\n",
      "7907818 variant(s) observed in base file, with: \n",
      "1173043 ambiguous variant(s) excluded \n",
      "6734775 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "2610 people (1659 male(s), 951 female(s)) observed \n",
      "2610 founder(s) included \n",
      "\n",
      "31084 variant(s) not found in previous data \n",
      "3 variant(s) with mismatch information \n",
      "77433 variant(s) included \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 6656 \n",
      "\n",
      "Start calculating the scores \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-04-25 12:32:44\n",
      "/home/jupyter/PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base /home/jupyter/sumstats/Chang2017_with23andMe_Sara/META4_hg38.tab \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --bp BP \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --interval 5e-08 \\\n",
      "    --lower 5e-08 \\\n",
      "    --no-regress  \\\n",
      "    --num-auto 22 \\\n",
      "    --out /home/jupyter/pathways/alpha_synuclein_PD_PRS_Chang23andMe \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --seed 1850210110 \\\n",
      "    --snp chrbp \\\n",
      "    --stat BETA \\\n",
      "    --target /home/jupyter/pathways/alpha_synuclein.updated_names.no_dupli \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.05\n",
      "\n",
      "Initializing Genotype file: \n",
      "/home/jupyter/pathways/alpha_synuclein.updated_names.no_dupli \n",
      "(bed) \n",
      "\n",
      "Start processing META4_hg38 \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "/home/jupyter/sumstats/Chang2017_with23andMe_Sara/META4_hg38.tab \n",
      "Header of file is: \n",
      "chrbp\tCHR\tBP\tA1\tA2\tP\tBETA \n",
      "\n",
      "Reading 100.00%\n",
      "7907818 variant(s) observed in base file, with: \n",
      "1173043 ambiguous variant(s) excluded \n",
      "6734775 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "2610 people (1659 male(s), 951 female(s)) observed \n",
      "2610 founder(s) included \n",
      "\n",
      "2458 variant(s) not found in previous data \n",
      "7774 variant(s) included \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 512 \n",
      "\n",
      "Start calculating the scores \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-04-25 12:33:50\n",
      "/home/jupyter/PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base /home/jupyter/sumstats/Chang2017_with23andMe_Sara/META4_hg38.tab \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --bp BP \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --interval 5e-08 \\\n",
      "    --lower 5e-08 \\\n",
      "    --no-regress  \\\n",
      "    --num-auto 22 \\\n",
      "    --out /home/jupyter/pathways/innate_immune_PD_PRS_Chang23andMe \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --seed 3852270234 \\\n",
      "    --snp chrbp \\\n",
      "    --stat BETA \\\n",
      "    --target /home/jupyter/pathways/innate_immune.updated_names.no_dupli \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.05\n",
      "\n",
      "Initializing Genotype file: \n",
      "/home/jupyter/pathways/innate_immune.updated_names.no_dupli \n",
      "(bed) \n",
      "\n",
      "Start processing META4_hg38 \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "/home/jupyter/sumstats/Chang2017_with23andMe_Sara/META4_hg38.tab \n",
      "Header of file is: \n",
      "chrbp\tCHR\tBP\tA1\tA2\tP\tBETA \n",
      "\n",
      "Reading 100.00%\n",
      "7907818 variant(s) observed in base file, with: \n",
      "1173043 ambiguous variant(s) excluded \n",
      "6734775 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "2610 people (1659 male(s), 951 female(s)) observed \n",
      "2610 founder(s) included \n",
      "\n",
      "37210 variant(s) not found in previous data \n",
      "4 variant(s) with mismatch information \n",
      "95626 variant(s) included \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 8676 \n",
      "\n",
      "Start calculating the scores \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-04-25 12:35:09\n",
      "/home/jupyter/PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base /home/jupyter/sumstats/Chang2017_with23andMe_Sara/META4_hg38.tab \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --bp BP \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --interval 5e-08 \\\n",
      "    --lower 5e-08 \\\n",
      "    --no-regress  \\\n",
      "    --num-auto 22 \\\n",
      "    --out /home/jupyter/pathways/lysosome_PD_PRS_Chang23andMe \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --seed 1002576195 \\\n",
      "    --snp chrbp \\\n",
      "    --stat BETA \\\n",
      "    --target /home/jupyter/pathways/lysosome.updated_names.no_dupli \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.05\n",
      "\n",
      "Initializing Genotype file: \n",
      "/home/jupyter/pathways/lysosome.updated_names.no_dupli \n",
      "(bed) \n",
      "\n",
      "Start processing META4_hg38 \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "/home/jupyter/sumstats/Chang2017_with23andMe_Sara/META4_hg38.tab \n",
      "Header of file is: \n",
      "chrbp\tCHR\tBP\tA1\tA2\tP\tBETA \n",
      "\n",
      "Reading 100.00%\n",
      "7907818 variant(s) observed in base file, with: \n",
      "1173043 ambiguous variant(s) excluded \n",
      "6734775 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "2610 people (1659 male(s), 951 female(s)) observed \n",
      "2610 founder(s) included \n",
      "\n",
      "4592 variant(s) not found in previous data \n",
      "12049 variant(s) included \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 1024 \n",
      "\n",
      "Start calculating the scores \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-04-25 12:36:16\n",
      "/home/jupyter/PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base /home/jupyter/sumstats/Chang2017_with23andMe_Sara/META4_hg38.tab \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --bp BP \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --interval 5e-08 \\\n",
      "    --lower 5e-08 \\\n",
      "    --no-regress  \\\n",
      "    --num-auto 22 \\\n",
      "    --out /home/jupyter/pathways/endocytosis_PD_PRS_Chang23andMe \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --seed 744243681 \\\n",
      "    --snp chrbp \\\n",
      "    --stat BETA \\\n",
      "    --target /home/jupyter/pathways/endocytosis.updated_names.no_dupli \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.05\n",
      "\n",
      "Initializing Genotype file: \n",
      "/home/jupyter/pathways/endocytosis.updated_names.no_dupli \n",
      "(bed) \n",
      "\n",
      "Start processing META4_hg38 \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "/home/jupyter/sumstats/Chang2017_with23andMe_Sara/META4_hg38.tab \n",
      "Header of file is: \n",
      "chrbp\tCHR\tBP\tA1\tA2\tP\tBETA \n",
      "\n",
      "Reading 100.00%\n",
      "7907818 variant(s) observed in base file, with: \n",
      "1173043 ambiguous variant(s) excluded \n",
      "6734775 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "2610 people (1659 male(s), 951 female(s)) observed \n",
      "2610 founder(s) included \n",
      "\n",
      "12815 variant(s) not found in previous data \n",
      "33964 variant(s) included \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 2596 \n",
      "\n",
      "Start calculating the scores \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-04-25 12:37:26\n",
      "/home/jupyter/PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base /home/jupyter/sumstats/Chang2017_with23andMe_Sara/META4_hg38.tab \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --bp BP \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --interval 5e-08 \\\n",
      "    --lower 5e-08 \\\n",
      "    --no-regress  \\\n",
      "    --num-auto 22 \\\n",
      "    --out /home/jupyter/pathways/mitochondria_PD_PRS_Chang23andMe \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --seed 2001005637 \\\n",
      "    --snp chrbp \\\n",
      "    --stat BETA \\\n",
      "    --target /home/jupyter/pathways/mitochondria.updated_names.no_dupli \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.05\n",
      "\n",
      "Initializing Genotype file: \n",
      "/home/jupyter/pathways/mitochondria.updated_names.no_dupli \n",
      "(bed) \n",
      "\n",
      "Start processing META4_hg38 \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "/home/jupyter/sumstats/Chang2017_with23andMe_Sara/META4_hg38.tab \n",
      "Header of file is: \n",
      "chrbp\tCHR\tBP\tA1\tA2\tP\tBETA \n",
      "\n",
      "Reading 100.00%\n",
      "7907818 variant(s) observed in base file, with: \n",
      "1173043 ambiguous variant(s) excluded \n",
      "6734775 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "2610 people (1659 male(s), 951 female(s)) observed \n",
      "2610 founder(s) included \n",
      "\n",
      "69019 variant(s) not found in previous data \n",
      "4 variant(s) with mismatch information \n",
      "184997 variant(s) included \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 14264 \n",
      "\n",
      "Start calculating the scores \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-04-25 12:38:55\n",
      "/home/jupyter/PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base /home/jupyter/sumstats/Chang2017_with23andMe_Sara/META4_hg38.tab \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --bp BP \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --interval 5e-08 \\\n",
      "    --lower 5e-08 \\\n",
      "    --no-regress  \\\n",
      "    --num-auto 22 \\\n",
      "    --out /home/jupyter/pathways/microglia_PD_PRS_Chang23andMe \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --seed 3790432398 \\\n",
      "    --snp chrbp \\\n",
      "    --stat BETA \\\n",
      "    --target /home/jupyter/pathways/microglia.updated_names.no_dupli \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.05\n",
      "\n",
      "Initializing Genotype file: \n",
      "/home/jupyter/pathways/microglia.updated_names.no_dupli \n",
      "(bed) \n",
      "\n",
      "Start processing META4_hg38 \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "/home/jupyter/sumstats/Chang2017_with23andMe_Sara/META4_hg38.tab \n",
      "Header of file is: \n",
      "chrbp\tCHR\tBP\tA1\tA2\tP\tBETA \n",
      "\n",
      "Reading 100.00%\n",
      "7907818 variant(s) observed in base file, with: \n",
      "1173043 ambiguous variant(s) excluded \n",
      "6734775 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "2610 people (1659 male(s), 951 female(s)) observed \n",
      "2610 founder(s) included \n",
      "\n",
      "21554 variant(s) not found in previous data \n",
      "1 variant(s) with mismatch information \n",
      "46105 variant(s) included \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 22143 \n",
      "\n",
      "Start calculating the scores \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n",
      "PRSice 2.3.3 (2020-08-05) \n",
      "https://github.com/choishingwan/PRSice\n",
      "(C) 2016-2020 Shing Wan (Sam) Choi and Paul F. O'Reilly\n",
      "GNU General Public License v3\n",
      "If you use PRSice in any published work, please cite:\n",
      "Choi SW, O'Reilly PF.\n",
      "PRSice-2: Polygenic Risk Score Software for Biobank-Scale Data.\n",
      "GigaScience 8, no. 7 (July 1, 2019)\n",
      "2023-04-25 12:40:20\n",
      "/home/jupyter/PRSice_linux \\\n",
      "    --a1 A1 \\\n",
      "    --a2 A2 \\\n",
      "    --bar-levels 0.001,0.05,0.1,0.2,0.3,0.4,0.5,1 \\\n",
      "    --base /home/jupyter/sumstats/Chang2017_with23andMe_Sara/META4_hg38.tab \\\n",
      "    --beta  \\\n",
      "    --binary-target F \\\n",
      "    --bp BP \\\n",
      "    --chr CHR \\\n",
      "    --clump-kb 250kb \\\n",
      "    --clump-p 1.000000 \\\n",
      "    --clump-r2 0.100000 \\\n",
      "    --interval 5e-08 \\\n",
      "    --lower 5e-08 \\\n",
      "    --no-regress  \\\n",
      "    --num-auto 22 \\\n",
      "    --out /home/jupyter/pathways/monocytes_PD_PRS_Chang23andMe \\\n",
      "    --print-snp  \\\n",
      "    --pvalue P \\\n",
      "    --seed 3320630746 \\\n",
      "    --snp chrbp \\\n",
      "    --stat BETA \\\n",
      "    --target /home/jupyter/pathways/monocytes.updated_names.no_dupli \\\n",
      "    --thread 1 \\\n",
      "    --upper 0.05\n",
      "\n",
      "Initializing Genotype file: \n",
      "/home/jupyter/pathways/monocytes.updated_names.no_dupli \n",
      "(bed) \n",
      "\n",
      "Start processing META4_hg38 \n",
      "================================================== \n",
      "\n",
      "Base file: \n",
      "/home/jupyter/sumstats/Chang2017_with23andMe_Sara/META4_hg38.tab \n",
      "Header of file is: \n",
      "chrbp\tCHR\tBP\tA1\tA2\tP\tBETA \n",
      "\n",
      "Reading 100.00%\n",
      "7907818 variant(s) observed in base file, with: \n",
      "1173043 ambiguous variant(s) excluded \n",
      "6734775 total variant(s) included from base file \n",
      "\n",
      "Loading Genotype info from target \n",
      "================================================== \n",
      "\n",
      "2610 people (1659 male(s), 951 female(s)) observed \n",
      "2610 founder(s) included \n",
      "\n",
      "20043 variant(s) not found in previous data \n",
      "1 variant(s) with mismatch information \n",
      "39574 variant(s) included \n",
      "\n",
      "Start performing clumping \n",
      "\n",
      "Clumping Progress: 100.00%\n",
      "Number of variant(s) after clumping : 18803 \n",
      "\n",
      "Start calculating the scores \n",
      "\n",
      "\n",
      "Start Processing\n",
      "Processing 100.00%\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "for PATHWAY in adaptive_immune alpha_synuclein innate_immune lysosome endocytosis mitochondria microglia monocytes\n",
    "do\n",
    "    \n",
    "    Rscript /home/jupyter/PRSice.R \\\n",
    "    --prsice /home/jupyter/PRSice_linux \\\n",
    "    --base /home/jupyter/sumstats/Chang2017_with23andMe_Sara/META4_hg38.tab \\\n",
    "    --pvalue P \\\n",
    "    --beta \\\n",
    "    --a1 A1 \\\n",
    "    --a2 A2 \\\n",
    "    --clump-kb 250 \\\n",
    "    --clump-r2 0.1 \\\n",
    "    --snp chrbp \\\n",
    "    --target /home/jupyter/pathways/\"$PATHWAY\".updated_names.no_dupli \\\n",
    "    --no-regress \\\n",
    "    --lower 5e-08 \\\n",
    "    --upper 0.05 \\\n",
    "    --interval 5e-08 \\\n",
    "    --print-snp \\\n",
    "    --out /home/jupyter/pathways/\"$PATHWAY\"_PD_PRS_Chang23andMe\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CHR\tSNP\tBP\tP\tBase\n",
      "1\t1:155235843\t155235843\t5.14e-15\t1\n",
      "1\t1:155236376\t155236376\t2.735e-15\t1\n",
      "3\t3:183133725\t183133725\t5.664e-15\t1\n",
      "4\t4:987108\t987108\t6.143e-12\t1\n",
      "4\t4:995234\t995234\t2.791e-09\t1\n",
      "4\t4:76226816\t76226816\t2.137e-11\t1\n",
      "12\t12:122941028\t122941028\t6.097e-09\t1\n",
      "17\t17:42494785\t42494785\t1.534e-08\t1\n",
      "4\t4:76182999\t76182999\t5.286e-07\t1\n",
      "4\t4:1002080\t1002080\t2.456e-06\t1\n",
      "4\t4:76163978\t76163978\t4.389e-06\t1\n",
      "14\t14:87941544\t87941544\t1.075e-05\t1\n",
      "4\t4:76222111\t76222111\t0.0001245\t1\n",
      "4\t4:988557\t988557\t0.0001553\t1\n",
      "11\t11:6616273\t6616273\t0.0001569\t1\n",
      "17\t17:7580783\t7580783\t0.0001886\t1\n",
      "16\t16:28479196\t28479196\t0.0003085\t1\n",
      "1\t1:155236246\t155236246\t0.0003095\t1\n",
      "17\t17:68319327\t68319327\t0.0003427\t1\n",
      "12\t12:123735244\t123735244\t0.0004496\t1\n",
      "15\t15:78924987\t78924987\t0.0005531\t1\n",
      "8\t8:53748385\t53748385\t0.0005545\t1\n",
      "17\t17:68421579\t68421579\t0.0005733\t1\n",
      "8\t8:18070100\t18070100\t0.0007517\t1\n",
      "19\t19:50359062\t50359062\t0.0007798\t1\n",
      "5\t5:74721674\t74721674\t0.0008597\t1\n",
      "8\t8:97852044\t97852044\t0.001007\t1\n",
      "4\t4:155941958\t155941958\t0.001027\t1\n",
      "5\t5:78112652\t78112652\t0.00159\t1\n",
      "16\t16:71770071\t71770071\t0.001636\t1\n",
      "5\t5:151231951\t151231951\t0.001956\t1\n",
      "14\t14:87864196\t87864196\t0.002132\t1\n",
      "17\t17:68322954\t68322954\t0.002155\t1\n",
      "15\t15:50962305\t50962305\t0.002371\t1\n",
      "14\t14:87906022\t87906022\t0.002522\t1\n",
      "7\t7:65960907\t65960907\t0.002564\t1\n",
      "5\t5:151226957\t151226957\t0.002594\t1\n",
      "3\t3:183155072\t183155072\t0.002652\t1\n",
      "3\t3:3740355\t3740355\t0.002654\t1\n",
      "4\t4:76224899\t76224899\t0.0028\t1\n",
      "19\t19:10576483\t10576483\t0.003625\t1\n",
      "8\t8:86046717\t86046717\t0.003837\t1\n",
      "4\t4:102637676\t102637676\t0.003866\t1\n",
      "8\t8:18070175\t18070175\t0.003891\t1\n",
      "19\t19:2122459\t2122459\t0.003983\t1\n",
      "14\t14:87977209\t87977209\t0.004336\t1\n",
      "16\t16:5028582\t5028582\t0.004522\t1\n",
      "5\t5:78142836\t78142836\t0.004737\t1\n",
      "3\t3:33042610\t33042610\t0.004996\t1\n",
      "4\t4:127930320\t127930320\t0.005185\t1\n",
      "5\t5:74642627\t74642627\t0.005276\t1\n",
      "15\t15:72344460\t72344460\t0.005392\t1\n",
      "5\t5:78819688\t78819688\t0.00548\t1\n",
      "3\t3:3735194\t3735194\t0.005554\t1\n",
      "16\t16:68248874\t68248874\t0.00602\t1\n",
      "4\t4:76222561\t76222561\t0.006079\t1\n",
      "10\t10:89225233\t89225233\t0.006344\t1\n",
      "16\t16:67438053\t67438053\t0.006595\t1\n",
      "3\t3:3717088\t3717088\t0.007169\t1\n",
      "9\t9:97131093\t97131093\t0.007204\t1\n",
      "17\t17:68307211\t68307211\t0.007473\t1\n",
      "12\t12:101803951\t101803951\t0.007501\t1\n",
      "22\t22:50627096\t50627096\t0.007619\t1\n",
      "12\t12:101828712\t101828712\t0.00839\t1\n",
      "16\t16:71793889\t71793889\t0.008418\t1\n",
      "3\t3:4389672\t4389672\t0.008802\t1\n",
      "6\t6:160055235\t160055235\t0.009174\t1\n",
      "8\t8:97797099\t97797099\t0.01051\t1\n",
      "8\t8:97827979\t97827979\t0.0109\t1\n",
      "16\t16:67438399\t67438399\t0.01097\t1\n",
      "8\t8:97804445\t97804445\t0.01125\t1\n",
      "16\t16:71777661\t71777661\t0.01126\t1\n",
      "11\t11:65880414\t65880414\t0.01252\t1\n",
      "22\t22:50624290\t50624290\t0.01252\t1\n",
      "15\t15:50990678\t50990678\t0.01265\t1\n",
      "10\t10:89229790\t89229790\t0.01269\t1\n",
      "1\t1:84399517\t84399517\t0.01291\t1\n",
      "18\t18:23560468\t23560468\t0.01325\t1\n",
      "19\t19:10584044\t10584044\t0.01376\t1\n",
      "5\t5:78809264\t78809264\t0.01412\t1\n",
      "4\t4:76229777\t76229777\t0.01423\t1\n",
      "22\t22:42059244\t42059244\t0.01484\t1\n",
      "19\t19:10580487\t10580487\t0.01503\t1\n",
      "6\t6:109378156\t109378156\t0.01526\t1\n",
      "19\t19:2151762\t2151762\t0.01563\t1\n",
      "17\t17:42497342\t42497342\t0.01676\t1\n",
      "3\t3:33051093\t33051093\t0.01682\t1\n",
      "22\t22:19208992\t19208992\t0.01707\t1\n",
      "19\t19:10583892\t10583892\t0.01755\t1\n",
      "6\t6:160043552\t160043552\t0.01795\t1\n",
      "10\t10:89309173\t89309173\t0.01852\t1\n",
      "3\t3:4347431\t4347431\t0.01876\t1\n",
      "10\t10:71817510\t71817510\t0.01945\t1\n",
      "5\t5:78804829\t78804829\t0.02053\t1\n",
      "12\t12:123739810\t123739810\t0.02065\t1\n",
      "17\t17:42538243\t42538243\t0.02071\t1\n",
      "12\t12:122959960\t122959960\t0.02075\t1\n",
      "11\t11:6616943\t6616943\t0.02082\t1\n",
      "3\t3:50302191\t50302191\t0.02252\t1\n",
      "7\t7:138726680\t138726680\t0.02273\t1\n",
      "17\t17:68368663\t68368663\t0.02333\t1\n",
      "16\t16:88855085\t88855085\t0.02385\t1\n",
      "5\t5:151259848\t151259848\t0.02448\t1\n",
      "3\t3:32998487\t32998487\t0.02458\t1\n",
      "4\t4:76203043\t76203043\t0.02506\t1\n",
      "19\t19:10586399\t10586399\t0.02506\t1\n",
      "3\t3:33091264\t33091264\t0.02588\t1\n",
      "19\t19:2114695\t2114695\t0.02705\t1\n",
      "1\t1:40080537\t40080537\t0.02758\t1\n",
      "2\t2:218395009\t218395009\t0.02761\t1\n",
      "10\t10:71833760\t71833760\t0.02809\t1\n",
      "8\t8:86033486\t86033486\t0.02869\t1\n",
      "4\t4:155950584\t155950584\t0.02877\t1\n",
      "7\t7:65967975\t65967975\t0.02902\t1\n",
      "14\t14:87949406\t87949406\t0.02964\t1\n",
      "3\t3:3701055\t3701055\t0.03015\t1\n",
      "3\t3:4041490\t4041490\t0.03053\t1\n",
      "3\t3:4449362\t4449362\t0.0313\t1\n",
      "3\t3:4421138\t4421138\t0.03139\t1\n",
      "5\t5:151251117\t151251117\t0.03145\t1\n",
      "12\t12:51010116\t51010116\t0.03218\t1\n",
      "3\t3:4395800\t4395800\t0.03359\t1\n",
      "7\t7:138711287\t138711287\t0.03374\t1\n",
      "14\t14:92722562\t92722562\t0.03447\t1\n",
      "12\t12:122968487\t122968487\t0.03467\t1\n",
      "5\t5:74648352\t74648352\t0.03493\t1\n",
      "22\t22:37610513\t37610513\t0.03543\t1\n",
      "8\t8:23402254\t23402254\t0.03569\t1\n",
      "1\t1:113901277\t113901277\t0.03678\t1\n",
      "5\t5:74694959\t74694959\t0.0374\t1\n",
      "8\t8:86131344\t86131344\t0.03754\t1\n",
      "5\t5:78961278\t78961278\t0.03768\t1\n",
      "1\t1:23854133\t23854133\t0.03826\t1\n",
      "22\t22:19248095\t19248095\t0.03897\t1\n",
      "1\t1:109372968\t109372968\t0.03916\t1\n",
      "16\t16:88850633\t88850633\t0.04011\t1\n",
      "10\t10:89321545\t89321545\t0.04067\t1\n",
      "15\t15:78947377\t78947377\t0.04135\t1\n",
      "17\t17:68276036\t68276036\t0.04186\t1\n",
      "10\t10:71849244\t71849244\t0.04214\t1\n",
      "3\t3:33094704\t33094704\t0.04233\t1\n",
      "5\t5:78859162\t78859162\t0.04317\t1\n",
      "20\t20:45893630\t45893630\t0.04325\t1\n",
      "9\t9:36293160\t36293160\t0.04474\t1\n",
      "12\t12:51015380\t51015380\t0.04487\t1\n",
      "1\t1:150757317\t150757317\t0.0455\t1\n",
      "8\t8:86152146\t86152146\t0.04581\t1\n",
      "4\t4:1002209\t1002209\t0.04669\t1\n",
      "5\t5:115857324\t115857324\t0.04791\t1\n",
      "16\t16:88816004\t88816004\t0.04821\t1\n",
      "19\t19:11576536\t11576536\t0.04822\t1\n",
      "19\t19:16203966\t16203966\t0.04884\t1\n",
      "9\t9:137014754\t137014754\t0.04896\t1\n",
      "5\t5:78207213\t78207213\t0.04966\t1\n",
      "1\t1:30734934\t30734934\t0.2768\t1\n",
      "1\t1:30738235\t30738235\t0.2497\t1\n",
      "1\t1:30740727\t30740727\t0.7724\t1\n",
      "1\t1:30746865\t30746865\t0.9003\t1\n",
      "1\t1:30751153\t30751153\t0.2675\t1\n",
      "1\t1:30751549\t30751549\t0.1271\t1\n",
      "1\t1:40077753\t40077753\t0.05713\t1\n",
      "1\t1:40084126\t40084126\t0.3655\t1\n",
      "1\t1:40090603\t40090603\t0.6199\t1\n",
      "1\t1:40094992\t40094992\t0.2184\t1\n",
      "1\t1:43977258\t43977258\t0.1361\t1\n",
      "1\t1:84400756\t84400756\t0.1935\t1\n",
      "1\t1:84401927\t84401927\t0.5159\t1\n",
      "1\t1:84402249\t84402249\t0.4683\t1\n",
      "1\t1:84404302\t84404302\t0.3028\t1\n",
      "1\t1:84404464\t84404464\t0.1809\t1\n",
      "1\t1:84404776\t84404776\t0.6223\t1\n",
      "1\t1:84410818\t84410818\t0.1963\t1\n",
      "1\t1:109313684\t109313684\t0.6399\t1\n",
      "1\t1:109319177\t109319177\t0.9365\t1\n",
      "1\t1:109329276\t109329276\t0.7109\t1\n",
      "1\t1:109336556\t109336556\t0.233\t1\n",
      "1\t1:109367880\t109367880\t0.4262\t1\n",
      "1\t1:109370442\t109370442\t0.7048\t1\n",
      "1\t1:109372187\t109372187\t0.4665\t1\n",
      "1\t1:109384627\t109384627\t0.4445\t1\n",
      "1\t1:109385002\t109385002\t0.6082\t1\n",
      "1\t1:109396059\t109396059\t0.5574\t1\n",
      "1\t1:113894438\t113894438\t0.7966\t1\n",
      "1\t1:113896403\t113896403\t0.4198\t1\n",
      "1\t1:113897063\t113897063\t0.2787\t1\n",
      "1\t1:113898297\t113898297\t0.9277\t1\n",
      "1\t1:113900721\t113900721\t0.4768\t1\n",
      "1\t1:113903258\t113903258\t0.7043\t1\n",
      "1\t1:113903866\t113903866\t0.9108\t1\n",
      "1\t1:150740618\t150740618\t0.07739\t1\n",
      "1\t1:150743192\t150743192\t0.7056\t1\n",
      "1\t1:150745012\t150745012\t0.2919\t1\n",
      "1\t1:150752055\t150752055\t0.549\t1\n",
      "1\t1:150754083\t150754083\t0.7314\t1\n",
      "1\t1:150764744\t150764744\t0.4664\t1\n",
      "1\t1:150804901\t150804901\t0.2124\t1\n",
      "1\t1:150806925\t150806925\t0.5244\t1\n",
      "1\t1:155237239\t155237239\t0.2377\t1\n",
      "1\t1:155237942\t155237942\t0.09994\t1\n",
      "2\t2:20033946\t20033946\t0.5391\t1\n",
      "2\t2:20034762\t20034762\t0.2541\t1\n",
      "2\t2:20034962\t20034962\t0.8818\t1\n",
      "2\t2:20036156\t20036156\t0.7423\t1\n",
      "2\t2:20038117\t20038117\t0.6925\t1\n",
      "2\t2:20039971\t20039971\t0.86\t1\n",
      "2\t2:20042655\t20042655\t0.4201\t1\n",
      "2\t2:20051136\t20051136\t0.3685\t1\n",
      "2\t2:218387485\t218387485\t0.4424\t1\n",
      "2\t2:218389174\t218389174\t0.3725\t1\n",
      "2\t2:218392703\t218392703\t0.8766\t1\n",
      "2\t2:218395928\t218395928\t0.284\t1\n",
      "2\t2:223755877\t223755877\t0.3441\t1\n",
      "2\t2:223759274\t223759274\t0.09949\t1\n",
      "2\t2:223759878\t223759878\t0.1687\t1\n",
      "2\t2:223767673\t223767673\t0.741\t1\n",
      "2\t2:223768080\t223768080\t0.5534\t1\n",
      "2\t2:223777862\t223777862\t0.1268\t1\n",
      "2\t2:223781461\t223781461\t0.6831\t1\n",
      "2\t2:223782183\t223782183\t0.3385\t1\n",
      "2\t2:223783448\t223783448\t0.6046\t1\n",
      "2\t2:223784823\t223784823\t0.101\t1\n",
      "2\t2:223822632\t223822632\t0.1232\t1\n",
      "2\t2:223822984\t223822984\t0.2137\t1\n",
      "2\t2:223829985\t223829985\t0.3017\t1\n",
      "3\t3:3713468\t3713468\t0.1147\t1\n",
      "3\t3:3729049\t3729049\t0.5037\t1\n",
      "3\t3:3743444\t3743444\t0.3527\t1\n",
      "3\t3:3745127\t3745127\t0.1911\t1\n",
      "3\t3:3756784\t3756784\t0.06251\t1\n",
      "3\t3:3759844\t3759844\t0.3496\t1\n",
      "3\t3:3764063\t3764063\t0.2872\t1\n",
      "3\t3:3768114\t3768114\t0.5445\t1\n",
      "3\t3:3776795\t3776795\t0.588\t1\n",
      "3\t3:3792111\t3792111\t0.278\t1\n",
      "3\t3:3793068\t3793068\t0.05005\t1\n",
      "3\t3:3798035\t3798035\t0.4586\t1\n",
      "3\t3:3806154\t3806154\t0.2063\t1\n",
      "3\t3:3807030\t3807030\t0.6622\t1\n",
      "3\t3:3852280\t3852280\t0.3123\t1\n",
      "3\t3:3861973\t3861973\t0.1157\t1\n",
      "3\t3:3869461\t3869461\t0.186\t1\n",
      "3\t3:3888861\t3888861\t0.2902\t1\n",
      "3\t3:3902526\t3902526\t0.3424\t1\n",
      "3\t3:3906908\t3906908\t0.0808\t1\n",
      "3\t3:3954314\t3954314\t0.1162\t1\n",
      "3\t3:3983209\t3983209\t0.4806\t1\n",
      "3\t3:3997196\t3997196\t0.2694\t1\n",
      "3\t3:4004914\t4004914\t0.7359\t1\n",
      "3\t3:4011580\t4011580\t0.2008\t1\n",
      "3\t3:4032962\t4032962\t0.556\t1\n",
      "3\t3:4034049\t4034049\t0.444\t1\n",
      "3\t3:4041494\t4041494\t0.8391\t1\n",
      "3\t3:4051044\t4051044\t0.1146\t1\n",
      "3\t3:4066015\t4066015\t0.2923\t1\n",
      "3\t3:4068029\t4068029\t0.293\t1\n",
      "3\t3:4081910\t4081910\t0.5863\t1\n",
      "3\t3:4096612\t4096612\t0.4135\t1\n",
      "3\t3:4102858\t4102858\t0.3892\t1\n",
      "3\t3:4110606\t4110606\t0.2382\t1\n",
      "3\t3:4114354\t4114354\t0.6452\t1\n",
      "3\t3:4118195\t4118195\t0.2401\t1\n",
      "3\t3:4137374\t4137374\t0.5143\t1\n",
      "3\t3:4163939\t4163939\t0.09834\t1\n",
      "3\t3:4176850\t4176850\t0.1473\t1\n",
      "3\t3:4196509\t4196509\t0.1606\t1\n",
      "3\t3:4201946\t4201946\t0.1238\t1\n",
      "3\t3:4212892\t4212892\t0.6974\t1\n",
      "3\t3:4223859\t4223859\t0.05329\t1\n",
      "3\t3:4240185\t4240185\t0.1763\t1\n",
      "3\t3:4252101\t4252101\t0.3219\t1\n",
      "3\t3:4268952\t4268952\t0.1907\t1\n",
      "3\t3:4272278\t4272278\t0.2883\t1\n",
      "3\t3:4275722\t4275722\t0.233\t1\n",
      "3\t3:4309913\t4309913\t0.1331\t1\n",
      "3\t3:4310408\t4310408\t0.7291\t1\n",
      "3\t3:4336458\t4336458\t0.8498\t1\n",
      "3\t3:4339087\t4339087\t0.3909\t1\n",
      "3\t3:4345832\t4345832\t0.5884\t1\n",
      "3\t3:4346333\t4346333\t0.834\t1\n",
      "3\t3:4347849\t4347849\t0.2514\t1\n",
      "3\t3:4351333\t4351333\t0.1848\t1\n",
      "3\t3:4353944\t4353944\t0.854\t1\n",
      "3\t3:4357141\t4357141\t0.3674\t1\n",
      "3\t3:4367692\t4367692\t0.2372\t1\n",
      "3\t3:4369491\t4369491\t0.4748\t1\n",
      "3\t3:4379312\t4379312\t0.1282\t1\n",
      "3\t3:4389164\t4389164\t0.1181\t1\n",
      "3\t3:4401655\t4401655\t0.53\t1\n",
      "3\t3:4436003\t4436003\t0.3199\t1\n",
      "3\t3:4443505\t4443505\t0.3153\t1\n",
      "3\t3:4447878\t4447878\t0.9778\t1\n",
      "3\t3:33003081\t33003081\t0.4647\t1\n",
      "3\t3:33017005\t33017005\t0.3316\t1\n",
      "3\t3:33051552\t33051552\t0.2075\t1\n",
      "3\t3:33081800\t33081800\t0.2816\t1\n",
      "3\t3:33081919\t33081919\t0.7245\t1\n",
      "3\t3:33090396\t33090396\t0.1597\t1\n",
      "3\t3:50306169\t50306169\t0.2858\t1\n",
      "3\t3:50310502\t50310502\t0.8899\t1\n",
      "3\t3:183124793\t183124793\t0.8101\t1\n",
      "3\t3:183127703\t183127703\t0.8834\t1\n",
      "3\t3:183130449\t183130449\t0.7215\t1\n",
      "3\t3:183143576\t183143576\t0.3071\t1\n",
      "3\t3:183147720\t183147720\t0.8569\t1\n",
      "4\t4:988559\t988559\t0.852\t1\n",
      "4\t4:990546\t990546\t0.1751\t1\n",
      "4\t4:993417\t993417\t0.7203\t1\n",
      "4\t4:993916\t993916\t0.4514\t1\n",
      "4\t4:1000568\t1000568\t0.3376\t1\n",
      "4\t4:7430495\t7430495\t0.4199\t1\n",
      "4\t4:7431456\t7431456\t0.2363\t1\n",
      "4\t4:7432246\t7432246\t0.5931\t1\n",
      "4\t4:7432290\t7432290\t0.1412\t1\n",
      "4\t4:7433716\t7433716\t0.1888\t1\n",
      "4\t4:7434070\t7434070\t0.1301\t1\n",
      "4\t4:7434759\t7434759\t0.4671\t1\n",
      "4\t4:76162161\t76162161\t0.06483\t1\n",
      "4\t4:76164231\t76164231\t0.6517\t1\n",
      "4\t4:76178666\t76178666\t0.4324\t1\n",
      "4\t4:76182311\t76182311\t0.6556\t1\n",
      "4\t4:76188080\t76188080\t0.4291\t1\n",
      "4\t4:76190738\t76190738\t0.1994\t1\n",
      "4\t4:76194965\t76194965\t0.3482\t1\n",
      "4\t4:76200050\t76200050\t0.6483\t1\n",
      "4\t4:76200612\t76200612\t0.1593\t1\n",
      "4\t4:76203653\t76203653\t0.9625\t1\n",
      "4\t4:102631701\t102631701\t0.05472\t1\n",
      "4\t4:102657436\t102657436\t0.6699\t1\n",
      "4\t4:102677228\t102677228\t0.4477\t1\n",
      "4\t4:102682549\t102682549\t0.1925\t1\n",
      "4\t4:102718986\t102718986\t0.2122\t1\n",
      "4\t4:102749387\t102749387\t0.2071\t1\n",
      "4\t4:102751212\t102751212\t0.2931\t1\n",
      "4\t4:102752750\t102752750\t0.5412\t1\n",
      "4\t4:102753381\t102753381\t0.3423\t1\n",
      "4\t4:127924475\t127924475\t0.2969\t1\n",
      "4\t4:127925971\t127925971\t0.4989\t1\n",
      "4\t4:127928603\t127928603\t0.3481\t1\n",
      "4\t4:127929245\t127929245\t0.1933\t1\n",
      "4\t4:127939122\t127939122\t0.9994\t1\n",
      "4\t4:127939978\t127939978\t0.8537\t1\n",
      "4\t4:127942776\t127942776\t0.8907\t1\n",
      "4\t4:127944886\t127944886\t0.2331\t1\n",
      "4\t4:127950021\t127950021\t0.7898\t1\n",
      "4\t4:127950147\t127950147\t0.2819\t1\n",
      "4\t4:155925495\t155925495\t0.7934\t1\n",
      "4\t4:155931924\t155931924\t0.5481\t1\n",
      "4\t4:155934079\t155934079\t0.06706\t1\n",
      "4\t4:155939099\t155939099\t0.4011\t1\n",
      "4\t4:155941309\t155941309\t0.5103\t1\n",
      "4\t4:155944462\t155944462\t0.4877\t1\n",
      "4\t4:155944576\t155944576\t0.1532\t1\n",
      "4\t4:155947099\t155947099\t0.2758\t1\n",
      "4\t4:155948106\t155948106\t0.4484\t1\n",
      "4\t4:155951806\t155951806\t0.1814\t1\n",
      "4\t4:155952876\t155952876\t0.5344\t1\n",
      "4\t4:177431525\t177431525\t0.5238\t1\n",
      "4\t4:177431726\t177431726\t0.5083\t1\n",
      "4\t4:177434150\t177434150\t0.9568\t1\n",
      "4\t4:177437915\t177437915\t0.9527\t1\n",
      "4\t4:177439111\t177439111\t0.8016\t1\n",
      "4\t4:177439657\t177439657\t0.7541\t1\n",
      "4\t4:177439897\t177439897\t0.3067\t1\n",
      "4\t4:177441857\t177441857\t0.3029\t1\n",
      "5\t5:74642434\t74642434\t0.9058\t1\n",
      "5\t5:74645346\t74645346\t0.9423\t1\n",
      "5\t5:74646903\t74646903\t0.5279\t1\n",
      "5\t5:74648989\t74648989\t0.4468\t1\n",
      "5\t5:74653184\t74653184\t0.6576\t1\n",
      "5\t5:74654678\t74654678\t0.1024\t1\n",
      "5\t5:74661316\t74661316\t0.1946\t1\n",
      "5\t5:74662141\t74662141\t0.9373\t1\n",
      "5\t5:74662903\t74662903\t0.05447\t1\n",
      "5\t5:74670732\t74670732\t0.6972\t1\n",
      "5\t5:74671076\t74671076\t0.06044\t1\n",
      "5\t5:74678662\t74678662\t0.3435\t1\n",
      "5\t5:74680005\t74680005\t0.1396\t1\n",
      "5\t5:74690823\t74690823\t0.1287\t1\n",
      "5\t5:74699794\t74699794\t0.4674\t1\n",
      "5\t5:74706569\t74706569\t0.7212\t1\n",
      "5\t5:74710136\t74710136\t0.3039\t1\n",
      "5\t5:74713502\t74713502\t0.6134\t1\n",
      "5\t5:74717786\t74717786\t0.4827\t1\n",
      "5\t5:78020999\t78020999\t0.4263\t1\n",
      "5\t5:78025111\t78025111\t0.807\t1\n",
      "5\t5:78028134\t78028134\t0.7629\t1\n",
      "5\t5:78039191\t78039191\t0.1837\t1\n",
      "5\t5:78057375\t78057375\t0.8462\t1\n",
      "5\t5:78059089\t78059089\t0.7903\t1\n",
      "5\t5:78075096\t78075096\t0.7522\t1\n",
      "5\t5:78082944\t78082944\t0.5525\t1\n",
      "5\t5:78094674\t78094674\t0.4883\t1\n",
      "5\t5:78100629\t78100629\t0.07962\t1\n",
      "5\t5:78114912\t78114912\t0.1922\t1\n",
      "5\t5:78119921\t78119921\t0.05937\t1\n",
      "5\t5:78134007\t78134007\t0.1235\t1\n",
      "5\t5:78136424\t78136424\t0.7459\t1\n",
      "5\t5:78158028\t78158028\t0.1144\t1\n",
      "5\t5:78169219\t78169219\t0.3326\t1\n",
      "5\t5:78203263\t78203263\t0.3157\t1\n",
      "5\t5:78204273\t78204273\t0.4251\t1\n",
      "5\t5:78237879\t78237879\t0.1022\t1\n",
      "5\t5:78290126\t78290126\t0.2825\t1\n",
      "5\t5:78777216\t78777216\t0.3962\t1\n",
      "5\t5:78778375\t78778375\t0.1488\t1\n",
      "5\t5:78779095\t78779095\t0.8926\t1\n",
      "5\t5:78790713\t78790713\t0.1119\t1\n",
      "5\t5:78792296\t78792296\t0.6204\t1\n",
      "5\t5:78797270\t78797270\t0.5829\t1\n",
      "5\t5:78803274\t78803274\t0.6527\t1\n",
      "5\t5:78804584\t78804584\t0.3814\t1\n",
      "5\t5:78808266\t78808266\t0.3959\t1\n",
      "5\t5:78814739\t78814739\t0.242\t1\n",
      "5\t5:78827399\t78827399\t0.2952\t1\n",
      "5\t5:78828362\t78828362\t0.7117\t1\n",
      "5\t5:78828980\t78828980\t0.5992\t1\n",
      "5\t5:78830458\t78830458\t0.1447\t1\n",
      "5\t5:78846506\t78846506\t0.7301\t1\n",
      "5\t5:78857676\t78857676\t0.06405\t1\n",
      "5\t5:78881527\t78881527\t0.0888\t1\n",
      "5\t5:78914217\t78914217\t0.4397\t1\n",
      "5\t5:78923585\t78923585\t0.3121\t1\n",
      "5\t5:78933108\t78933108\t0.2754\t1\n",
      "5\t5:78944862\t78944862\t0.4968\t1\n",
      "5\t5:78967089\t78967089\t0.5255\t1\n",
      "5\t5:78969943\t78969943\t0.5305\t1\n",
      "5\t5:78976568\t78976568\t0.2277\t1\n",
      "5\t5:78977119\t78977119\t0.4929\t1\n",
      "5\t5:78977265\t78977265\t0.2033\t1\n",
      "5\t5:78984770\t78984770\t0.703\t1\n",
      "5\t5:115850710\t115850710\t0.4588\t1\n",
      "5\t5:115850779\t115850779\t0.06159\t1\n",
      "5\t5:115853803\t115853803\t0.3473\t1\n",
      "5\t5:115856145\t115856145\t0.4177\t1\n",
      "5\t5:115856709\t115856709\t0.5017\t1\n",
      "5\t5:115856933\t115856933\t0.1956\t1\n",
      "5\t5:115860691\t115860691\t0.5189\t1\n",
      "5\t5:115863670\t115863670\t0.1017\t1\n",
      "5\t5:115863743\t115863743\t0.8108\t1\n",
      "5\t5:115879813\t115879813\t0.1535\t1\n",
      "5\t5:115893444\t115893444\t0.9888\t1\n",
      "5\t5:115896390\t115896390\t0.4736\t1\n",
      "5\t5:115899393\t115899393\t0.8895\t1\n",
      "5\t5:115903480\t115903480\t0.6974\t1\n",
      "5\t5:115909934\t115909934\t0.3799\t1\n",
      "5\t5:151212226\t151212226\t0.2743\t1\n",
      "5\t5:151217375\t151217375\t0.4088\t1\n",
      "5\t5:151219129\t151219129\t0.8392\t1\n",
      "5\t5:151221426\t151221426\t0.05231\t1\n",
      "5\t5:151225475\t151225475\t0.5592\t1\n",
      "5\t5:151229227\t151229227\t0.07508\t1\n",
      "5\t5:151233007\t151233007\t0.2884\t1\n",
      "5\t5:151235980\t151235980\t0.845\t1\n",
      "5\t5:151240604\t151240604\t0.2303\t1\n",
      "5\t5:151249639\t151249639\t0.3527\t1\n",
      "5\t5:176392950\t176392950\t0.3487\t1\n",
      "5\t5:176393587\t176393587\t0.9828\t1\n",
      "5\t5:176411695\t176411695\t0.3899\t1\n",
      "5\t5:176413922\t176413922\t0.05428\t1\n",
      "6\t6:73629554\t73629554\t0.1815\t1\n",
      "6\t6:73630804\t73630804\t0.6351\t1\n",
      "6\t6:73643192\t73643192\t0.8819\t1\n",
      "6\t6:73643898\t73643898\t0.1014\t1\n",
      "6\t6:73643930\t73643930\t0.3909\t1\n",
      "6\t6:73646415\t73646415\t0.8191\t1\n",
      "6\t6:73649735\t73649735\t0.398\t1\n",
      "6\t6:109370125\t109370125\t0.3307\t1\n",
      "6\t6:109379853\t109379853\t0.06304\t1\n",
      "6\t6:109381573\t109381573\t0.8062\t1\n",
      "6\t6:109381824\t109381824\t0.2912\t1\n",
      "6\t6:159973159\t159973159\t0.7288\t1\n",
      "6\t6:159979703\t159979703\t0.09043\t1\n",
      "6\t6:159983800\t159983800\t0.1094\t1\n",
      "6\t6:159984814\t159984814\t0.06923\t1\n",
      "6\t6:159989177\t159989177\t0.09836\t1\n",
      "6\t6:159994522\t159994522\t0.2589\t1\n",
      "6\t6:160007826\t160007826\t0.4262\t1\n",
      "6\t6:160008660\t160008660\t0.1837\t1\n",
      "6\t6:160011241\t160011241\t0.2425\t1\n",
      "6\t6:160040053\t160040053\t0.1792\t1\n",
      "6\t6:160043496\t160043496\t0.3948\t1\n",
      "6\t6:160044389\t160044389\t0.7964\t1\n",
      "6\t6:160057440\t160057440\t0.284\t1\n",
      "6\t6:160074023\t160074023\t0.2983\t1\n",
      "6\t6:160075287\t160075287\t0.3125\t1\n",
      "6\t6:160075400\t160075400\t0.213\t1\n",
      "6\t6:160096449\t160096449\t0.05502\t1\n",
      "6\t6:160097494\t160097494\t0.552\t1\n",
      "6\t6:160098196\t160098196\t0.1372\t1\n",
      "6\t6:160102223\t160102223\t0.4806\t1\n",
      "6\t6:160102283\t160102283\t0.8864\t1\n",
      "6\t6:160107602\t160107602\t0.4967\t1\n",
      "6\t6:160110469\t160110469\t0.5849\t1\n",
      "7\t7:65960725\t65960725\t0.3344\t1\n",
      "7\t7:100101813\t100101813\t0.05127\t1\n",
      "7\t7:100105324\t100105324\t0.3271\t1\n",
      "7\t7:100105882\t100105882\t0.6772\t1\n",
      "7\t7:100107697\t100107697\t0.8036\t1\n",
      "7\t7:100108831\t100108831\t0.5798\t1\n",
      "7\t7:101156292\t101156292\t0.4614\t1\n",
      "7\t7:138711380\t138711380\t0.4954\t1\n",
      "7\t7:138718994\t138718994\t0.1703\t1\n",
      "7\t7:138719372\t138719372\t0.4077\t1\n",
      "7\t7:138725043\t138725043\t0.3022\t1\n",
      "7\t7:138729897\t138729897\t0.1103\t1\n",
      "7\t7:138732414\t138732414\t0.5258\t1\n",
      "7\t7:138735214\t138735214\t0.07245\t1\n",
      "7\t7:138745106\t138745106\t0.7068\t1\n",
      "7\t7:138748658\t138748658\t0.9279\t1\n",
      "7\t7:138748967\t138748967\t0.9921\t1\n",
      "7\t7:138754103\t138754103\t0.1222\t1\n",
      "7\t7:138765825\t138765825\t0.776\t1\n",
      "7\t7:138767695\t138767695\t0.3101\t1\n",
      "7\t7:138769677\t138769677\t0.1958\t1\n",
      "7\t7:138770706\t138770706\t0.1273\t1\n",
      "7\t7:138773192\t138773192\t0.1389\t1\n",
      "7\t7:138778667\t138778667\t0.07301\t1\n",
      "7\t7:138783383\t138783383\t0.2671\t1\n",
      "7\t7:138792115\t138792115\t0.7742\t1\n",
      "7\t7:138796804\t138796804\t0.1283\t1\n",
      "7\t7:138797312\t138797312\t0.4882\t1\n",
      "7\t7:138799320\t138799320\t0.8716\t1\n",
      "8\t8:18056244\t18056244\t0.8594\t1\n",
      "8\t8:18057200\t18057200\t0.5014\t1\n",
      "8\t8:18060357\t18060357\t0.6197\t1\n",
      "8\t8:18061104\t18061104\t0.8725\t1\n",
      "8\t8:18061234\t18061234\t0.13\t1\n",
      "8\t8:18064349\t18064349\t0.3835\t1\n",
      "8\t8:18065511\t18065511\t0.8866\t1\n",
      "8\t8:18066080\t18066080\t0.2224\t1\n",
      "8\t8:18067575\t18067575\t0.3702\t1\n",
      "8\t8:18081065\t18081065\t0.1232\t1\n",
      "8\t8:18081685\t18081685\t0.9276\t1\n",
      "8\t8:23387191\t23387191\t0.2153\t1\n",
      "8\t8:23388555\t23388555\t0.8509\t1\n",
      "8\t8:23394248\t23394248\t0.2027\t1\n",
      "8\t8:23397733\t23397733\t0.508\t1\n",
      "8\t8:23410896\t23410896\t0.823\t1\n",
      "8\t8:23413160\t23413160\t0.6456\t1\n",
      "8\t8:23414608\t23414608\t0.2292\t1\n",
      "8\t8:23415632\t23415632\t0.3899\t1\n",
      "8\t8:23416303\t23416303\t0.3733\t1\n",
      "8\t8:23421668\t23421668\t0.3153\t1\n",
      "8\t8:23426830\t23426830\t0.5294\t1\n",
      "8\t8:23435145\t23435145\t0.06615\t1\n",
      "8\t8:23446078\t23446078\t0.08416\t1\n",
      "8\t8:23451386\t23451386\t0.1045\t1\n",
      "8\t8:42161438\t42161438\t0.8589\t1\n",
      "8\t8:42162640\t42162640\t0.5425\t1\n",
      "8\t8:42166736\t42166736\t0.4084\t1\n",
      "8\t8:43140947\t43140947\t0.7369\t1\n",
      "8\t8:43142601\t43142601\t0.7615\t1\n",
      "8\t8:43145715\t43145715\t0.7071\t1\n",
      "8\t8:43158037\t43158037\t0.4257\t1\n",
      "8\t8:43183279\t43183279\t0.3124\t1\n",
      "8\t8:43199282\t43199282\t0.6318\t1\n",
      "8\t8:53723036\t53723036\t0.1326\t1\n",
      "8\t8:53738282\t53738282\t0.8866\t1\n",
      "8\t8:53752379\t53752379\t0.1607\t1\n",
      "8\t8:53826658\t53826658\t0.9617\t1\n",
      "8\t8:53841352\t53841352\t0.2565\t1\n",
      "8\t8:86014296\t86014296\t0.8808\t1\n",
      "8\t8:86016947\t86016947\t0.09445\t1\n",
      "8\t8:86017052\t86017052\t0.453\t1\n",
      "8\t8:86031045\t86031045\t0.1361\t1\n",
      "8\t8:86042282\t86042282\t0.0592\t1\n",
      "8\t8:86053020\t86053020\t0.6715\t1\n",
      "8\t8:86061345\t86061345\t0.2335\t1\n",
      "8\t8:86064581\t86064581\t0.985\t1\n",
      "8\t8:86072008\t86072008\t0.4282\t1\n",
      "8\t8:86076220\t86076220\t0.3176\t1\n",
      "8\t8:86087045\t86087045\t0.1091\t1\n",
      "8\t8:86089974\t86089974\t0.5445\t1\n",
      "8\t8:86092478\t86092478\t0.2613\t1\n",
      "8\t8:86118514\t86118514\t0.9909\t1\n",
      "8\t8:86124973\t86124973\t0.4638\t1\n",
      "8\t8:86127368\t86127368\t0.3305\t1\n",
      "8\t8:86152234\t86152234\t0.08091\t1\n",
      "8\t8:97775353\t97775353\t0.5223\t1\n",
      "8\t8:97780993\t97780993\t0.1375\t1\n",
      "8\t8:97783971\t97783971\t0.1544\t1\n",
      "8\t8:97812837\t97812837\t0.3742\t1\n",
      "8\t8:97834687\t97834687\t0.4202\t1\n",
      "8\t8:97835491\t97835491\t0.3499\t1\n",
      "8\t8:97838887\t97838887\t0.3063\t1\n",
      "8\t8:97841246\t97841246\t0.1785\t1\n",
      "8\t8:97850027\t97850027\t0.5409\t1\n",
      "9\t9:36195242\t36195242\t0.9678\t1\n",
      "9\t9:36204642\t36204642\t0.7827\t1\n",
      "9\t9:36209085\t36209085\t0.7019\t1\n",
      "9\t9:36210412\t36210412\t0.6903\t1\n",
      "9\t9:36211034\t36211034\t0.63\t1\n",
      "9\t9:36217801\t36217801\t0.4723\t1\n",
      "9\t9:36219884\t36219884\t0.7918\t1\n",
      "9\t9:36220848\t36220848\t0.1026\t1\n",
      "9\t9:36224914\t36224914\t0.1297\t1\n",
      "9\t9:36230082\t36230082\t0.1358\t1\n",
      "9\t9:36242046\t36242046\t0.4673\t1\n",
      "9\t9:36247561\t36247561\t0.5113\t1\n",
      "9\t9:36277137\t36277137\t0.6956\t1\n",
      "9\t9:36281153\t36281153\t0.2233\t1\n",
      "9\t9:36281843\t36281843\t0.4892\t1\n",
      "9\t9:36289085\t36289085\t0.5434\t1\n",
      "9\t9:36293268\t36293268\t0.3574\t1\n",
      "9\t9:36294687\t36294687\t0.233\t1\n",
      "9\t9:36299648\t36299648\t0.05883\t1\n",
      "9\t9:36300142\t36300142\t0.2479\t1\n",
      "9\t9:36300887\t36300887\t0.3611\t1\n",
      "9\t9:87724615\t87724615\t0.2648\t1\n",
      "9\t9:87726001\t87726001\t0.4609\t1\n",
      "9\t9:87728590\t87728590\t0.1896\t1\n",
      "9\t9:87728888\t87728888\t0.3063\t1\n",
      "9\t9:87730131\t87730131\t0.3018\t1\n",
      "9\t9:97040085\t97040085\t0.3262\t1\n",
      "9\t9:97094115\t97094115\t0.1386\t1\n",
      "9\t9:97144183\t97144183\t0.3383\t1\n",
      "9\t9:97155072\t97155072\t0.6309\t1\n",
      "9\t9:137007714\t137007714\t0.55\t1\n",
      "9\t9:137007762\t137007762\t0.09613\t1\n",
      "9\t9:137010689\t137010689\t0.3167\t1\n",
      "9\t9:137017531\t137017531\t0.06385\t1\n",
      "10\t10:71818915\t71818915\t0.1824\t1\n",
      "10\t10:71822277\t71822277\t0.8877\t1\n",
      "10\t10:71829541\t71829541\t0.983\t1\n",
      "10\t10:71830428\t71830428\t0.3281\t1\n",
      "10\t10:71834561\t71834561\t0.6174\t1\n",
      "10\t10:71842892\t71842892\t0.1219\t1\n",
      "10\t10:71843015\t71843015\t0.6562\t1\n",
      "10\t10:71843340\t71843340\t0.679\t1\n",
      "10\t10:71845090\t71845090\t0.3122\t1\n",
      "10\t10:71850499\t71850499\t0.1327\t1\n",
      "10\t10:74125062\t74125062\t0.5073\t1\n",
      "10\t10:74129312\t74129312\t0.2987\t1\n",
      "10\t10:74137739\t74137739\t0.4067\t1\n",
      "10\t10:74148456\t74148456\t0.409\t1\n",
      "10\t10:74149876\t74149876\t0.2158\t1\n",
      "10\t10:89217426\t89217426\t0.1862\t1\n",
      "10\t10:89218513\t89218513\t0.8255\t1\n",
      "10\t10:89218846\t89218846\t0.9223\t1\n",
      "10\t10:89219920\t89219920\t0.9412\t1\n",
      "10\t10:89225806\t89225806\t0.256\t1\n",
      "10\t10:89227250\t89227250\t0.463\t1\n",
      "10\t10:89227701\t89227701\t0.07048\t1\n",
      "10\t10:89229974\t89229974\t0.3789\t1\n",
      "10\t10:89237707\t89237707\t0.8096\t1\n",
      "10\t10:89247582\t89247582\t0.5695\t1\n",
      "10\t10:89255979\t89255979\t0.2703\t1\n",
      "10\t10:89256998\t89256998\t0.5479\t1\n",
      "10\t10:89258141\t89258141\t0.4602\t1\n",
      "10\t10:89258829\t89258829\t0.3306\t1\n",
      "10\t10:89268769\t89268769\t0.4587\t1\n",
      "10\t10:89286035\t89286035\t0.728\t1\n",
      "10\t10:89296417\t89296417\t0.3785\t1\n",
      "10\t10:89296863\t89296863\t0.2135\t1\n",
      "10\t10:89317338\t89317338\t0.7943\t1\n",
      "10\t10:89322331\t89322331\t0.6537\t1\n",
      "10\t10:89324963\t89324963\t0.7722\t1\n",
      "10\t10:89333093\t89333093\t0.1429\t1\n",
      "10\t10:89339120\t89339120\t0.2976\t1\n",
      "10\t10:89348067\t89348067\t0.3309\t1\n",
      "10\t10:89348216\t89348216\t0.9676\t1\n",
      "10\t10:89348513\t89348513\t0.8862\t1\n",
      "10\t10:89356949\t89356949\t0.2084\t1\n",
      "10\t10:89357609\t89357609\t0.2736\t1\n",
      "10\t10:89361287\t89361287\t0.7846\t1\n",
      "10\t10:89369538\t89369538\t0.3509\t1\n",
      "10\t10:89387853\t89387853\t0.4778\t1\n",
      "10\t10:89390932\t89390932\t0.1063\t1\n",
      "10\t10:89399103\t89399103\t0.4742\t1\n",
      "10\t10:89399997\t89399997\t0.7009\t1\n",
      "10\t10:89401752\t89401752\t0.08523\t1\n",
      "10\t10:89403214\t89403214\t0.477\t1\n",
      "11\t11:1755102\t1755102\t0.1911\t1\n",
      "11\t11:1757266\t1757266\t0.1535\t1\n",
      "11\t11:1762562\t1762562\t0.8463\t1\n",
      "11\t11:6394162\t6394162\t0.2362\t1\n",
      "11\t11:6394233\t6394233\t0.6325\t1\n",
      "11\t11:6615069\t6615069\t0.36\t1\n",
      "11\t11:6616092\t6616092\t0.08777\t1\n",
      "11\t11:6616346\t6616346\t0.572\t1\n",
      "11\t11:6618120\t6618120\t0.5619\t1\n",
      "11\t11:47240211\t47240211\t0.3349\t1\n",
      "11\t11:47245257\t47245257\t0.7707\t1\n",
      "11\t11:47246642\t47246642\t0.3004\t1\n",
      "11\t11:47246759\t47246759\t0.3316\t1\n",
      "11\t11:65880664\t65880664\t0.08895\t1\n",
      "11\t11:65882676\t65882676\t0.9909\t1\n",
      "11\t11:65883044\t65883044\t0.07994\t1\n",
      "11\t11:66566520\t66566520\t0.08606\t1\n",
      "11\t11:66567837\t66567837\t0.193\t1\n",
      "11\t11:68041801\t68041801\t0.07136\t1\n",
      "11\t11:68042874\t68042874\t0.1685\t1\n",
      "11\t11:68046463\t68046463\t0.1179\t1\n",
      "11\t11:68047376\t68047376\t0.1064\t1\n",
      "12\t12:8942386\t8942386\t0.4872\t1\n",
      "12\t12:8944802\t8944802\t0.2394\t1\n",
      "12\t12:8946972\t8946972\t0.5235\t1\n",
      "12\t12:8949383\t8949383\t0.2697\t1\n",
      "12\t12:50981535\t50981535\t0.8546\t1\n",
      "12\t12:50982420\t50982420\t0.4428\t1\n",
      "12\t12:51009122\t51009122\t0.05814\t1\n",
      "12\t12:51027032\t51027032\t0.5443\t1\n",
      "12\t12:55726312\t55726312\t0.6461\t1\n",
      "12\t12:55729581\t55729581\t0.1717\t1\n",
      "12\t12:64716330\t64716330\t0.08686\t1\n",
      "12\t12:64720248\t64720248\t0.8918\t1\n",
      "12\t12:64747808\t64747808\t0.2145\t1\n",
      "12\t12:64751599\t64751599\t0.3022\t1\n",
      "12\t12:64753079\t64753079\t0.7446\t1\n",
      "12\t12:64755825\t64755825\t0.5368\t1\n",
      "12\t12:64758162\t64758162\t0.1235\t1\n",
      "12\t12:101758237\t101758237\t0.7439\t1\n",
      "12\t12:101764986\t101764986\t0.601\t1\n",
      "12\t12:101771646\t101771646\t0.5638\t1\n",
      "12\t12:101773024\t101773024\t0.3887\t1\n",
      "12\t12:101777146\t101777146\t0.1581\t1\n",
      "12\t12:101785688\t101785688\t0.8861\t1\n",
      "12\t12:101795523\t101795523\t0.3655\t1\n",
      "12\t12:101796879\t101796879\t0.4878\t1\n",
      "12\t12:101797442\t101797442\t0.7971\t1\n",
      "12\t12:101797443\t101797443\t0.1035\t1\n",
      "12\t12:101830344\t101830344\t0.2481\t1\n",
      "12\t12:122945535\t122945535\t0.6114\t1\n",
      "12\t12:122954690\t122954690\t0.842\t1\n",
      "12\t12:122966320\t122966320\t0.1627\t1\n",
      "12\t12:122974562\t122974562\t0.5016\t1\n",
      "12\t12:123716466\t123716466\t0.1838\t1\n",
      "12\t12:123718153\t123718153\t0.4464\t1\n",
      "12\t12:123730579\t123730579\t0.3919\t1\n",
      "12\t12:123756959\t123756959\t0.1237\t1\n",
      "12\t12:123758010\t123758010\t0.859\t1\n",
      "12\t12:123760389\t123760389\t0.5398\t1\n",
      "13\t13:76994597\t76994597\t0.1566\t1\n",
      "13\t13:76995766\t76995766\t0.3244\t1\n",
      "13\t13:76995943\t76995943\t0.06781\t1\n",
      "13\t13:77000471\t77000471\t0.5188\t1\n",
      "13\t13:77003413\t77003413\t0.6919\t1\n",
      "13\t13:77008337\t77008337\t0.711\t1\n",
      "13\t13:77009272\t77009272\t0.6565\t1\n",
      "13\t13:77014323\t77014323\t0.3421\t1\n",
      "13\t13:77016801\t77016801\t0.8845\t1\n",
      "13\t13:113320975\t113320975\t0.4601\t1\n",
      "14\t14:24573862\t24573862\t0.928\t1\n",
      "14\t14:24574465\t24574465\t0.3103\t1\n",
      "14\t14:24575006\t24575006\t0.6249\t1\n",
      "14\t14:31026140\t31026140\t0.06343\t1\n",
      "14\t14:31027006\t31027006\t0.05719\t1\n",
      "14\t14:31033275\t31033275\t0.1809\t1\n",
      "14\t14:31037715\t31037715\t0.2666\t1\n",
      "14\t14:31049789\t31049789\t0.0558\t1\n",
      "14\t14:31049866\t31049866\t0.3747\t1\n",
      "14\t14:31056982\t31056982\t0.6703\t1\n",
      "14\t14:31065776\t31065776\t0.5758\t1\n",
      "14\t14:31068159\t31068159\t0.1005\t1\n",
      "14\t14:31077435\t31077435\t0.3207\t1\n",
      "14\t14:31080539\t31080539\t0.3005\t1\n",
      "14\t14:31097242\t31097242\t0.2738\t1\n",
      "14\t14:31103392\t31103392\t0.887\t1\n",
      "14\t14:74481405\t74481405\t0.5557\t1\n",
      "14\t14:74482940\t74482940\t0.1096\t1\n",
      "14\t14:74487372\t74487372\t0.3411\t1\n",
      "14\t14:74488976\t74488976\t0.9523\t1\n",
      "14\t14:74490216\t74490216\t0.09852\t1\n",
      "14\t14:74490290\t74490290\t0.5056\t1\n",
      "14\t14:74492898\t74492898\t0.2848\t1\n",
      "14\t14:87854385\t87854385\t0.4329\t1\n",
      "14\t14:87855449\t87855449\t0.5391\t1\n",
      "14\t14:87855563\t87855563\t0.1449\t1\n",
      "14\t14:87858796\t87858796\t0.07236\t1\n",
      "14\t14:87862544\t87862544\t0.9724\t1\n",
      "14\t14:87879199\t87879199\t0.544\t1\n",
      "14\t14:87887569\t87887569\t0.3156\t1\n",
      "14\t14:87888508\t87888508\t0.3473\t1\n",
      "14\t14:87891480\t87891480\t0.257\t1\n",
      "14\t14:87892442\t87892442\t0.4227\t1\n",
      "14\t14:87897309\t87897309\t0.06873\t1\n",
      "14\t14:87897614\t87897614\t0.4967\t1\n",
      "14\t14:87916292\t87916292\t0.4765\t1\n",
      "14\t14:87917317\t87917317\t0.5866\t1\n",
      "14\t14:87923465\t87923465\t0.06516\t1\n",
      "14\t14:87927180\t87927180\t0.07025\t1\n",
      "14\t14:87935116\t87935116\t0.1081\t1\n",
      "14\t14:87949460\t87949460\t0.4271\t1\n",
      "14\t14:87953580\t87953580\t0.5595\t1\n",
      "14\t14:87954935\t87954935\t0.1459\t1\n",
      "14\t14:87977409\t87977409\t0.8884\t1\n",
      "14\t14:87984426\t87984426\t0.1966\t1\n",
      "14\t14:92707296\t92707296\t0.9833\t1\n",
      "14\t14:92708106\t92708106\t0.3231\t1\n",
      "14\t14:92709697\t92709697\t0.5356\t1\n",
      "14\t14:92710743\t92710743\t0.4003\t1\n",
      "14\t14:92711055\t92711055\t0.2856\t1\n",
      "14\t14:92720665\t92720665\t0.9975\t1\n",
      "14\t14:92721769\t92721769\t0.122\t1\n",
      "14\t14:92724160\t92724160\t0.206\t1\n",
      "14\t14:92729029\t92729029\t0.1605\t1\n",
      "14\t14:92732224\t92732224\t0.8824\t1\n",
      "14\t14:92732567\t92732567\t0.3554\t1\n",
      "14\t14:92732735\t92732735\t0.1113\t1\n",
      "14\t14:92733507\t92733507\t0.8605\t1\n",
      "14\t14:92738313\t92738313\t0.3668\t1\n",
      "14\t14:92741955\t92741955\t0.05447\t1\n",
      "14\t14:92742661\t92742661\t0.3918\t1\n",
      "15\t15:50917203\t50917203\t0.4778\t1\n",
      "15\t15:50926648\t50926648\t0.9688\t1\n",
      "15\t15:50932290\t50932290\t0.1065\t1\n",
      "15\t15:50938643\t50938643\t0.6425\t1\n",
      "15\t15:50943638\t50943638\t0.2058\t1\n",
      "15\t15:50947936\t50947936\t0.4361\t1\n",
      "15\t15:50955644\t50955644\t0.4621\t1\n",
      "15\t15:50958413\t50958413\t0.1395\t1\n",
      "15\t15:50960338\t50960338\t0.2622\t1\n",
      "15\t15:50963020\t50963020\t0.06614\t1\n",
      "15\t15:50964687\t50964687\t0.2257\t1\n",
      "15\t15:50973914\t50973914\t0.841\t1\n",
      "15\t15:50974530\t50974530\t0.9019\t1\n",
      "15\t15:50985615\t50985615\t0.05926\t1\n",
      "15\t15:50997987\t50997987\t0.2086\t1\n",
      "15\t15:51001539\t51001539\t0.6885\t1\n",
      "15\t15:51001752\t51001752\t0.9926\t1\n",
      "15\t15:72363900\t72363900\t0.05544\t1\n",
      "15\t15:72369307\t72369307\t0.5401\t1\n",
      "15\t15:78921873\t78921873\t0.3994\t1\n",
      "15\t15:78922050\t78922050\t0.05477\t1\n",
      "15\t15:78936921\t78936921\t0.2209\t1\n",
      "15\t15:78948129\t78948129\t0.622\t1\n",
      "15\t15:82662121\t82662121\t0.6016\t1\n",
      "15\t15:82663400\t82663400\t0.3639\t1\n",
      "15\t15:82666104\t82666104\t0.4168\t1\n",
      "15\t15:82672130\t82672130\t0.7253\t1\n",
      "15\t15:82677644\t82677644\t0.8756\t1\n",
      "15\t15:82682759\t82682759\t0.9034\t1\n",
      "15\t15:82683159\t82683159\t0.2334\t1\n",
      "15\t15:82690211\t82690211\t0.5124\t1\n",
      "15\t15:82691298\t82691298\t0.9823\t1\n",
      "15\t15:82696186\t82696186\t0.09014\t1\n",
      "15\t15:82697195\t82697195\t0.4325\t1\n",
      "15\t15:82702814\t82702814\t0.5038\t1\n",
      "15\t15:82706286\t82706286\t0.7505\t1\n",
      "15\t15:82707304\t82707304\t0.4902\t1\n",
      "15\t15:89831965\t89831965\t0.411\t1\n",
      "15\t15:89842906\t89842906\t0.3704\t1\n",
      "15\t15:89852047\t89852047\t0.9537\t1\n",
      "15\t15:89870907\t89870907\t0.705\t1\n",
      "15\t15:89882091\t89882091\t0.2235\t1\n",
      "15\t15:89893236\t89893236\t0.321\t1\n",
      "16\t16:1354725\t1354725\t0.2971\t1\n",
      "16\t16:1355969\t1355969\t0.404\t1\n",
      "16\t16:1359079\t1359079\t0.1386\t1\n",
      "16\t16:1364497\t1364497\t0.1915\t1\n",
      "16\t16:1364909\t1364909\t0.8989\t1\n",
      "16\t16:5024951\t5024951\t0.07238\t1\n",
      "16\t16:5029465\t5029465\t0.4768\t1\n",
      "16\t16:5030557\t5030557\t0.6847\t1\n",
      "16\t16:23475432\t23475432\t0.2864\t1\n",
      "16\t16:23505979\t23505979\t0.2961\t1\n",
      "16\t16:23517851\t23517851\t0.206\t1\n",
      "16\t16:23518768\t23518768\t0.3419\t1\n",
      "16\t16:23520845\t23520845\t0.1174\t1\n",
      "16\t16:67441899\t67441899\t0.9721\t1\n",
      "16\t16:67442483\t67442483\t0.06437\t1\n",
      "16\t16:67447255\t67447255\t0.3151\t1\n",
      "16\t16:67449000\t67449000\t0.3997\t1\n",
      "16\t16:67472344\t67472344\t0.637\t1\n",
      "16\t16:68246414\t68246414\t0.3499\t1\n",
      "16\t16:68252091\t68252091\t0.05887\t1\n",
      "16\t16:68255836\t68255836\t0.6426\t1\n",
      "16\t16:68259714\t68259714\t0.6053\t1\n",
      "16\t16:71740196\t71740196\t0.5745\t1\n",
      "16\t16:71743049\t71743049\t0.08234\t1\n",
      "16\t16:71768090\t71768090\t0.322\t1\n",
      "16\t16:71773329\t71773329\t0.06398\t1\n",
      "16\t16:71778493\t71778493\t0.405\t1\n",
      "16\t16:71790769\t71790769\t0.06527\t1\n",
      "16\t16:71792122\t71792122\t0.2014\t1\n",
      "16\t16:71792682\t71792682\t0.757\t1\n",
      "16\t16:71793847\t71793847\t0.3043\t1\n",
      "16\t16:71805794\t71805794\t0.961\t1\n",
      "16\t16:71806623\t71806623\t0.6897\t1\n",
      "16\t16:71807947\t71807947\t0.296\t1\n",
      "16\t16:88815227\t88815227\t0.948\t1\n",
      "16\t16:88824332\t88824332\t0.4995\t1\n",
      "16\t16:88829887\t88829887\t0.5111\t1\n",
      "16\t16:88830349\t88830349\t0.9311\t1\n",
      "16\t16:88832378\t88832378\t0.7865\t1\n",
      "16\t16:88834884\t88834884\t0.1812\t1\n",
      "16\t16:88836614\t88836614\t0.2001\t1\n",
      "16\t16:88841898\t88841898\t0.1956\t1\n",
      "16\t16:88848460\t88848460\t0.9288\t1\n",
      "17\t17:3637461\t3637461\t0.1503\t1\n",
      "17\t17:3638868\t3638868\t0.9146\t1\n",
      "17\t17:3649998\t3649998\t0.2084\t1\n",
      "17\t17:3658102\t3658102\t0.3667\t1\n",
      "17\t17:3661084\t3661084\t0.358\t1\n",
      "17\t17:3661964\t3661964\t0.587\t1\n",
      "17\t17:3662038\t3662038\t0.47\t1\n",
      "17\t17:7581464\t7581464\t0.878\t1\n",
      "17\t17:7581813\t7581813\t0.2425\t1\n",
      "17\t17:42480711\t42480711\t0.4726\t1\n",
      "17\t17:42501949\t42501949\t0.5811\t1\n",
      "17\t17:42537437\t42537437\t0.8183\t1\n",
      "17\t17:59623220\t59623220\t0.6717\t1\n",
      "17\t17:59670964\t59670964\t0.09579\t1\n",
      "17\t17:59692171\t59692171\t0.616\t1\n",
      "17\t17:68260815\t68260815\t0.5013\t1\n",
      "17\t17:68262585\t68262585\t0.8228\t1\n",
      "17\t17:68276733\t68276733\t0.4216\t1\n",
      "17\t17:68278422\t68278422\t0.0547\t1\n",
      "17\t17:68279252\t68279252\t0.1759\t1\n",
      "17\t17:68288666\t68288666\t0.2384\t1\n",
      "17\t17:68312607\t68312607\t0.6311\t1\n",
      "17\t17:68316097\t68316097\t0.1636\t1\n",
      "17\t17:68322325\t68322325\t0.8369\t1\n",
      "17\t17:68323143\t68323143\t0.2364\t1\n",
      "17\t17:68327781\t68327781\t0.9725\t1\n",
      "17\t17:68338167\t68338167\t0.6651\t1\n",
      "17\t17:68341562\t68341562\t0.8098\t1\n",
      "17\t17:68349104\t68349104\t0.5199\t1\n",
      "17\t17:68370405\t68370405\t0.6372\t1\n",
      "17\t17:68372487\t68372487\t0.6445\t1\n",
      "17\t17:68378729\t68378729\t0.4115\t1\n",
      "17\t17:68380509\t68380509\t0.3193\t1\n",
      "17\t17:68388866\t68388866\t0.07142\t1\n",
      "17\t17:68393883\t68393883\t0.7894\t1\n",
      "17\t17:68395135\t68395135\t0.5094\t1\n",
      "17\t17:68395173\t68395173\t0.3083\t1\n",
      "17\t17:68395685\t68395685\t0.07831\t1\n",
      "17\t17:68410774\t68410774\t0.438\t1\n",
      "17\t17:75242127\t75242127\t0.721\t1\n",
      "17\t17:75251579\t75251579\t0.05286\t1\n",
      "17\t17:75252915\t75252915\t0.5833\t1\n",
      "17\t17:75258978\t75258978\t0.7134\t1\n",
      "17\t17:80102206\t80102206\t0.1013\t1\n",
      "17\t17:80104857\t80104857\t0.2342\t1\n",
      "17\t17:80108051\t80108051\t0.3196\t1\n",
      "17\t17:80111230\t80111230\t0.5465\t1\n",
      "17\t17:80112392\t80112392\t0.9054\t1\n",
      "17\t17:80113242\t80113242\t0.1677\t1\n",
      "17\t17:80113973\t80113973\t0.462\t1\n",
      "17\t17:80114716\t80114716\t0.5606\t1\n",
      "17\t17:80119422\t80119422\t0.5394\t1\n",
      "17\t17:80208391\t80208391\t0.6983\t1\n",
      "17\t17:80208576\t80208576\t0.4291\t1\n",
      "17\t17:80210802\t80210802\t0.9822\t1\n",
      "17\t17:80213996\t80213996\t0.05689\t1\n",
      "17\t17:80215210\t80215210\t0.1167\t1\n",
      "18\t18:23520784\t23520784\t0.1405\t1\n",
      "18\t18:23539568\t23539568\t0.4879\t1\n",
      "18\t18:23560403\t23560403\t0.2953\t1\n",
      "18\t18:23573978\t23573978\t0.2301\t1\n",
      "18\t18:23579569\t23579569\t0.3909\t1\n",
      "18\t18:23582197\t23582197\t0.6222\t1\n",
      "19\t19:2103256\t2103256\t0.6522\t1\n",
      "19\t19:2113519\t2113519\t0.1957\t1\n",
      "19\t19:2120155\t2120155\t0.3364\t1\n",
      "19\t19:2127832\t2127832\t0.2398\t1\n",
      "19\t19:2132363\t2132363\t0.8523\t1\n",
      "19\t19:2159572\t2159572\t0.815\t1\n",
      "19\t19:2160530\t2160530\t0.103\t1\n",
      "19\t19:7529273\t7529273\t0.5524\t1\n",
      "19\t19:7530941\t7530941\t0.7692\t1\n",
      "19\t19:10578342\t10578342\t0.2182\t1\n",
      "19\t19:11575590\t11575590\t0.1791\t1\n",
      "19\t19:11577385\t11577385\t0.1771\t1\n",
      "19\t19:11577724\t11577724\t0.6346\t1\n",
      "19\t19:12651959\t12651959\t0.2903\t1\n",
      "19\t19:12651969\t12651969\t0.1474\t1\n",
      "19\t19:12657035\t12657035\t0.09628\t1\n",
      "19\t19:12658134\t12658134\t0.4\t1\n",
      "19\t19:12658850\t12658850\t0.3185\t1\n",
      "19\t19:12664048\t12664048\t0.4367\t1\n",
      "19\t19:12877779\t12877779\t0.4415\t1\n",
      "19\t19:12878746\t12878746\t0.7682\t1\n",
      "19\t19:16202311\t16202311\t0.9224\t1\n",
      "19\t19:16211373\t16211373\t0.6124\t1\n",
      "19\t19:16215855\t16215855\t0.1124\t1\n",
      "19\t19:16216920\t16216920\t0.4164\t1\n",
      "19\t19:16224482\t16224482\t0.1303\t1\n",
      "19\t19:50359511\t50359511\t0.3112\t1\n",
      "19\t19:50362278\t50362278\t0.1749\t1\n",
      "20\t20:45891004\t45891004\t0.07701\t1\n",
      "20\t20:45891103\t45891103\t0.1743\t1\n",
      "20\t20:45895658\t45895658\t0.08547\t1\n",
      "20\t20:45897980\t45897980\t0.6789\t1\n",
      "20\t20:58986361\t58986361\t0.3615\t1\n",
      "20\t20:58988554\t58988554\t0.335\t1\n",
      "20\t20:58988752\t58988752\t0.2324\t1\n",
      "20\t20:58995401\t58995401\t0.6584\t1\n",
      "20\t20:58998666\t58998666\t0.8796\t1\n",
      "20\t20:59003556\t59003556\t0.3509\t1\n",
      "20\t20:59004920\t59004920\t0.6187\t1\n",
      "20\t20:59005013\t59005013\t0.6908\t1\n",
      "22\t22:19180775\t19180775\t0.09397\t1\n",
      "22\t22:19208170\t19208170\t0.343\t1\n",
      "22\t22:19209105\t19209105\t0.7344\t1\n",
      "22\t22:19268675\t19268675\t0.26\t1\n",
      "22\t22:19282285\t19282285\t0.1081\t1\n",
      "22\t22:19285016\t19285016\t0.3219\t1\n",
      "22\t22:29328803\t29328803\t0.584\t1\n",
      "22\t22:29337521\t29337521\t0.2217\t1\n",
      "22\t22:29347790\t29347790\t0.08624\t1\n",
      "22\t22:29349083\t29349083\t0.5227\t1\n",
      "22\t22:29353965\t29353965\t0.3463\t1\n",
      "22\t22:29358459\t29358459\t0.7499\t1\n",
      "22\t22:29361874\t29361874\t0.8874\t1\n",
      "22\t22:29362974\t29362974\t0.8973\t1\n",
      "22\t22:29363537\t29363537\t0.4333\t1\n",
      "22\t22:29379808\t29379808\t0.2883\t1\n",
      "22\t22:37609259\t37609259\t0.1041\t1\n",
      "22\t22:37617802\t37617802\t0.5431\t1\n",
      "22\t22:37630052\t37630052\t0.5446\t1\n",
      "22\t22:37630327\t37630327\t0.1171\t1\n",
      "22\t22:42058349\t42058349\t0.4321\t1\n",
      "22\t22:42058583\t42058583\t0.6952\t1\n",
      "22\t22:42059176\t42059176\t0.2384\t1\n",
      "22\t22:42063618\t42063618\t0.8442\t1\n",
      "22\t22:42065183\t42065183\t0.1714\t1\n",
      "22\t22:42068641\t42068641\t0.7557\t1\n",
      "22\t22:50622771\t50622771\t0.675\t1\n",
      "22\t22:50623776\t50623776\t0.4391\t1\n",
      "22\t22:50624404\t50624404\t0.7944\t1\n",
      "22\t22:50625741\t50625741\t0.5163\t1\n",
      "22\t22:50626801\t50626801\t0.8063\t1\n",
      "22\t22:50626933\t50626933\t0.344\t1\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cat /home/jupyter/pathways/lysosome_PD_PRS_Chang23andMe.snp"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Get cohort information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "cohort_query = f\"\"\"\n",
    "\n",
    "SELECT\n",
    "participant_id, study\n",
    "FROM `{BQ_RELEASE_DATASET}.amp_pd_participants`\n",
    "\n",
    "\"\"\"\n",
    "\n",
    "cohort = bq_query(cohort_query)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "LBD        4586\n",
       "PPMI       1945\n",
       "PDBP       1606\n",
       "HBS        1189\n",
       "LCC         638\n",
       "Steady      334\n",
       "Sure        261\n",
       "BioFIND     213\n",
       "Name: study, dtype: int64"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cohort.study.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Write full file as output\n",
    "with open('/home/jupyter/clinical_data_v2-5/cohort.tab', 'w') as f:\n",
    "    f.write(cohort.to_csv(index=False, sep='\\t'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LBD\n",
      "PPMI\n",
      "PDBP\n",
      "HBS\n",
      "LCC\n",
      "Steady\n",
      "Sure\n",
      "BioFIND\n"
     ]
    }
   ],
   "source": [
    "cohorts = [\"LBD\", \"PPMI\", \"PDBP\", \"HBS\", \"LCC\", \"Steady\", \"Sure\", \"BioFIND\"]\n",
    "\n",
    "#Loop over cohorts - select just individuals in each cohort and write as separate output\n",
    "#This is so we can extract each cohort individuals from plink or PRSice2 files\n",
    "for COHORT in cohorts:\n",
    "    cohort_select = cohort.copy()\n",
    "    cohort_select = cohort_select[cohort_select['study'] == COHORT]\n",
    "    print(COHORT)\n",
    "    \n",
    "    with open(f'/home/jupyter/clinical_data_v2-5/{COHORT}.tab', 'w') as f:\n",
    "        f.write(cohort_select.to_csv(index=False, sep='\\t'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "participant_id\tstudy\n",
      "LB-00001\tLBD\n",
      "LB-00002\tLBD\n",
      "LB-00003\tLBD\n",
      "LB-00004\tLBD\n",
      "LB-00005\tLBD\n",
      "LB-00006\tLBD\n",
      "LB-00007\tLBD\n",
      "LB-00008\tLBD\n",
      "LB-00009\tLBD\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head /home/jupyter/clinical_data_v2-5/LBD.tab"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Standardise PRSs within each cohort"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Note that the final p-value threshold used depends on the number of SNPs - so it is not always labelled as Pt_0.05."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LBD\n",
      "LBD dataFrame is empty!\n",
      "LBD dataFrame is empty!\n",
      "LBD dataFrame is empty!\n",
      "LBD dataFrame is empty!\n",
      "LBD dataFrame is empty!\n",
      "LBD dataFrame is empty!\n",
      "LBD dataFrame is empty!\n",
      "LBD dataFrame is empty!\n",
      "PPMI\n",
      "PDBP\n",
      "HBS\n",
      "LCC\n",
      "Steady\n",
      "Sure\n",
      "BioFIND\n"
     ]
    }
   ],
   "source": [
    "from sklearn.preprocessing import StandardScaler\n",
    "\n",
    "cohorts = [\"LBD\", \"PPMI\", \"PDBP\", \"HBS\", \"LCC\", \"Steady\", \"Sure\", \"BioFIND\"]\n",
    "\n",
    "pathways = [\"adaptive_immune\", \"alpha_synuclein\", \"innate_immune\", \"lysosome\", \"endocytosis\", \"mitochondria\", \"microglia\", \"monocytes\"]\n",
    "\n",
    "for COHORT in cohorts:\n",
    "    \n",
    "    #Read in cohort file\n",
    "    cohort = pd.read_csv(f'/home/jupyter/clinical_data_v2-5/{COHORT}.tab',\n",
    "                         delim_whitespace=True)\n",
    "    \n",
    "    print(COHORT)\n",
    "    \n",
    "    for PATHWAY in pathways:\n",
    "        \n",
    "        #Read in PRS file\n",
    "        PRS = pd.read_csv(f'/home/jupyter/pathways/{PATHWAY}_PD_PRS_Chang23andMe.all_score',\n",
    "                     delim_whitespace=True)\n",
    "        \n",
    "        #Merge PRS with selected cohort - inner join\n",
    "        merged = pd.merge(cohort, PRS, left_on='participant_id', right_on='FID', how = 'inner')\n",
    "        \n",
    "        #If there is no match and merged dataframe is empty - print warning\n",
    "        #This will print multiple times - once for each pathway\n",
    "        if merged.empty:\n",
    "            print(f'{COHORT} dataFrame is empty!')\n",
    "        \n",
    "        else:\n",
    "            PRS_edit = merged.copy()\n",
    "        \n",
    "            #The last column is either Pt_1 or Pt_0.05 (or closest threshold) as this depends on the number of SNPs included\n",
    "            #If the last column is Pt_1, then remove this column, if not keep everything\n",
    "            #This is because the columns are not consistently named - it depends on the SNPs in each pathway PRS\n",
    "            if PRS_edit.columns[-1] == \"Pt_1\":\n",
    "                PRS_edit = PRS_edit.drop(\"Pt_1\", axis = 1)\n",
    "            else :\n",
    "                PRS_edit = PRS_edit\n",
    "\n",
    "            #Select just relevant columns - PRS at p value 0.05 which is now the last column\n",
    "            PRS_selected = PRS_edit.copy()\n",
    "            PRS_selected = PRS_selected.iloc[:,[0,1,-1]]\n",
    "\n",
    "            #Standardise the PRS - make new column with z-transformed PRS\n",
    "            scaler = StandardScaler()\n",
    "            PRS_selected['PRS_z'] = scaler.fit_transform(PRS_selected.iloc[:,[-1]])\n",
    "\n",
    "            #Save as file\n",
    "            with open('/home/jupyter/pathways/' + COHORT + '.' + PATHWAY + '_PRS_final.txt', 'w') as f:\n",
    "                f.write(PRS_selected.to_csv(index=False, sep='\\t'))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Make a results dataframe to store the final p threshold (the one closest to 0.05) and number of SNPs in the PRS for each pathway. This is the same for each cohort."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "prsice_results = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "pathways = [\"adaptive_immune\", \"alpha_synuclein\", \"innate_immune\", \"lysosome\", \"endocytosis\", \"mitochondria\", \"microglia\", \"monocytes\"]\n",
    "\n",
    "for index,pathway in enumerate(pathways):\n",
    "    \n",
    "    #Read in prsice file\n",
    "    prsice = pd.read_csv(f'/home/jupyter/pathways/' + pathway + '_PD_PRS_Chang23andMe.prsice',\n",
    "                 delim_whitespace=True)\n",
    "    \n",
    "    prsice_edit = prsice.copy()\n",
    "    \n",
    "    #The last row is either Pt_1 or Pt_0.05 (or closest threshold) as this depends on the number of SNPs included\n",
    "    #If the last row threshold is 1, then remove this last row, if not keep everything\n",
    "    if prsice_edit.iloc[-1,2] == 1.0 :\n",
    "        prsice_edit = prsice_edit.drop(prsice_edit.index[-1], axis = 0)\n",
    "    else :\n",
    "        prsice_edit = prsice_edit\n",
    "    \n",
    "    #From the prsice results, get the PRS threshold and the number of SNPs\n",
    "    list = prsice_edit.iloc[-1,2:4].values.tolist()\n",
    "    \n",
    "    #Add the pathway name to this list\n",
    "    list.append(pathway)\n",
    "    \n",
    "    #Now join to the main results dataframe to make a list of lists\n",
    "    prsice_results.append(list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[[0.04998, 733, 'adaptive_immune'],\n",
       " [0.04863, 75, 'alpha_synuclein'],\n",
       " [0.05, 1049, 'innate_immune'],\n",
       " [0.04966, 154, 'lysosome'],\n",
       " [0.04989, 350, 'endocytosis'],\n",
       " [0.04998, 1677, 'mitochondria'],\n",
       " [0.04995, 2001, 'microglia'],\n",
       " [0.04995, 1705, 'monocytes']]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Show the list of lists\n",
    "prsice_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Convert list to dataframe\n",
    "prsice_results_df = pd.DataFrame(prsice_results, columns = ['p_threshold_closest_to_0.05', 'n_snps', 'pathway'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>p_threshold_closest_to_0.05</th>\n",
       "      <th>n_snps</th>\n",
       "      <th>pathway</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.04998</td>\n",
       "      <td>733</td>\n",
       "      <td>adaptive_immune</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.04863</td>\n",
       "      <td>75</td>\n",
       "      <td>alpha_synuclein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.05000</td>\n",
       "      <td>1049</td>\n",
       "      <td>innate_immune</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.04966</td>\n",
       "      <td>154</td>\n",
       "      <td>lysosome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.04989</td>\n",
       "      <td>350</td>\n",
       "      <td>endocytosis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>0.04998</td>\n",
       "      <td>1677</td>\n",
       "      <td>mitochondria</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>0.04995</td>\n",
       "      <td>2001</td>\n",
       "      <td>microglia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>0.04995</td>\n",
       "      <td>1705</td>\n",
       "      <td>monocytes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   p_threshold_closest_to_0.05  n_snps          pathway\n",
       "0                      0.04998     733  adaptive_immune\n",
       "1                      0.04863      75  alpha_synuclein\n",
       "2                      0.05000    1049    innate_immune\n",
       "3                      0.04966     154         lysosome\n",
       "4                      0.04989     350      endocytosis\n",
       "5                      0.04998    1677     mitochondria\n",
       "6                      0.04995    2001        microglia\n",
       "7                      0.04995    1705        monocytes"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prsice_results_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Write as output\n",
    "with open('/home/jupyter/pathways/n_snps_in_pathwayPRS_summary.tab', 'w') as f:\n",
    "    f.write(prsice_results_df.to_csv(index=False, sep='\\t'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Get age at diagnosis data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Query clinical data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "age_dx_query = f\"\"\"\n",
    "\n",
    "SELECT \n",
    "*\n",
    "FROM `{BQ_RELEASE_DATASET}.PD_Medical_History`\n",
    "\n",
    "\"\"\"\n",
    "\n",
    "age_dx = bq_query(age_dx_query)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>GUID</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>initial_diagnosis</th>\n",
       "      <th>most_recent_diagnosis</th>\n",
       "      <th>change_in_diagnosis</th>\n",
       "      <th>change_in_diagnosis_months_after_baseline</th>\n",
       "      <th>surgery_for_parkinson_disease</th>\n",
       "      <th>pd_diagnosis_months_after_baseline</th>\n",
       "      <th>age_at_diagnosis</th>\n",
       "      <th>pd_medication_initiation_months_after_baseline</th>\n",
       "      <th>pd_medication_start_months_after_baseline</th>\n",
       "      <th>use_of_pd_medication</th>\n",
       "      <th>pd_medication_recent_use_months_after_baseline</th>\n",
       "      <th>on_levodopa</th>\n",
       "      <th>on_dopamine_agonist</th>\n",
       "      <th>on_other_pd_medications</th>\n",
       "      <th>diagnosis_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6936</th>\n",
       "      <td>SY-NIH_INVAA791MKCET</td>\n",
       "      <td>NIH_INVAA791MKCET</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Parkinson's Disease</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6937</th>\n",
       "      <td>SY-NIH_INVAA976PRYBT</td>\n",
       "      <td>NIH_INVAA976PRYBT</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Parkinson's Disease</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6938</th>\n",
       "      <td>SY-NIH_INVBC118NRWC4</td>\n",
       "      <td>NIH_INVBC118NRWC4</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Parkinson's Disease</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6939</th>\n",
       "      <td>SY-NIH_INVCW212TXMFX</td>\n",
       "      <td>NIH_INVCW212TXMFX</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Parkinson's Disease</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6940</th>\n",
       "      <td>SY-NIH_INVCX656ULCDX</td>\n",
       "      <td>NIH_INVCX656ULCDX</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Parkinson's Disease</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24079</th>\n",
       "      <td>SY-PDXZ650DHD</td>\n",
       "      <td>PDXZ650DHD</td>\n",
       "      <td>M42</td>\n",
       "      <td>42.0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>None</td>\n",
       "      <td>No</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24080</th>\n",
       "      <td>SY-PDZJ450XYU</td>\n",
       "      <td>PDZJ450XYU</td>\n",
       "      <td>M42</td>\n",
       "      <td>42.0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>None</td>\n",
       "      <td>No</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24081</th>\n",
       "      <td>SY-PDZN008XD9</td>\n",
       "      <td>PDZN008XD9</td>\n",
       "      <td>M42</td>\n",
       "      <td>42.0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>None</td>\n",
       "      <td>No</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24082</th>\n",
       "      <td>SY-PDZP715EGW</td>\n",
       "      <td>PDZP715EGW</td>\n",
       "      <td>M42</td>\n",
       "      <td>42.0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>None</td>\n",
       "      <td>No</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24083</th>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>PDZZ260HUM</td>\n",
       "      <td>M42</td>\n",
       "      <td>42.0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>None</td>\n",
       "      <td>No</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1409 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             participant_id               GUID visit_name  visit_month  \\\n",
       "6936   SY-NIH_INVAA791MKCET  NIH_INVAA791MKCET         M0          0.0   \n",
       "6937   SY-NIH_INVAA976PRYBT  NIH_INVAA976PRYBT         M0          0.0   \n",
       "6938   SY-NIH_INVBC118NRWC4  NIH_INVBC118NRWC4         M0          0.0   \n",
       "6939   SY-NIH_INVCW212TXMFX  NIH_INVCW212TXMFX         M0          0.0   \n",
       "6940   SY-NIH_INVCX656ULCDX  NIH_INVCX656ULCDX         M0          0.0   \n",
       "...                     ...                ...        ...          ...   \n",
       "24079         SY-PDXZ650DHD         PDXZ650DHD        M42         42.0   \n",
       "24080         SY-PDZJ450XYU         PDZJ450XYU        M42         42.0   \n",
       "24081         SY-PDZN008XD9         PDZN008XD9        M42         42.0   \n",
       "24082         SY-PDZP715EGW         PDZP715EGW        M42         42.0   \n",
       "24083         SY-PDZZ260HUM         PDZZ260HUM        M42         42.0   \n",
       "\n",
       "                 diagnosis initial_diagnosis most_recent_diagnosis  \\\n",
       "6936   Parkinson's Disease              None                  None   \n",
       "6937   Parkinson's Disease              None                  None   \n",
       "6938   Parkinson's Disease              None                  None   \n",
       "6939   Parkinson's Disease              None                  None   \n",
       "6940   Parkinson's Disease              None                  None   \n",
       "...                    ...               ...                   ...   \n",
       "24079                 None              None                  None   \n",
       "24080                 None              None                  None   \n",
       "24081                 None              None                  None   \n",
       "24082                 None              None                  None   \n",
       "24083                 None              None                  None   \n",
       "\n",
       "      change_in_diagnosis  change_in_diagnosis_months_after_baseline  \\\n",
       "6936                 None                                        NaN   \n",
       "6937                 None                                        NaN   \n",
       "6938                 None                                        NaN   \n",
       "6939                 None                                        NaN   \n",
       "6940                 None                                        NaN   \n",
       "...                   ...                                        ...   \n",
       "24079                None                                        NaN   \n",
       "24080                None                                        NaN   \n",
       "24081                None                                        NaN   \n",
       "24082                None                                        NaN   \n",
       "24083                None                                        NaN   \n",
       "\n",
       "      surgery_for_parkinson_disease  pd_diagnosis_months_after_baseline  \\\n",
       "6936                           None                                 NaN   \n",
       "6937                           None                                 NaN   \n",
       "6938                           None                                 NaN   \n",
       "6939                           None                                 NaN   \n",
       "6940                           None                                 NaN   \n",
       "...                             ...                                 ...   \n",
       "24079                          None                                 NaN   \n",
       "24080                          None                                 NaN   \n",
       "24081                          None                                 NaN   \n",
       "24082                          None                                 NaN   \n",
       "24083                          None                                 NaN   \n",
       "\n",
       "       age_at_diagnosis  pd_medication_initiation_months_after_baseline  \\\n",
       "6936                NaN                                             NaN   \n",
       "6937                NaN                                             NaN   \n",
       "6938                NaN                                             NaN   \n",
       "6939                NaN                                             NaN   \n",
       "6940                NaN                                             NaN   \n",
       "...                 ...                                             ...   \n",
       "24079               NaN                                             NaN   \n",
       "24080               NaN                                             NaN   \n",
       "24081               NaN                                             NaN   \n",
       "24082               NaN                                             NaN   \n",
       "24083               NaN                                             NaN   \n",
       "\n",
       "       pd_medication_start_months_after_baseline use_of_pd_medication  \\\n",
       "6936                                         NaN                 None   \n",
       "6937                                         NaN                 None   \n",
       "6938                                         NaN                 None   \n",
       "6939                                         NaN                 None   \n",
       "6940                                         NaN                 None   \n",
       "...                                          ...                  ...   \n",
       "24079                                        NaN                 None   \n",
       "24080                                        NaN                 None   \n",
       "24081                                        NaN                 None   \n",
       "24082                                        NaN                 None   \n",
       "24083                                        NaN                 None   \n",
       "\n",
       "       pd_medication_recent_use_months_after_baseline on_levodopa  \\\n",
       "6936                                              NaN        None   \n",
       "6937                                              NaN        None   \n",
       "6938                                              NaN        None   \n",
       "6939                                              NaN        None   \n",
       "6940                                              NaN        None   \n",
       "...                                               ...         ...   \n",
       "24079                                             NaN         Yes   \n",
       "24080                                             NaN         Yes   \n",
       "24081                                             NaN         Yes   \n",
       "24082                                             NaN         Yes   \n",
       "24083                                             NaN         Yes   \n",
       "\n",
       "      on_dopamine_agonist on_other_pd_medications diagnosis_type  \n",
       "6936                 None                    None           None  \n",
       "6937                 None                    None           None  \n",
       "6938                 None                    None           None  \n",
       "6939                 None                    None           None  \n",
       "6940                 None                    None           None  \n",
       "...                   ...                     ...            ...  \n",
       "24079                None                      No           None  \n",
       "24080                None                      No           None  \n",
       "24081                None                      No           None  \n",
       "24082                None                      No           None  \n",
       "24083                None                      No           None  \n",
       "\n",
       "[1409 rows x 20 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Check STEADY-PD participants - they seem to be all missing age at diagnosis\n",
    "age_dx[age_dx.participant_id.str.contains(\"SY\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>GUID</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>initial_diagnosis</th>\n",
       "      <th>most_recent_diagnosis</th>\n",
       "      <th>change_in_diagnosis</th>\n",
       "      <th>change_in_diagnosis_months_after_baseline</th>\n",
       "      <th>surgery_for_parkinson_disease</th>\n",
       "      <th>pd_diagnosis_months_after_baseline</th>\n",
       "      <th>age_at_diagnosis</th>\n",
       "      <th>pd_medication_initiation_months_after_baseline</th>\n",
       "      <th>pd_medication_start_months_after_baseline</th>\n",
       "      <th>use_of_pd_medication</th>\n",
       "      <th>pd_medication_recent_use_months_after_baseline</th>\n",
       "      <th>on_levodopa</th>\n",
       "      <th>on_dopamine_agonist</th>\n",
       "      <th>on_other_pd_medications</th>\n",
       "      <th>diagnosis_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BF-1002</td>\n",
       "      <td>PDCB969UGG</td>\n",
       "      <td>LOG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-39.0</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BF-1003</td>\n",
       "      <td>PDLW805AHT</td>\n",
       "      <td>LOG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-57.0</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BF-1004</td>\n",
       "      <td>PDKW284DYW</td>\n",
       "      <td>LOG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-76.0</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BF-1006</td>\n",
       "      <td>PDKY484YDC</td>\n",
       "      <td>LOG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-13.0</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BF-1008</td>\n",
       "      <td>PDEA056CRM</td>\n",
       "      <td>LOG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-74.0</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  participant_id        GUID visit_name  visit_month diagnosis  \\\n",
       "0        BF-1002  PDCB969UGG        LOG          NaN      None   \n",
       "1        BF-1003  PDLW805AHT        LOG          NaN      None   \n",
       "2        BF-1004  PDKW284DYW        LOG          NaN      None   \n",
       "3        BF-1006  PDKY484YDC        LOG          NaN      None   \n",
       "4        BF-1008  PDEA056CRM        LOG          NaN      None   \n",
       "\n",
       "  initial_diagnosis most_recent_diagnosis change_in_diagnosis  \\\n",
       "0              None                  None                None   \n",
       "1              None                  None                None   \n",
       "2              None                  None                None   \n",
       "3              None                  None                None   \n",
       "4              None                  None                None   \n",
       "\n",
       "   change_in_diagnosis_months_after_baseline surgery_for_parkinson_disease  \\\n",
       "0                                        NaN                          None   \n",
       "1                                        NaN                          None   \n",
       "2                                        NaN                          None   \n",
       "3                                        NaN                          None   \n",
       "4                                        NaN                          None   \n",
       "\n",
       "   pd_diagnosis_months_after_baseline  age_at_diagnosis  \\\n",
       "0                                 NaN               NaN   \n",
       "1                                 NaN               NaN   \n",
       "2                                 NaN               NaN   \n",
       "3                                 NaN               NaN   \n",
       "4                                 NaN               NaN   \n",
       "\n",
       "   pd_medication_initiation_months_after_baseline  \\\n",
       "0                                             NaN   \n",
       "1                                             NaN   \n",
       "2                                             NaN   \n",
       "3                                             NaN   \n",
       "4                                             NaN   \n",
       "\n",
       "   pd_medication_start_months_after_baseline use_of_pd_medication  \\\n",
       "0                                      -39.0                 None   \n",
       "1                                      -57.0                 None   \n",
       "2                                      -76.0                 None   \n",
       "3                                      -13.0                 None   \n",
       "4                                      -74.0                 None   \n",
       "\n",
       "   pd_medication_recent_use_months_after_baseline on_levodopa  \\\n",
       "0                                             NaN        None   \n",
       "1                                             NaN        None   \n",
       "2                                             NaN        None   \n",
       "3                                             NaN        None   \n",
       "4                                             NaN        None   \n",
       "\n",
       "  on_dopamine_agonist on_other_pd_medications diagnosis_type  \n",
       "0                None                    None           None  \n",
       "1                None                    None           None  \n",
       "2                None                    None           None  \n",
       "3                None                    None           None  \n",
       "4                None                    None           None  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "age_dx.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>GUID</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>initial_diagnosis</th>\n",
       "      <th>most_recent_diagnosis</th>\n",
       "      <th>change_in_diagnosis</th>\n",
       "      <th>change_in_diagnosis_months_after_baseline</th>\n",
       "      <th>surgery_for_parkinson_disease</th>\n",
       "      <th>pd_diagnosis_months_after_baseline</th>\n",
       "      <th>age_at_diagnosis</th>\n",
       "      <th>pd_medication_initiation_months_after_baseline</th>\n",
       "      <th>pd_medication_start_months_after_baseline</th>\n",
       "      <th>use_of_pd_medication</th>\n",
       "      <th>pd_medication_recent_use_months_after_baseline</th>\n",
       "      <th>on_levodopa</th>\n",
       "      <th>on_dopamine_agonist</th>\n",
       "      <th>on_other_pd_medications</th>\n",
       "      <th>diagnosis_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BF-1002</td>\n",
       "      <td>PDCB969UGG</td>\n",
       "      <td>LOG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-39.0</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9190</th>\n",
       "      <td>BF-1002</td>\n",
       "      <td>PDCB969UGG</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Idiopathic PD</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-61.0</td>\n",
       "      <td>61.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Yes</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     participant_id        GUID visit_name  visit_month      diagnosis  \\\n",
       "0           BF-1002  PDCB969UGG        LOG          NaN           None   \n",
       "9190        BF-1002  PDCB969UGG         M0          0.0  Idiopathic PD   \n",
       "\n",
       "     initial_diagnosis most_recent_diagnosis change_in_diagnosis  \\\n",
       "0                 None                  None                None   \n",
       "9190              None                  None                None   \n",
       "\n",
       "      change_in_diagnosis_months_after_baseline surgery_for_parkinson_disease  \\\n",
       "0                                           NaN                          None   \n",
       "9190                                        NaN                          None   \n",
       "\n",
       "      pd_diagnosis_months_after_baseline  age_at_diagnosis  \\\n",
       "0                                    NaN               NaN   \n",
       "9190                               -61.0              61.0   \n",
       "\n",
       "      pd_medication_initiation_months_after_baseline  \\\n",
       "0                                                NaN   \n",
       "9190                                             NaN   \n",
       "\n",
       "      pd_medication_start_months_after_baseline use_of_pd_medication  \\\n",
       "0                                         -39.0                 None   \n",
       "9190                                        NaN                  Yes   \n",
       "\n",
       "      pd_medication_recent_use_months_after_baseline on_levodopa  \\\n",
       "0                                                NaN        None   \n",
       "9190                                             0.0         Yes   \n",
       "\n",
       "     on_dopamine_agonist on_other_pd_medications diagnosis_type  \n",
       "0                   None                    None           None  \n",
       "9190                  No                      No           None  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "age_dx[age_dx.participant_id == \"BF-1002\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Going to use the age of diagnosis variable. There used to be a date of diagnosis variable in v2.0 but this has been removed in v2.5 of the dataset."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Note this data has multiple rows per participant as data is recorded at each visit. Want to remove duplicates but take just the most frequently reported age at diagnosis."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#First remove rows missing age at diagnosis\n",
    "age_dx_nomiss = age_dx[age_dx['age_at_diagnosis'].notnull()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>GUID</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>diagnosis</th>\n",
       "      <th>initial_diagnosis</th>\n",
       "      <th>most_recent_diagnosis</th>\n",
       "      <th>change_in_diagnosis</th>\n",
       "      <th>change_in_diagnosis_months_after_baseline</th>\n",
       "      <th>surgery_for_parkinson_disease</th>\n",
       "      <th>pd_diagnosis_months_after_baseline</th>\n",
       "      <th>age_at_diagnosis</th>\n",
       "      <th>pd_medication_initiation_months_after_baseline</th>\n",
       "      <th>pd_medication_start_months_after_baseline</th>\n",
       "      <th>use_of_pd_medication</th>\n",
       "      <th>pd_medication_recent_use_months_after_baseline</th>\n",
       "      <th>on_levodopa</th>\n",
       "      <th>on_dopamine_agonist</th>\n",
       "      <th>on_other_pd_medications</th>\n",
       "      <th>diagnosis_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5685</th>\n",
       "      <td>LC-1000006</td>\n",
       "      <td>None</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Parkinson's Disease</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-108.0</td>\n",
       "      <td>63.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5686</th>\n",
       "      <td>LC-100001</td>\n",
       "      <td>None</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Parkinson's Disease</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-36.0</td>\n",
       "      <td>71.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5689</th>\n",
       "      <td>LC-100005</td>\n",
       "      <td>None</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Parkinson's Disease</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-108.0</td>\n",
       "      <td>52.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5690</th>\n",
       "      <td>LC-100006</td>\n",
       "      <td>None</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Parkinson's Disease</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-264.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5691</th>\n",
       "      <td>LC-100008</td>\n",
       "      <td>None</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Parkinson's Disease</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>-84.0</td>\n",
       "      <td>51.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     participant_id  GUID visit_name  visit_month            diagnosis  \\\n",
       "5685     LC-1000006  None         M0          0.0  Parkinson's Disease   \n",
       "5686      LC-100001  None         M0          0.0  Parkinson's Disease   \n",
       "5689      LC-100005  None         M0          0.0  Parkinson's Disease   \n",
       "5690      LC-100006  None         M0          0.0  Parkinson's Disease   \n",
       "5691      LC-100008  None         M0          0.0  Parkinson's Disease   \n",
       "\n",
       "     initial_diagnosis most_recent_diagnosis change_in_diagnosis  \\\n",
       "5685              None                  None                None   \n",
       "5686              None                  None                None   \n",
       "5689              None                  None                None   \n",
       "5690              None                  None                None   \n",
       "5691              None                  None                None   \n",
       "\n",
       "      change_in_diagnosis_months_after_baseline surgery_for_parkinson_disease  \\\n",
       "5685                                        NaN                          None   \n",
       "5686                                        NaN                          None   \n",
       "5689                                        NaN                          None   \n",
       "5690                                        NaN                          None   \n",
       "5691                                        NaN                          None   \n",
       "\n",
       "      pd_diagnosis_months_after_baseline  age_at_diagnosis  \\\n",
       "5685                              -108.0              63.0   \n",
       "5686                               -36.0              71.0   \n",
       "5689                              -108.0              52.0   \n",
       "5690                              -264.0              50.0   \n",
       "5691                               -84.0              51.0   \n",
       "\n",
       "      pd_medication_initiation_months_after_baseline  \\\n",
       "5685                                             NaN   \n",
       "5686                                             NaN   \n",
       "5689                                             NaN   \n",
       "5690                                             NaN   \n",
       "5691                                             NaN   \n",
       "\n",
       "      pd_medication_start_months_after_baseline use_of_pd_medication  \\\n",
       "5685                                        NaN                 None   \n",
       "5686                                        NaN                 None   \n",
       "5689                                        NaN                 None   \n",
       "5690                                        NaN                 None   \n",
       "5691                                        NaN                 None   \n",
       "\n",
       "      pd_medication_recent_use_months_after_baseline on_levodopa  \\\n",
       "5685                                             NaN        None   \n",
       "5686                                             NaN        None   \n",
       "5689                                             NaN        None   \n",
       "5690                                             NaN        None   \n",
       "5691                                             NaN        None   \n",
       "\n",
       "     on_dopamine_agonist on_other_pd_medications diagnosis_type  \n",
       "5685                None                    None           None  \n",
       "5686                None                    None           None  \n",
       "5689                None                    None           None  \n",
       "5690                None                    None           None  \n",
       "5691                None                    None           None  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "age_dx_nomiss.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Note there are some individuals who report different age at diagnosis at different visits. Here we want to take the most frequently reported age at diagnosis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Now get the most frequent age at diagnosis reported for each participant (the mode)\n",
    "age_dx_nomiss_nodupli = age_dx_nomiss.groupby('participant_id').age_at_diagnosis.agg(lambda x : x.mode()[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "participant_id\n",
       "BF-1002    61.0\n",
       "BF-1003    56.0\n",
       "BF-1004    55.0\n",
       "BF-1006    52.0\n",
       "BF-1008    65.0\n",
       "Name: age_at_diagnosis, dtype: float64"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "age_dx_nomiss_nodupli.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Extract just PD cases (PD at study entry and latest diagnosis, excluding known genetic PD cases). This is the same list of individuals we have extracted from the genetic data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>age_at_diagnosis</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BF-1002</td>\n",
       "      <td>BF-1002</td>\n",
       "      <td>61.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BF-1003</td>\n",
       "      <td>BF-1003</td>\n",
       "      <td>56.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BF-1004</td>\n",
       "      <td>BF-1004</td>\n",
       "      <td>55.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BF-1006</td>\n",
       "      <td>BF-1006</td>\n",
       "      <td>52.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BF-1008</td>\n",
       "      <td>BF-1008</td>\n",
       "      <td>65.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       FID      IID  age_at_diagnosis\n",
       "0  BF-1002  BF-1002              61.0\n",
       "1  BF-1003  BF-1003              56.0\n",
       "2  BF-1004  BF-1004              55.0\n",
       "3  BF-1006  BF-1006              52.0\n",
       "4  BF-1008  BF-1008              65.0"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PDcases_nongenetic = pd.read_csv(f'/home/jupyter/clinical_data_v2-5/PD_cases_start_end_dx_nongenetic.tab',\n",
    "                 delimiter=\"\\t\")\n",
    "\n",
    "PDcases_agedx = pd.merge(PDcases_nongenetic, age_dx_nomiss_nodupli, left_on='FID', right_on='participant_id', how = 'left')\n",
    "\n",
    "PDcases_agedx.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Check if there are any cases missing age at diagnosis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>age_at_diagnosis</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1281</th>\n",
       "      <td>LC-130003</td>\n",
       "      <td>LC-130003</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1284</th>\n",
       "      <td>LC-180003</td>\n",
       "      <td>LC-180003</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1285</th>\n",
       "      <td>LC-190003</td>\n",
       "      <td>LC-190003</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1293</th>\n",
       "      <td>LC-280003</td>\n",
       "      <td>LC-280003</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1297</th>\n",
       "      <td>LC-340003</td>\n",
       "      <td>LC-340003</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2831</th>\n",
       "      <td>SY-PDZW544CEL</td>\n",
       "      <td>SY-PDZW544CEL</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2832</th>\n",
       "      <td>SY-PDZX554VNB</td>\n",
       "      <td>SY-PDZX554VNB</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2833</th>\n",
       "      <td>SY-PDZX943HWN</td>\n",
       "      <td>SY-PDZX943HWN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2834</th>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2835</th>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>365 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                FID            IID  age_at_diagnosis\n",
       "1281      LC-130003      LC-130003               NaN\n",
       "1284      LC-180003      LC-180003               NaN\n",
       "1285      LC-190003      LC-190003               NaN\n",
       "1293      LC-280003      LC-280003               NaN\n",
       "1297      LC-340003      LC-340003               NaN\n",
       "...             ...            ...               ...\n",
       "2831  SY-PDZW544CEL  SY-PDZW544CEL               NaN\n",
       "2832  SY-PDZX554VNB  SY-PDZX554VNB               NaN\n",
       "2833  SY-PDZX943HWN  SY-PDZX943HWN               NaN\n",
       "2834  SY-PDZY968RFA  SY-PDZY968RFA               NaN\n",
       "2835  SY-PDZZ260HUM  SY-PDZZ260HUM               NaN\n",
       "\n",
       "[365 rows x 3 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PDcases_agedx[PDcases_agedx['age_at_diagnosis'].isnull()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Write as output\n",
    "with open('/home/jupyter/clinical_data_v2-5/PD_cases_agedx.tab', 'w') as f:\n",
    "    f.write(PDcases_agedx.to_csv(index=False, sep='\\t'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Generate genetic principal components to include as covariates"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Generate genetic PCs in each cohort separately. Excluding LBD as we excluded these samples from the genetic dataset."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "First prune for independent SNPs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/PPMI.pruning.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --indep-pairwise 50 5 0.01\n",
      "  --keep-fam /home/jupyter/clinical_data_v2-5/PPMI.tab\n",
      "  --out /home/jupyter/genetic_data_v2-5/PPMI.pruning\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:46:49 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--keep-fam: 465 samples remaining.\n",
      "465 samples (165 females, 300 males; 465 founders) remaining after main\n",
      "filters.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "--indep-pairwise (13 compute threads): 0%\b\b10%\b\b\b20%\b\b\b30%\b\b\b40%\b\b\b50%\b\b\b60%\b\b\b71%\b\b\b81%\b\b\b91%\b\b\b7575491/7737986 variants removed.\n",
      "Variant lists written to /home/jupyter/genetic_data_v2-5/PPMI.pruning.prune.in\n",
      "and /home/jupyter/genetic_data_v2-5/PPMI.pruning.prune.out .\n",
      "End time: Thu Jun 23 11:47:34 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/PDBP.pruning.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --indep-pairwise 50 5 0.01\n",
      "  --keep-fam /home/jupyter/clinical_data_v2-5/PDBP.tab\n",
      "  --out /home/jupyter/genetic_data_v2-5/PDBP.pruning\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:47:34 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--keep-fam: 804 samples remaining.\n",
      "804 samples (290 females, 514 males; 804 founders) remaining after main\n",
      "filters.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "--indep-pairwise (13 compute threads): 0%\b\b10%\b\b\b20%\b\b\b30%\b\b\b40%\b\b\b50%\b\b\b60%\b\b\b71%\b\b\b81%\b\b\b91%\b\b\b7573395/7737986 variants removed.\n",
      "Variant lists written to /home/jupyter/genetic_data_v2-5/PDBP.pruning.prune.in\n",
      "and /home/jupyter/genetic_data_v2-5/PDBP.pruning.prune.out .\n",
      "End time: Thu Jun 23 11:47:41 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/HBS.pruning.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --indep-pairwise 50 5 0.01\n",
      "  --keep-fam /home/jupyter/clinical_data_v2-5/HBS.tab\n",
      "  --out /home/jupyter/genetic_data_v2-5/HBS.pruning\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:47:41 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--keep-fam: 616 samples remaining.\n",
      "616 samples (221 females, 395 males; 616 founders) remaining after main\n",
      "filters.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "--indep-pairwise (13 compute threads): 0%\b\b10%\b\b\b20%\b\b\b30%\b\b\b40%\b\b\b50%\b\b\b60%\b\b\b71%\b\b\b81%\b\b\b91%\b\b\b7572658/7737986 variants removed.\n",
      "Variant lists written to /home/jupyter/genetic_data_v2-5/HBS.pruning.prune.in\n",
      "and /home/jupyter/genetic_data_v2-5/HBS.pruning.prune.out .\n",
      "End time: Thu Jun 23 11:47:47 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/LCC.pruning.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --indep-pairwise 50 5 0.01\n",
      "  --keep-fam /home/jupyter/clinical_data_v2-5/LCC.tab\n",
      "  --out /home/jupyter/genetic_data_v2-5/LCC.pruning\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:47:47 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--keep-fam: 79 samples remaining.\n",
      "79 samples (29 females, 50 males; 79 founders) remaining after main filters.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "--indep-pairwise (13 compute threads): 0%\b\b10%\b\b\b20%\b\b\b30%\b\b\b40%\b\b\b50%\b\b\b60%\b\b\b71%\b\b\b81%\b\b\b91%\b\b\b7606542/7737986 variants removed.\n",
      "Variant lists written to /home/jupyter/genetic_data_v2-5/LCC.pruning.prune.in\n",
      "and /home/jupyter/genetic_data_v2-5/LCC.pruning.prune.out .\n",
      "End time: Thu Jun 23 11:47:53 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/Steady.pruning.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --indep-pairwise 50 5 0.01\n",
      "  --keep-fam /home/jupyter/clinical_data_v2-5/Steady.tab\n",
      "  --out /home/jupyter/genetic_data_v2-5/Steady.pruning\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:47:53 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--keep-fam: 307 samples remaining.\n",
      "307 samples (93 females, 214 males; 307 founders) remaining after main filters.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "--indep-pairwise (13 compute threads): 0%\b\b10%\b\b\b20%\b\b\b30%\b\b\b40%\b\b\b50%\b\b\b60%\b\b\b71%\b\b\b81%\b\b\b91%\b\b\b7582685/7737986 variants removed.\n",
      "Variant lists written to\n",
      "/home/jupyter/genetic_data_v2-5/Steady.pruning.prune.in and\n",
      "/home/jupyter/genetic_data_v2-5/Steady.pruning.prune.out .\n",
      "End time: Thu Jun 23 11:47:59 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/Sure.pruning.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --indep-pairwise 50 5 0.01\n",
      "  --keep-fam /home/jupyter/clinical_data_v2-5/Sure.tab\n",
      "  --out /home/jupyter/genetic_data_v2-5/Sure.pruning\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:48:00 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--keep-fam: 247 samples remaining.\n",
      "247 samples (120 females, 127 males; 247 founders) remaining after main\n",
      "filters.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "--indep-pairwise (13 compute threads): 0%\b\b10%\b\b\b20%\b\b\b30%\b\b\b40%\b\b\b50%\b\b\b60%\b\b\b71%\b\b\b81%\b\b\b91%\b\b\b7586472/7737986 variants removed.\n",
      "Variant lists written to /home/jupyter/genetic_data_v2-5/Sure.pruning.prune.in\n",
      "and /home/jupyter/genetic_data_v2-5/Sure.pruning.prune.out .\n",
      "End time: Thu Jun 23 11:48:06 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/BioFIND.pruning.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --indep-pairwise 50 5 0.01\n",
      "  --keep-fam /home/jupyter/clinical_data_v2-5/BioFIND.tab\n",
      "  --out /home/jupyter/genetic_data_v2-5/BioFIND.pruning\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:48:06 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--keep-fam: 92 samples remaining.\n",
      "92 samples (33 females, 59 males; 92 founders) remaining after main filters.\n",
      "Calculating allele frequencies... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "--indep-pairwise (13 compute threads): 0%\b\b10%\b\b\b20%\b\b\b30%\b\b\b40%\b\b\b50%\b\b\b60%\b\b\b71%\b\b\b81%\b\b\b91%\b\b\b7603955/7737986 variants removed.\n",
      "Variant lists written to\n",
      "/home/jupyter/genetic_data_v2-5/BioFIND.pruning.prune.in and\n",
      "/home/jupyter/genetic_data_v2-5/BioFIND.pruning.prune.out .\n",
      "End time: Thu Jun 23 11:48:11 2022\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "for COHORT in PPMI PDBP HBS LCC Steady Sure BioFIND\n",
    "do\n",
    "/home/jupyter/plink2 --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf \\\n",
    "                --keep-fam /home/jupyter/clinical_data_v2-5/\"$COHORT\".tab \\\n",
    "                --indep-pairwise 50 5 0.01 \\\n",
    "                --out /home/jupyter/genetic_data_v2-5/\"$COHORT\".pruning \\\n",
    "                --threads 16\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Extract just independent SNPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/PPMI.pruned.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --extract /home/jupyter/genetic_data_v2-5/PPMI.pruning.prune.in\n",
      "  --keep-fam /home/jupyter/clinical_data_v2-5/PPMI.tab\n",
      "  --make-bed\n",
      "  --out /home/jupyter/genetic_data_v2-5/PPMI.pruned\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:49:03 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 162495 variants remaining.\n",
      "--keep-fam: 465 samples remaining.\n",
      "465 samples (165 females, 300 males; 465 founders) remaining after main\n",
      "filters.\n",
      "162495 variants remaining after main filters.\n",
      "Writing /home/jupyter/genetic_data_v2-5/PPMI.pruned.fam ... done.\n",
      "Writing /home/jupyter/genetic_data_v2-5/PPMI.pruned.bim ... done.\n",
      "Writing /home/jupyter/genetic_data_v2-5/PPMI.pruned.bed ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "End time: Thu Jun 23 11:49:04 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/PDBP.pruned.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --extract /home/jupyter/genetic_data_v2-5/PDBP.pruning.prune.in\n",
      "  --keep-fam /home/jupyter/clinical_data_v2-5/PDBP.tab\n",
      "  --make-bed\n",
      "  --out /home/jupyter/genetic_data_v2-5/PDBP.pruned\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:49:04 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 164591 variants remaining.\n",
      "--keep-fam: 804 samples remaining.\n",
      "804 samples (290 females, 514 males; 804 founders) remaining after main\n",
      "filters.\n",
      "164591 variants remaining after main filters.\n",
      "Writing /home/jupyter/genetic_data_v2-5/PDBP.pruned.fam ... done.\n",
      "Writing /home/jupyter/genetic_data_v2-5/PDBP.pruned.bim ... done.\n",
      "Writing /home/jupyter/genetic_data_v2-5/PDBP.pruned.bed ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "End time: Thu Jun 23 11:49:06 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/HBS.pruned.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --extract /home/jupyter/genetic_data_v2-5/HBS.pruning.prune.in\n",
      "  --keep-fam /home/jupyter/clinical_data_v2-5/HBS.tab\n",
      "  --make-bed\n",
      "  --out /home/jupyter/genetic_data_v2-5/HBS.pruned\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:49:06 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 165328 variants remaining.\n",
      "--keep-fam: 616 samples remaining.\n",
      "616 samples (221 females, 395 males; 616 founders) remaining after main\n",
      "filters.\n",
      "165328 variants remaining after main filters.\n",
      "Writing /home/jupyter/genetic_data_v2-5/HBS.pruned.fam ... done.\n",
      "Writing /home/jupyter/genetic_data_v2-5/HBS.pruned.bim ... done.\n",
      "Writing /home/jupyter/genetic_data_v2-5/HBS.pruned.bed ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "End time: Thu Jun 23 11:49:08 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/LCC.pruned.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --extract /home/jupyter/genetic_data_v2-5/LCC.pruning.prune.in\n",
      "  --keep-fam /home/jupyter/clinical_data_v2-5/LCC.tab\n",
      "  --make-bed\n",
      "  --out /home/jupyter/genetic_data_v2-5/LCC.pruned\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:49:08 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 131444 variants remaining.\n",
      "--keep-fam: 79 samples remaining.\n",
      "79 samples (29 females, 50 males; 79 founders) remaining after main filters.\n",
      "131444 variants remaining after main filters.\n",
      "Writing /home/jupyter/genetic_data_v2-5/LCC.pruned.fam ... done.\n",
      "Writing /home/jupyter/genetic_data_v2-5/LCC.pruned.bim ... done.\n",
      "Writing /home/jupyter/genetic_data_v2-5/LCC.pruned.bed ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "End time: Thu Jun 23 11:49:09 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/Steady.pruned.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --extract /home/jupyter/genetic_data_v2-5/Steady.pruning.prune.in\n",
      "  --keep-fam /home/jupyter/clinical_data_v2-5/Steady.tab\n",
      "  --make-bed\n",
      "  --out /home/jupyter/genetic_data_v2-5/Steady.pruned\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:49:09 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 155301 variants remaining.\n",
      "--keep-fam: 307 samples remaining.\n",
      "307 samples (93 females, 214 males; 307 founders) remaining after main filters.\n",
      "155301 variants remaining after main filters.\n",
      "Writing /home/jupyter/genetic_data_v2-5/Steady.pruned.fam ... done.\n",
      "Writing /home/jupyter/genetic_data_v2-5/Steady.pruned.bim ... done.\n",
      "Writing /home/jupyter/genetic_data_v2-5/Steady.pruned.bed ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "End time: Thu Jun 23 11:49:11 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/Sure.pruned.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --extract /home/jupyter/genetic_data_v2-5/Sure.pruning.prune.in\n",
      "  --keep-fam /home/jupyter/clinical_data_v2-5/Sure.tab\n",
      "  --make-bed\n",
      "  --out /home/jupyter/genetic_data_v2-5/Sure.pruned\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:49:11 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 151514 variants remaining.\n",
      "--keep-fam: 247 samples remaining.\n",
      "247 samples (120 females, 127 males; 247 founders) remaining after main\n",
      "filters.\n",
      "151514 variants remaining after main filters.\n",
      "Writing /home/jupyter/genetic_data_v2-5/Sure.pruned.fam ... done.\n",
      "Writing /home/jupyter/genetic_data_v2-5/Sure.pruned.bim ... done.\n",
      "Writing /home/jupyter/genetic_data_v2-5/Sure.pruned.bed ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "End time: Thu Jun 23 11:49:13 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/BioFIND.pruned.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf\n",
      "  --extract /home/jupyter/genetic_data_v2-5/BioFIND.pruning.prune.in\n",
      "  --keep-fam /home/jupyter/clinical_data_v2-5/BioFIND.tab\n",
      "  --make-bed\n",
      "  --out /home/jupyter/genetic_data_v2-5/BioFIND.pruned\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:49:13 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "2610 samples (951 females, 1659 males; 2610 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.fam.\n",
      "7737986 variants loaded from\n",
      "/home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf.bim.\n",
      "Note: No phenotype data present.\n",
      "--extract: 134031 variants remaining.\n",
      "--keep-fam: 92 samples remaining.\n",
      "92 samples (33 females, 59 males; 92 founders) remaining after main filters.\n",
      "134031 variants remaining after main filters.\n",
      "Writing /home/jupyter/genetic_data_v2-5/BioFIND.pruned.fam ... done.\n",
      "Writing /home/jupyter/genetic_data_v2-5/BioFIND.pruned.bim ... done.\n",
      "Writing /home/jupyter/genetic_data_v2-5/BioFIND.pruned.bed ... 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "End time: Thu Jun 23 11:49:14 2022\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "for COHORT in PPMI PDBP HBS LCC Steady Sure BioFIND\n",
    "do\n",
    "/home/jupyter/plink2 --bfile /home/jupyter/genetic_data_v2-5/merged.PDonly.updated_sex.mind_0.05.het.binary.xsplit.nongenetic.no_dupli.auto_snps.european.pihat.geno.missing_hap.hwe.maf \\\n",
    "                --keep-fam /home/jupyter/clinical_data_v2-5/\"$COHORT\".tab \\\n",
    "                --extract /home/jupyter/genetic_data_v2-5/\"$COHORT\".pruning.prune.in \\\n",
    "                --make-bed \\\n",
    "                --out /home/jupyter/genetic_data_v2-5/\"$COHORT\".pruned \\\n",
    "                --threads 16\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Perform PCA on independent SNPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/PPMI.pca.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/PPMI.pruned\n",
      "  --out /home/jupyter/genetic_data_v2-5/PPMI.pca\n",
      "  --pca\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:49:35 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "465 samples (165 females, 300 males; 465 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/PPMI.pruned.fam.\n",
      "162495 variants loaded from /home/jupyter/genetic_data_v2-5/PPMI.pruned.bim.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... 0%\b\b40%\b\b\b80%\b\b\bdone.\n",
      "Constructing GRM: 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "Correcting for missingness... 0%\b\b9%\b\b18%\b\b\b28%\b\b\b37%\b\b\b47%\b\b\b56%\b\b\b66%\b\b\b75%\b\b\b85%\b\b\b94%\b\b\bdone.\n",
      "Extracting eigenvalues and eigenvectors... done.\n",
      "--pca: Eigenvectors written to\n",
      "/home/jupyter/genetic_data_v2-5/PPMI.pca.eigenvec , and eigenvalues written to\n",
      "/home/jupyter/genetic_data_v2-5/PPMI.pca.eigenval .\n",
      "End time: Thu Jun 23 11:49:36 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/PDBP.pca.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/PDBP.pruned\n",
      "  --out /home/jupyter/genetic_data_v2-5/PDBP.pca\n",
      "  --pca\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:49:36 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "804 samples (290 females, 514 males; 804 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/PDBP.pruned.fam.\n",
      "164591 variants loaded from /home/jupyter/genetic_data_v2-5/PDBP.pruned.bim.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... 0%\b\b39%\b\b\b79%\b\b\bdone.\n",
      "Constructing GRM: 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "Correcting for missingness... 0%\b\b5%\b\b11%\b\b\b16%\b\b\b22%\b\b\b27%\b\b\b33%\b\b\b38%\b\b\b44%\b\b\b49%\b\b\b55%\b\b\b60%\b\b\b66%\b\b\b71%\b\b\b77%\b\b\b82%\b\b\b88%\b\b\b93%\b\b\b99%\b\b\bdone.\n",
      "Extracting eigenvalues and eigenvectors... done.\n",
      "--pca: Eigenvectors written to\n",
      "/home/jupyter/genetic_data_v2-5/PDBP.pca.eigenvec , and eigenvalues written to\n",
      "/home/jupyter/genetic_data_v2-5/PDBP.pca.eigenval .\n",
      "End time: Thu Jun 23 11:49:36 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/HBS.pca.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/HBS.pruned\n",
      "  --out /home/jupyter/genetic_data_v2-5/HBS.pca\n",
      "  --pca\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:49:36 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "616 samples (221 females, 395 males; 616 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/HBS.pruned.fam.\n",
      "165328 variants loaded from /home/jupyter/genetic_data_v2-5/HBS.pruned.bim.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... 0%\b\b39%\b\b\b79%\b\b\bdone.\n",
      "Constructing GRM: 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "Correcting for missingness... 0%\b\b5%\b\b11%\b\b\b16%\b\b\b22%\b\b\b27%\b\b\b33%\b\b\b38%\b\b\b44%\b\b\b49%\b\b\b55%\b\b\b60%\b\b\b66%\b\b\b71%\b\b\b77%\b\b\b82%\b\b\b88%\b\b\b93%\b\b\b99%\b\b\bdone.\n",
      "Extracting eigenvalues and eigenvectors... done.\n",
      "--pca: Eigenvectors written to /home/jupyter/genetic_data_v2-5/HBS.pca.eigenvec\n",
      ", and eigenvalues written to /home/jupyter/genetic_data_v2-5/HBS.pca.eigenval .\n",
      "End time: Thu Jun 23 11:49:37 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/LCC.pca.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/LCC.pruned\n",
      "  --out /home/jupyter/genetic_data_v2-5/LCC.pca\n",
      "  --pca\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:49:37 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "79 samples (29 females, 50 males; 79 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/LCC.pruned.fam.\n",
      "131444 variants loaded from /home/jupyter/genetic_data_v2-5/LCC.pruned.bim.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... 0%\b\b49%\b\b\b99%\b\b\bdone.\n",
      "Constructing GRM: 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "Correcting for missingness... 0%\b\b53%\b\b\bdone.\n",
      "Extracting eigenvalues and eigenvectors... done.\n",
      "--pca: Eigenvectors written to /home/jupyter/genetic_data_v2-5/LCC.pca.eigenvec\n",
      ", and eigenvalues written to /home/jupyter/genetic_data_v2-5/LCC.pca.eigenval .\n",
      "End time: Thu Jun 23 11:49:37 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/Steady.pca.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/Steady.pruned\n",
      "  --out /home/jupyter/genetic_data_v2-5/Steady.pca\n",
      "  --pca\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:49:37 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "307 samples (93 females, 214 males; 307 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/Steady.pruned.fam.\n",
      "155301 variants loaded from /home/jupyter/genetic_data_v2-5/Steady.pruned.bim.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... 0%\b\b42%\b\b\b84%\b\b\bdone.\n",
      "Constructing GRM: 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "Correcting for missingness... 0%\b\b5%\b\b10%\b\b\b16%\b\b\b21%\b\b\b27%\b\b\b32%\b\b\b38%\b\b\b43%\b\b\b48%\b\b\b54%\b\b\b59%\b\b\b65%\b\b\b70%\b\b\b76%\b\b\b81%\b\b\b86%\b\b\b92%\b\b\b97%\b\b\bdone.\n",
      "Extracting eigenvalues and eigenvectors... done.\n",
      "--pca: Eigenvectors written to\n",
      "/home/jupyter/genetic_data_v2-5/Steady.pca.eigenvec , and eigenvalues written\n",
      "to /home/jupyter/genetic_data_v2-5/Steady.pca.eigenval .\n",
      "End time: Thu Jun 23 11:49:37 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/Sure.pca.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/Sure.pruned\n",
      "  --out /home/jupyter/genetic_data_v2-5/Sure.pca\n",
      "  --pca\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:49:37 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "247 samples (120 females, 127 males; 247 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/Sure.pruned.fam.\n",
      "151514 variants loaded from /home/jupyter/genetic_data_v2-5/Sure.pruned.bim.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... 0%\b\b43%\b\b\b86%\b\b\bdone.\n",
      "Constructing GRM: 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b91%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "Correcting for missingness... 0%\b\b18%\b\b\b37%\b\b\b55%\b\b\b74%\b\b\b93%\b\b\bdone.\n",
      "Extracting eigenvalues and eigenvectors... done.\n",
      "--pca: Eigenvectors written to\n",
      "/home/jupyter/genetic_data_v2-5/Sure.pca.eigenvec , and eigenvalues written to\n",
      "/home/jupyter/genetic_data_v2-5/Sure.pca.eigenval .\n",
      "End time: Thu Jun 23 11:49:38 2022\n",
      "PLINK v2.00a3LM AVX2 Intel (26 Aug 2021)       www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2021 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /home/jupyter/genetic_data_v2-5/BioFIND.pca.log.\n",
      "Options in effect:\n",
      "  --bfile /home/jupyter/genetic_data_v2-5/BioFIND.pruned\n",
      "  --out /home/jupyter/genetic_data_v2-5/BioFIND.pca\n",
      "  --pca\n",
      "  --threads 16\n",
      "\n",
      "Start time: Thu Jun 23 11:49:38 2022\n",
      "60292 MiB RAM detected; reserving 30146 MiB for main workspace.\n",
      "Using up to 16 threads (change this with --threads).\n",
      "92 samples (33 females, 59 males; 92 founders) loaded from\n",
      "/home/jupyter/genetic_data_v2-5/BioFIND.pruned.fam.\n",
      "134031 variants loaded from /home/jupyter/genetic_data_v2-5/BioFIND.pruned.bim.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... 0%\b\b48%\b\b\b97%\b\b\bdone.\n",
      "Constructing GRM: 0%\b\b1%\b\b2%\b\b3%\b\b4%\b\b5%\b\b6%\b\b7%\b\b8%\b\b9%\b\b10%\b\b\b11%\b\b\b12%\b\b\b12%\b\b\b13%\b\b\b14%\b\b\b15%\b\b\b16%\b\b\b17%\b\b\b18%\b\b\b19%\b\b\b20%\b\b\b21%\b\b\b22%\b\b\b23%\b\b\b24%\b\b\b25%\b\b\b25%\b\b\b26%\b\b\b27%\b\b\b28%\b\b\b29%\b\b\b30%\b\b\b31%\b\b\b32%\b\b\b33%\b\b\b34%\b\b\b35%\b\b\b36%\b\b\b37%\b\b\b38%\b\b\b38%\b\b\b39%\b\b\b40%\b\b\b41%\b\b\b42%\b\b\b43%\b\b\b44%\b\b\b45%\b\b\b46%\b\b\b47%\b\b\b48%\b\b\b49%\b\b\b50%\b\b\b51%\b\b\b51%\b\b\b52%\b\b\b53%\b\b\b54%\b\b\b55%\b\b\b56%\b\b\b57%\b\b\b58%\b\b\b59%\b\b\b60%\b\b\b61%\b\b\b62%\b\b\b63%\b\b\b64%\b\b\b64%\b\b\b65%\b\b\b66%\b\b\b67%\b\b\b68%\b\b\b69%\b\b\b70%\b\b\b71%\b\b\b72%\b\b\b73%\b\b\b74%\b\b\b75%\b\b\b76%\b\b\b77%\b\b\b77%\b\b\b78%\b\b\b79%\b\b\b80%\b\b\b81%\b\b\b82%\b\b\b83%\b\b\b84%\b\b\b85%\b\b\b86%\b\b\b87%\b\b\b88%\b\b\b89%\b\b\b90%\b\b\b90%\b\b\b91%\b\b\b92%\b\b\b93%\b\b\b94%\b\b\b95%\b\b\b96%\b\b\b97%\b\b\b98%\b\b\b99%\b\b\bdone.\n",
      "Correcting for missingness... 0%\b\b47%\b\b\b94%\b\b\bdone.\n",
      "Extracting eigenvalues and eigenvectors... done.\n",
      "--pca: Eigenvectors written to\n",
      "/home/jupyter/genetic_data_v2-5/BioFIND.pca.eigenvec , and eigenvalues written\n",
      "to /home/jupyter/genetic_data_v2-5/BioFIND.pca.eigenval .\n",
      "End time: Thu Jun 23 11:49:38 2022\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "for COHORT in PPMI PDBP HBS LCC Steady Sure BioFIND\n",
    "do\n",
    "/home/jupyter/plink2 --bfile /home/jupyter/genetic_data_v2-5/\"$COHORT\".pruned \\\n",
    "                --pca \\\n",
    "                --out /home/jupyter/genetic_data_v2-5/\"$COHORT\".pca \\\n",
    "                --threads 16\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Prepare for R analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%bash\n",
    "mkdir -p /home/jupyter/R_packages\n",
    "#Make directory R_packages if it does not already exist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "pack <- \"/home/jupyter/R_packages\"\n",
    "\n",
    "#COMMENT OUT PACKAGES ALREADY INSTALLED\n",
    "\n",
    "#install.packages(\"survival\",lib = pack)\n",
    "#install.packages(\"survminer\", lib = pack)\n",
    "#install.packages(\"dplyr\", lib = pack)\n",
    "#install.packages(\"data.table\", lib = pack)\n",
    "#install.packages(\"tidyr\", lib = pack)\n",
    "#install.packages(\"ggpubr\", lib = pack) \n",
    "#install.packages(\"ggtext\", lib = pack)\n",
    "#install.packages(\"lme4\", lib = pack)\n",
    "#install.packages(\"ggplot2\", lib = pack)\n",
    "#install.packages(\"lmerTest\", lib = pack)\n",
    "#install.packages(\"tidyr\", lib = pack)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "hidden": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: \n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "R[write to console]: The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "R[write to console]: The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "R[write to console]: \n",
      "Attaching package: ‘ggplot2’\n",
      "\n",
      "\n",
      "R[write to console]: The following object is masked _by_ ‘.GlobalEnv’:\n",
      "\n",
      "    expand_scale\n",
      "\n",
      "\n",
      "R[write to console]: Loading required package: Matrix\n",
      "\n",
      "R[write to console]: \n",
      "Attaching package: ‘lmerTest’\n",
      "\n",
      "\n",
      "R[write to console]: The following object is masked from ‘package:lme4’:\n",
      "\n",
      "    lmer\n",
      "\n",
      "\n",
      "R[write to console]: The following object is masked from ‘package:stats’:\n",
      "\n",
      "    step\n",
      "\n",
      "\n",
      "R[write to console]: \n",
      "Attaching package: ‘tidyr’\n",
      "\n",
      "\n",
      "R[write to console]: The following objects are masked from ‘package:Matrix’:\n",
      "\n",
      "    expand, pack, unpack\n",
      "\n",
      "\n",
      "R[write to console]: \n",
      "Attaching package: ‘survminer’\n",
      "\n",
      "\n",
      "R[write to console]: The following object is masked from ‘package:survival’:\n",
      "\n",
      "    myeloma\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "pack <- \"/home/jupyter/R_packages\"\n",
    "library(dplyr, lib.loc = pack)\n",
    "library(ggplot2, lib.loc = pack)\n",
    "library(lme4, lib.loc = pack)\n",
    "library(lmerTest, lib.loc = pack)\n",
    "library(tidyr, lib.loc = pack)\n",
    "library(ggpubr, lib.loc = pack)\n",
    "library(survival,lib.loc = pack)\n",
    "library(survminer, lib.loc = pack)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Analyse AAO vs pathways in R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "#Make list of pathways\n",
    "pathways <- c(\"adaptive_immune\", \"alpha_synuclein\", \"innate_immune\", \"lysosome\", \"endocytosis\",\n",
    "              \"microglia\", \"monocytes\", \"mitochondria\")\n",
    "\n",
    "#Make list of cohorts - excluding LBD as these samples have been excluded from the genetic dataset\n",
    "cohorts <- c(\"PPMI\", \"PDBP\", \"HBS\", \"LCC\", \"Steady\", \"Sure\", \"BioFIND\")\n",
    "\n",
    "#Make function to analyse\n",
    "analyse_aao <- function(pathway, cohort) {\n",
    "    \n",
    "        #Read in data for PRS, age at diagnosis, cohort, and genetic PCs\n",
    "        PRS <- read.table(paste(\"/home/jupyter/pathways/\", cohort, \".\", pathway, \"_PRS_final.txt\", sep = \"\"), \n",
    "                          sep = \"\\t\", header = T)\n",
    "\n",
    "        aao <- read.table(\"/home/jupyter/clinical_data_v2-5/PD_cases_agedx.tab\", sep = \"\\t\", header = T)\n",
    "\n",
    "        demo <- read.table(\"/home/jupyter/clinical_data_v2-5/PD_cases_start_end_dx_sex.tab\", sep = \"\\t\", header = T)\n",
    "\n",
    "        #Read in genetic PCs\n",
    "        pcs <- read.table(paste(\"/home/jupyter/genetic_data_v2-5/\", cohort, \".pca.eigenvec\", sep = \"\"))\n",
    "        colnames(pcs) <- c(\"FID\", \"IID\", paste(\"PC\", 1:10, sep = \"\"))\n",
    "\n",
    "        #Merge\n",
    "        merged <- PRS %>%\n",
    "        inner_join(aao, by = c(\"participant_id\" = \"IID\")) %>%\n",
    "        inner_join(demo, by = c(\"participant_id\" = \"IID\")) %>%\n",
    "        inner_join(pcs, by = c(\"participant_id\" = \"IID\"))\n",
    "        \n",
    "        #Select just relevant columns\n",
    "        merged_selected <- merged %>%\n",
    "            select(participant_id, study, PRS_z, sex, age_at_diagnosis, PC1:PC5)\n",
    "        \n",
    "        #Remove individuals missing any rows\n",
    "        merged_selected_nomiss <- merged_selected %>%\n",
    "            drop_na()\n",
    "    \n",
    "        N <- dim(merged_selected_nomiss)[1]\n",
    "                \n",
    "        #If the final dataframe is not empty, run GLM\n",
    "        if (dim(merged_selected_nomiss)[1] != 0) {\n",
    "            \n",
    "            #Run GLM\n",
    "            glm_AAO_PRS <- glm(age_at_diagnosis ~ PRS_z + sex + PC1 + PC2 + PC3 + PC4 + PC5,\n",
    "                          data = merged_selected_nomiss)\n",
    "            \n",
    "            #Get summary from coefficients\n",
    "            summary <- c(pathway,\n",
    "                        cohort,\n",
    "                        summary(glm_AAO_PRS)$coefficients[2,1],\n",
    "                        summary(glm_AAO_PRS)$coefficients[2,2],\n",
    "                        summary(glm_AAO_PRS)$coefficients[2,4],\n",
    "                        N)\n",
    "            \n",
    "        } else {\n",
    "            \n",
    "            #If the dataframe is empty, print NA for summary stat\n",
    "            summary <- c(pathway,\n",
    "                        cohort,\n",
    "                        rep(NA, 3),\n",
    "                        N)\n",
    "            }\n",
    "    \n",
    "    return(summary)\n",
    "        \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           pathway  cohort                Coeff                se\n",
      "1  adaptive_immune    PPMI     0.11754008310212 0.446742353872611\n",
      "2  adaptive_immune    PDBP   -0.332335186986961 0.367177480640906\n",
      "3  adaptive_immune     HBS   -0.261082261339774 0.424091577967974\n",
      "4  adaptive_immune     LCC   -0.317705278196143  1.17640874019523\n",
      "5  adaptive_immune  Steady                 <NA>              <NA>\n",
      "6  adaptive_immune    Sure    0.432857079106381 0.616280311399835\n",
      "7  adaptive_immune BioFIND     0.65651521882082 0.681548832826391\n",
      "8  alpha_synuclein    PPMI   -0.895683264513345 0.445336129686749\n",
      "9  alpha_synuclein    PDBP   -0.224143381442031 0.364606977516569\n",
      "10 alpha_synuclein     HBS   -0.470040328419086 0.427452936204207\n",
      "11 alpha_synuclein     LCC     0.81491485331913  1.23696805918159\n",
      "12 alpha_synuclein  Steady                 <NA>              <NA>\n",
      "13 alpha_synuclein    Sure   -0.919593274652152 0.606858978170641\n",
      "14 alpha_synuclein BioFIND    0.545375716464445 0.678939189025328\n",
      "15   innate_immune    PPMI    0.470891368563763 0.453455858087763\n",
      "16   innate_immune    PDBP    0.112143782656244 0.368962774142204\n",
      "17   innate_immune     HBS   -0.107132640382175  0.42525270739704\n",
      "18   innate_immune     LCC    0.388058356556641  1.19578618636834\n",
      "19   innate_immune  Steady                 <NA>              <NA>\n",
      "20   innate_immune    Sure     0.73756067483427 0.609218129577813\n",
      "21   innate_immune BioFIND    0.707286860002631 0.697999588082121\n",
      "22        lysosome    PPMI   -0.820195297321081 0.444416685800239\n",
      "23        lysosome    PDBP   -0.274152550692425 0.365159625221371\n",
      "24        lysosome     HBS   -0.540410040396168 0.430028166765531\n",
      "25        lysosome     LCC   -0.302896604274209   1.3255158621218\n",
      "26        lysosome  Steady                 <NA>              <NA>\n",
      "27        lysosome    Sure    0.162707843910197 0.618178956527253\n",
      "28        lysosome BioFIND    -1.19029716852363 0.653919678690888\n",
      "29     endocytosis    PPMI    -0.20877709700642 0.447824098602034\n",
      "30     endocytosis    PDBP   0.0213108420954438 0.366725361483181\n",
      "31     endocytosis     HBS   -0.724348296090305 0.427142802067408\n",
      "32     endocytosis     LCC    0.742631626202097  1.22589299796138\n",
      "33     endocytosis  Steady                 <NA>              <NA>\n",
      "34     endocytosis    Sure    0.713434786836271  0.61767311292743\n",
      "35     endocytosis BioFIND -0.00722268648597844 0.681669794380371\n",
      "36       microglia    PPMI   -0.448238983747422 0.454040126513676\n",
      "37       microglia    PDBP   -0.648402798323591 0.365146431197885\n",
      "38       microglia     HBS   -0.702828285601217 0.428363221437813\n",
      "39       microglia     LCC     2.40216444293173  1.30364662591763\n",
      "40       microglia  Steady                 <NA>              <NA>\n",
      "41       microglia    Sure    -0.39329340350388 0.626402850423098\n",
      "42       microglia BioFIND  -0.0910662202214838 0.678614951377902\n",
      "43       monocytes    PPMI    -0.60055420623886 0.448095193268122\n",
      "44       monocytes    PDBP   -0.329590425700923 0.365668519542949\n",
      "45       monocytes     HBS   -0.342321139491971 0.426373190953864\n",
      "46       monocytes     LCC    0.891469136082504  1.24595707237826\n",
      "47       monocytes  Steady                 <NA>              <NA>\n",
      "48       monocytes    Sure    -1.34983269169621 0.608366613781044\n",
      "49       monocytes BioFIND  -0.0726705109384102 0.692660337231255\n",
      "50    mitochondria    PPMI    -0.20844149153818 0.450458792668578\n",
      "51    mitochondria    PDBP   -0.654830427784813 0.368540977634392\n",
      "52    mitochondria     HBS   -0.683890439876145 0.433127491314318\n",
      "53    mitochondria     LCC  0.00500146456061263  1.29326140529454\n",
      "54    mitochondria  Steady                 <NA>              <NA>\n",
      "55    mitochondria    Sure     0.32640405370588 0.624243149485222\n",
      "56    mitochondria BioFIND    0.120484269090679 0.757594830632491\n",
      "               Pvalue   N\n",
      "1   0.792588338532287 465\n",
      "2   0.365687066610229 790\n",
      "3   0.538371046912622 616\n",
      "4   0.788104029185174  64\n",
      "5                <NA>   0\n",
      "6   0.483131659529219 247\n",
      "7   0.338177174921148  92\n",
      "8    0.04488548611182 465\n",
      "9   0.538896413058281 790\n",
      "10  0.271928310989317 616\n",
      "11  0.512723849805161  64\n",
      "12               <NA>   0\n",
      "13  0.131009578845384 247\n",
      "14  0.424082269558086  92\n",
      "15  0.299609581318365 465\n",
      "16  0.761251903114821 790\n",
      "17  0.801182620050917 616\n",
      "18  0.746752375154975  64\n",
      "19               <NA>   0\n",
      "20  0.227218681346888 247\n",
      "21  0.313824129096833  92\n",
      "22 0.0656035520662395 465\n",
      "23  0.453014200953906 790\n",
      "24  0.209350229660201 616\n",
      "25  0.820080452041053  64\n",
      "26               <NA>   0\n",
      "27  0.792619281380441 247\n",
      "28 0.0722825783359915  92\n",
      "29  0.641292082005077 465\n",
      "30  0.953674917599328 790\n",
      "31 0.0904359067372402 616\n",
      "32  0.547102931242529  64\n",
      "33               <NA>   0\n",
      "34  0.249229104168258 247\n",
      "35  0.991571233172258  92\n",
      "36  0.324055576139453 465\n",
      "37 0.0761656553732349 790\n",
      "38  0.101370708447341 616\n",
      "39  0.070675956357568  64\n",
      "40               <NA>   0\n",
      "41  0.530694989704022 247\n",
      "42  0.893569982073501  92\n",
      "43  0.180833910258827 465\n",
      "44  0.367686879344884 790\n",
      "45  0.422365104897528 616\n",
      "46  0.477279369035242  64\n",
      "47               <NA>   0\n",
      "48 0.0274416278570505 247\n",
      "49    0.9166933132673  92\n",
      "50   0.64377705703052 465\n",
      "51 0.0759867803748157 790\n",
      "52  0.114865612926349 616\n",
      "53  0.996928071334019  64\n",
      "54               <NA>   0\n",
      "55  0.601542378610713 247\n",
      "56  0.874022869047398  92\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "results_list <- list()\n",
    "\n",
    "#Loop over each pathway and cohort\n",
    "for (pathway in pathways) {\n",
    "    for (cohort in cohorts) {\n",
    "        i <- length(results_list)\n",
    "        results_list[[i+1]] <- analyse_aao(pathway, cohort)\n",
    "    }\n",
    "}\n",
    "\n",
    "#Combine lists into a single dataframe\n",
    "aao_results <- do.call(rbind, results_list)\n",
    "\n",
    "#Set column names\n",
    "colnames(aao_results) <- c(\"pathway\", \"cohort\", \"Coeff\", \"se\", \"Pvalue\", \"N\")\n",
    "\n",
    "#Make into dataframe\n",
    "aao_results <- as.data.frame(aao_results)\n",
    "\n",
    "#Save output\n",
    "write.table(aao_results, \"/home/jupyter/pathways/pathways_agediagnosis_results.txt\",\n",
    "           sep = \"\\t\", row.names = F, col.names = T, quote = F)\n",
    "\n",
    "aao_results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Get cohort demographics for TABLE 1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Count the number of individuals in each cohort - after genetic QC, only individuals who have at least sex and age at diagnosis. The number of individuals who have available outcomes may be less."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "participant_id\tstudy\n",
      "BF-1001\tBioFIND\n",
      "BF-1002\tBioFIND\n",
      "BF-1003\tBioFIND\n",
      "BF-1004\tBioFIND\n",
      "BF-1005\tBioFIND\n",
      "BF-1006\tBioFIND\n",
      "BF-1008\tBioFIND\n",
      "BF-1009\tBioFIND\n",
      "BF-1010\tBioFIND\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "head /home/jupyter/clinical_data_v2-5/cohort.tab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 2610   10\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "#Read in one of the pathway fam files - can be any one as they should all be the same\n",
    "fam <- read.table(\"/home/jupyter/pathways/adaptive_immune.fam\")\n",
    "colnames(fam) <- c(\"FID\", \"IID\", \"m\", \"f\", \"sex_fam\", \"pheno\")\n",
    "\n",
    "#Read in AAO data\n",
    "aao <- read.table(\"/home/jupyter/clinical_data_v2-5/PD_cases_agedx.tab\", sep = \"\\t\", header = T)\n",
    "\n",
    "#Read in demographic data\n",
    "demo <- read.table(\"/home/jupyter/clinical_data_v2-5/PD_cases_start_end_dx_sex.tab\", sep = \"\\t\", header = T)\n",
    "\n",
    "#Read in cohort data\n",
    "cohort <- read.table(\"/home/jupyter/clinical_data_v2-5/cohort.tab\", header = T)\n",
    "\n",
    "#Merge\n",
    "merged <- fam %>%\n",
    "    inner_join(aao, by = c(\"FID\", \"IID\")) %>%\n",
    "    inner_join(demo, by = c(\"FID\", \"IID\")) %>%\n",
    "    inner_join(cohort, by = c(\"IID\" = \"participant_id\"))\n",
    "\n",
    "print(dim(merged))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                     FID                  IID m f sex_fam pheno\n",
      "1   SY-NIH_INVAA791MKCET SY-NIH_INVAA791MKCET 0 0       1    -9\n",
      "2   SY-NIH_INVAA976PRYBT SY-NIH_INVAA976PRYBT 0 0       1    -9\n",
      "3   SY-NIH_INVAL031EKGJ6 SY-NIH_INVAL031EKGJ6 0 0       1    -9\n",
      "4   SY-NIH_INVAN212WJYBM SY-NIH_INVAN212WJYBM 0 0       1    -9\n",
      "5   SY-NIH_INVBB604RTZN4 SY-NIH_INVBB604RTZN4 0 0       1    -9\n",
      "6   SY-NIH_INVBC118NRWC4 SY-NIH_INVBC118NRWC4 0 0       1    -9\n",
      "7   SY-NIH_INVBP682BYGHF SY-NIH_INVBP682BYGHF 0 0       1    -9\n",
      "8   SY-NIH_INVCB901YLVWG SY-NIH_INVCB901YLVWG 0 0       1    -9\n",
      "9   SY-NIH_INVCW212TXMFX SY-NIH_INVCW212TXMFX 0 0       2    -9\n",
      "10  SY-NIH_INVCX656ULCDX SY-NIH_INVCX656ULCDX 0 0       1    -9\n",
      "11  SY-NIH_INVDE149YELY9 SY-NIH_INVDE149YELY9 0 0       1    -9\n",
      "12  SY-NIH_INVEF231BNLJU SY-NIH_INVEF231BNLJU 0 0       1    -9\n",
      "13  SY-NIH_INVEG915UKFH3 SY-NIH_INVEG915UKFH3 0 0       1    -9\n",
      "14  SY-NIH_INVEP886EEYYL SY-NIH_INVEP886EEYYL 0 0       1    -9\n",
      "15  SY-NIH_INVEY006JDFDF SY-NIH_INVEY006JDFDF 0 0       2    -9\n",
      "16  SY-NIH_INVFM717GWDX4 SY-NIH_INVFM717GWDX4 0 0       2    -9\n",
      "17  SY-NIH_INVGB457MUMVF SY-NIH_INVGB457MUMVF 0 0       1    -9\n",
      "18  SY-NIH_INVGE792FENNX SY-NIH_INVGE792FENNX 0 0       1    -9\n",
      "19  SY-NIH_INVGJ141REHUN SY-NIH_INVGJ141REHUN 0 0       2    -9\n",
      "20  SY-NIH_INVGM431TEJZT SY-NIH_INVGM431TEJZT 0 0       1    -9\n",
      "21  SY-NIH_INVGX335TZRJR SY-NIH_INVGX335TZRJR 0 0       2    -9\n",
      "22  SY-NIH_INVHG261TPDN1 SY-NIH_INVHG261TPDN1 0 0       2    -9\n",
      "23  SY-NIH_INVJH118EENMJ SY-NIH_INVJH118EENMJ 0 0       1    -9\n",
      "24  SY-NIH_INVJM199EFPWA SY-NIH_INVJM199EFPWA 0 0       2    -9\n",
      "25  SY-NIH_INVKC092VHYYD SY-NIH_INVKC092VHYYD 0 0       1    -9\n",
      "26  SY-NIH_INVKN070BCHXF SY-NIH_INVKN070BCHXF 0 0       1    -9\n",
      "27  SY-NIH_INVKU234TEHCA SY-NIH_INVKU234TEHCA 0 0       1    -9\n",
      "28  SY-NIH_INVKY455DKJJN SY-NIH_INVKY455DKJJN 0 0       2    -9\n",
      "29  SY-NIH_INVKZ251GAYH9 SY-NIH_INVKZ251GAYH9 0 0       1    -9\n",
      "30  SY-NIH_INVLB876LTWEL SY-NIH_INVLB876LTWEL 0 0       1    -9\n",
      "31  SY-NIH_INVLD826LLAN9 SY-NIH_INVLD826LLAN9 0 0       1    -9\n",
      "32  SY-NIH_INVLE239PDMDT SY-NIH_INVLE239PDMDT 0 0       1    -9\n",
      "33  SY-NIH_INVLF737FXTMC SY-NIH_INVLF737FXTMC 0 0       1    -9\n",
      "34  SY-NIH_INVLU546LFCFN SY-NIH_INVLU546LFCFN 0 0       1    -9\n",
      "35  SY-NIH_INVMC691AGLBK SY-NIH_INVMC691AGLBK 0 0       1    -9\n",
      "36  SY-NIH_INVND417JLBRD SY-NIH_INVND417JLBRD 0 0       2    -9\n",
      "37  SY-NIH_INVNH467LWRZQ SY-NIH_INVNH467LWRZQ 0 0       1    -9\n",
      "38  SY-NIH_INVNN611MKKN9 SY-NIH_INVNN611MKKN9 0 0       1    -9\n",
      "39  SY-NIH_INVNV021EXHAL SY-NIH_INVNV021EXHAL 0 0       1    -9\n",
      "40  SY-NIH_INVPA076WWCFL SY-NIH_INVPA076WWCFL 0 0       2    -9\n",
      "41  SY-NIH_INVRW457MWNLE SY-NIH_INVRW457MWNLE 0 0       1    -9\n",
      "42  SY-NIH_INVTE240HLPVG SY-NIH_INVTE240HLPVG 0 0       1    -9\n",
      "43  SY-NIH_INVTL441KZJEK SY-NIH_INVTL441KZJEK 0 0       1    -9\n",
      "44  SY-NIH_INVTU048FRKCV SY-NIH_INVTU048FRKCV 0 0       1    -9\n",
      "45  SY-NIH_INVTU213XNUYV SY-NIH_INVTU213XNUYV 0 0       2    -9\n",
      "46  SY-NIH_INVTV572ZNPL5 SY-NIH_INVTV572ZNPL5 0 0       1    -9\n",
      "47  SY-NIH_INVUB827RAVLJ SY-NIH_INVUB827RAVLJ 0 0       2    -9\n",
      "48  SY-NIH_INVUR161RNWE8 SY-NIH_INVUR161RNWE8 0 0       1    -9\n",
      "49  SY-NIH_INVUT097LFWP3 SY-NIH_INVUT097LFWP3 0 0       2    -9\n",
      "50  SY-NIH_INVUU315DKLU5 SY-NIH_INVUU315DKLU5 0 0       2    -9\n",
      "51  SY-NIH_INVVN107RBPFT SY-NIH_INVVN107RBPFT 0 0       1    -9\n",
      "52  SY-NIH_INVVV249KZYC7 SY-NIH_INVVV249KZYC7 0 0       1    -9\n",
      "53  SY-NIH_INVVX457EUNNP SY-NIH_INVVX457EUNNP 0 0       2    -9\n",
      "54  SY-NIH_INVWA963GYCMG SY-NIH_INVWA963GYCMG 0 0       2    -9\n",
      "55  SY-NIH_INVWE763JVNRR SY-NIH_INVWE763JVNRR 0 0       1    -9\n",
      "56  SY-NIH_INVWW719VAYJ7 SY-NIH_INVWW719VAYJ7 0 0       1    -9\n",
      "57  SY-NIH_INVWY269KWAD8 SY-NIH_INVWY269KWAD8 0 0       2    -9\n",
      "58  SY-NIH_INVXD132ZRCB5 SY-NIH_INVXD132ZRCB5 0 0       1    -9\n",
      "59  SY-NIH_INVXH367JZZBN SY-NIH_INVXH367JZZBN 0 0       1    -9\n",
      "60  SY-NIH_INVXH620VNDGM SY-NIH_INVXH620VNDGM 0 0       1    -9\n",
      "61  SY-NIH_INVXR659XFEFH SY-NIH_INVXR659XFEFH 0 0       1    -9\n",
      "62  SY-NIH_INVYD129CCPMU SY-NIH_INVYD129CCPMU 0 0       1    -9\n",
      "63  SY-NIH_INVYF099KLUV0 SY-NIH_INVYF099KLUV0 0 0       1    -9\n",
      "64  SY-NIH_INVYK292ZAVCE SY-NIH_INVYK292ZAVCE 0 0       1    -9\n",
      "65         SY-PDAA303ML7        SY-PDAA303ML7 0 0       2    -9\n",
      "66         SY-PDAC092WHX        SY-PDAC092WHX 0 0       1    -9\n",
      "67         SY-PDAL671CXE        SY-PDAL671CXE 0 0       1    -9\n",
      "68         SY-PDAM498RLN        SY-PDAM498RLN 0 0       1    -9\n",
      "69         SY-PDAM806RP7        SY-PDAM806RP7 0 0       2    -9\n",
      "70         SY-PDAP235YZ9        SY-PDAP235YZ9 0 0       1    -9\n",
      "71         SY-PDAT402PC9        SY-PDAT402PC9 0 0       2    -9\n",
      "72         SY-PDAW037JR3        SY-PDAW037JR3 0 0       2    -9\n",
      "73         SY-PDBB338BL7        SY-PDBB338BL7 0 0       1    -9\n",
      "74         SY-PDBD215UJJ        SY-PDBD215UJJ 0 0       1    -9\n",
      "75         SY-PDBE205LW8        SY-PDBE205LW8 0 0       1    -9\n",
      "76         SY-PDBE889AAA        SY-PDBE889AAA 0 0       2    -9\n",
      "77         SY-PDBG899XA4        SY-PDBG899XA4 0 0       2    -9\n",
      "78         SY-PDBJ898HCX        SY-PDBJ898HCX 0 0       2    -9\n",
      "79         SY-PDBL117YP2        SY-PDBL117YP2 0 0       1    -9\n",
      "80         SY-PDBN480ZM4        SY-PDBN480ZM4 0 0       2    -9\n",
      "81         SY-PDBP035HXQ        SY-PDBP035HXQ 0 0       2    -9\n",
      "82         SY-PDBP093XHZ        SY-PDBP093XHZ 0 0       1    -9\n",
      "83         SY-PDBU289VUV        SY-PDBU289VUV 0 0       1    -9\n",
      "84         SY-PDBV973XVQ        SY-PDBV973XVQ 0 0       2    -9\n",
      "85         SY-PDBY057RY0        SY-PDBY057RY0 0 0       1    -9\n",
      "86         SY-PDCA047BYG        SY-PDCA047BYG 0 0       2    -9\n",
      "87         SY-PDCA710AKR        SY-PDCA710AKR 0 0       1    -9\n",
      "88         SY-PDCD423RAF        SY-PDCD423RAF 0 0       1    -9\n",
      "89         SY-PDCK827URL        SY-PDCK827URL 0 0       1    -9\n",
      "90         SY-PDCP453KR7        SY-PDCP453KR7 0 0       1    -9\n",
      "91         SY-PDCV595BXH        SY-PDCV595BXH 0 0       1    -9\n",
      "92         SY-PDCV789PZ0        SY-PDCV789PZ0 0 0       1    -9\n",
      "93         SY-PDCY011DK8        SY-PDCY011DK8 0 0       2    -9\n",
      "94         SY-PDDC840UCT        SY-PDDC840UCT 0 0       2    -9\n",
      "95         SY-PDDE536HX9        SY-PDDE536HX9 0 0       1    -9\n",
      "96         SY-PDDF547AW8        SY-PDDF547AW8 0 0       1    -9\n",
      "97         SY-PDDG798EYE        SY-PDDG798EYE 0 0       2    -9\n",
      "98         SY-PDDJ151DRG        SY-PDDJ151DRG 0 0       2    -9\n",
      "99         SY-PDDJ328RUZ        SY-PDDJ328RUZ 0 0       1    -9\n",
      "100        SY-PDDM154XKN        SY-PDDM154XKN 0 0       1    -9\n",
      "101        SY-PDDN072RJP        SY-PDDN072RJP 0 0       1    -9\n",
      "102        SY-PDDR519UC8        SY-PDDR519UC8 0 0       1    -9\n",
      "103        SY-PDDU181LUY        SY-PDDU181LUY 0 0       1    -9\n",
      "104        SY-PDDV638ATW        SY-PDDV638ATW 0 0       2    -9\n",
      "105        SY-PDDX116DAR        SY-PDDX116DAR 0 0       1    -9\n",
      "106        SY-PDDX938GFK        SY-PDDX938GFK 0 0       1    -9\n",
      "107        SY-PDEC217LX6        SY-PDEC217LX6 0 0       2    -9\n",
      "108        SY-PDEC378HZJ        SY-PDEC378HZJ 0 0       1    -9\n",
      "109        SY-PDED475CD6        SY-PDED475CD6 0 0       1    -9\n",
      "110        SY-PDEE672XB6        SY-PDEE672XB6 0 0       2    -9\n",
      "111        SY-PDEL268YC3        SY-PDEL268YC3 0 0       2    -9\n",
      "112        SY-PDEP248MFK        SY-PDEP248MFK 0 0       1    -9\n",
      "113        SY-PDEP388KKG        SY-PDEP388KKG 0 0       1    -9\n",
      "114        SY-PDET315VVM        SY-PDET315VVM 0 0       1    -9\n",
      "115        SY-PDEY398MJ9        SY-PDEY398MJ9 0 0       2    -9\n",
      "116        SY-PDEZ499EJW        SY-PDEZ499EJW 0 0       1    -9\n",
      "117        SY-PDFA420XBE        SY-PDFA420XBE 0 0       1    -9\n",
      "118        SY-PDFB561WBC        SY-PDFB561WBC 0 0       2    -9\n",
      "119        SY-PDFB947NXY        SY-PDFB947NXY 0 0       2    -9\n",
      "120        SY-PDFC734MB0        SY-PDFC734MB0 0 0       2    -9\n",
      "121        SY-PDFE466KZ7        SY-PDFE466KZ7 0 0       1    -9\n",
      "122        SY-PDFK027EMJ        SY-PDFK027EMJ 0 0       2    -9\n",
      "123        SY-PDFK572NTV        SY-PDFK572NTV 0 0       2    -9\n",
      "124        SY-PDFM424ACF        SY-PDFM424ACF 0 0       2    -9\n",
      "125        SY-PDFP586PTW        SY-PDFP586PTW 0 0       1    -9\n",
      "126        SY-PDFU242FB5        SY-PDFU242FB5 0 0       2    -9\n",
      "127        SY-PDFX633XN8        SY-PDFX633XN8 0 0       1    -9\n",
      "128        SY-PDFY440AHU        SY-PDFY440AHU 0 0       1    -9\n",
      "129        SY-PDGC257HUC        SY-PDGC257HUC 0 0       2    -9\n",
      "130        SY-PDGC898NW1        SY-PDGC898NW1 0 0       1    -9\n",
      "131        SY-PDGK154AAA        SY-PDGK154AAA 0 0       1    -9\n",
      "132        SY-PDGM203WLH        SY-PDGM203WLH 0 0       1    -9\n",
      "133        SY-PDGM962WE0        SY-PDGM962WE0 0 0       2    -9\n",
      "134        SY-PDGP840YXV        SY-PDGP840YXV 0 0       1    -9\n",
      "135        SY-PDGR527WAJ        SY-PDGR527WAJ 0 0       1    -9\n",
      "136        SY-PDGT120PCD        SY-PDGT120PCD 0 0       1    -9\n",
      "137        SY-PDGT137BDP        SY-PDGT137BDP 0 0       1    -9\n",
      "138        SY-PDGT904GUV        SY-PDGT904GUV 0 0       1    -9\n",
      "139        SY-PDGX683XN4        SY-PDGX683XN4 0 0       2    -9\n",
      "140        SY-PDGZ475MV7        SY-PDGZ475MV7 0 0       1    -9\n",
      "141        SY-PDHB986ZP1        SY-PDHB986ZP1 0 0       2    -9\n",
      "142        SY-PDHD219YRF        SY-PDHD219YRF 0 0       1    -9\n",
      "143        SY-PDHL121FTL        SY-PDHL121FTL 0 0       1    -9\n",
      "144        SY-PDHM538CPR        SY-PDHM538CPR 0 0       1    -9\n",
      "145        SY-PDHU960FL7        SY-PDHU960FL7 0 0       2    -9\n",
      "146        SY-PDHX876WCT        SY-PDHX876WCT 0 0       1    -9\n",
      "147        SY-PDHZ036PXX        SY-PDHZ036PXX 0 0       1    -9\n",
      "148        SY-PDJB470UNA        SY-PDJB470UNA 0 0       1    -9\n",
      "149        SY-PDJC327HP1        SY-PDJC327HP1 0 0       1    -9\n",
      "150        SY-PDJC608VWK        SY-PDJC608VWK 0 0       1    -9\n",
      "151        SY-PDJC949KH5        SY-PDJC949KH5 0 0       1    -9\n",
      "152        SY-PDJF204HHG        SY-PDJF204HHG 0 0       1    -9\n",
      "153        SY-PDJJ489LPF        SY-PDJJ489LPF 0 0       1    -9\n",
      "154        SY-PDJK841KD3        SY-PDJK841KD3 0 0       1    -9\n",
      "155        SY-PDJL010ZAN        SY-PDJL010ZAN 0 0       2    -9\n",
      "156        SY-PDJL745HC6        SY-PDJL745HC6 0 0       1    -9\n",
      "157        SY-PDJM068DZ4        SY-PDJM068DZ4 0 0       1    -9\n",
      "158        SY-PDJR756TZR        SY-PDJR756TZR 0 0       2    -9\n",
      "159        SY-PDJX091AFX        SY-PDJX091AFX 0 0       1    -9\n",
      "160        SY-PDKF900ZLV        SY-PDKF900ZLV 0 0       2    -9\n",
      "161        SY-PDKJ659ZM9        SY-PDKJ659ZM9 0 0       1    -9\n",
      "162        SY-PDKL261NW4        SY-PDKL261NW4 0 0       1    -9\n",
      "163        SY-PDKL664FP1        SY-PDKL664FP1 0 0       1    -9\n",
      "164        SY-PDKR735UZ2        SY-PDKR735UZ2 0 0       1    -9\n",
      "165        SY-PDKT778AGJ        SY-PDKT778AGJ 0 0       1    -9\n",
      "166        SY-PDKY219GFB        SY-PDKY219GFB 0 0       1    -9\n",
      "167        SY-PDKY344HUU        SY-PDKY344HUU 0 0       1    -9\n",
      "168        SY-PDLA635RV5        SY-PDLA635RV5 0 0       1    -9\n",
      "169        SY-PDLF179RP6        SY-PDLF179RP6 0 0       1    -9\n",
      "170        SY-PDLJ410GY9        SY-PDLJ410GY9 0 0       2    -9\n",
      "171        SY-PDLN246VJR        SY-PDLN246VJR 0 0       2    -9\n",
      "172        SY-PDLP356HZT        SY-PDLP356HZT 0 0       1    -9\n",
      "173        SY-PDLP386MNR        SY-PDLP386MNR 0 0       2    -9\n",
      "174        SY-PDLR732JK8        SY-PDLR732JK8 0 0       1    -9\n",
      "175        SY-PDLR914FEX        SY-PDLR914FEX 0 0       2    -9\n",
      "176        SY-PDLT248VD1        SY-PDLT248VD1 0 0       2    -9\n",
      "177        SY-PDLT933TJW        SY-PDLT933TJW 0 0       1    -9\n",
      "178        SY-PDLZ486TMV        SY-PDLZ486TMV 0 0       1    -9\n",
      "179        SY-PDMA876HBK        SY-PDMA876HBK 0 0       1    -9\n",
      "180        SY-PDMB735TY8        SY-PDMB735TY8 0 0       1    -9\n",
      "181        SY-PDMC549RNY        SY-PDMC549RNY 0 0       1    -9\n",
      "182        SY-PDMG210BM3        SY-PDMG210BM3 0 0       1    -9\n",
      "183        SY-PDMJ024DH2        SY-PDMJ024DH2 0 0       2    -9\n",
      "184        SY-PDMJ741KYZ        SY-PDMJ741KYZ 0 0       2    -9\n",
      "185        SY-PDMP675WHN        SY-PDMP675WHN 0 0       1    -9\n",
      "186        SY-PDMT950HHZ        SY-PDMT950HHZ 0 0       1    -9\n",
      "187        SY-PDMU271MR2        SY-PDMU271MR2 0 0       1    -9\n",
      "188        SY-PDMW411JHM        SY-PDMW411JHM 0 0       1    -9\n",
      "189        SY-PDMW532RFW        SY-PDMW532RFW 0 0       1    -9\n",
      "190        SY-PDMW764ZJ2        SY-PDMW764ZJ2 0 0       1    -9\n",
      "191        SY-PDNB025TVP        SY-PDNB025TVP 0 0       1    -9\n",
      "192        SY-PDNC843PUF        SY-PDNC843PUF 0 0       1    -9\n",
      "193        SY-PDNE693YUB        SY-PDNE693YUB 0 0       1    -9\n",
      "194        SY-PDNF475NVF        SY-PDNF475NVF 0 0       2    -9\n",
      "195        SY-PDNG871RZD        SY-PDNG871RZD 0 0       1    -9\n",
      "196        SY-PDNJ340MG3        SY-PDNJ340MG3 0 0       1    -9\n",
      "197        SY-PDNK239KGR        SY-PDNK239KGR 0 0       1    -9\n",
      "198        SY-PDNM553BMC        SY-PDNM553BMC 0 0       1    -9\n",
      "199        SY-PDNR183XEG        SY-PDNR183XEG 0 0       1    -9\n",
      "200        SY-PDNR386JND        SY-PDNR386JND 0 0       1    -9\n",
      "201        SY-PDNV666TYG        SY-PDNV666TYG 0 0       1    -9\n",
      "202        SY-PDNX988LZ0        SY-PDNX988LZ0 0 0       1    -9\n",
      "203        SY-PDNY365REY        SY-PDNY365REY 0 0       2    -9\n",
      "204        SY-PDNZ608ALK        SY-PDNZ608ALK 0 0       1    -9\n",
      "205        SY-PDPC279JY2        SY-PDPC279JY2 0 0       1    -9\n",
      "206        SY-PDPC501RUX        SY-PDPC501RUX 0 0       2    -9\n",
      "207        SY-PDPE166JAC        SY-PDPE166JAC 0 0       1    -9\n",
      "208        SY-PDPG763MPH        SY-PDPG763MPH 0 0       2    -9\n",
      "209        SY-PDPH970THZ        SY-PDPH970THZ 0 0       1    -9\n",
      "210        SY-PDPK714GMB        SY-PDPK714GMB 0 0       1    -9\n",
      "211        SY-PDPL807RAU        SY-PDPL807RAU 0 0       2    -9\n",
      "212        SY-PDPR510HJ0        SY-PDPR510HJ0 0 0       1    -9\n",
      "213        SY-PDPT351UXN        SY-PDPT351UXN 0 0       2    -9\n",
      "214        SY-PDPU784TAZ        SY-PDPU784TAZ 0 0       2    -9\n",
      "215        SY-PDPV423CNM        SY-PDPV423CNM 0 0       2    -9\n",
      "216        SY-PDPV528NPV        SY-PDPV528NPV 0 0       1    -9\n",
      "217        SY-PDPV787NPG        SY-PDPV787NPG 0 0       1    -9\n",
      "218        SY-PDPW188FZT        SY-PDPW188FZT 0 0       1    -9\n",
      "219        SY-PDRA286VUP        SY-PDRA286VUP 0 0       2    -9\n",
      "220        SY-PDRD344UZP        SY-PDRD344UZP 0 0       2    -9\n",
      "221        SY-PDRF965LCT        SY-PDRF965LCT 0 0       1    -9\n",
      "222        SY-PDRH878JR9        SY-PDRH878JR9 0 0       1    -9\n",
      "223        SY-PDRJ196DZU        SY-PDRJ196DZU 0 0       1    -9\n",
      "224        SY-PDRT879DEU        SY-PDRT879DEU 0 0       1    -9\n",
      "225        SY-PDRW958ZX5        SY-PDRW958ZX5 0 0       2    -9\n",
      "226        SY-PDRX123WPH        SY-PDRX123WPH 0 0       2    -9\n",
      "227        SY-PDRY066RUY        SY-PDRY066RUY 0 0       1    -9\n",
      "228        SY-PDRZ083MLJ        SY-PDRZ083MLJ 0 0       1    -9\n",
      "229        SY-PDTA108ZNG        SY-PDTA108ZNG 0 0       1    -9\n",
      "230        SY-PDTB123HUP        SY-PDTB123HUP 0 0       2    -9\n",
      "231        SY-PDTD556NZW        SY-PDTD556NZW 0 0       1    -9\n",
      "232        SY-PDTK671UAF        SY-PDTK671UAF 0 0       2    -9\n",
      "233        SY-PDTL303ZHA        SY-PDTL303ZHA 0 0       2    -9\n",
      "234        SY-PDTP690JJV        SY-PDTP690JJV 0 0       1    -9\n",
      "235        SY-PDTT002PBC        SY-PDTT002PBC 0 0       1    -9\n",
      "236        SY-PDTT287RCE        SY-PDTT287RCE 0 0       1    -9\n",
      "237        SY-PDTU302AVZ        SY-PDTU302AVZ 0 0       1    -9\n",
      "238        SY-PDTW883TFJ        SY-PDTW883TFJ 0 0       1    -9\n",
      "239        SY-PDUE624VMC        SY-PDUE624VMC 0 0       2    -9\n",
      "240        SY-PDUH392PEM        SY-PDUH392PEM 0 0       1    -9\n",
      "241        SY-PDUJ353KZ5        SY-PDUJ353KZ5 0 0       1    -9\n",
      "242        SY-PDUL755LRF        SY-PDUL755LRF 0 0       1    -9\n",
      "243        SY-PDUP890GY0        SY-PDUP890GY0 0 0       1    -9\n",
      "244        SY-PDUR240JAP        SY-PDUR240JAP 0 0       1    -9\n",
      "245        SY-PDVC085CEF        SY-PDVC085CEF 0 0       1    -9\n",
      "246        SY-PDVC271HCN        SY-PDVC271HCN 0 0       2    -9\n",
      "247        SY-PDVD904ZLL        SY-PDVD904ZLL 0 0       1    -9\n",
      "248        SY-PDVE127VA1        SY-PDVE127VA1 0 0       1    -9\n",
      "249        SY-PDVF912GT3        SY-PDVF912GT3 0 0       2    -9\n",
      "250        SY-PDVG331VY8        SY-PDVG331VY8 0 0       1    -9\n",
      "251        SY-PDVK241MYL        SY-PDVK241MYL 0 0       2    -9\n",
      "252        SY-PDVN602XTZ        SY-PDVN602XTZ 0 0       1    -9\n",
      "253        SY-PDVR967ABF        SY-PDVR967ABF 0 0       1    -9\n",
      "254        SY-PDVT618YMA        SY-PDVT618YMA 0 0       2    -9\n",
      "255        SY-PDVV007KB1        SY-PDVV007KB1 0 0       2    -9\n",
      "256        SY-PDVV928BM6        SY-PDVV928BM6 0 0       1    -9\n",
      "257        SY-PDVW660TK7        SY-PDVW660TK7 0 0       1    -9\n",
      "258        SY-PDVX201JNJ        SY-PDVX201JNJ 0 0       1    -9\n",
      "259        SY-PDVZ048AZN        SY-PDVZ048AZN 0 0       1    -9\n",
      "260        SY-PDWJ270UA9        SY-PDWJ270UA9 0 0       1    -9\n",
      "261        SY-PDWJ972PW0        SY-PDWJ972PW0 0 0       1    -9\n",
      "262        SY-PDWL063MJY        SY-PDWL063MJY 0 0       2    -9\n",
      "263        SY-PDWM132ZJG        SY-PDWM132ZJG 0 0       1    -9\n",
      "264        SY-PDWM375MR4        SY-PDWM375MR4 0 0       1    -9\n",
      "265        SY-PDWT248KPP        SY-PDWT248KPP 0 0       2    -9\n",
      "266        SY-PDWT850FPR        SY-PDWT850FPR 0 0       2    -9\n",
      "267        SY-PDWU635DMK        SY-PDWU635DMK 0 0       1    -9\n",
      "268        SY-PDWW140KR1        SY-PDWW140KR1 0 0       1    -9\n",
      "269        SY-PDWY946KH3        SY-PDWY946KH3 0 0       1    -9\n",
      "270        SY-PDWZ607RUN        SY-PDWZ607RUN 0 0       2    -9\n",
      "271        SY-PDXB171UFV        SY-PDXB171UFV 0 0       1    -9\n",
      "272        SY-PDXB360APE        SY-PDXB360APE 0 0       1    -9\n",
      "273        SY-PDXC021PH0        SY-PDXC021PH0 0 0       1    -9\n",
      "274        SY-PDXD243JZ8        SY-PDXD243JZ8 0 0       1    -9\n",
      "275        SY-PDXD727ER4        SY-PDXD727ER4 0 0       2    -9\n",
      "276        SY-PDXF113AWF        SY-PDXF113AWF 0 0       1    -9\n",
      "277        SY-PDXK969CV3        SY-PDXK969CV3 0 0       2    -9\n",
      "278        SY-PDXM231AJX        SY-PDXM231AJX 0 0       1    -9\n",
      "279        SY-PDXP355GHP        SY-PDXP355GHP 0 0       2    -9\n",
      "280        SY-PDXR916GN6        SY-PDXR916GN6 0 0       1    -9\n",
      "281        SY-PDXU049YTD        SY-PDXU049YTD 0 0       1    -9\n",
      "282        SY-PDXU284BKT        SY-PDXU284BKT 0 0       1    -9\n",
      "283        SY-PDXV507VWG        SY-PDXV507VWG 0 0       1    -9\n",
      "284        SY-PDXW097DTR        SY-PDXW097DTR 0 0       2    -9\n",
      "285        SY-PDXX265ZZE        SY-PDXX265ZZE 0 0       1    -9\n",
      "286        SY-PDXZ650DHD        SY-PDXZ650DHD 0 0       1    -9\n",
      "287        SY-PDYA640KX7        SY-PDYA640KX7 0 0       2    -9\n",
      "288        SY-PDYE953YNZ        SY-PDYE953YNZ 0 0       1    -9\n",
      "289        SY-PDYR637WCG        SY-PDYR637WCG 0 0       1    -9\n",
      "290        SY-PDYT949VRE        SY-PDYT949VRE 0 0       1    -9\n",
      "291        SY-PDYV487GW8        SY-PDYV487GW8 0 0       1    -9\n",
      "292        SY-PDYX586ZYN        SY-PDYX586ZYN 0 0       1    -9\n",
      "293        SY-PDYY549CXW        SY-PDYY549CXW 0 0       2    -9\n",
      "294        SY-PDYY682GYY        SY-PDYY682GYY 0 0       1    -9\n",
      "295        SY-PDYZ339GYL        SY-PDYZ339GYL 0 0       2    -9\n",
      "296        SY-PDYZ663VX3        SY-PDYZ663VX3 0 0       1    -9\n",
      "297        SY-PDZH104KR2        SY-PDZH104KR2 0 0       1    -9\n",
      "298        SY-PDZJ450XYU        SY-PDZJ450XYU 0 0       1    -9\n",
      "299        SY-PDZK538BK5        SY-PDZK538BK5 0 0       2    -9\n",
      "300        SY-PDZK862NFW        SY-PDZK862NFW 0 0       1    -9\n",
      "301        SY-PDZN008XD9        SY-PDZN008XD9 0 0       1    -9\n",
      "302        SY-PDZU300RYJ        SY-PDZU300RYJ 0 0       1    -9\n",
      "303        SY-PDZV652BE6        SY-PDZV652BE6 0 0       1    -9\n",
      "304        SY-PDZW544CEL        SY-PDZW544CEL 0 0       1    -9\n",
      "305        SY-PDZX554VNB        SY-PDZX554VNB 0 0       1    -9\n",
      "306        SY-PDZX943HWN        SY-PDZX943HWN 0 0       1    -9\n",
      "307        SY-PDZZ260HUM        SY-PDZZ260HUM 0 0       2    -9\n",
      "    age_at_diagnosis sex age_at_baseline  study\n",
      "1                 NA   1              69 Steady\n",
      "2                 NA   1              74 Steady\n",
      "3                 NA   1              73 Steady\n",
      "4                 NA   1              56 Steady\n",
      "5                 NA   1              67 Steady\n",
      "6                 NA   1              53 Steady\n",
      "7                 NA   1              71 Steady\n",
      "8                 NA   1              69 Steady\n",
      "9                 NA   2              63 Steady\n",
      "10                NA   1              56 Steady\n",
      "11                NA   1              71 Steady\n",
      "12                NA   1              55 Steady\n",
      "13                NA   1              72 Steady\n",
      "14                NA   1              61 Steady\n",
      "15                NA   2              35 Steady\n",
      "16                NA   2              32 Steady\n",
      "17                NA   1              65 Steady\n",
      "18                NA   1              60 Steady\n",
      "19                NA   2              61 Steady\n",
      "20                NA   1              67 Steady\n",
      "21                NA   2              52 Steady\n",
      "22                NA   2              73 Steady\n",
      "23                NA   1              59 Steady\n",
      "24                NA   2              71 Steady\n",
      "25                NA   1              71 Steady\n",
      "26                NA   1              76 Steady\n",
      "27                NA   1              64 Steady\n",
      "28                NA   2              64 Steady\n",
      "29                NA   1              40 Steady\n",
      "30                NA   1              64 Steady\n",
      "31                NA   1              79 Steady\n",
      "32                NA   1              64 Steady\n",
      "33                NA   1              72 Steady\n",
      "34                NA   1              50 Steady\n",
      "35                NA   1              58 Steady\n",
      "36                NA   2              57 Steady\n",
      "37                NA   1              60 Steady\n",
      "38                NA   1              67 Steady\n",
      "39                NA   1              43 Steady\n",
      "40                NA   2              50 Steady\n",
      "41                NA   1              61 Steady\n",
      "42                NA   1              59 Steady\n",
      "43                NA   1              59 Steady\n",
      "44                NA   1              46 Steady\n",
      "45                NA   2              68 Steady\n",
      "46                NA   1              52 Steady\n",
      "47                NA   2              73 Steady\n",
      "48                NA   1              43 Steady\n",
      "49                NA   2              69 Steady\n",
      "50                NA   2              83 Steady\n",
      "51                NA   1              53 Steady\n",
      "52                NA   1              58 Steady\n",
      "53                NA   2              64 Steady\n",
      "54                NA   2              68 Steady\n",
      "55                NA   1              48 Steady\n",
      "56                NA   1              50 Steady\n",
      "57                NA   2              65 Steady\n",
      "58                NA   1              48 Steady\n",
      "59                NA   1              45 Steady\n",
      "60                NA   1              78 Steady\n",
      "61                NA   1              61 Steady\n",
      "62                NA   1              72 Steady\n",
      "63                NA   1              69 Steady\n",
      "64                NA   1              67 Steady\n",
      "65                NA   2              60 Steady\n",
      "66                NA   1              47 Steady\n",
      "67                NA   1              66 Steady\n",
      "68                NA   1              77 Steady\n",
      "69                NA   2              68 Steady\n",
      "70                NA   1              38 Steady\n",
      "71                NA   2              66 Steady\n",
      "72                NA   2              52 Steady\n",
      "73                NA   1              69 Steady\n",
      "74                NA   1              69 Steady\n",
      "75                NA   1              51 Steady\n",
      "76                NA   2              57 Steady\n",
      "77                NA   2              68 Steady\n",
      "78                NA   2              65 Steady\n",
      "79                NA   1              74 Steady\n",
      "80                NA   2              66 Steady\n",
      "81                NA   2              53 Steady\n",
      "82                NA   1              80 Steady\n",
      "83                NA   1              72 Steady\n",
      "84                NA   2              61 Steady\n",
      "85                NA   1              62 Steady\n",
      "86                NA   2              61 Steady\n",
      "87                NA   1              52 Steady\n",
      "88                NA   1              76 Steady\n",
      "89                NA   1              44 Steady\n",
      "90                NA   1              73 Steady\n",
      "91                NA   1              51 Steady\n",
      "92                NA   1              65 Steady\n",
      "93                NA   2              72 Steady\n",
      "94                NA   2              56 Steady\n",
      "95                NA   1              48 Steady\n",
      "96                NA   1              60 Steady\n",
      "97                NA   2              67 Steady\n",
      "98                NA   2              80 Steady\n",
      "99                NA   1              68 Steady\n",
      "100               NA   1              56 Steady\n",
      "101               NA   1              61 Steady\n",
      "102               NA   1              45 Steady\n",
      "103               NA   1              68 Steady\n",
      "104               NA   2              41 Steady\n",
      "105               NA   1              72 Steady\n",
      "106               NA   1              61 Steady\n",
      "107               NA   2              66 Steady\n",
      "108               NA   1              64 Steady\n",
      "109               NA   1              71 Steady\n",
      "110               NA   2              67 Steady\n",
      "111               NA   2              61 Steady\n",
      "112               NA   1              63 Steady\n",
      "113               NA   1              59 Steady\n",
      "114               NA   1              55 Steady\n",
      "115               NA   2              71 Steady\n",
      "116               NA   1              63 Steady\n",
      "117               NA   1              75 Steady\n",
      "118               NA   2              55 Steady\n",
      "119               NA   2              75 Steady\n",
      "120               NA   2              50 Steady\n",
      "121               NA   1              65 Steady\n",
      "122               NA   2              59 Steady\n",
      "123               NA   2              53 Steady\n",
      "124               NA   2              68 Steady\n",
      "125               NA   1              54 Steady\n",
      "126               NA   2              64 Steady\n",
      "127               NA   1              59 Steady\n",
      "128               NA   1              64 Steady\n",
      "129               NA   2              63 Steady\n",
      "130               NA   1              51 Steady\n",
      "131               NA   1              56 Steady\n",
      "132               NA   1              58 Steady\n",
      "133               NA   2              64 Steady\n",
      "134               NA   1              69 Steady\n",
      "135               NA   1              77 Steady\n",
      "136               NA   1              53 Steady\n",
      "137               NA   1              60 Steady\n",
      "138               NA   1              56 Steady\n",
      "139               NA   2              53 Steady\n",
      "140               NA   1              67 Steady\n",
      "141               NA   2              60 Steady\n",
      "142               NA   1              59 Steady\n",
      "143               NA   1              65 Steady\n",
      "144               NA   1              59 Steady\n",
      "145               NA   2              62 Steady\n",
      "146               NA   1              53 Steady\n",
      "147               NA   1              62 Steady\n",
      "148               NA   1              65 Steady\n",
      "149               NA   1              37 Steady\n",
      "150               NA   1              59 Steady\n",
      "151               NA   1              70 Steady\n",
      "152               NA   1              67 Steady\n",
      "153               NA   1              64 Steady\n",
      "154               NA   1              69 Steady\n",
      "155               NA   2              61 Steady\n",
      "156               NA   1              64 Steady\n",
      "157               NA   1              49 Steady\n",
      "158               NA   2              68 Steady\n",
      "159               NA   1              62 Steady\n",
      "160               NA   2              44 Steady\n",
      "161               NA   1              67 Steady\n",
      "162               NA   1              70 Steady\n",
      "163               NA   1              63 Steady\n",
      "164               NA   1              54 Steady\n",
      "165               NA   1              64 Steady\n",
      "166               NA   1              60 Steady\n",
      "167               NA   1              61 Steady\n",
      "168               NA   1              47 Steady\n",
      "169               NA   1              68 Steady\n",
      "170               NA   2              65 Steady\n",
      "171               NA   2              57 Steady\n",
      "172               NA   1              63 Steady\n",
      "173               NA   2              56 Steady\n",
      "174               NA   1              63 Steady\n",
      "175               NA   2              69 Steady\n",
      "176               NA   2              78 Steady\n",
      "177               NA   1              68 Steady\n",
      "178               NA   1              46 Steady\n",
      "179               NA   1              69 Steady\n",
      "180               NA   1              63 Steady\n",
      "181               NA   1              72 Steady\n",
      "182               NA   1              61 Steady\n",
      "183               NA   2              62 Steady\n",
      "184               NA   2              67 Steady\n",
      "185               NA   1              50 Steady\n",
      "186               NA   1              70 Steady\n",
      "187               NA   1              65 Steady\n",
      "188               NA   1              68 Steady\n",
      "189               NA   1              63 Steady\n",
      "190               NA   1              77 Steady\n",
      "191               NA   1              72 Steady\n",
      "192               NA   1              76 Steady\n",
      "193               NA   1              66 Steady\n",
      "194               NA   2              56 Steady\n",
      "195               NA   1              71 Steady\n",
      "196               NA   1              64 Steady\n",
      "197               NA   1              74 Steady\n",
      "198               NA   1              56 Steady\n",
      "199               NA   1              69 Steady\n",
      "200               NA   1              54 Steady\n",
      "201               NA   1              61 Steady\n",
      "202               NA   1              61 Steady\n",
      "203               NA   2              66 Steady\n",
      "204               NA   1              56 Steady\n",
      "205               NA   1              76 Steady\n",
      "206               NA   2              59 Steady\n",
      "207               NA   1              78 Steady\n",
      "208               NA   2              54 Steady\n",
      "209               NA   1              63 Steady\n",
      "210               NA   1              74 Steady\n",
      "211               NA   2              65 Steady\n",
      "212               NA   1              64 Steady\n",
      "213               NA   2              47 Steady\n",
      "214               NA   2              78 Steady\n",
      "215               NA   2              67 Steady\n",
      "216               NA   1              66 Steady\n",
      "217               NA   1              59 Steady\n",
      "218               NA   1              67 Steady\n",
      "219               NA   2              72 Steady\n",
      "220               NA   2              69 Steady\n",
      "221               NA   1              63 Steady\n",
      "222               NA   1              73 Steady\n",
      "223               NA   1              77 Steady\n",
      "224               NA   1              36 Steady\n",
      "225               NA   2              57 Steady\n",
      "226               NA   2              78 Steady\n",
      "227               NA   1              65 Steady\n",
      "228               NA   1              68 Steady\n",
      "229               NA   1              65 Steady\n",
      "230               NA   2              65 Steady\n",
      "231               NA   1              65 Steady\n",
      "232               NA   2              65 Steady\n",
      "233               NA   2              66 Steady\n",
      "234               NA   1              70 Steady\n",
      "235               NA   1              50 Steady\n",
      "236               NA   1              65 Steady\n",
      "237               NA   1              67 Steady\n",
      "238               NA   1              79 Steady\n",
      "239               NA   2              63 Steady\n",
      "240               NA   1              65 Steady\n",
      "241               NA   1              71 Steady\n",
      "242               NA   1              79 Steady\n",
      "243               NA   1              59 Steady\n",
      "244               NA   1              61 Steady\n",
      "245               NA   1              56 Steady\n",
      "246               NA   2              59 Steady\n",
      "247               NA   1              73 Steady\n",
      "248               NA   1              63 Steady\n",
      "249               NA   2              52 Steady\n",
      "250               NA   1              83 Steady\n",
      "251               NA   2              66 Steady\n",
      "252               NA   1              62 Steady\n",
      "253               NA   1              68 Steady\n",
      "254               NA   2              77 Steady\n",
      "255               NA   2              53 Steady\n",
      "256               NA   1              52 Steady\n",
      "257               NA   1              56 Steady\n",
      "258               NA   1              61 Steady\n",
      "259               NA   1              66 Steady\n",
      "260               NA   1              63 Steady\n",
      "261               NA   1              68 Steady\n",
      "262               NA   2              54 Steady\n",
      "263               NA   1              55 Steady\n",
      "264               NA   1              64 Steady\n",
      "265               NA   2              66 Steady\n",
      "266               NA   2              63 Steady\n",
      "267               NA   1              65 Steady\n",
      "268               NA   1              60 Steady\n",
      "269               NA   1              74 Steady\n",
      "270               NA   2              55 Steady\n",
      "271               NA   1              70 Steady\n",
      "272               NA   1              47 Steady\n",
      "273               NA   1              60 Steady\n",
      "274               NA   1              65 Steady\n",
      "275               NA   2              65 Steady\n",
      "276               NA   1              75 Steady\n",
      "277               NA   2              64 Steady\n",
      "278               NA   1              65 Steady\n",
      "279               NA   2              71 Steady\n",
      "280               NA   1              68 Steady\n",
      "281               NA   1              65 Steady\n",
      "282               NA   1              60 Steady\n",
      "283               NA   1              68 Steady\n",
      "284               NA   2              60 Steady\n",
      "285               NA   1              71 Steady\n",
      "286               NA   1              71 Steady\n",
      "287               NA   2              60 Steady\n",
      "288               NA   1              64 Steady\n",
      "289               NA   1              66 Steady\n",
      "290               NA   1              62 Steady\n",
      "291               NA   1              61 Steady\n",
      "292               NA   1              74 Steady\n",
      "293               NA   2              63 Steady\n",
      "294               NA   1              68 Steady\n",
      "295               NA   2              62 Steady\n",
      "296               NA   1              55 Steady\n",
      "297               NA   1              49 Steady\n",
      "298               NA   1              62 Steady\n",
      "299               NA   2              58 Steady\n",
      "300               NA   1              63 Steady\n",
      "301               NA   1              64 Steady\n",
      "302               NA   1              64 Steady\n",
      "303               NA   1              65 Steady\n",
      "304               NA   1              62 Steady\n",
      "305               NA   1              74 Steady\n",
      "306               NA   1              46 Steady\n",
      "307               NA   2              70 Steady\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "#Looks like all the Steady-PD cohort are missing age at diagnosis - have checked against the raw data \n",
    "merged %>%\n",
    "filter(study == \"Steady\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      FID     IID m f sex_fam pheno age_at_diagnosis sex age_at_baseline\n",
      "1 BF-1002 BF-1002 0 0       2    -9               61   2              66\n",
      "2 BF-1003 BF-1003 0 0       1    -9               56   1              61\n",
      "3 BF-1004 BF-1004 0 0       1    -9               55   1              62\n",
      "4 BF-1006 BF-1006 0 0       1    -9               52   1              59\n",
      "5 BF-1008 BF-1008 0 0       2    -9               65   2              69\n",
      "6 BF-1010 BF-1010 0 0       1    -9               58   1              65\n",
      "    study\n",
      "1 BioFIND\n",
      "2 BioFIND\n",
      "3 BioFIND\n",
      "4 BioFIND\n",
      "5 BioFIND\n",
      "6 BioFIND\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "head(merged)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# A tibble: 6 × 6\n",
      "  study   count mean_aad sd_aad mean_entry sd_entry\n",
      "  <chr>   <int>    <dbl>  <dbl>      <dbl>    <dbl>\n",
      "1 BioFIND    92     60.9   6.20       67.7     6.26\n",
      "2 HBS       616     61.9  10.5        66.0     9.97\n",
      "3 LCC        64     60.2   9.31       68.0     9.67\n",
      "4 PDBP      790     58.7  10.2        64.5     9.02\n",
      "5 PPMI      465     61.3   9.65       61.8     9.71\n",
      "6 Sure      247     62.0   9.52       62.7     9.50\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "#Get final cohort demographics for TABLE 1 - after genetic QC, only individuals who have at least sex and age at diagnosis\n",
    "merged %>%\n",
    "group_by(study) %>%\n",
    "filter(!is.na(sex)) %>%\n",
    "filter(!is.na(age_at_diagnosis)) %>%\n",
    "summarise(count = n(),\n",
    "          mean_aad = mean(age_at_diagnosis),\n",
    "          sd_aad = sd(age_at_diagnosis),\n",
    "          mean_entry = mean(age_at_baseline),\n",
    "          sd_entry = sd(age_at_baseline))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  count\n",
      "1  2274\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "#Check total number of individuals (not separated by cohort) that have the key data\n",
    "merged %>%\n",
    "filter(!is.na(sex)) %>%\n",
    "filter(!is.na(age_at_diagnosis)) %>%\n",
    "summarise(count = n())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "`summarise()` has grouped output by 'study'. You can override using the `.groups` argument.\n",
      "# A tibble: 12 × 3\n",
      "# Groups:   study [6]\n",
      "   study     sex count\n",
      "   <chr>   <int> <int>\n",
      " 1 BioFIND     1    59\n",
      " 2 BioFIND     2    33\n",
      " 3 HBS         1   395\n",
      " 4 HBS         2   221\n",
      " 5 LCC         1    42\n",
      " 6 LCC         2    22\n",
      " 7 PDBP        1   503\n",
      " 8 PDBP        2   287\n",
      " 9 PPMI        1   300\n",
      "10 PPMI        2   165\n",
      "11 Sure        1   127\n",
      "12 Sure        2   120\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "#Get counts of men vs. women in each cohort\n",
    "merged %>%\n",
    "filter(!is.na(sex)) %>%\n",
    "filter(!is.na(age_at_diagnosis)) %>%\n",
    "group_by(study, sex) %>%\n",
    "summarise(count = n())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Analyse pathways vs MDS-UPDRSIII"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Query MDS-UPDRS Part III data, including Hoehn and Yahr stage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "mdsupdrs3_query = f\"\"\"\n",
    "\n",
    "SELECT \n",
    "participant_id, visit_name, visit_month, mds_updrs_part_iii_summary_score, upd2hy_hoehn_and_yahr_stage\n",
    "FROM `{BQ_RELEASE_DATASET}.MDS_UPDRS_Part_III`\n",
    "\n",
    "\"\"\"\n",
    "\n",
    "mdsupdrs3 = bq_query(mdsupdrs3_query)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>mds_updrs_part_iii_summary_score</th>\n",
       "      <th>upd2hy_hoehn_and_yahr_stage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>854</th>\n",
       "      <td>BF-1001</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Stage 0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2239</th>\n",
       "      <td>BF-1002</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>Stage 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21518</th>\n",
       "      <td>BF-1002</td>\n",
       "      <td>M0_5</td>\n",
       "      <td>0.5</td>\n",
       "      <td>16.0</td>\n",
       "      <td>Stage 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2240</th>\n",
       "      <td>BF-1003</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>Stage 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21525</th>\n",
       "      <td>BF-1003</td>\n",
       "      <td>M0_5</td>\n",
       "      <td>0.5</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Stage 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17382</th>\n",
       "      <td>SY-PDZX943HWN</td>\n",
       "      <td>M36</td>\n",
       "      <td>36.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>Stage 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2663</th>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>Stage 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16964</th>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>M36</td>\n",
       "      <td>36.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>Stage 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3089</th>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>Stage 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16965</th>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>M36</td>\n",
       "      <td>36.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>Stage 2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>23353 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      participant_id visit_name  visit_month  \\\n",
       "854          BF-1001         M0          0.0   \n",
       "2239         BF-1002         M0          0.0   \n",
       "21518        BF-1002       M0_5          0.5   \n",
       "2240         BF-1003         M0          0.0   \n",
       "21525        BF-1003       M0_5          0.5   \n",
       "...              ...        ...          ...   \n",
       "17382  SY-PDZX943HWN        M36         36.0   \n",
       "2663   SY-PDZY968RFA         M0          0.0   \n",
       "16964  SY-PDZY968RFA        M36         36.0   \n",
       "3089   SY-PDZZ260HUM         M0          0.0   \n",
       "16965  SY-PDZZ260HUM        M36         36.0   \n",
       "\n",
       "       mds_updrs_part_iii_summary_score upd2hy_hoehn_and_yahr_stage  \n",
       "854                                 0.0                     Stage 0  \n",
       "2239                               15.0                     Stage 1  \n",
       "21518                              16.0                     Stage 1  \n",
       "2240                               12.0                     Stage 1  \n",
       "21525                              24.0                     Stage 2  \n",
       "...                                 ...                         ...  \n",
       "17382                              10.0                     Stage 2  \n",
       "2663                               11.0                     Stage 1  \n",
       "16964                              22.0                     Stage 2  \n",
       "3089                               13.0                     Stage 2  \n",
       "16965                              14.0                     Stage 2  \n",
       "\n",
       "[23353 rows x 5 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mdsupdrs3 = mdsupdrs3.sort_values(by = ['participant_id', 'visit_month'])\n",
    "mdsupdrs3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       " 0.0      5047\n",
       " 12.0     2426\n",
       " 24.0     2227\n",
       " 6.0      1953\n",
       " 36.0     1826\n",
       " 18.0     1718\n",
       " 30.0     1454\n",
       " 48.0     1135\n",
       "-1.0       988\n",
       " 60.0      869\n",
       " 42.0      795\n",
       " 84.0      527\n",
       " 3.0       519\n",
       " 72.0      497\n",
       " 54.0      475\n",
       " 9.0       463\n",
       " 96.0      266\n",
       " 0.5       118\n",
       " 108.0      42\n",
       "-2.0         8\n",
       "Name: visit_month, dtype: int64"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Look at values of visit month\n",
    "mdsupdrs3.visit_month.value_counts()\n",
    "#Could still include negative visit month values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>mds_updrs_part_iii_summary_score</th>\n",
       "      <th>upd2hy_hoehn_and_yahr_stage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>19051</th>\n",
       "      <td>PP-3001</td>\n",
       "      <td>M48</td>\n",
       "      <td>48.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>Stage 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22585</th>\n",
       "      <td>PP-3001</td>\n",
       "      <td>M48#2</td>\n",
       "      <td>48.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>Stage 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20241</th>\n",
       "      <td>PP-3001</td>\n",
       "      <td>M60</td>\n",
       "      <td>60.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>Stage 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22840</th>\n",
       "      <td>PP-3001</td>\n",
       "      <td>M60#2</td>\n",
       "      <td>60.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>Stage 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21089</th>\n",
       "      <td>PP-3001</td>\n",
       "      <td>M84</td>\n",
       "      <td>84.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>Stage 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21691</th>\n",
       "      <td>PP-75524</td>\n",
       "      <td>M12#2</td>\n",
       "      <td>12.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>Stage 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4811</th>\n",
       "      <td>PP-75562</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>Stage 3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21511</th>\n",
       "      <td>PP-75562</td>\n",
       "      <td>M0#2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>Stage 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11096</th>\n",
       "      <td>PP-75562</td>\n",
       "      <td>M12</td>\n",
       "      <td>12.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>Stage 2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21894</th>\n",
       "      <td>PP-75562</td>\n",
       "      <td>M12#2</td>\n",
       "      <td>12.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>Stage 2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4031 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      participant_id visit_name  visit_month  \\\n",
       "19051        PP-3001        M48         48.0   \n",
       "22585        PP-3001      M48#2         48.0   \n",
       "20241        PP-3001        M60         60.0   \n",
       "22840        PP-3001      M60#2         60.0   \n",
       "21089        PP-3001        M84         84.0   \n",
       "...              ...        ...          ...   \n",
       "21691       PP-75524      M12#2         12.0   \n",
       "4811        PP-75562         M0          0.0   \n",
       "21511       PP-75562       M0#2          0.0   \n",
       "11096       PP-75562        M12         12.0   \n",
       "21894       PP-75562      M12#2         12.0   \n",
       "\n",
       "       mds_updrs_part_iii_summary_score upd2hy_hoehn_and_yahr_stage  \n",
       "19051                              39.0                     Stage 2  \n",
       "22585                              35.0                     Stage 2  \n",
       "20241                              34.0                     Stage 2  \n",
       "22840                              26.0                     Stage 2  \n",
       "21089                              36.0                     Stage 2  \n",
       "...                                 ...                         ...  \n",
       "21691                               8.0                     Stage 1  \n",
       "4811                               38.0                     Stage 3  \n",
       "21511                              17.0                     Stage 2  \n",
       "11096                              41.0                     Stage 2  \n",
       "21894                              29.0                     Stage 2  \n",
       "\n",
       "[4031 rows x 5 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Check for duplicated entries by participant id and visit month\n",
    "mdsupdrs3[mdsupdrs3.duplicated(['participant_id', 'visit_month'], keep = False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#There are some duplicates with a second entry at the same visit (with #2 in the visit name)\n",
    "#Remove rows with #\n",
    "mdsupdrs3_nodupli = mdsupdrs3.copy()\n",
    "mdsupdrs3_nodupli = mdsupdrs3_nodupli[~mdsupdrs3_nodupli.visit_name.str.contains('#')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>mds_updrs_part_iii_summary_score</th>\n",
       "      <th>upd2hy_hoehn_and_yahr_stage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [participant_id, visit_name, visit_month, mds_updrs_part_iii_summary_score, upd2hy_hoehn_and_yahr_stage]\n",
       "Index: []"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Check for any other duplicated entries by participant id and visit month\n",
    "mdsupdrs3_nodupli[mdsupdrs3_nodupli.duplicated(['participant_id', 'visit_month'], keep = False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Write as output\n",
    "with open('/home/jupyter/clinical_data_v2-5/mdsupdrs3_allparticipants.tab', 'w') as f:\n",
    "    f.write(mdsupdrs3_nodupli.to_csv(index=False, sep='\\t'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "#Make list of pathways\n",
    "pathways <- c(\"adaptive_immune\", \"alpha_synuclein\", \"innate_immune\", \"lysosome\", \"endocytosis\",\n",
    "              \"microglia\", \"monocytes\", \"mitochondria\")\n",
    "\n",
    "#Make list of cohorts - excluding LBD as these samples have been excluded from the genetic dataset\n",
    "cohorts <- c(\"PPMI\", \"PDBP\", \"HBS\", \"LCC\", \"Steady\", \"Sure\", \"BioFIND\")\n",
    "\n",
    "#Make function to analyse longitudinal outcomes using mixed effects models\n",
    "#df is the name of the .tab file in /home/jupyter/clinical_data_v2-5/ which has the data in long format\n",
    "#df = \"mdsupdrs3_allparticipants\"\n",
    "#outcome is the variable name of the outcome within df\n",
    "#outcome = \"mds_updrs_part_ii_summary_score\"\n",
    "\n",
    "analyse_cont <- function(pathway, cohort, df, outcome) {\n",
    "    \n",
    "        #Read in data for PRS, age at diagnosis, cohort, and genetic PCs\n",
    "        PRS <- read.table(paste(\"/home/jupyter/pathways/\", cohort, \".\", pathway, \"_PRS_final.txt\", sep = \"\"), \n",
    "                          sep = \"\\t\", header = T)\n",
    "    \n",
    "        data <- read.table(paste(\"/home/jupyter/clinical_data_v2-5/\", df, \".tab\", sep = \"\"), \n",
    "                             sep = \"\\t\", header = T)\n",
    "    \n",
    "        aao <- read.table(\"/home/jupyter/clinical_data_v2-5/PD_cases_agedx.tab\", sep = \"\\t\", header = T)\n",
    "\n",
    "        demo <- read.table(\"/home/jupyter/clinical_data_v2-5/PD_cases_start_end_dx_sex.tab\", sep = \"\\t\", header = T)\n",
    "\n",
    "        #Read in genetic PCs\n",
    "        pcs <- read.table(paste(\"/home/jupyter/genetic_data_v2-5/\", cohort, \".pca.eigenvec\", sep = \"\"))\n",
    "        colnames(pcs) <- c(\"FID\", \"IID\", paste(\"PC\", 1:10, sep = \"\"))\n",
    "\n",
    "        #Merge - in long format, starting from the longitudinal outcome data\n",
    "        merged <- data %>%\n",
    "        inner_join(PRS, by = \"participant_id\") %>%\n",
    "        inner_join(aao, by = c(\"participant_id\" = \"FID\")) %>%\n",
    "        inner_join(demo, by = \"IID\") %>%\n",
    "        inner_join(pcs, by = c(\"FID\", \"IID\"))\n",
    "\n",
    "        #Select the outcome variable from the function argument\n",
    "        merged$y <- merged[[outcome]]\n",
    "    \n",
    "        #Select just relevant columns\n",
    "        #This cannot be done with dplyr::select\n",
    "        merged_selected <- merged[, c('y', 'participant_id', 'study', 'PRS_z', 'sex', \n",
    "                                      'age_at_baseline', 'age_at_diagnosis', 'visit_month',\n",
    "                                      'PC1', 'PC2', 'PC3', 'PC4', 'PC5')]\n",
    "    \n",
    "        #Remove individuals missing any rows\n",
    "        merged_selected_nomiss <- merged_selected %>%\n",
    "            drop_na()\n",
    "    \n",
    "        #Get mean and SD of outcome at baseline\n",
    "        print(merged_selected_nomiss %>% \n",
    "            filter(visit_month == 0) %>%\n",
    "            summarise(mean = mean(y),\n",
    "                     sd = sd(y)))\n",
    "    \n",
    "        #Make variable years from diagnosis\n",
    "        merged_selected_nomiss <- merged_selected_nomiss %>%\n",
    "        mutate(disease_duration_at_entry = age_at_baseline - age_at_diagnosis) %>%\n",
    "        mutate(years_from_diagnosis = disease_duration_at_entry + visit_month/12)\n",
    "        \n",
    "        #Count total number of observations\n",
    "        N_obs <- dim(merged_selected_nomiss)[1]\n",
    "        \n",
    "        #Count total number of individuals - removing duplicated rows by participant_id\n",
    "        uniq_id <- merged_selected_nomiss %>%\n",
    "        distinct(participant_id)\n",
    "            \n",
    "        N_inds <- dim(uniq_id)[1]\n",
    "\n",
    "        #For mixed effects models, we usually need 2 x number of individuals because we are estimating random effects for both the intercept and slope\n",
    "        #If the number of observations is more than 2 times the number of individuals (this is because we have N_inds * 2 random effects, one for the intercept and one for the slope)\n",
    "        if (N_obs > N_inds*2) {\n",
    "            \n",
    "            #Run GLM with random intercept and slope per individual\n",
    "            lmer_PRS <- lmer(y ~ PRS_z + PRS_z*years_from_diagnosis\n",
    "                                + sex + sex*years_from_diagnosis\n",
    "                                + age_at_diagnosis\n",
    "                                + PC1 + PC2 + PC3 + PC4 + PC5\n",
    "                                + (years_from_diagnosis | participant_id),\n",
    "                                data = merged_selected_nomiss,\n",
    "                                REML = TRUE)\n",
    "\n",
    "            \n",
    "            #Save model coefficients to results table\n",
    "            #Want the slope coefficient PRS x time from diagnosis\n",
    "            summary <- c(pathway,\n",
    "                         cohort,\n",
    "                        summary(lmer_PRS)$coefficients['PRS_z:years_from_diagnosis',1],\n",
    "                        summary(lmer_PRS)$coefficients['PRS_z:years_from_diagnosis',2],\n",
    "                        summary(lmer_PRS)$coefficients['PRS_z:years_from_diagnosis',3],\n",
    "                        summary(lmer_PRS)$coefficients['PRS_z:years_from_diagnosis',4],\n",
    "                        summary(lmer_PRS)$coefficients['PRS_z:years_from_diagnosis',5],\n",
    "                        N_obs,\n",
    "                        N_inds)\n",
    "            \n",
    "            } else if (N_obs > N_inds) {\n",
    "            \n",
    "            #If there are not enough observations to fit both random intercept and slope (N_obs < 2*N_inds)\n",
    "            #Run GLM with random intercept only\n",
    "            lmer_PRS <- lmer(y ~ PRS_z + PRS_z*years_from_diagnosis\n",
    "                                + sex + sex*years_from_diagnosis\n",
    "                                + age_at_diagnosis\n",
    "                                + PC1 + PC2 + PC3 + PC4 + PC5\n",
    "                                + (1 | participant_id),\n",
    "                                data = merged_selected_nomiss,\n",
    "                                REML = TRUE)\n",
    "\n",
    "            \n",
    "            #Save model coefficients to results table\n",
    "            #Want the slope coefficient PRS x time from diagnosis\n",
    "            summary <- c(pathway,\n",
    "                         cohort,\n",
    "                        summary(lmer_PRS)$coefficients['PRS_z:years_from_diagnosis',1],\n",
    "                        summary(lmer_PRS)$coefficients['PRS_z:years_from_diagnosis',2],\n",
    "                        summary(lmer_PRS)$coefficients['PRS_z:years_from_diagnosis',3],\n",
    "                        summary(lmer_PRS)$coefficients['PRS_z:years_from_diagnosis',4],\n",
    "                        summary(lmer_PRS)$coefficients['PRS_z:years_from_diagnosis',5],\n",
    "                        N_obs,\n",
    "                        N_inds)\n",
    "            \n",
    "            } else {\n",
    "            \n",
    "        \n",
    "            #If the dataframe is empty, print NA for summary stat\n",
    "            summary <- c(pathway,\n",
    "                        cohort,\n",
    "                        rep(NA, 5),\n",
    "                        N_obs,\n",
    "                        N_inds)\n",
    "            \n",
    "            }\n",
    "        \n",
    "                     \n",
    "    return(summary)\n",
    "        \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"PPMI\"\n",
      "      mean       sd\n",
      "1 20.04662 9.259633\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 25.46819 13.48902\n",
      "[1] \"HBS\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"LCC\"\n",
      "      mean       sd\n",
      "1 24.66667 15.01111\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 22.27935 9.121966\n",
      "[1] \"BioFIND\"\n",
      "      mean      sd\n",
      "1 28.26087 14.3893\n",
      "[1] \"PPMI\"\n",
      "      mean       sd\n",
      "1 20.04662 9.259633\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 25.46819 13.48902\n",
      "[1] \"HBS\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"LCC\"\n",
      "      mean       sd\n",
      "1 24.66667 15.01111\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 22.27935 9.121966\n",
      "[1] \"BioFIND\"\n",
      "      mean      sd\n",
      "1 28.26087 14.3893\n",
      "[1] \"PPMI\"\n",
      "      mean       sd\n",
      "1 20.04662 9.259633\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 25.46819 13.48902\n",
      "[1] \"HBS\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"LCC\"\n",
      "      mean       sd\n",
      "1 24.66667 15.01111\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 22.27935 9.121966\n",
      "[1] \"BioFIND\"\n",
      "      mean      sd\n",
      "1 28.26087 14.3893\n",
      "[1] \"PPMI\"\n",
      "      mean       sd\n",
      "1 20.04662 9.259633\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 25.46819 13.48902\n",
      "[1] \"HBS\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"LCC\"\n",
      "      mean       sd\n",
      "1 24.66667 15.01111\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 22.27935 9.121966\n",
      "[1] \"BioFIND\"\n",
      "      mean      sd\n",
      "1 28.26087 14.3893\n",
      "[1] \"PPMI\"\n",
      "      mean       sd\n",
      "1 20.04662 9.259633\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 25.46819 13.48902\n",
      "[1] \"HBS\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"LCC\"\n",
      "      mean       sd\n",
      "1 24.66667 15.01111\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 22.27935 9.121966\n",
      "[1] \"BioFIND\"\n",
      "      mean      sd\n",
      "1 28.26087 14.3893\n",
      "[1] \"PPMI\"\n",
      "      mean       sd\n",
      "1 20.04662 9.259633\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 25.46819 13.48902\n",
      "[1] \"HBS\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"LCC\"\n",
      "      mean       sd\n",
      "1 24.66667 15.01111\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 22.27935 9.121966\n",
      "[1] \"BioFIND\"\n",
      "      mean      sd\n",
      "1 28.26087 14.3893\n",
      "[1] \"PPMI\"\n",
      "      mean       sd\n",
      "1 20.04662 9.259633\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 25.46819 13.48902\n",
      "[1] \"HBS\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"LCC\"\n",
      "      mean       sd\n",
      "1 24.66667 15.01111\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 22.27935 9.121966\n",
      "[1] \"BioFIND\"\n",
      "      mean      sd\n",
      "1 28.26087 14.3893\n",
      "[1] \"PPMI\"\n",
      "      mean       sd\n",
      "1 20.04662 9.259633\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 25.46819 13.48902\n",
      "[1] \"HBS\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"LCC\"\n",
      "      mean       sd\n",
      "1 24.66667 15.01111\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 22.27935 9.121966\n",
      "[1] \"BioFIND\"\n",
      "      mean      sd\n",
      "1 28.26087 14.3893\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "updrs3_results_list <- list()\n",
    "\n",
    "#Loop over each pathway and cohort\n",
    "for (pathway in pathways) {\n",
    "    for (cohort in cohorts) {\n",
    "        print(cohort)\n",
    "        i <- length(updrs3_results_list)\n",
    "        updrs3_results_list[[i+1]] <- analyse_cont(pathway, cohort, \"mdsupdrs3_allparticipants\", \"mds_updrs_part_iii_summary_score\")\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           pathway  cohort               Coeff                 se\n",
      "1  adaptive_immune    PPMI  0.0721015112189424  0.102865127038644\n",
      "2  adaptive_immune    PDBP   0.204767173955193 0.0919642819515198\n",
      "3  adaptive_immune     HBS                <NA>               <NA>\n",
      "4  adaptive_immune     LCC                <NA>               <NA>\n",
      "5  adaptive_immune  Steady                <NA>               <NA>\n",
      "6  adaptive_immune    Sure   0.294400922273215  0.235417361481696\n",
      "7  adaptive_immune BioFIND   0.417312692183422  0.392696002966269\n",
      "8  alpha_synuclein    PPMI  0.0693572600950412  0.107291291622245\n",
      "9  alpha_synuclein    PDBP  0.0537062026147313 0.0935439218821116\n",
      "10 alpha_synuclein     HBS                <NA>               <NA>\n",
      "11 alpha_synuclein     LCC                <NA>               <NA>\n",
      "12 alpha_synuclein  Steady                <NA>               <NA>\n",
      "13 alpha_synuclein    Sure   0.116111593194773  0.234422621200671\n",
      "14 alpha_synuclein BioFIND   0.246137505035106  0.369339441828035\n",
      "15   innate_immune    PPMI  0.0759003110997098  0.106917517061901\n",
      "16   innate_immune    PDBP   0.147564591187306 0.0885050241897939\n",
      "17   innate_immune     HBS                <NA>               <NA>\n",
      "18   innate_immune     LCC                <NA>               <NA>\n",
      "19   innate_immune  Steady                <NA>               <NA>\n",
      "20   innate_immune    Sure   -0.14237400419273  0.229214367764768\n",
      "21   innate_immune BioFIND  -0.624082180444006  0.435075964751612\n",
      "22        lysosome    PPMI -0.0475776384174448  0.103389959462461\n",
      "23        lysosome    PDBP   0.138147747203893 0.0903050400162387\n",
      "24        lysosome     HBS                <NA>               <NA>\n",
      "25        lysosome     LCC                <NA>               <NA>\n",
      "26        lysosome  Steady                <NA>               <NA>\n",
      "27        lysosome    Sure   0.239517860418466  0.236528495377645\n",
      "28        lysosome BioFIND  -0.499538385797322  0.403516685222135\n",
      "29     endocytosis    PPMI 0.00958039037217941  0.106083414304301\n",
      "30     endocytosis    PDBP   0.112424336739535 0.0911038696087938\n",
      "31     endocytosis     HBS                <NA>               <NA>\n",
      "32     endocytosis     LCC                <NA>               <NA>\n",
      "33     endocytosis  Steady                <NA>               <NA>\n",
      "34     endocytosis    Sure   0.145952800483109  0.238995305971402\n",
      "35     endocytosis BioFIND  0.0164158516855939  0.410500906688303\n",
      "36       microglia    PPMI  0.0631627857415372   0.10707387082565\n",
      "37       microglia    PDBP -0.0681121745788321 0.0931494698215704\n",
      "38       microglia     HBS                <NA>               <NA>\n",
      "39       microglia     LCC                <NA>               <NA>\n",
      "40       microglia  Steady                <NA>               <NA>\n",
      "41       microglia    Sure  -0.010131426420969  0.233241917501812\n",
      "42       microglia BioFIND  -0.317025231938629  0.384848978975049\n",
      "43       monocytes    PPMI  0.0144009778096812  0.106723338727988\n",
      "44       monocytes    PDBP  -0.082679364380479 0.0921261270447425\n",
      "45       monocytes     HBS                <NA>               <NA>\n",
      "46       monocytes     LCC                <NA>               <NA>\n",
      "47       monocytes  Steady                <NA>               <NA>\n",
      "48       monocytes    Sure  -0.239108147227544  0.232264189694972\n",
      "49       monocytes BioFIND   0.490001248337283  0.460243791654704\n",
      "50    mitochondria    PPMI  -0.011121768733043  0.106385501563977\n",
      "51    mitochondria    PDBP 0.00492043023836047 0.0895177457469975\n",
      "52    mitochondria     HBS                <NA>               <NA>\n",
      "53    mitochondria     LCC                <NA>               <NA>\n",
      "54    mitochondria  Steady                <NA>               <NA>\n",
      "55    mitochondria    Sure  0.0425138608842627  0.236534956297232\n",
      "56    mitochondria BioFIND  -0.514788002520053  0.443664878068863\n",
      "                 df             t_value             Pvalue N_obs N_inds\n",
      "1  335.488392192724   0.700932505453044  0.483831000511307  5669    465\n",
      "2  413.382986405136     2.2265946040131 0.0265125268860296  2937    787\n",
      "3              <NA>                <NA>               <NA>     0      0\n",
      "4              <NA>                <NA>               <NA>     3      3\n",
      "5              <NA>                <NA>               <NA>     0      0\n",
      "6  219.148999790441    1.25054889928373  0.212432675367478  1262    247\n",
      "7  79.8247832824888    1.06268637580013  0.291128584285046   183     92\n",
      "8  342.814069408208   0.646438858609672  0.518427715202727  5669    465\n",
      "9  421.729131995571   0.574128190631288  0.566187138616724  2937    787\n",
      "10             <NA>                <NA>               <NA>     0      0\n",
      "11             <NA>                <NA>               <NA>     3      3\n",
      "12             <NA>                <NA>               <NA>     0      0\n",
      "13  214.63317071886    0.49530882557353  0.620888850476585  1262    247\n",
      "14  79.888549075824   0.666426265813516  0.507058714963879   183     92\n",
      "15  338.83359313304   0.709895938340645  0.478256921947707  5669    465\n",
      "16 353.186106905693    1.66730185702071 0.0963408052592058  2937    787\n",
      "17             <NA>                <NA>               <NA>     0      0\n",
      "18             <NA>                <NA>               <NA>     3      3\n",
      "19             <NA>                <NA>               <NA>     0      0\n",
      "20 213.545238171658  -0.621139091677019  0.535170555309373  1262    247\n",
      "21 79.9791046722596    -1.4344211838967  0.155350712617574   183     92\n",
      "22 330.234011679332  -0.460176584504025  0.645692321183143  5669    465\n",
      "23 393.766170180894    1.52979000041472  0.126871636360627  2937    787\n",
      "24             <NA>                <NA>               <NA>     0      0\n",
      "25             <NA>                <NA>               <NA>     3      3\n",
      "26             <NA>                <NA>               <NA>     0      0\n",
      "27 223.114672275196    1.01263849852868  0.312329753802526  1262    247\n",
      "28 79.8424523872933   -1.23796215644051  0.219358023294519   183     92\n",
      "29 347.038186262644   0.090309973854141  0.928092994500219  5669    465\n",
      "30  409.48956267746    1.23402372722798  0.217901823251155  2937    787\n",
      "31             <NA>                <NA>               <NA>     0      0\n",
      "32             <NA>                <NA>               <NA>     3      3\n",
      "33             <NA>                <NA>               <NA>     0      0\n",
      "34 225.971346830669   0.610693167758589  0.542016472918069  1262    247\n",
      "35  80.083464285954  0.0399898061566491  0.968200780903874   183     92\n",
      "36 344.342360987883    0.58989915330871  0.555645140370611  5669    465\n",
      "37 399.639458934546  -0.731213765459988  0.465077098475993  2937    787\n",
      "38             <NA>                <NA>               <NA>     0      0\n",
      "39             <NA>                <NA>               <NA>     3      3\n",
      "40             <NA>                <NA>               <NA>     0      0\n",
      "41 221.503491908409 -0.0434374169509658   0.96539198284051  1262    247\n",
      "42 79.9033914582911  -0.823765293032471  0.412524993456414   183     92\n",
      "43 344.991153772107   0.134937474607927  0.892740001479115  5669    465\n",
      "44 381.177232115225  -0.897458376170795  0.370040859078511  2937    787\n",
      "45             <NA>                <NA>               <NA>     0      0\n",
      "46             <NA>                <NA>               <NA>     3      3\n",
      "47             <NA>                <NA>               <NA>     0      0\n",
      "48 218.962153556969   -1.02946626228331   0.30439644629053  1262    247\n",
      "49 79.8318898697002    1.06465585679188  0.290241305150107   183     92\n",
      "50 342.800474755433  -0.104542146904808  0.916800261971131  5669    465\n",
      "51 360.007082589113  0.0549659757101904  0.956196060805588  2937    787\n",
      "52             <NA>                <NA>               <NA>     0      0\n",
      "53             <NA>                <NA>               <NA>     3      3\n",
      "54             <NA>                <NA>               <NA>     0      0\n",
      "55 224.319852894302   0.179736059100032  0.857522035946331  1262    247\n",
      "56 79.8328047882221   -1.16030821452617  0.249381516173099   183     92\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "#Combine lists into a single dataframe\n",
    "updrs3_results <- do.call(rbind, updrs3_results_list)\n",
    "\n",
    "#Set column names\n",
    "colnames(updrs3_results) <- c(\"pathway\", \"cohort\", \"Coeff\", \"se\", \"df\", \"t_value\", \"Pvalue\", \"N_obs\", \"N_inds\")\n",
    "\n",
    "#Make into dataframe\n",
    "updrs3_results <- as.data.frame(updrs3_results)\n",
    "\n",
    "#Save output\n",
    "write.table(updrs3_results, \"/home/jupyter/pathways/pathways_mdsupdrs3_results.txt\",\n",
    "           sep = \"\\t\", row.names = F, col.names = T, quote = F)\n",
    "\n",
    "updrs3_results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Analyse pathways vs MDS-UPDRSII"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "mdsupdrs2_query = f\"\"\"\n",
    "\n",
    "SELECT \n",
    "participant_id, visit_name, visit_month, mds_updrs_part_ii_summary_score\n",
    "FROM `{BQ_RELEASE_DATASET}.MDS_UPDRS_Part_II`\n",
    "\n",
    "\"\"\"\n",
    "\n",
    "mdsupdrs2 = bq_query(mdsupdrs2_query)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>mds_updrs_part_ii_summary_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3335</th>\n",
       "      <td>BF-1002</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149</th>\n",
       "      <td>BF-1003</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4220</th>\n",
       "      <td>BF-1004</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3624</th>\n",
       "      <td>BF-1006</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1400</th>\n",
       "      <td>BF-1008</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16062</th>\n",
       "      <td>SY-PDZX943HWN</td>\n",
       "      <td>M36</td>\n",
       "      <td>36.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3333</th>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16300</th>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>M36</td>\n",
       "      <td>36.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3334</th>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15964</th>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>M36</td>\n",
       "      <td>36.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21220 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      participant_id visit_name  visit_month  mds_updrs_part_ii_summary_score\n",
       "3335         BF-1002         M0          0.0                             11.0\n",
       "1149         BF-1003         M0          0.0                             11.0\n",
       "4220         BF-1004         M0          0.0                             11.0\n",
       "3624         BF-1006         M0          0.0                             14.0\n",
       "1400         BF-1008         M0          0.0                              3.0\n",
       "...              ...        ...          ...                              ...\n",
       "16062  SY-PDZX943HWN        M36         36.0                              5.0\n",
       "3333   SY-PDZY968RFA         M0          0.0                              0.0\n",
       "16300  SY-PDZY968RFA        M36         36.0                              5.0\n",
       "3334   SY-PDZZ260HUM         M0          0.0                              1.0\n",
       "15964  SY-PDZZ260HUM        M36         36.0                              4.0\n",
       "\n",
       "[21220 rows x 4 columns]"
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mdsupdrs2 = mdsupdrs2.sort_values(by = ['participant_id', 'visit_month'])\n",
    "mdsupdrs2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>mds_updrs_part_ii_summary_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>11398</th>\n",
       "      <td>PP-3066</td>\n",
       "      <td>M18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11399</th>\n",
       "      <td>PP-3066</td>\n",
       "      <td>M18#2</td>\n",
       "      <td>18.0</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11109</th>\n",
       "      <td>PP-3067</td>\n",
       "      <td>M18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11570</th>\n",
       "      <td>PP-3067</td>\n",
       "      <td>M18#2</td>\n",
       "      <td>18.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11217</th>\n",
       "      <td>PP-3078</td>\n",
       "      <td>M18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11402</th>\n",
       "      <td>PP-3078</td>\n",
       "      <td>M18#2</td>\n",
       "      <td>18.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4970</th>\n",
       "      <td>PP-3251</td>\n",
       "      <td>M3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5109</th>\n",
       "      <td>PP-3251</td>\n",
       "      <td>M3#2</td>\n",
       "      <td>3.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5273</th>\n",
       "      <td>PP-3252</td>\n",
       "      <td>M3#2</td>\n",
       "      <td>3.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5274</th>\n",
       "      <td>PP-3252</td>\n",
       "      <td>M3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11243</th>\n",
       "      <td>PP-3552</td>\n",
       "      <td>M18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11244</th>\n",
       "      <td>PP-3552</td>\n",
       "      <td>M18#2</td>\n",
       "      <td>18.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10190</th>\n",
       "      <td>PP-3653</td>\n",
       "      <td>M18#2</td>\n",
       "      <td>18.0</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11004</th>\n",
       "      <td>PP-3653</td>\n",
       "      <td>M18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5156</th>\n",
       "      <td>PP-3709</td>\n",
       "      <td>M3#2</td>\n",
       "      <td>3.0</td>\n",
       "      <td>13.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5221</th>\n",
       "      <td>PP-3709</td>\n",
       "      <td>M3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>10.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10546</th>\n",
       "      <td>PP-3900</td>\n",
       "      <td>M18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11272</th>\n",
       "      <td>PP-3900</td>\n",
       "      <td>M18#2</td>\n",
       "      <td>18.0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10654</th>\n",
       "      <td>PP-42445</td>\n",
       "      <td>M18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10655</th>\n",
       "      <td>PP-42445</td>\n",
       "      <td>M18#2</td>\n",
       "      <td>18.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      participant_id visit_name  visit_month  mds_updrs_part_ii_summary_score\n",
       "11398        PP-3066        M18         18.0                             13.0\n",
       "11399        PP-3066      M18#2         18.0                             15.0\n",
       "11109        PP-3067        M18         18.0                             14.0\n",
       "11570        PP-3067      M18#2         18.0                              9.0\n",
       "11217        PP-3078        M18         18.0                              5.0\n",
       "11402        PP-3078      M18#2         18.0                              7.0\n",
       "4970         PP-3251         M3          3.0                              7.0\n",
       "5109         PP-3251       M3#2          3.0                             11.0\n",
       "5273         PP-3252       M3#2          3.0                              9.0\n",
       "5274         PP-3252         M3          3.0                              9.0\n",
       "11243        PP-3552        M18         18.0                              5.0\n",
       "11244        PP-3552      M18#2         18.0                              3.0\n",
       "10190        PP-3653      M18#2         18.0                             15.0\n",
       "11004        PP-3653        M18         18.0                             13.0\n",
       "5156         PP-3709       M3#2          3.0                             13.0\n",
       "5221         PP-3709         M3          3.0                             10.0\n",
       "10546        PP-3900        M18         18.0                              5.0\n",
       "11272        PP-3900      M18#2         18.0                              7.0\n",
       "10654       PP-42445        M18         18.0                              0.0\n",
       "10655       PP-42445      M18#2         18.0                              0.0"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Check for duplicated entries by participant id and visit month\n",
    "mdsupdrs2[mdsupdrs2.duplicated(['participant_id', 'visit_month'], keep = False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Remove duplicated visit entries - rows with #\n",
    "mdsupdrs2_nodupli = mdsupdrs2.copy()\n",
    "mdsupdrs2_nodupli = mdsupdrs2_nodupli[~mdsupdrs2_nodupli.visit_name.str.contains('#')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>mds_updrs_part_ii_summary_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [participant_id, visit_name, visit_month, mds_updrs_part_ii_summary_score]\n",
       "Index: []"
      ]
     },
     "execution_count": 139,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Check for any other duplicated entries by participant id and visit month\n",
    "mdsupdrs2_nodupli[mdsupdrs2_nodupli.duplicated(['participant_id', 'visit_month'], keep = False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Write as output\n",
    "with open('/home/jupyter/clinical_data_v2-5/mdsupdrs2_allparticipants.tab', 'w') as f:\n",
    "    f.write(mdsupdrs2_nodupli.to_csv(index=False, sep='\\t'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"PPMI\"\n",
      "      mean       sd\n",
      "1 5.771028 4.138572\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 10.79625 7.767168\n",
      "[1] \"HBS\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"LCC\"\n",
      "      mean       sd\n",
      "1 11.66667 13.42882\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 5.789474 4.316283\n",
      "[1] \"BioFIND\"\n",
      "      mean      sd\n",
      "1 11.04348 6.35906\n",
      "[1] \"PPMI\"\n",
      "      mean       sd\n",
      "1 5.771028 4.138572\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 10.79625 7.767168\n",
      "[1] \"HBS\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"LCC\"\n",
      "      mean       sd\n",
      "1 11.66667 13.42882\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 5.789474 4.316283\n",
      "[1] \"BioFIND\"\n",
      "      mean      sd\n",
      "1 11.04348 6.35906\n",
      "[1] \"PPMI\"\n",
      "      mean       sd\n",
      "1 5.771028 4.138572\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 10.79625 7.767168\n",
      "[1] \"HBS\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"LCC\"\n",
      "      mean       sd\n",
      "1 11.66667 13.42882\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 5.789474 4.316283\n",
      "[1] \"BioFIND\"\n",
      "      mean      sd\n",
      "1 11.04348 6.35906\n",
      "[1] \"PPMI\"\n",
      "      mean       sd\n",
      "1 5.771028 4.138572\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 10.79625 7.767168\n",
      "[1] \"HBS\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"LCC\"\n",
      "      mean       sd\n",
      "1 11.66667 13.42882\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 5.789474 4.316283\n",
      "[1] \"BioFIND\"\n",
      "      mean      sd\n",
      "1 11.04348 6.35906\n",
      "[1] \"PPMI\"\n",
      "      mean       sd\n",
      "1 5.771028 4.138572\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 10.79625 7.767168\n",
      "[1] \"HBS\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"LCC\"\n",
      "      mean       sd\n",
      "1 11.66667 13.42882\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 5.789474 4.316283\n",
      "[1] \"BioFIND\"\n",
      "      mean      sd\n",
      "1 11.04348 6.35906\n",
      "[1] \"PPMI\"\n",
      "      mean       sd\n",
      "1 5.771028 4.138572\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 10.79625 7.767168\n",
      "[1] \"HBS\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"LCC\"\n",
      "      mean       sd\n",
      "1 11.66667 13.42882\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 5.789474 4.316283\n",
      "[1] \"BioFIND\"\n",
      "      mean      sd\n",
      "1 11.04348 6.35906\n",
      "[1] \"PPMI\"\n",
      "      mean       sd\n",
      "1 5.771028 4.138572\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 10.79625 7.767168\n",
      "[1] \"HBS\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"LCC\"\n",
      "      mean       sd\n",
      "1 11.66667 13.42882\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 5.789474 4.316283\n",
      "[1] \"BioFIND\"\n",
      "      mean      sd\n",
      "1 11.04348 6.35906\n",
      "[1] \"PPMI\"\n",
      "      mean       sd\n",
      "1 5.771028 4.138572\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 10.79625 7.767168\n",
      "[1] \"HBS\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"LCC\"\n",
      "      mean       sd\n",
      "1 11.66667 13.42882\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 5.789474 4.316283\n",
      "[1] \"BioFIND\"\n",
      "      mean      sd\n",
      "1 11.04348 6.35906\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "updrs2_results_list <- list()\n",
    "\n",
    "#Loop over each pathway and cohort\n",
    "for (pathway in pathways) {\n",
    "    for (cohort in cohorts) {\n",
    "        print(cohort)\n",
    "        i <- length(updrs2_results_list)\n",
    "        updrs2_results_list[[i+1]] <- analyse_cont(pathway, cohort, \"mdsupdrs2_allparticipants\", \"mds_updrs_part_ii_summary_score\")\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           pathway  cohort               Coeff                 se\n",
      "1  adaptive_immune    PPMI  0.0435532363926763 0.0524325288887569\n",
      "2  adaptive_immune    PDBP  0.0692447472877802 0.0516395557084703\n",
      "3  adaptive_immune     HBS                <NA>               <NA>\n",
      "4  adaptive_immune     LCC                <NA>               <NA>\n",
      "5  adaptive_immune  Steady                <NA>               <NA>\n",
      "6  adaptive_immune    Sure  0.0400043398960736 0.0983470886906584\n",
      "7  adaptive_immune BioFIND                <NA>               <NA>\n",
      "8  alpha_synuclein    PPMI 0.00244144009967074 0.0546595102967342\n",
      "9  alpha_synuclein    PDBP  0.0494776409269386 0.0526424916445889\n",
      "10 alpha_synuclein     HBS                <NA>               <NA>\n",
      "11 alpha_synuclein     LCC                <NA>               <NA>\n",
      "12 alpha_synuclein  Steady                <NA>               <NA>\n",
      "13 alpha_synuclein    Sure  0.0483134294830268 0.0980916166758742\n",
      "14 alpha_synuclein BioFIND                <NA>               <NA>\n",
      "15   innate_immune    PPMI  0.0126278783316729 0.0544209617297168\n",
      "16   innate_immune    PDBP -0.0242078406751316 0.0497844321266854\n",
      "17   innate_immune     HBS                <NA>               <NA>\n",
      "18   innate_immune     LCC                <NA>               <NA>\n",
      "19   innate_immune  Steady                <NA>               <NA>\n",
      "20   innate_immune    Sure  -0.133826820752486  0.095469301673083\n",
      "21   innate_immune BioFIND                <NA>               <NA>\n",
      "22        lysosome    PPMI  0.0536581682467098 0.0527849314333855\n",
      "23        lysosome    PDBP 0.00416868474750996  0.051165484836527\n",
      "24        lysosome     HBS                <NA>               <NA>\n",
      "25        lysosome     LCC                <NA>               <NA>\n",
      "26        lysosome  Steady                <NA>               <NA>\n",
      "27        lysosome    Sure -0.0274621941119495 0.0990516737371417\n",
      "28        lysosome BioFIND                <NA>               <NA>\n",
      "29     endocytosis    PPMI  0.0289257286456244 0.0538631352620975\n",
      "30     endocytosis    PDBP  0.0431346312131801  0.051088499544596\n",
      "31     endocytosis     HBS                <NA>               <NA>\n",
      "32     endocytosis     LCC                <NA>               <NA>\n",
      "33     endocytosis  Steady                <NA>               <NA>\n",
      "34     endocytosis    Sure  0.0866561130196984  0.100077565660805\n",
      "35     endocytosis BioFIND                <NA>               <NA>\n",
      "36       microglia    PPMI 0.00804603793501585 0.0543204119382124\n",
      "37       microglia    PDBP -0.0468634217984853 0.0525706261923791\n",
      "38       microglia     HBS                <NA>               <NA>\n",
      "39       microglia     LCC                <NA>               <NA>\n",
      "40       microglia  Steady                <NA>               <NA>\n",
      "41       microglia    Sure  0.0617892861174982 0.0973072564323127\n",
      "42       microglia BioFIND                <NA>               <NA>\n",
      "43       monocytes    PPMI  0.0191046054000922 0.0542119753701256\n",
      "44       monocytes    PDBP -0.0723069957316134 0.0522590775473835\n",
      "45       monocytes     HBS                <NA>               <NA>\n",
      "46       monocytes     LCC                <NA>               <NA>\n",
      "47       monocytes  Steady                <NA>               <NA>\n",
      "48       monocytes    Sure  0.0991438835005254 0.0972082459808794\n",
      "49       monocytes BioFIND                <NA>               <NA>\n",
      "50    mitochondria    PPMI -0.0248022450685781 0.0540443429699711\n",
      "51    mitochondria    PDBP  0.0337324575023054 0.0504561591988745\n",
      "52    mitochondria     HBS                <NA>               <NA>\n",
      "53    mitochondria     LCC                <NA>               <NA>\n",
      "54    mitochondria  Steady                <NA>               <NA>\n",
      "55    mitochondria    Sure -0.0115123564388154 0.0987582046687296\n",
      "56    mitochondria BioFIND                <NA>               <NA>\n",
      "                 df            t_value            Pvalue N_obs N_inds\n",
      "1  300.159875010733    0.8306529804252 0.406829913637994  5670    465\n",
      "2  407.819223664464   1.34092453619662 0.180691328340483  2878    747\n",
      "3              <NA>               <NA>              <NA>     0      0\n",
      "4              <NA>               <NA>              <NA>     3      3\n",
      "5              <NA>               <NA>              <NA>     0      0\n",
      "6  227.999605056552  0.406766894970359 0.684560783360912  1263    247\n",
      "7              <NA>               <NA>              <NA>    92     92\n",
      "8   308.10836960687 0.0446663368628205 0.964402200438138  5670    465\n",
      "9  416.711685018563  0.939880301657879 0.347823442366852  2878    747\n",
      "10             <NA>               <NA>              <NA>     0      0\n",
      "11             <NA>               <NA>              <NA>     3      3\n",
      "12             <NA>               <NA>              <NA>     0      0\n",
      "13 225.775605680109  0.492533726329231 0.622820707473556  1263    247\n",
      "14             <NA>               <NA>              <NA>    92     92\n",
      "15 305.182281021989   0.23204070509429 0.816661917274177  5670    465\n",
      "16 363.109139031248 -0.486253224974635 0.627080966421732  2878    747\n",
      "17             <NA>               <NA>              <NA>     0      0\n",
      "18             <NA>               <NA>              <NA>     3      3\n",
      "19             <NA>               <NA>              <NA>     0      0\n",
      "20 225.666317862634   -1.4017785655409 0.162355299756858  1263    247\n",
      "21             <NA>               <NA>              <NA>    92     92\n",
      "22 295.458378466166   1.01654329729359 0.310202429127357  5670    465\n",
      "23 405.143963798734 0.0814745479463128 0.935104799029503  2878    747\n",
      "24             <NA>               <NA>              <NA>     0      0\n",
      "25             <NA>               <NA>              <NA>     3      3\n",
      "26             <NA>               <NA>              <NA>     0      0\n",
      "27 232.937817021609 -0.277251186939327 0.781833125268032  1263    247\n",
      "28             <NA>               <NA>              <NA>    92     92\n",
      "29 310.605317916571  0.537022743011005 0.591636375745556  5670    465\n",
      "30 404.844787634133   0.84431196057201 0.398993741610702  2878    747\n",
      "31             <NA>               <NA>              <NA>     0      0\n",
      "32             <NA>               <NA>              <NA>     3      3\n",
      "33             <NA>               <NA>              <NA>     0      0\n",
      "34 237.780538752707  0.865889497286572 0.387423586189497  1263    247\n",
      "35             <NA>               <NA>              <NA>    92     92\n",
      "36 311.228509011243  0.148121813659439 0.882342553179201  5670    465\n",
      "37 409.655267302398 -0.891437389902706 0.373217844707528  2878    747\n",
      "38             <NA>               <NA>              <NA>     0      0\n",
      "39             <NA>               <NA>              <NA>     3      3\n",
      "40             <NA>               <NA>              <NA>     0      0\n",
      "41 232.174463299187  0.634991555439435 0.526059194292245  1263    247\n",
      "42             <NA>               <NA>              <NA>    92     92\n",
      "43 311.047871477871  0.352405631221845 0.724772736594048  5670    465\n",
      "44 400.711892440714  -1.38362556564555  0.16724340258306  2878    747\n",
      "45             <NA>               <NA>              <NA>     0      0\n",
      "46             <NA>               <NA>              <NA>     3      3\n",
      "47             <NA>               <NA>              <NA>     0      0\n",
      "48 231.820449736198   1.01991227698962 0.308833259913962  1263    247\n",
      "49             <NA>               <NA>              <NA>    92     92\n",
      "50 310.144866646805 -0.458923981782128 0.646610209655391  5670    465\n",
      "51 385.550049161831  0.668549846795669 0.504182735808137  2878    747\n",
      "52             <NA>               <NA>              <NA>     0      0\n",
      "53             <NA>               <NA>              <NA>     3      3\n",
      "54             <NA>               <NA>              <NA>     0      0\n",
      "55 234.920196890677 -0.116571139354264 0.907299501867861  1263    247\n",
      "56             <NA>               <NA>              <NA>    92     92\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "#Combine lists into a single dataframe\n",
    "updrs2_results <- do.call(rbind, updrs2_results_list)\n",
    "\n",
    "#Set column names\n",
    "colnames(updrs2_results) <- c(\"pathway\", \"cohort\", \"Coeff\", \"se\", \"df\", \"t_value\", \"Pvalue\", \"N_obs\", \"N_inds\")\n",
    "\n",
    "#Make into dataframe\n",
    "updrs2_results <- as.data.frame(updrs2_results)\n",
    "\n",
    "#Save output\n",
    "write.table(updrs2_results, \"/home/jupyter/pathways/pathways_mdsupdrs2_results.txt\",\n",
    "           sep = \"\\t\", row.names = F, col.names = T, quote = F)\n",
    "\n",
    "updrs2_results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Analyse pathways vs H&Y3+"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Read in H&Y data (in MDS-UPDRS dataset)\n",
    "mdsupdrs3_nodupli = pd.read_csv(f'/home/jupyter/notebooks/clinical_data_v2-5/mdsupdrs3_allparticipants.tab',\n",
    "                               delimiter = \"\\t\")\n",
    "\n",
    "#Read in age at diagnosis and demographics data\n",
    "demo = pd.read_csv(f'/home/jupyter/notebooks/clinical_data_v2-5/PD_cases_start_end_dx_sex.tab',\n",
    "                    delimiter = \"\\t\")\n",
    "    \n",
    "aao = pd.read_csv(f'/home/jupyter/notebooks/clinical_data_v2-5/PD_cases_agedx.tab',\n",
    "                   delimiter = \"\\t\").loc[:, ['FID', 'IID', 'age_at_diagnosis']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Merge with demographics and age at diagnosis data to get time from diagnosis\n",
    "temp = mdsupdrs3_nodupli.merge(demo, left_on = 'participant_id', right_on = 'IID')\n",
    "merged = temp.merge(aao, on = (['FID', 'IID']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Calculate time from diagnosis for each visit\n",
    "merged['YEARS_DIAGNOSIS_TO_BL'] = merged.age_at_baseline - merged.age_at_diagnosis\n",
    "merged['YEARS_FROM_DIAGNOSIS'] = merged.YEARS_DIAGNOSIS_TO_BL + merged.visit_month/12"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Make new variable for H&Y stage as numeric\n",
    "merged['HOEHN_YAHR_STAGE'] = merged.upd2hy_hoehn_and_yahr_stage.str.replace(r'Stage ', '').astype(float)\n",
    "\n",
    "#Make new variable coding whether H&Y stage is >=3 at each visit\n",
    "merged['HY3'] = [np.nan if pd.isna(i)\n",
    "                else 1 if i >=3\n",
    "                else 0 for i in merged.HOEHN_YAHR_STAGE]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Make a variable for H&Y3 EVER  - whether a patient ever has a HY3 greater than one in the course of their study \n",
    "HYEV = merged.loc[:, [\"participant_id\", 'HY3']].groupby(\"participant_id\").agg(func=max)\n",
    "\n",
    "#Join to main HY dataframe \n",
    "HY_merged = merged.join(HYEV, on = \"participant_id\", rsuffix=\"_EV\").reset_index(drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>mds_updrs_part_iii_summary_score</th>\n",
       "      <th>upd2hy_hoehn_and_yahr_stage</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>sex</th>\n",
       "      <th>age_at_baseline</th>\n",
       "      <th>age_at_diagnosis</th>\n",
       "      <th>YEARS_DIAGNOSIS_TO_BL</th>\n",
       "      <th>YEARS_FROM_DIAGNOSIS</th>\n",
       "      <th>HOEHN_YAHR_STAGE</th>\n",
       "      <th>HY3</th>\n",
       "      <th>HY3_EV</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BF-1002</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>Stage 1</td>\n",
       "      <td>BF-1002</td>\n",
       "      <td>BF-1002</td>\n",
       "      <td>2</td>\n",
       "      <td>66</td>\n",
       "      <td>61.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BF-1002</td>\n",
       "      <td>M0_5</td>\n",
       "      <td>0.5</td>\n",
       "      <td>16.0</td>\n",
       "      <td>Stage 1</td>\n",
       "      <td>BF-1002</td>\n",
       "      <td>BF-1002</td>\n",
       "      <td>2</td>\n",
       "      <td>66</td>\n",
       "      <td>61.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.041667</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BF-1003</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>Stage 1</td>\n",
       "      <td>BF-1003</td>\n",
       "      <td>BF-1003</td>\n",
       "      <td>1</td>\n",
       "      <td>61</td>\n",
       "      <td>56.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.000000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BF-1003</td>\n",
       "      <td>M0_5</td>\n",
       "      <td>0.5</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Stage 2</td>\n",
       "      <td>BF-1003</td>\n",
       "      <td>BF-1003</td>\n",
       "      <td>1</td>\n",
       "      <td>61</td>\n",
       "      <td>56.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.041667</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BF-1004</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>Stage 2</td>\n",
       "      <td>BF-1004</td>\n",
       "      <td>BF-1004</td>\n",
       "      <td>1</td>\n",
       "      <td>62</td>\n",
       "      <td>55.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.000000</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14427</th>\n",
       "      <td>SY-PDZX943HWN</td>\n",
       "      <td>M36</td>\n",
       "      <td>36.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>Stage 2</td>\n",
       "      <td>SY-PDZX943HWN</td>\n",
       "      <td>SY-PDZX943HWN</td>\n",
       "      <td>1</td>\n",
       "      <td>46</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14428</th>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>Stage 1</td>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>1</td>\n",
       "      <td>41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14429</th>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>M36</td>\n",
       "      <td>36.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>Stage 2</td>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>SY-PDZY968RFA</td>\n",
       "      <td>1</td>\n",
       "      <td>41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14430</th>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>Stage 2</td>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>2</td>\n",
       "      <td>70</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14431</th>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>M36</td>\n",
       "      <td>36.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>Stage 2</td>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>SY-PDZZ260HUM</td>\n",
       "      <td>2</td>\n",
       "      <td>70</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14432 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      participant_id visit_name  visit_month  \\\n",
       "0            BF-1002         M0          0.0   \n",
       "1            BF-1002       M0_5          0.5   \n",
       "2            BF-1003         M0          0.0   \n",
       "3            BF-1003       M0_5          0.5   \n",
       "4            BF-1004         M0          0.0   \n",
       "...              ...        ...          ...   \n",
       "14427  SY-PDZX943HWN        M36         36.0   \n",
       "14428  SY-PDZY968RFA         M0          0.0   \n",
       "14429  SY-PDZY968RFA        M36         36.0   \n",
       "14430  SY-PDZZ260HUM         M0          0.0   \n",
       "14431  SY-PDZZ260HUM        M36         36.0   \n",
       "\n",
       "       mds_updrs_part_iii_summary_score upd2hy_hoehn_and_yahr_stage  \\\n",
       "0                                  15.0                     Stage 1   \n",
       "1                                  16.0                     Stage 1   \n",
       "2                                  12.0                     Stage 1   \n",
       "3                                  24.0                     Stage 2   \n",
       "4                                  33.0                     Stage 2   \n",
       "...                                 ...                         ...   \n",
       "14427                              10.0                     Stage 2   \n",
       "14428                              11.0                     Stage 1   \n",
       "14429                              22.0                     Stage 2   \n",
       "14430                              13.0                     Stage 2   \n",
       "14431                              14.0                     Stage 2   \n",
       "\n",
       "                 FID            IID  sex  age_at_baseline  age_at_diagnosis  \\\n",
       "0            BF-1002        BF-1002    2               66              61.0   \n",
       "1            BF-1002        BF-1002    2               66              61.0   \n",
       "2            BF-1003        BF-1003    1               61              56.0   \n",
       "3            BF-1003        BF-1003    1               61              56.0   \n",
       "4            BF-1004        BF-1004    1               62              55.0   \n",
       "...              ...            ...  ...              ...               ...   \n",
       "14427  SY-PDZX943HWN  SY-PDZX943HWN    1               46               NaN   \n",
       "14428  SY-PDZY968RFA  SY-PDZY968RFA    1               41               NaN   \n",
       "14429  SY-PDZY968RFA  SY-PDZY968RFA    1               41               NaN   \n",
       "14430  SY-PDZZ260HUM  SY-PDZZ260HUM    2               70               NaN   \n",
       "14431  SY-PDZZ260HUM  SY-PDZZ260HUM    2               70               NaN   \n",
       "\n",
       "       YEARS_DIAGNOSIS_TO_BL  YEARS_FROM_DIAGNOSIS  HOEHN_YAHR_STAGE  HY3  \\\n",
       "0                        5.0              5.000000               1.0  0.0   \n",
       "1                        5.0              5.041667               1.0  0.0   \n",
       "2                        5.0              5.000000               1.0  0.0   \n",
       "3                        5.0              5.041667               2.0  0.0   \n",
       "4                        7.0              7.000000               2.0  0.0   \n",
       "...                      ...                   ...               ...  ...   \n",
       "14427                    NaN                   NaN               2.0  0.0   \n",
       "14428                    NaN                   NaN               1.0  0.0   \n",
       "14429                    NaN                   NaN               2.0  0.0   \n",
       "14430                    NaN                   NaN               2.0  0.0   \n",
       "14431                    NaN                   NaN               2.0  0.0   \n",
       "\n",
       "       HY3_EV  \n",
       "0         0.0  \n",
       "1         0.0  \n",
       "2         0.0  \n",
       "3         0.0  \n",
       "4         0.0  \n",
       "...       ...  \n",
       "14427     0.0  \n",
       "14428     0.0  \n",
       "14429     0.0  \n",
       "14430     0.0  \n",
       "14431     0.0  \n",
       "\n",
       "[14432 rows x 15 columns]"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "HY_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {
    "code_folding": [],
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Now want to calculate the time to event for H&Y3+\n",
    "\n",
    "#First filter for individuals who do not ever meet H&Y3+, and take the last visit age where H&Y3+ was not met\n",
    "HY3_no = (HY_merged[(HY_merged.HY3_EV == 0) & (HY_merged.HY3 == 0)].drop_duplicates(subset=[\"participant_id\"],keep=\"last\")\n",
    "                               .rename({\"YEARS_FROM_DIAGNOSIS\":\"HY3_TIME_FROM_DIAGNOSIS\"}, axis=1))\n",
    "\n",
    "#Secondly filter for individuals who have meet H&Y3+, and take the FIRST visit age where H&Y3+ was met\n",
    "#Here need to filter for H&Y3+ EVER as well as H&Y3 at each visit\n",
    "HY3_yes = (HY_merged[(HY_merged.HY3_EV == 1) & (HY_merged.HY3 == 1)].drop_duplicates(subset=[\"participant_id\"],keep=\"first\")\n",
    "                               .rename({\"YEARS_FROM_DIAGNOSIS\":\"HY3_TIME_FROM_DIAGNOSIS\"}, axis=1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 duplicated\n"
     ]
    }
   ],
   "source": [
    "#Join dataframes - concatenate (add rows)\n",
    "HY3 = pd.concat([HY3_no, HY3_yes]).reset_index(drop = True)\n",
    "\n",
    "#Check for duplicates\n",
    "HY3_dup = HY3[HY3.duplicated(['participant_id'], keep = False)]\n",
    "n_dup = len(HY3_dup)\n",
    "print(str(n_dup) + ' duplicated')\n",
    "\n",
    "#If a patient has H&Y3+ at baseline (visit_month == 0), they are left censored - code as 2 \n",
    "HY3.loc[:,\"HY3_EV\"] = HY3.HY3_EV.mask((HY3.visit_month == 0) & (HY3.HY3 == 1), 2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Now select relevant columns - only one row per participant\n",
    "HY3_export = HY3.copy()\n",
    "HY3_export = HY3_export.loc[:, ['FID', 'IID', 'age_at_diagnosis', 'age_at_baseline', 'sex', 'HY3_EV', 'HY3_TIME_FROM_DIAGNOSIS']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Write as output\n",
    "#Note this is all the participants with clinical data, not just our list of PD only participants\n",
    "with open('/home/jupyter/notebooks/clinical_data_v2-5/HY3_allparticipants.tab', 'w') as f:\n",
    "    f.write(HY3_export.to_csv(index=False, sep='\\t'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>age_at_diagnosis</th>\n",
       "      <th>age_at_baseline</th>\n",
       "      <th>sex</th>\n",
       "      <th>HY3_EV</th>\n",
       "      <th>HY3_TIME_FROM_DIAGNOSIS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BF-1002</td>\n",
       "      <td>BF-1002</td>\n",
       "      <td>61.0</td>\n",
       "      <td>66</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.041667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BF-1003</td>\n",
       "      <td>BF-1003</td>\n",
       "      <td>56.0</td>\n",
       "      <td>61</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.041667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BF-1004</td>\n",
       "      <td>BF-1004</td>\n",
       "      <td>55.0</td>\n",
       "      <td>62</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.041667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BF-1006</td>\n",
       "      <td>BF-1006</td>\n",
       "      <td>52.0</td>\n",
       "      <td>59</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7.041667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BF-1008</td>\n",
       "      <td>BF-1008</td>\n",
       "      <td>65.0</td>\n",
       "      <td>69</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.041667</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       FID      IID  age_at_diagnosis  age_at_baseline  sex  HY3_EV  \\\n",
       "0  BF-1002  BF-1002              61.0               66    2     0.0   \n",
       "1  BF-1003  BF-1003              56.0               61    1     0.0   \n",
       "2  BF-1004  BF-1004              55.0               62    1     0.0   \n",
       "3  BF-1006  BF-1006              52.0               59    1     0.0   \n",
       "4  BF-1008  BF-1008              65.0               69    2     0.0   \n",
       "\n",
       "   HY3_TIME_FROM_DIAGNOSIS  \n",
       "0                 5.041667  \n",
       "1                 5.041667  \n",
       "2                 7.041667  \n",
       "3                 7.041667  \n",
       "4                 4.041667  "
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "HY3_export.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Now run survival analysis in R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "pack <- \"/home/jupyter/notebooks/R_packages\"\n",
    "library(dplyr, lib.loc = pack)\n",
    "library(ggplot2, lib.loc = pack)\n",
    "library(ggpubr, lib.loc = pack)\n",
    "library(survival,lib.loc = pack)\n",
    "library(survminer, lib.loc = pack)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "#Make list of pathways\n",
    "pathways <- c(\"adaptive_immune\", \"alpha_synuclein\", \"innate_immune\", \"lysosome\", \"endocytosis\",\n",
    "              \"microglia\", \"monocytes\", \"mitochondria\")\n",
    "\n",
    "#Make list of cohorts - excluding LBD as these samples have been excluded from the genetic dataset\n",
    "cohorts <- c(\"PPMI\", \"PDBP\", \"HBS\", \"LCC\", \"Steady\", \"Sure\", \"BioFIND\")\n",
    "\n",
    "#Make function to analyse longitudinal binomial outcomes using survival Cox proportional hazards models\n",
    "#df is the name of the .tab file in /home/jupyter/clinical_data_v2-5/ which has the data in long format\n",
    "#df = \"HY3_allparticipants\"\n",
    "#outcome is the variable name of the outcome within df\n",
    "#outcome = \"HY3\"\n",
    "\n",
    "analyse_surv <- function(pathway, cohort, df, outcome) {\n",
    "    \n",
    "        #Read in data for PRS, clinical data, and genetic PCs\n",
    "        PRS <- read.table(paste(\"/home/jupyter/pathways/\", cohort, \".\", pathway, \"_PRS_final.txt\", sep = \"\"), \n",
    "                          sep = \"\\t\", header = T)\n",
    "    \n",
    "        data <- read.table(paste(\"/home/jupyter/clinical_data_v2-5/\", df, \".tab\", sep = \"\"), \n",
    "                             sep = \"\\t\", header = T)\n",
    "    \n",
    "        #Read in genetic PCs\n",
    "        pcs <- read.table(paste(\"/home/jupyter/genetic_data_v2-5/\", cohort, \".pca.eigenvec\", sep = \"\"))\n",
    "        colnames(pcs) <- c(\"FID\", \"IID\", paste(\"PC\", 1:10, sep = \"\"))\n",
    "\n",
    "        #Merge\n",
    "        merged <- PRS %>%\n",
    "        inner_join(data, by = c(\"participant_id\" = \"IID\")) %>%\n",
    "        inner_join(pcs, by = c(\"FID\"))\n",
    "        \n",
    "        #Select the outcome variable from the function argument\n",
    "        merged$event <- merged[[paste(outcome, \"_EV\", sep =\"\")]]\n",
    "    \n",
    "        #Select the time to event variable from the function argument\n",
    "        merged$time_to_event <- merged[[paste(outcome, \"_TIME_FROM_DIAGNOSIS\", sep =\"\")]]\n",
    "\n",
    "        #Filter out individuals who met outcome at baseline (left censored, coded as 2)\n",
    "        #Otherwise the Surv function to make the survival object misinterprets the codes\n",
    "        merged <- merged %>%\n",
    "            filter(event <= 1)\n",
    "        \n",
    "        #Select just relevant columns\n",
    "        #This cannot be done with dplyr::select\n",
    "        merged_selected <- merged[, c('event', 'time_to_event', 'participant_id', 'study', 'PRS_z', 'sex', \n",
    "                                      'age_at_diagnosis',\n",
    "                                      'PC1', 'PC2', 'PC3', 'PC4', 'PC5')]\n",
    "    \n",
    "        #Remove individuals missing any rows\n",
    "        merged_selected_nomiss <- merged_selected %>%\n",
    "            drop_na()\n",
    "    \n",
    "        #Number of individuals in dataset\n",
    "        N_inds <- dim(merged_selected_nomiss)[1]\n",
    "    \n",
    "        #If < 20 individuals have met the outcome of interest\n",
    "        if (sum(merged_selected_nomiss$event == 1) > 20) {\n",
    "            \n",
    "            #Run Cox proportional hazards analysis for HY3 adjusting for age at diagnosis and gender\n",
    "            surv_object_HY3 <- Surv(time = merged_selected_nomiss$time_to_event, event = merged_selected_nomiss$event)\n",
    "    \n",
    "            #Fit Cox Proportional Hazards model with covariates age at diagnosis and gender and PC1-5\n",
    "            model.cox_HY3 = coxph(surv_object_HY3 ~ PRS_z + age_at_diagnosis + sex \n",
    "                              + PC1 + PC2 + PC3 + PC4 + PC5, \n",
    "                              data = merged_selected_nomiss)\n",
    "    \n",
    "            kmz <- cox.zph(model.cox_HY3, transform = \"km\")\n",
    "    \n",
    "            #Save model coefficients to results table\n",
    "            summary <- c(pathway,\n",
    "                         cohort,\n",
    "                         summary(model.cox_HY3)$coefficients[1,1],\n",
    "                         summary(model.cox_HY3)$coefficients[1,3],\n",
    "                         summary(model.cox_HY3)$coefficients[1,5],\n",
    "                         kmz$table[1,3],\n",
    "                         summary(model.cox_HY3)$logtest[[1]],\n",
    "                         summary(model.cox_HY3)$sctest[[1]],\n",
    "                         summary(model.cox_HY3)$rsq[[1]],\n",
    "                         N_inds) # nagelkerke r square\n",
    "\n",
    "    \n",
    "        } else {\n",
    "            \n",
    "            #If not enough individuals met the outcome or the dataframe is empty, print NA for summary stat\n",
    "            summary <- c(pathway,\n",
    "                        cohort,\n",
    "                        rep(NA, 7),\n",
    "                        N_inds)\n",
    "        }\n",
    "    \n",
    "    return(summary)\n",
    "\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "HY3_results_list <- list()\n",
    "\n",
    "#Loop over each pathway and cohort\n",
    "for (pathway in pathways) {\n",
    "    for (cohort in cohorts) {\n",
    "        i <- length(HY3_results_list)\n",
    "        HY3_results_list[[i+1]] <- analyse_surv(pathway, cohort, \"HY3_allparticipants\", \"HY3\")\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           pathway   study               Coeff                 se\n",
      "1  adaptive_immune    PPMI   0.100677662295459 0.0946301767819111\n",
      "2  adaptive_immune    PDBP   0.143502378099667  0.125561194430677\n",
      "3  adaptive_immune     HBS                <NA>               <NA>\n",
      "4  adaptive_immune     LCC                <NA>               <NA>\n",
      "5  adaptive_immune  Steady                <NA>               <NA>\n",
      "6  adaptive_immune    Sure                <NA>               <NA>\n",
      "7  adaptive_immune BioFIND                <NA>               <NA>\n",
      "8  alpha_synuclein    PPMI -0.0536558363959514   0.10663081240186\n",
      "9  alpha_synuclein    PDBP  -0.164685029248054   0.13746711273961\n",
      "10 alpha_synuclein     HBS                <NA>               <NA>\n",
      "11 alpha_synuclein     LCC                <NA>               <NA>\n",
      "12 alpha_synuclein  Steady                <NA>               <NA>\n",
      "13 alpha_synuclein    Sure                <NA>               <NA>\n",
      "14 alpha_synuclein BioFIND                <NA>               <NA>\n",
      "15   innate_immune    PPMI  0.0546271271134055  0.101462490131038\n",
      "16   innate_immune    PDBP -0.0429401912271681  0.141245303178374\n",
      "17   innate_immune     HBS                <NA>               <NA>\n",
      "18   innate_immune     LCC                <NA>               <NA>\n",
      "19   innate_immune  Steady                <NA>               <NA>\n",
      "20   innate_immune    Sure                <NA>               <NA>\n",
      "21   innate_immune BioFIND                <NA>               <NA>\n",
      "22        lysosome    PPMI  0.0325612552737421 0.0985870753778845\n",
      "23        lysosome    PDBP   0.118771975319927  0.129135766334707\n",
      "24        lysosome     HBS                <NA>               <NA>\n",
      "25        lysosome     LCC                <NA>               <NA>\n",
      "26        lysosome  Steady                <NA>               <NA>\n",
      "27        lysosome    Sure                <NA>               <NA>\n",
      "28        lysosome BioFIND                <NA>               <NA>\n",
      "29     endocytosis    PPMI  0.0279274317754326  0.102791192854332\n",
      "30     endocytosis    PDBP -0.0142281477175121  0.131236239935421\n",
      "31     endocytosis     HBS                <NA>               <NA>\n",
      "32     endocytosis     LCC                <NA>               <NA>\n",
      "33     endocytosis  Steady                <NA>               <NA>\n",
      "34     endocytosis    Sure                <NA>               <NA>\n",
      "35     endocytosis BioFIND                <NA>               <NA>\n",
      "36       microglia    PPMI -0.0777521717752385  0.100795261157736\n",
      "37       microglia    PDBP   0.114982659598318  0.132159662922686\n",
      "38       microglia     HBS                <NA>               <NA>\n",
      "39       microglia     LCC                <NA>               <NA>\n",
      "40       microglia  Steady                <NA>               <NA>\n",
      "41       microglia    Sure                <NA>               <NA>\n",
      "42       microglia BioFIND                <NA>               <NA>\n",
      "43       monocytes    PPMI  0.0498160126567836 0.0963798815992689\n",
      "44       monocytes    PDBP  0.0758894338390537  0.140539316504974\n",
      "45       monocytes     HBS                <NA>               <NA>\n",
      "46       monocytes     LCC                <NA>               <NA>\n",
      "47       monocytes  Steady                <NA>               <NA>\n",
      "48       monocytes    Sure                <NA>               <NA>\n",
      "49       monocytes BioFIND                <NA>               <NA>\n",
      "50    mitochondria    PPMI -0.0248361014608991 0.0983849109077766\n",
      "51    mitochondria    PDBP -0.0643678062095002  0.143969138846749\n",
      "52    mitochondria     HBS                <NA>               <NA>\n",
      "53    mitochondria     LCC                <NA>               <NA>\n",
      "54    mitochondria  Steady                <NA>               <NA>\n",
      "55    mitochondria    Sure                <NA>               <NA>\n",
      "56    mitochondria BioFIND                <NA>               <NA>\n",
      "              Pvalue       Cox.zphPVal     ov.lik.ratio          logrank\n",
      "1  0.287371048003942 0.331022393705869  47.554627439669 46.5803677162447\n",
      "2  0.253085109620323 0.537505341730598 67.9334298943519 64.8294358161696\n",
      "3               <NA>              <NA>             <NA>             <NA>\n",
      "4               <NA>              <NA>             <NA>             <NA>\n",
      "5               <NA>              <NA>             <NA>             <NA>\n",
      "6               <NA>              <NA>             <NA>             <NA>\n",
      "7               <NA>              <NA>             <NA>             <NA>\n",
      "8   0.61482886266132 0.788611240468314 46.6690176853981 45.6441882129136\n",
      "9  0.230918636355404 0.236717459621458 68.0444356078277  65.408132105695\n",
      "10              <NA>              <NA>             <NA>             <NA>\n",
      "11              <NA>              <NA>             <NA>             <NA>\n",
      "12              <NA>              <NA>             <NA>             <NA>\n",
      "13              <NA>              <NA>             <NA>             <NA>\n",
      "14              <NA>              <NA>             <NA>             <NA>\n",
      "15 0.590302814660612 0.443604675857291 46.7060432897017 45.5896138111738\n",
      "16 0.761119158214199 0.898428908754455 66.7153975913658 63.5216306396247\n",
      "17              <NA>              <NA>             <NA>             <NA>\n",
      "18              <NA>              <NA>             <NA>             <NA>\n",
      "19              <NA>              <NA>             <NA>             <NA>\n",
      "20              <NA>              <NA>             <NA>             <NA>\n",
      "21              <NA>              <NA>             <NA>             <NA>\n",
      "22 0.741189047121464  0.56688601298667 46.5252702887794  45.579440243862\n",
      "23 0.357706031227613 0.612600895443608 67.4631809530899 63.9695623545978\n",
      "24              <NA>              <NA>             <NA>             <NA>\n",
      "25              <NA>              <NA>             <NA>             <NA>\n",
      "26              <NA>              <NA>             <NA>             <NA>\n",
      "27              <NA>              <NA>             <NA>             <NA>\n",
      "28              <NA>              <NA>             <NA>             <NA>\n",
      "29 0.785859698363633 0.314846010534425 46.4900039181923 45.4734352114156\n",
      "30 0.913665470862658 0.185282105218108  66.634760893347 63.3800767403972\n",
      "31              <NA>              <NA>             <NA>             <NA>\n",
      "32              <NA>              <NA>             <NA>             <NA>\n",
      "33              <NA>              <NA>             <NA>             <NA>\n",
      "34              <NA>              <NA>             <NA>             <NA>\n",
      "35              <NA>              <NA>             <NA>             <NA>\n",
      "36 0.440477474489158 0.173721136850993 47.0116224723217 45.7760807349904\n",
      "37 0.384284886095466  0.62949262604293 67.3932962171001 64.6145084113447\n",
      "38              <NA>              <NA>             <NA>             <NA>\n",
      "39              <NA>              <NA>             <NA>             <NA>\n",
      "40              <NA>              <NA>             <NA>             <NA>\n",
      "41              <NA>              <NA>             <NA>             <NA>\n",
      "42              <NA>              <NA>             <NA>             <NA>\n",
      "43 0.605245873328895 0.688474431912506 46.6842474399627 45.6462565871976\n",
      "44 0.589205850824613  0.24501332842118 66.9157658506509 63.8671578104904\n",
      "45              <NA>              <NA>             <NA>             <NA>\n",
      "46              <NA>              <NA>             <NA>             <NA>\n",
      "47              <NA>              <NA>             <NA>             <NA>\n",
      "48              <NA>              <NA>             <NA>             <NA>\n",
      "49              <NA>              <NA>             <NA>             <NA>\n",
      "50 0.800702442287146 0.371310349894007 46.4799455592738 45.5478295872991\n",
      "51 0.654806849747672 0.318194647951465 66.8232833425618   64.27881964165\n",
      "52              <NA>              <NA>             <NA>             <NA>\n",
      "53              <NA>              <NA>             <NA>             <NA>\n",
      "54              <NA>              <NA>             <NA>             <NA>\n",
      "55              <NA>              <NA>             <NA>             <NA>\n",
      "56              <NA>              <NA>             <NA>             <NA>\n",
      "                   r2   N\n",
      "1  0.0978117646270072 462\n",
      "2  0.0965592399273905 669\n",
      "3                <NA> 538\n",
      "4                <NA>   0\n",
      "5                <NA>   0\n",
      "6                <NA> 247\n",
      "7                <NA>  81\n",
      "8   0.096080697567529 462\n",
      "9  0.0967091334496719 669\n",
      "10               <NA> 538\n",
      "11               <NA>   0\n",
      "12               <NA>   0\n",
      "13               <NA> 247\n",
      "14               <NA>  81\n",
      "15 0.0961531365661167 462\n",
      "16   0.09491286863882 669\n",
      "17               <NA> 538\n",
      "18               <NA>   0\n",
      "19               <NA>   0\n",
      "20               <NA> 247\n",
      "21               <NA>  81\n",
      "22 0.0957994069553511 462\n",
      "23 0.0959239762361236 669\n",
      "24               <NA> 538\n",
      "25               <NA>   0\n",
      "26               <NA>   0\n",
      "27               <NA> 247\n",
      "28               <NA>  81\n",
      "29 0.0957303829503793 462\n",
      "30 0.0948037690329607 669\n",
      "31               <NA> 538\n",
      "32               <NA>   0\n",
      "33               <NA>   0\n",
      "34               <NA> 247\n",
      "35               <NA>  81\n",
      "36 0.0967507674403564 462\n",
      "37 0.0958295301758141 669\n",
      "38               <NA> 538\n",
      "39               <NA>   0\n",
      "40               <NA>   0\n",
      "41               <NA> 247\n",
      "42               <NA>  81\n",
      "43 0.0961104946286114 462\n",
      "44 0.0951839053773508 669\n",
      "45               <NA> 538\n",
      "46               <NA>   0\n",
      "47               <NA>   0\n",
      "48               <NA> 247\n",
      "49               <NA>  81\n",
      "50 0.0957106955751014 462\n",
      "51 0.0950588150245979 669\n",
      "52               <NA> 538\n",
      "53               <NA>   0\n",
      "54               <NA>   0\n",
      "55               <NA> 247\n",
      "56               <NA>  81\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "#Combine lists into a single dataframe\n",
    "HY3_results <- do.call(rbind, HY3_results_list)\n",
    "\n",
    "#Set column names\n",
    "colnames(HY3_results) <- c(\"pathway\", \"study\", \"Coeff\", \"se\", \"Pvalue\", \"Cox.zphPVal\", \"ov.lik.ratio\",\"logrank\", \"r2\", \"N\")\n",
    "\n",
    "#Make into dataframe\n",
    "HY3_results <- as.data.frame(HY3_results)\n",
    "\n",
    "#Save output\n",
    "write.table(HY3_results, \"/home/jupyter/pathways/pathways_HY3_results.txt\",\n",
    "           sep = \"\\t\", row.names = F, col.names = T, quote = F)\n",
    "\n",
    "HY3_results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Analyse pathways vs MoCA"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Query MoCA data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "moca_query = f\"\"\"\n",
    "\n",
    "SELECT \n",
    "*\n",
    "FROM `{BQ_RELEASE_DATASET}.MOCA`\n",
    "\"\"\"\n",
    "moca = bq_query(moca_query)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>GUID</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>moca01_alternating_trail_making</th>\n",
       "      <th>moca02_visuoconstr_skills_cube</th>\n",
       "      <th>moca03_visuoconstr_skills_clock_cont</th>\n",
       "      <th>moca04_visuoconstr_skills_clock_num</th>\n",
       "      <th>moca05_visuoconstr_skills_clock_hands</th>\n",
       "      <th>moca_visuospatial_executive_subscore</th>\n",
       "      <th>...</th>\n",
       "      <th>moca22_orientation_date_score</th>\n",
       "      <th>moca23_orientation_month_score</th>\n",
       "      <th>moca24_orientation_year_score</th>\n",
       "      <th>moca25_orientation_day_score</th>\n",
       "      <th>moca26_orientation_place_score</th>\n",
       "      <th>moca27_orientation_city_score</th>\n",
       "      <th>moca_orientation_subscore</th>\n",
       "      <th>code_education_12years_complete</th>\n",
       "      <th>education_12years_complete</th>\n",
       "      <th>moca_total_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LC-100001</td>\n",
       "      <td>None</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>...</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>None</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LC-100001</td>\n",
       "      <td>None</td>\n",
       "      <td>M12</td>\n",
       "      <td>12.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>...</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>None</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LC-100001</td>\n",
       "      <td>None</td>\n",
       "      <td>M24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>...</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>None</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LC-100001</td>\n",
       "      <td>None</td>\n",
       "      <td>M36</td>\n",
       "      <td>36.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>...</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>None</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LC-10080001</td>\n",
       "      <td>None</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>...</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>None</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 43 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  participant_id  GUID visit_name  visit_month  \\\n",
       "0      LC-100001  None         M0          0.0   \n",
       "1      LC-100001  None        M12         12.0   \n",
       "2      LC-100001  None        M24         24.0   \n",
       "3      LC-100001  None        M36         36.0   \n",
       "4    LC-10080001  None         M0          0.0   \n",
       "\n",
       "   moca01_alternating_trail_making  moca02_visuoconstr_skills_cube  \\\n",
       "0                             <NA>                            <NA>   \n",
       "1                             <NA>                            <NA>   \n",
       "2                             <NA>                            <NA>   \n",
       "3                             <NA>                            <NA>   \n",
       "4                             <NA>                            <NA>   \n",
       "\n",
       "   moca03_visuoconstr_skills_clock_cont  moca04_visuoconstr_skills_clock_num  \\\n",
       "0                                  <NA>                                 <NA>   \n",
       "1                                  <NA>                                 <NA>   \n",
       "2                                  <NA>                                 <NA>   \n",
       "3                                  <NA>                                 <NA>   \n",
       "4                                  <NA>                                 <NA>   \n",
       "\n",
       "   moca05_visuoconstr_skills_clock_hands  \\\n",
       "0                                   <NA>   \n",
       "1                                   <NA>   \n",
       "2                                   <NA>   \n",
       "3                                   <NA>   \n",
       "4                                   <NA>   \n",
       "\n",
       "   moca_visuospatial_executive_subscore  ...  moca22_orientation_date_score  \\\n",
       "0                                  <NA>  ...                           <NA>   \n",
       "1                                  <NA>  ...                           <NA>   \n",
       "2                                  <NA>  ...                           <NA>   \n",
       "3                                  <NA>  ...                           <NA>   \n",
       "4                                  <NA>  ...                           <NA>   \n",
       "\n",
       "   moca23_orientation_month_score  moca24_orientation_year_score  \\\n",
       "0                            <NA>                           <NA>   \n",
       "1                            <NA>                           <NA>   \n",
       "2                            <NA>                           <NA>   \n",
       "3                            <NA>                           <NA>   \n",
       "4                            <NA>                           <NA>   \n",
       "\n",
       "   moca25_orientation_day_score  moca26_orientation_place_score  \\\n",
       "0                          <NA>                            <NA>   \n",
       "1                          <NA>                            <NA>   \n",
       "2                          <NA>                            <NA>   \n",
       "3                          <NA>                            <NA>   \n",
       "4                          <NA>                            <NA>   \n",
       "\n",
       "   moca27_orientation_city_score  moca_orientation_subscore  \\\n",
       "0                           <NA>                       <NA>   \n",
       "1                           <NA>                       <NA>   \n",
       "2                           <NA>                       <NA>   \n",
       "3                           <NA>                       <NA>   \n",
       "4                           <NA>                       <NA>   \n",
       "\n",
       "   code_education_12years_complete  education_12years_complete  \\\n",
       "0                             <NA>                        None   \n",
       "1                             <NA>                        None   \n",
       "2                             <NA>                        None   \n",
       "3                             <NA>                        None   \n",
       "4                             <NA>                        None   \n",
       "\n",
       "   moca_total_score  \n",
       "0                24  \n",
       "1                23  \n",
       "2                18  \n",
       "3                22  \n",
       "4                28  \n",
       "\n",
       "[5 rows x 43 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "moca.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Note that there is no data on individual items, so will just use the total score - assume this has already been adjusted for education"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "moca = moca.sort_values(by = ['participant_id', 'visit_month'])\n",
    "moca = moca.loc[:, ['participant_id', 'visit_name', 'visit_month', 'moca_total_score']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>moca_total_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4706</th>\n",
       "      <td>BF-1001</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8735</th>\n",
       "      <td>BF-1002</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10591</th>\n",
       "      <td>BF-1003</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7216</th>\n",
       "      <td>BF-1004</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4707</th>\n",
       "      <td>BF-1005</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      participant_id visit_name  visit_month  moca_total_score\n",
       "4706         BF-1001         M0          0.0                28\n",
       "8735         BF-1002         M0          0.0                29\n",
       "10591        BF-1003         M0          0.0                30\n",
       "7216         BF-1004         M0          0.0                28\n",
       "4707         BF-1005         M0          0.0                27"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "moca.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Remove duplicated visit entries - rows with #\n",
    "moca_nodupli = moca.copy()\n",
    "moca_nodupli = moca_nodupli[~moca_nodupli.visit_name.str.contains('#')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>moca_total_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10349</th>\n",
       "      <td>SU-32242</td>\n",
       "      <td>M24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13069</th>\n",
       "      <td>SU-32242</td>\n",
       "      <td>M24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3829</th>\n",
       "      <td>SU-33107</td>\n",
       "      <td>M24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5496</th>\n",
       "      <td>SU-33107</td>\n",
       "      <td>M24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5497</th>\n",
       "      <td>SU-33113</td>\n",
       "      <td>M24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13288</th>\n",
       "      <td>SU-33113</td>\n",
       "      <td>M24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13290</th>\n",
       "      <td>SU-33114</td>\n",
       "      <td>M24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13291</th>\n",
       "      <td>SU-33114</td>\n",
       "      <td>M24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10490</th>\n",
       "      <td>SY-NIH_INVLD826LLAN9</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10491</th>\n",
       "      <td>SY-NIH_INVLD826LLAN9</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1982</th>\n",
       "      <td>SY-NIH_INVRB171EXGUK</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4677</th>\n",
       "      <td>SY-NIH_INVRB171EXGUK</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13389</th>\n",
       "      <td>SY-PDAC092WHX</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13390</th>\n",
       "      <td>SY-PDAC092WHX</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6325</th>\n",
       "      <td>SY-PDDJ151DRG</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13426</th>\n",
       "      <td>SY-PDDJ151DRG</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3846</th>\n",
       "      <td>SY-PDNC843PUF</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3847</th>\n",
       "      <td>SY-PDNC843PUF</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1592</th>\n",
       "      <td>SY-PDNE693YUB</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3849</th>\n",
       "      <td>SY-PDNE693YUB</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8696</th>\n",
       "      <td>SY-PDNK239KGR</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10542</th>\n",
       "      <td>SY-PDNK239KGR</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5930</th>\n",
       "      <td>SY-PDPC279JY2</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8700</th>\n",
       "      <td>SY-PDPC279JY2</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2614</th>\n",
       "      <td>SY-PDTT287RCE</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7207</th>\n",
       "      <td>SY-PDTT287RCE</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             participant_id visit_name  visit_month  moca_total_score\n",
       "10349              SU-32242        M24         24.0                28\n",
       "13069              SU-32242        M24         24.0                29\n",
       "3829               SU-33107        M24         24.0                25\n",
       "5496               SU-33107        M24         24.0                27\n",
       "5497               SU-33113        M24         24.0                29\n",
       "13288              SU-33113        M24         24.0                30\n",
       "13290              SU-33114        M24         24.0                26\n",
       "13291              SU-33114        M24         24.0                30\n",
       "10490  SY-NIH_INVLD826LLAN9         M0          0.0                27\n",
       "10491  SY-NIH_INVLD826LLAN9         M0          0.0                29\n",
       "1982   SY-NIH_INVRB171EXGUK         M0          0.0                25\n",
       "4677   SY-NIH_INVRB171EXGUK         M0          0.0                27\n",
       "13389         SY-PDAC092WHX         M0          0.0                27\n",
       "13390         SY-PDAC092WHX         M0          0.0                28\n",
       "6325          SY-PDDJ151DRG         M0          0.0                24\n",
       "13426         SY-PDDJ151DRG         M0          0.0                28\n",
       "3846          SY-PDNC843PUF         M0          0.0                25\n",
       "3847          SY-PDNC843PUF         M0          0.0                27\n",
       "1592          SY-PDNE693YUB         M0          0.0                26\n",
       "3849          SY-PDNE693YUB         M0          0.0                25\n",
       "8696          SY-PDNK239KGR         M0          0.0                26\n",
       "10542         SY-PDNK239KGR         M0          0.0                25\n",
       "5930          SY-PDPC279JY2         M0          0.0                25\n",
       "8700          SY-PDPC279JY2         M0          0.0                28\n",
       "2614          SY-PDTT287RCE         M0          0.0                23\n",
       "7207          SY-PDTT287RCE         M0          0.0                26"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Check for any other duplicated entries by participant id and visit month\n",
    "moca_nodupli[moca_nodupli.duplicated(['participant_id', 'visit_month'], keep = False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#There are still a few individuals with duplicated entries by participant_id and visit_month with different MoCA scores\n",
    "#Just keep the first datapoint\n",
    "moca_nodupli_final = moca_nodupli.copy()\n",
    "moca_nodupli_final = moca_nodupli_final.drop_duplicates(subset = ['participant_id', 'visit_month'], keep = 'first')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>moca_total_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4706</th>\n",
       "      <td>BF-1001</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8735</th>\n",
       "      <td>BF-1002</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10591</th>\n",
       "      <td>BF-1003</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7216</th>\n",
       "      <td>BF-1004</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4707</th>\n",
       "      <td>BF-1005</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      participant_id visit_name  visit_month  moca_total_score\n",
       "4706         BF-1001         M0          0.0                28\n",
       "8735         BF-1002         M0          0.0                29\n",
       "10591        BF-1003         M0          0.0                30\n",
       "7216         BF-1004         M0          0.0                28\n",
       "4707         BF-1005         M0          0.0                27"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "moca_nodupli_final.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Write as output\n",
    "with open('/home/jupyter/clinical_data_v2-5/moca_allparticipants.tab', 'w') as f:\n",
    "    f.write(moca_nodupli_final.to_csv(index=False, sep='\\t'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: boundary (singular) fit: see ?isSingular\n",
      "\n",
      "R[write to console]: boundary (singular) fit: see ?isSingular\n",
      "\n",
      "R[write to console]: boundary (singular) fit: see ?isSingular\n",
      "\n",
      "R[write to console]: boundary (singular) fit: see ?isSingular\n",
      "\n",
      "R[write to console]: boundary (singular) fit: see ?isSingular\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "moca_results_list <- list()\n",
    "\n",
    "#Loop over each pathway and cohort\n",
    "for (pathway in pathways) {\n",
    "    for (cohort in cohorts) {\n",
    "        i <- length(moca_results_list)\n",
    "        moca_results_list[[i+1]] <- analyse_cont(pathway, cohort, \"moca_allparticipants\", \"moca_total_score\")\n",
    "\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           pathway  cohort                Coeff                 se\n",
      "1  adaptive_immune    PPMI -0.00331208833891141 0.0283583864000862\n",
      "2  adaptive_immune    PDBP  -0.0374590940531551 0.0235467372857831\n",
      "3  adaptive_immune     HBS                 <NA>               <NA>\n",
      "4  adaptive_immune     LCC    0.027228585154695  0.101372211103009\n",
      "5  adaptive_immune  Steady                 <NA>               <NA>\n",
      "6  adaptive_immune    Sure  -0.0573211717939767 0.0490121898218669\n",
      "7  adaptive_immune BioFIND                 <NA>               <NA>\n",
      "8  alpha_synuclein    PPMI -0.00130287689163538 0.0296109535366347\n",
      "9  alpha_synuclein    PDBP  -0.0247545864977931 0.0238875290784027\n",
      "10 alpha_synuclein     HBS                 <NA>               <NA>\n",
      "11 alpha_synuclein     LCC  -0.0490108368325559  0.096440187026592\n",
      "12 alpha_synuclein  Steady                 <NA>               <NA>\n",
      "13 alpha_synuclein    Sure  -0.0346465700283331 0.0483450228982943\n",
      "14 alpha_synuclein BioFIND                 <NA>               <NA>\n",
      "15   innate_immune    PPMI    0.035188807408259  0.029467894274155\n",
      "16   innate_immune    PDBP  -0.0342761789601284 0.0220016527358828\n",
      "17   innate_immune     HBS                 <NA>               <NA>\n",
      "18   innate_immune     LCC   -0.033736238594289 0.0972783844381547\n",
      "19   innate_immune  Steady                 <NA>               <NA>\n",
      "20   innate_immune    Sure   0.0144234887070205 0.0485889127683609\n",
      "21   innate_immune BioFIND                 <NA>               <NA>\n",
      "22        lysosome    PPMI   0.0287978489624366 0.0285227213235616\n",
      "23        lysosome    PDBP  -0.0374315169076443 0.0234483360214081\n",
      "24        lysosome     HBS                 <NA>               <NA>\n",
      "25        lysosome     LCC  -0.0354597845541441  0.112444462505713\n",
      "26        lysosome  Steady                 <NA>               <NA>\n",
      "27        lysosome    Sure   0.0454281945537864 0.0515226425978347\n",
      "28        lysosome BioFIND                 <NA>               <NA>\n",
      "29     endocytosis    PPMI -0.00209549301890968 0.0293765096526167\n",
      "30     endocytosis    PDBP -0.00788498422434294 0.0233201897822008\n",
      "31     endocytosis     HBS                 <NA>               <NA>\n",
      "32     endocytosis     LCC   0.0585374839526415 0.0877447764029992\n",
      "33     endocytosis  Steady                 <NA>               <NA>\n",
      "34     endocytosis    Sure  -0.0755677664968793 0.0497777255905863\n",
      "35     endocytosis BioFIND                 <NA>               <NA>\n",
      "36       microglia    PPMI   0.0137677534363559 0.0296781382339513\n",
      "37       microglia    PDBP  0.00339114347604773 0.0236867137339622\n",
      "38       microglia     HBS                 <NA>               <NA>\n",
      "39       microglia     LCC  -0.0953532532066547  0.100724019745798\n",
      "40       microglia  Steady                 <NA>               <NA>\n",
      "41       microglia    Sure  -0.0532777887308421 0.0501125037662621\n",
      "42       microglia BioFIND                 <NA>               <NA>\n",
      "43       monocytes    PPMI -0.00841185121444902 0.0295396503651903\n",
      "44       monocytes    PDBP 6.55200592076685e-05  0.023273615280606\n",
      "45       monocytes     HBS                 <NA>               <NA>\n",
      "46       monocytes     LCC  -0.0382026175137155  0.100580710681125\n",
      "47       monocytes  Steady                 <NA>               <NA>\n",
      "48       monocytes    Sure  -0.0985180291117606 0.0497482251464393\n",
      "49       monocytes BioFIND                 <NA>               <NA>\n",
      "50    mitochondria    PPMI  -0.0655816655294649 0.0289433230961387\n",
      "51    mitochondria    PDBP   0.0149121323191364  0.022402797943698\n",
      "52    mitochondria     HBS                 <NA>               <NA>\n",
      "53    mitochondria     LCC    0.115270453625741 0.0879134557847755\n",
      "54    mitochondria  Steady                 <NA>               <NA>\n",
      "55    mitochondria    Sure  0.00832742681534717 0.0497508951449876\n",
      "56    mitochondria BioFIND                 <NA>               <NA>\n",
      "                 df             t_value             Pvalue N_obs N_inds\n",
      "1  262.457683601433  -0.116793963245431  0.907112648606781  2755    465\n",
      "2  657.185405623314   -1.59084010657272  0.112126453977244  1825    784\n",
      "3              <NA>                <NA>               <NA>     0      0\n",
      "4  21.6546628470658   0.268600091271826  0.790779772950749   158     63\n",
      "5              <NA>                <NA>               <NA>     0      0\n",
      "6  186.750271956895   -1.16952888663634  0.243681122636136   881    246\n",
      "7              <NA>                <NA>               <NA>    92     92\n",
      "8  268.071135868237 -0.0439998289829964  0.964937296638335  2755    465\n",
      "9  694.528097043751   -1.03629749299497  0.300424043741076  1825    784\n",
      "10             <NA>                <NA>               <NA>     0      0\n",
      "11 14.5376455505026  -0.508199313415287  0.618937448427498   158     63\n",
      "12             <NA>                <NA>               <NA>     0      0\n",
      "13 177.167566514295  -0.716652262244673  0.474532343934408   881    246\n",
      "14             <NA>                <NA>               <NA>    92     92\n",
      "15 263.191540367808    1.19414054770522  0.233498119889383  2755    465\n",
      "16 529.787638216844   -1.55789109898217  0.119856205494669  1825    784\n",
      "17             <NA>                <NA>               <NA>     0      0\n",
      "18  14.305613092739  -0.346800975254035  0.733789425984501   158     63\n",
      "19             <NA>                <NA>               <NA>     0      0\n",
      "20  181.16160515452   0.296847323499128  0.766923142084595   881    246\n",
      "21             <NA>                <NA>               <NA>    92     92\n",
      "22 260.842471481161     1.0096459112633  0.313600616321117  2755    465\n",
      "23  699.23715035491   -1.59634000781418  0.110864716996469  1825    784\n",
      "24             <NA>                <NA>               <NA>     0      0\n",
      "25 20.5782904341422  -0.315353764551478  0.755668790642776   158     63\n",
      "26             <NA>                <NA>               <NA>     0      0\n",
      "27 208.190934846145   0.881713209246289  0.378948738953092   881    246\n",
      "28             <NA>                <NA>               <NA>    92     92\n",
      "29 273.242234986041 -0.0713322666201438  0.943185497163802  2755    465\n",
      "30 709.232452080831  -0.338118355724583  0.735374058952248  1825    784\n",
      "31             <NA>                <NA>               <NA>     0      0\n",
      "32 10.2024104052049   0.667133547457996  0.519489951439109   158     63\n",
      "33             <NA>                <NA>               <NA>     0      0\n",
      "34 189.500721104896   -1.51810404353168  0.130654702143968   881    246\n",
      "35             <NA>                <NA>               <NA>    92     92\n",
      "36 271.821833342596   0.463902193858165  0.643089099898584  2755    465\n",
      "37 590.612091246047   0.143166482025976  0.886207530524957  1825    784\n",
      "38             <NA>                <NA>               <NA>     0      0\n",
      "39 17.8664434281671  -0.946678393567909  0.356431633743149   158     63\n",
      "40             <NA>                <NA>               <NA>     0      0\n",
      "41 201.056979229766   -1.06316357648669  0.288983037775433   881    246\n",
      "42             <NA>                <NA>               <NA>    92     92\n",
      "43 269.053965712007  -0.284764752136728  0.776043439251876  2755    465\n",
      "44 607.962300706665 0.00281520762536049  0.997754715744288  1825    784\n",
      "45             <NA>                <NA>               <NA>     0      0\n",
      "46 16.7136922028973  -0.379820516826837  0.708861721144134   158     63\n",
      "47             <NA>                <NA>               <NA>     0      0\n",
      "48 199.760716087705   -1.98033254094517 0.0490387906851641   881    246\n",
      "49             <NA>                <NA>               <NA>    92     92\n",
      "50 265.938640296445   -2.26586509474491 0.0242637881846844  2755    465\n",
      "51 529.591015278815   0.665637049292375  0.505932922384208  1825    784\n",
      "52             <NA>                <NA>               <NA>     0      0\n",
      "53 6.36918038042926    1.31118100860395  0.235083277929155   158     63\n",
      "54             <NA>                <NA>               <NA>     0      0\n",
      "55 190.769613954348   0.167382451935363  0.867246430229614   881    246\n",
      "56             <NA>                <NA>               <NA>    92     92\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "#Combine lists into a single dataframe\n",
    "moca_results <- do.call(rbind, moca_results_list)\n",
    "\n",
    "#Set column names\n",
    "colnames(moca_results) <- c(\"pathway\", \"cohort\", \"Coeff\", \"se\", \"df\", \"t_value\", \"Pvalue\", \"N_obs\", \"N_inds\")\n",
    "\n",
    "#Make into dataframe\n",
    "moca_results <- as.data.frame(moca_results)\n",
    "\n",
    "#Save output\n",
    "write.table(moca_results, \"/home/jupyter/pathways/pathways_moca_results.txt\",\n",
    "           sep = \"\\t\", row.names = F, col.names = T, quote = F)\n",
    "\n",
    "moca_results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Analyse pathways vs dementia"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "MOCA_cutoff = 21\n",
    "MMSE_cutoff = 26\n",
    "#These cutoffs should be inclusive - i.e. MoCA <=21 and MMSE <= 26"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Query MMSE data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "mmse_query = f\"\"\"\n",
    "\n",
    "SELECT\n",
    "participant_id, visit_name, visit_month, mms112_total_score\n",
    "FROM `{BQ_RELEASE_DATASET}.MMSE_corrected_20220506`\n",
    "\n",
    "\"\"\"\n",
    "mmse = bq_query(mmse_query)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "mmse = mmse.sort_values(by = ['participant_id', 'visit_month'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>mms112_total_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>HB-PD_INVAA223GY7</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>380</th>\n",
       "      <td>HB-PD_INVAA769TBF</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>701</th>\n",
       "      <td>HB-PD_INVAB109VHC</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>702</th>\n",
       "      <td>HB-PD_INVAB289LG3</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>559</th>\n",
       "      <td>HB-PD_INVAB465GYE</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        participant_id visit_name  visit_month  mms112_total_score\n",
       "124  HB-PD_INVAA223GY7         M0          0.0                  27\n",
       "380  HB-PD_INVAA769TBF         M0          0.0                  29\n",
       "701  HB-PD_INVAB109VHC         M0          0.0                  30\n",
       "702  HB-PD_INVAB289LG3         M0          0.0                  30\n",
       "559  HB-PD_INVAB465GYE         M0          0.0                  29"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mmse.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Remove duplicated visit entries - rows with #\n",
    "mmse_nodupli = mmse.copy()\n",
    "mmse_nodupli = mmse_nodupli[~mmse_nodupli.visit_name.str.contains('#')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>mms112_total_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [participant_id, visit_name, visit_month, mms112_total_score]\n",
       "Index: []"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Check for any other duplicated entries by participant id and visit month\n",
    "mmse_nodupli[mmse_nodupli.duplicated(['participant_id', 'visit_month'], keep = False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Write as output\n",
    "with open('/home/jupyter/clinical_data_v2-5/mmse_allparticipants.tab', 'w') as f:\n",
    "    f.write(mmse_nodupli.to_csv(index=False, sep='\\t'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>moca_total_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BF-1001</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BF-1002</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BF-1003</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BF-1004</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BF-1005</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  participant_id visit_name  visit_month  moca_total_score\n",
       "0        BF-1001         M0          0.0              28.0\n",
       "1        BF-1002         M0          0.0              29.0\n",
       "2        BF-1003         M0          0.0              30.0\n",
       "3        BF-1004         M0          0.0              28.0\n",
       "4        BF-1005         M0          0.0              27.0"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Read in MoCA data\n",
    "moca_nodupli = pd.read_csv('/home/jupyter/clinical_data_v2-5/moca_allparticipants.tab',\n",
    "                          sep = '\\t')\n",
    "moca_nodupli.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month_x</th>\n",
       "      <th>moca_total_score</th>\n",
       "      <th>visit_month_y</th>\n",
       "      <th>mms112_total_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PD-PDAA955TCY</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PD-PDAJ440URR</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PD-PDBK615WC6</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PD-PDBY238FWE</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PD-PDCF053TPC</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PD-PDCR944LPD</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PD-PDDC904HWE</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PD-PDDF170BDZ</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PD-PDED633MRU</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PD-PDEH654NL2</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PD-PDEM427WA7</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>PD-PDEN441WED</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>PD-PDEW817GNG</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>PD-PDEY957GBY</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PD-PDFE792DK0</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>PD-PDFK551GW2</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>PD-PDFV450FRF</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PD-PDFW311WR6</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>PD-PDFY781HVX</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>PD-PDGB158UUN</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>PD-PDGE693UKX</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>PD-PDGU974LXR</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>PD-PDGV403MFR</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>PD-PDHB484JY8</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PD-PDHF618LTF</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>PD-PDJK502DEH</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>PD-PDJL832MZN</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>PD-PDKD167AZ1</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>PD-PDKK308HHV</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>PD-PDKY229EYC</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>PD-PDLF380FA3</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>PD-PDLG906XEB</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>PD-PDLW898UEJ</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>PD-PDMX081PLW</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>PD-PDNA400GW6</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>PD-PDNB563CBA</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>PD-PDNF237HNT</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>PD-PDNV525LXN</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>PD-PDPP280DG5</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>PD-PDRG728DC7</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>PD-PDTA859UDF</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>PD-PDTH897LN9</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>PD-PDTN576BY4</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>PD-PDTT264VT6</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>PD-PDUC469GHX</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>PD-PDUK666PGT</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>PD-PDVF215FR5</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>PD-PDVL381PKJ</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>PD-PDVX986CEU</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>PD-PDWB112FNQ</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>PD-PDWE635JYP</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>PD-PDWK436LWD</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>PD-PDXA570BN6</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>PD-PDXJ740DLT</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>PD-PDXV036MCF</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>PD-PDYG322MCF</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>PD-PDZG969FZP</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>PD-PDZL311WBN</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>PD-PDZZ075NW6</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   participant_id visit_name  visit_month_x  moca_total_score  visit_month_y  \\\n",
       "0   PD-PDAA955TCY         M0            0.0              25.0            0.0   \n",
       "1   PD-PDAJ440URR         M0            0.0              29.0            0.0   \n",
       "2   PD-PDBK615WC6         M0            0.0              27.0            0.0   \n",
       "3   PD-PDBY238FWE         M0            0.0              22.0            0.0   \n",
       "4   PD-PDCF053TPC         M0            0.0              28.0            0.0   \n",
       "5   PD-PDCR944LPD         M0            0.0              28.0            0.0   \n",
       "6   PD-PDDC904HWE         M0            0.0              28.0            0.0   \n",
       "7   PD-PDDF170BDZ         M0            0.0              27.0            0.0   \n",
       "8   PD-PDED633MRU         M0            0.0              25.0            0.0   \n",
       "9   PD-PDEH654NL2         M0            0.0              27.0            0.0   \n",
       "10  PD-PDEM427WA7         M0            0.0              20.0            0.0   \n",
       "11  PD-PDEN441WED         M0            0.0              26.0            0.0   \n",
       "12  PD-PDEW817GNG         M0            0.0              26.0            0.0   \n",
       "13  PD-PDEY957GBY         M0            0.0              29.0            0.0   \n",
       "14  PD-PDFE792DK0         M0            0.0              30.0            0.0   \n",
       "15  PD-PDFK551GW2         M0            0.0              18.0            0.0   \n",
       "16  PD-PDFV450FRF         M0            0.0              21.0            0.0   \n",
       "17  PD-PDFW311WR6         M0            0.0              24.0            0.0   \n",
       "18  PD-PDFY781HVX         M0            0.0              29.0            0.0   \n",
       "19  PD-PDGB158UUN         M0            0.0              24.0            0.0   \n",
       "20  PD-PDGE693UKX         M0            0.0              28.0            0.0   \n",
       "21  PD-PDGU974LXR         M0            0.0              28.0            0.0   \n",
       "22  PD-PDGV403MFR         M0            0.0              26.0            0.0   \n",
       "23  PD-PDHB484JY8         M0            0.0              30.0            0.0   \n",
       "24  PD-PDHF618LTF         M0            0.0              26.0            0.0   \n",
       "25  PD-PDJK502DEH         M0            0.0              24.0            0.0   \n",
       "26  PD-PDJL832MZN         M0            0.0              28.0            0.0   \n",
       "27  PD-PDKD167AZ1         M0            0.0              30.0            0.0   \n",
       "28  PD-PDKK308HHV         M0            0.0              28.0            0.0   \n",
       "29  PD-PDKY229EYC         M0            0.0              20.0            0.0   \n",
       "30  PD-PDLF380FA3         M0            0.0              25.0            0.0   \n",
       "31  PD-PDLG906XEB         M0            0.0              24.0            0.0   \n",
       "32  PD-PDLW898UEJ         M0            0.0              27.0            0.0   \n",
       "33  PD-PDMX081PLW         M0            0.0              28.0            0.0   \n",
       "34  PD-PDNA400GW6         M0            0.0              26.0            0.0   \n",
       "35  PD-PDNB563CBA         M0            0.0              28.0            0.0   \n",
       "36  PD-PDNF237HNT         M0            0.0              29.0            0.0   \n",
       "37  PD-PDNV525LXN         M0            0.0              29.0            0.0   \n",
       "38  PD-PDPP280DG5         M0            0.0              29.0            0.0   \n",
       "39  PD-PDRG728DC7         M0            0.0              25.0            0.0   \n",
       "40  PD-PDTA859UDF         M0            0.0              25.0            0.0   \n",
       "41  PD-PDTH897LN9         M0            0.0              23.0            0.0   \n",
       "42  PD-PDTN576BY4         M0            0.0              28.0            0.0   \n",
       "43  PD-PDTT264VT6         M0            0.0              29.0            0.0   \n",
       "44  PD-PDUC469GHX         M0            0.0              27.0            0.0   \n",
       "45  PD-PDUK666PGT         M0            0.0              23.0            0.0   \n",
       "46  PD-PDVF215FR5         M0            0.0              24.0            0.0   \n",
       "47  PD-PDVL381PKJ         M0            0.0              28.0            0.0   \n",
       "48  PD-PDVX986CEU         M0            0.0              23.0            0.0   \n",
       "49  PD-PDWB112FNQ         M0            0.0              26.0            0.0   \n",
       "50  PD-PDWE635JYP         M0            0.0              20.0            0.0   \n",
       "51  PD-PDWK436LWD         M0            0.0              30.0            0.0   \n",
       "52  PD-PDXA570BN6         M0            0.0              30.0            0.0   \n",
       "53  PD-PDXJ740DLT         M0            0.0              24.0            0.0   \n",
       "54  PD-PDXV036MCF         M0            0.0              30.0            0.0   \n",
       "55  PD-PDYG322MCF         M0            0.0              27.0            0.0   \n",
       "56  PD-PDZG969FZP         M0            0.0              23.0            0.0   \n",
       "57  PD-PDZL311WBN         M0            0.0              29.0            0.0   \n",
       "58  PD-PDZZ075NW6         M0            0.0              29.0            0.0   \n",
       "\n",
       "    mms112_total_score  \n",
       "0                   29  \n",
       "1                   30  \n",
       "2                   30  \n",
       "3                   28  \n",
       "4                   29  \n",
       "5                   29  \n",
       "6                   30  \n",
       "7                   30  \n",
       "8                   28  \n",
       "9                   30  \n",
       "10                  27  \n",
       "11                  30  \n",
       "12                  28  \n",
       "13                  27  \n",
       "14                  30  \n",
       "15                  24  \n",
       "16                  27  \n",
       "17                  25  \n",
       "18                  29  \n",
       "19                  28  \n",
       "20                  29  \n",
       "21                  27  \n",
       "22                  28  \n",
       "23                  30  \n",
       "24                  29  \n",
       "25                  28  \n",
       "26                  30  \n",
       "27                  29  \n",
       "28                  30  \n",
       "29                  27  \n",
       "30                  27  \n",
       "31                  26  \n",
       "32                  29  \n",
       "33                  27  \n",
       "34                  30  \n",
       "35                  29  \n",
       "36                  30  \n",
       "37                  29  \n",
       "38                  30  \n",
       "39                  26  \n",
       "40                  28  \n",
       "41                  29  \n",
       "42                  29  \n",
       "43                  29  \n",
       "44                  27  \n",
       "45                  30  \n",
       "46                  26  \n",
       "47                  29  \n",
       "48                  29  \n",
       "49                  27  \n",
       "50                  28  \n",
       "51                  30  \n",
       "52                  30  \n",
       "53                  28  \n",
       "54                  29  \n",
       "55                  25  \n",
       "56                  29  \n",
       "57                  30  \n",
       "58                  29  "
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Check whether there are individuals who have done both the MoCA and MMSE\n",
    "both_moca_mmse = moca_nodupli.merge(mmse_nodupli, on = ['participant_id', 'visit_name'])\n",
    "both_moca_mmse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>moca_total_score</th>\n",
       "      <th>mms112_total_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BF-1001</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BF-1002</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BF-1003</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BF-1004</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BF-1005</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  participant_id visit_name  visit_month  moca_total_score  mms112_total_score\n",
       "0        BF-1001         M0          0.0              28.0                <NA>\n",
       "1        BF-1002         M0          0.0              29.0                <NA>\n",
       "2        BF-1003         M0          0.0              30.0                <NA>\n",
       "3        BF-1004         M0          0.0              28.0                <NA>\n",
       "4        BF-1005         M0          0.0              27.0                <NA>"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Merge all individuals by participant_id and visit_name\n",
    "moca_mmse = moca_nodupli.merge(mmse_nodupli, on = ['participant_id', 'visit_name', 'visit_month'], how = 'outer')\n",
    "moca_mmse.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Read in age at diagnosis and demographics data\n",
    "demo = pd.read_csv(f'/home/jupyter/clinical_data_v2-5/PD_cases_start_end_dx_sex.tab',\n",
    "                    delimiter = \"\\t\")\n",
    "\n",
    "aao = pd.read_csv(f'/home/jupyter/clinical_data_v2-5/PD_cases_agedx.tab',\n",
    "                   delimiter = \"\\t\").loc[:, ['FID', 'IID', 'age_at_diagnosis']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Merge with demographics and age at diagnosis data to get time from diagnosis\n",
    "temp = moca_mmse.merge(demo, left_on = 'participant_id', right_on = 'IID')\n",
    "merged = temp.merge(aao, on = (['FID', 'IID']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>moca_total_score</th>\n",
       "      <th>mms112_total_score</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>sex</th>\n",
       "      <th>age_at_baseline</th>\n",
       "      <th>age_at_diagnosis</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BF-1002</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>BF-1002</td>\n",
       "      <td>BF-1002</td>\n",
       "      <td>2</td>\n",
       "      <td>66</td>\n",
       "      <td>61.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BF-1003</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>BF-1003</td>\n",
       "      <td>BF-1003</td>\n",
       "      <td>1</td>\n",
       "      <td>61</td>\n",
       "      <td>56.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BF-1004</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>BF-1004</td>\n",
       "      <td>BF-1004</td>\n",
       "      <td>1</td>\n",
       "      <td>62</td>\n",
       "      <td>55.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BF-1006</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>BF-1006</td>\n",
       "      <td>BF-1006</td>\n",
       "      <td>1</td>\n",
       "      <td>59</td>\n",
       "      <td>52.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BF-1008</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>BF-1008</td>\n",
       "      <td>BF-1008</td>\n",
       "      <td>2</td>\n",
       "      <td>69</td>\n",
       "      <td>65.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  participant_id visit_name  visit_month  moca_total_score  \\\n",
       "0        BF-1002         M0          0.0              29.0   \n",
       "1        BF-1003         M0          0.0              30.0   \n",
       "2        BF-1004         M0          0.0              28.0   \n",
       "3        BF-1006         M0          0.0              25.0   \n",
       "4        BF-1008         M0          0.0              29.0   \n",
       "\n",
       "   mms112_total_score      FID      IID  sex  age_at_baseline  \\\n",
       "0                <NA>  BF-1002  BF-1002    2               66   \n",
       "1                <NA>  BF-1003  BF-1003    1               61   \n",
       "2                <NA>  BF-1004  BF-1004    1               62   \n",
       "3                <NA>  BF-1006  BF-1006    1               59   \n",
       "4                <NA>  BF-1008  BF-1008    2               69   \n",
       "\n",
       "   age_at_diagnosis  \n",
       "0              61.0  \n",
       "1              56.0  \n",
       "2              55.0  \n",
       "3              52.0  \n",
       "4              65.0  "
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>moca_total_score</th>\n",
       "      <th>mms112_total_score</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>sex</th>\n",
       "      <th>age_at_baseline</th>\n",
       "      <th>age_at_diagnosis</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [participant_id, visit_name, visit_month, moca_total_score, mms112_total_score, FID, IID, sex, age_at_baseline, age_at_diagnosis]\n",
       "Index: []"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Check whether there are any individuals with multiple entires for visit_name and visit_month\n",
    "#Because we merged on participant_id + visit_name + visit_month, there may be some individuals with the same visit name but different visit months\n",
    "merged[merged.duplicated(['participant_id', 'visit_name'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Calculate time from diagnosis for each visit\n",
    "merged['YEARS_DIAGNOSIS_TO_BL'] = merged.age_at_baseline - merged.age_at_diagnosis\n",
    "merged['YEARS_FROM_DIAGNOSIS'] = merged.YEARS_DIAGNOSIS_TO_BL + merged.visit_month/12"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Make new variable coding whether MoCA <=21 (or whatever predefined cutoff) at each visit\n",
    "merged['MOCA21'] = [np.nan if pd.isna(i)\n",
    "                else 1 if i <= MOCA_cutoff\n",
    "                else 0 for i in merged.moca_total_score]\n",
    "\n",
    "merged['MMSE26'] = [np.nan if pd.isna(i)\n",
    "                else 1 if i <= MMSE_cutoff\n",
    "                else 0 for i in merged.mms112_total_score]\n",
    "\n",
    "#Make variable which is the sum of the MOCA and MMSE cutoff variables\n",
    "merged['MOCA_MMSE_cutoffs'] = merged.fillna(0)['MOCA21'] + merged.fillna(0)['MMSE26']\n",
    "\n",
    "#Now make a combined variable which codes dementia if EITHER MOCA or MMSE are below cutoffs\n",
    "merged['dementia'] = [np.nan if pd.isna(i)\n",
    "                else 1 if i >= 1\n",
    "                else 0 for i in merged.MOCA_MMSE_cutoffs]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>moca_total_score</th>\n",
       "      <th>mms112_total_score</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>sex</th>\n",
       "      <th>age_at_baseline</th>\n",
       "      <th>age_at_diagnosis</th>\n",
       "      <th>YEARS_DIAGNOSIS_TO_BL</th>\n",
       "      <th>YEARS_FROM_DIAGNOSIS</th>\n",
       "      <th>MOCA21</th>\n",
       "      <th>MMSE26</th>\n",
       "      <th>MOCA_MMSE_cutoffs</th>\n",
       "      <th>dementia</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>BF-1032</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>BF-1032</td>\n",
       "      <td>BF-1032</td>\n",
       "      <td>1</td>\n",
       "      <td>70</td>\n",
       "      <td>63.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>BF-1177</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>BF-1177</td>\n",
       "      <td>BF-1177</td>\n",
       "      <td>1</td>\n",
       "      <td>73</td>\n",
       "      <td>62.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>BF-1210</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>BF-1210</td>\n",
       "      <td>BF-1210</td>\n",
       "      <td>1</td>\n",
       "      <td>73</td>\n",
       "      <td>65.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>LC-1490001</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>LC-1490001</td>\n",
       "      <td>LC-1490001</td>\n",
       "      <td>2</td>\n",
       "      <td>71</td>\n",
       "      <td>64.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>LC-1490001</td>\n",
       "      <td>M36</td>\n",
       "      <td>36.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>LC-1490001</td>\n",
       "      <td>LC-1490001</td>\n",
       "      <td>2</td>\n",
       "      <td>71</td>\n",
       "      <td>64.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7427</th>\n",
       "      <td>HB-PD_INVYU764JYK</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24</td>\n",
       "      <td>HB-PD_INVYU764JYK</td>\n",
       "      <td>HB-PD_INVYU764JYK</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>80.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7440</th>\n",
       "      <td>HB-PD_INVZF591WW3</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25</td>\n",
       "      <td>HB-PD_INVZF591WW3</td>\n",
       "      <td>HB-PD_INVZF591WW3</td>\n",
       "      <td>2</td>\n",
       "      <td>69</td>\n",
       "      <td>65.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7443</th>\n",
       "      <td>HB-PD_INVZJ012MP6</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26</td>\n",
       "      <td>HB-PD_INVZJ012MP6</td>\n",
       "      <td>HB-PD_INVZJ012MP6</td>\n",
       "      <td>2</td>\n",
       "      <td>64</td>\n",
       "      <td>60.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7454</th>\n",
       "      <td>HB-PD_INVZW010GB2</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13</td>\n",
       "      <td>HB-PD_INVZW010GB2</td>\n",
       "      <td>HB-PD_INVZW010GB2</td>\n",
       "      <td>1</td>\n",
       "      <td>75</td>\n",
       "      <td>65.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7458</th>\n",
       "      <td>HB-PD_INVZY881DN4</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23</td>\n",
       "      <td>HB-PD_INVZY881DN4</td>\n",
       "      <td>HB-PD_INVZY881DN4</td>\n",
       "      <td>1</td>\n",
       "      <td>70</td>\n",
       "      <td>70.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>603 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         participant_id visit_name  visit_month  moca_total_score  \\\n",
       "10              BF-1032         M0          0.0              18.0   \n",
       "88              BF-1177         M0          0.0              20.0   \n",
       "100             BF-1210         M0          0.0              18.0   \n",
       "127          LC-1490001         M0          0.0              21.0   \n",
       "129          LC-1490001        M36         36.0              20.0   \n",
       "...                 ...        ...          ...               ...   \n",
       "7427  HB-PD_INVYU764JYK         M0          0.0               NaN   \n",
       "7440  HB-PD_INVZF591WW3         M0          0.0               NaN   \n",
       "7443  HB-PD_INVZJ012MP6         M0          0.0               NaN   \n",
       "7454  HB-PD_INVZW010GB2         M0          0.0               NaN   \n",
       "7458  HB-PD_INVZY881DN4         M0          0.0               NaN   \n",
       "\n",
       "      mms112_total_score                FID                IID  sex  \\\n",
       "10                  <NA>            BF-1032            BF-1032    1   \n",
       "88                  <NA>            BF-1177            BF-1177    1   \n",
       "100                 <NA>            BF-1210            BF-1210    1   \n",
       "127                 <NA>         LC-1490001         LC-1490001    2   \n",
       "129                 <NA>         LC-1490001         LC-1490001    2   \n",
       "...                  ...                ...                ...  ...   \n",
       "7427                  24  HB-PD_INVYU764JYK  HB-PD_INVYU764JYK    1   \n",
       "7440                  25  HB-PD_INVZF591WW3  HB-PD_INVZF591WW3    2   \n",
       "7443                  26  HB-PD_INVZJ012MP6  HB-PD_INVZJ012MP6    2   \n",
       "7454                  13  HB-PD_INVZW010GB2  HB-PD_INVZW010GB2    1   \n",
       "7458                  23  HB-PD_INVZY881DN4  HB-PD_INVZY881DN4    1   \n",
       "\n",
       "      age_at_baseline  age_at_diagnosis  YEARS_DIAGNOSIS_TO_BL  \\\n",
       "10                 70              63.0                    7.0   \n",
       "88                 73              62.0                   11.0   \n",
       "100                73              65.0                    8.0   \n",
       "127                71              64.0                    7.0   \n",
       "129                71              64.0                    7.0   \n",
       "...               ...               ...                    ...   \n",
       "7427               87              80.0                    7.0   \n",
       "7440               69              65.0                    4.0   \n",
       "7443               64              60.0                    4.0   \n",
       "7454               75              65.0                   10.0   \n",
       "7458               70              70.0                    0.0   \n",
       "\n",
       "      YEARS_FROM_DIAGNOSIS  MOCA21  MMSE26  MOCA_MMSE_cutoffs  dementia  \n",
       "10                     7.0     1.0     NaN                1.0         1  \n",
       "88                    11.0     1.0     NaN                1.0         1  \n",
       "100                    8.0     1.0     NaN                1.0         1  \n",
       "127                    7.0     1.0     NaN                1.0         1  \n",
       "129                   10.0     1.0     NaN                1.0         1  \n",
       "...                    ...     ...     ...                ...       ...  \n",
       "7427                   7.0     NaN     1.0                1.0         1  \n",
       "7440                   4.0     NaN     1.0                1.0         1  \n",
       "7443                   4.0     NaN     1.0                1.0         1  \n",
       "7454                  10.0     NaN     1.0                1.0         1  \n",
       "7458                   0.0     NaN     1.0                1.0         1  \n",
       "\n",
       "[603 rows x 16 columns]"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged[merged.dementia == 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>moca_total_score</th>\n",
       "      <th>mms112_total_score</th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>sex</th>\n",
       "      <th>age_at_baseline</th>\n",
       "      <th>age_at_diagnosis</th>\n",
       "      <th>YEARS_DIAGNOSIS_TO_BL</th>\n",
       "      <th>YEARS_FROM_DIAGNOSIS</th>\n",
       "      <th>MOCA21</th>\n",
       "      <th>MMSE26</th>\n",
       "      <th>MOCA_MMSE_cutoffs</th>\n",
       "      <th>dementia</th>\n",
       "      <th>dementia_EV</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BF-1002</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>BF-1002</td>\n",
       "      <td>BF-1002</td>\n",
       "      <td>2</td>\n",
       "      <td>66</td>\n",
       "      <td>61.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BF-1003</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>BF-1003</td>\n",
       "      <td>BF-1003</td>\n",
       "      <td>1</td>\n",
       "      <td>61</td>\n",
       "      <td>56.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BF-1004</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>BF-1004</td>\n",
       "      <td>BF-1004</td>\n",
       "      <td>1</td>\n",
       "      <td>62</td>\n",
       "      <td>55.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BF-1006</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>BF-1006</td>\n",
       "      <td>BF-1006</td>\n",
       "      <td>1</td>\n",
       "      <td>59</td>\n",
       "      <td>52.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BF-1008</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29.0</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "      <td>BF-1008</td>\n",
       "      <td>BF-1008</td>\n",
       "      <td>2</td>\n",
       "      <td>69</td>\n",
       "      <td>65.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7455</th>\n",
       "      <td>HB-PD_INVZW684KJU</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29</td>\n",
       "      <td>HB-PD_INVZW684KJU</td>\n",
       "      <td>HB-PD_INVZW684KJU</td>\n",
       "      <td>1</td>\n",
       "      <td>70</td>\n",
       "      <td>69.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7456</th>\n",
       "      <td>HB-PD_INVZX194BGG</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29</td>\n",
       "      <td>HB-PD_INVZX194BGG</td>\n",
       "      <td>HB-PD_INVZX194BGG</td>\n",
       "      <td>1</td>\n",
       "      <td>70</td>\n",
       "      <td>62.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7457</th>\n",
       "      <td>HB-PD_INVZX600VYV</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30</td>\n",
       "      <td>HB-PD_INVZX600VYV</td>\n",
       "      <td>HB-PD_INVZX600VYV</td>\n",
       "      <td>2</td>\n",
       "      <td>61</td>\n",
       "      <td>59.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7458</th>\n",
       "      <td>HB-PD_INVZY881DN4</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23</td>\n",
       "      <td>HB-PD_INVZY881DN4</td>\n",
       "      <td>HB-PD_INVZY881DN4</td>\n",
       "      <td>1</td>\n",
       "      <td>70</td>\n",
       "      <td>70.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7459</th>\n",
       "      <td>HB-PD_INVZY899BLD</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30</td>\n",
       "      <td>HB-PD_INVZY899BLD</td>\n",
       "      <td>HB-PD_INVZY899BLD</td>\n",
       "      <td>1</td>\n",
       "      <td>58</td>\n",
       "      <td>56.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7460 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         participant_id visit_name  visit_month  moca_total_score  \\\n",
       "0               BF-1002         M0          0.0              29.0   \n",
       "1               BF-1003         M0          0.0              30.0   \n",
       "2               BF-1004         M0          0.0              28.0   \n",
       "3               BF-1006         M0          0.0              25.0   \n",
       "4               BF-1008         M0          0.0              29.0   \n",
       "...                 ...        ...          ...               ...   \n",
       "7455  HB-PD_INVZW684KJU         M0          0.0               NaN   \n",
       "7456  HB-PD_INVZX194BGG         M0          0.0               NaN   \n",
       "7457  HB-PD_INVZX600VYV         M0          0.0               NaN   \n",
       "7458  HB-PD_INVZY881DN4         M0          0.0               NaN   \n",
       "7459  HB-PD_INVZY899BLD         M0          0.0               NaN   \n",
       "\n",
       "      mms112_total_score                FID                IID  sex  \\\n",
       "0                   <NA>            BF-1002            BF-1002    2   \n",
       "1                   <NA>            BF-1003            BF-1003    1   \n",
       "2                   <NA>            BF-1004            BF-1004    1   \n",
       "3                   <NA>            BF-1006            BF-1006    1   \n",
       "4                   <NA>            BF-1008            BF-1008    2   \n",
       "...                  ...                ...                ...  ...   \n",
       "7455                  29  HB-PD_INVZW684KJU  HB-PD_INVZW684KJU    1   \n",
       "7456                  29  HB-PD_INVZX194BGG  HB-PD_INVZX194BGG    1   \n",
       "7457                  30  HB-PD_INVZX600VYV  HB-PD_INVZX600VYV    2   \n",
       "7458                  23  HB-PD_INVZY881DN4  HB-PD_INVZY881DN4    1   \n",
       "7459                  30  HB-PD_INVZY899BLD  HB-PD_INVZY899BLD    1   \n",
       "\n",
       "      age_at_baseline  age_at_diagnosis  YEARS_DIAGNOSIS_TO_BL  \\\n",
       "0                  66              61.0                    5.0   \n",
       "1                  61              56.0                    5.0   \n",
       "2                  62              55.0                    7.0   \n",
       "3                  59              52.0                    7.0   \n",
       "4                  69              65.0                    4.0   \n",
       "...               ...               ...                    ...   \n",
       "7455               70              69.0                    1.0   \n",
       "7456               70              62.0                    8.0   \n",
       "7457               61              59.0                    2.0   \n",
       "7458               70              70.0                    0.0   \n",
       "7459               58              56.0                    2.0   \n",
       "\n",
       "      YEARS_FROM_DIAGNOSIS  MOCA21  MMSE26  MOCA_MMSE_cutoffs  dementia  \\\n",
       "0                      5.0     0.0     NaN                0.0         0   \n",
       "1                      5.0     0.0     NaN                0.0         0   \n",
       "2                      7.0     0.0     NaN                0.0         0   \n",
       "3                      7.0     0.0     NaN                0.0         0   \n",
       "4                      4.0     0.0     NaN                0.0         0   \n",
       "...                    ...     ...     ...                ...       ...   \n",
       "7455                   1.0     NaN     0.0                0.0         0   \n",
       "7456                   8.0     NaN     0.0                0.0         0   \n",
       "7457                   2.0     NaN     0.0                0.0         0   \n",
       "7458                   0.0     NaN     1.0                1.0         1   \n",
       "7459                   2.0     NaN     0.0                0.0         0   \n",
       "\n",
       "      dementia_EV  \n",
       "0               0  \n",
       "1               0  \n",
       "2               0  \n",
       "3               0  \n",
       "4               0  \n",
       "...           ...  \n",
       "7455            0  \n",
       "7456            0  \n",
       "7457            0  \n",
       "7458            1  \n",
       "7459            0  \n",
       "\n",
       "[7460 rows x 17 columns]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Make a variable for dementia EVER  - whether a patient ever has a dementia in the course of their study \n",
    "DEM_EV = merged.loc[:, [\"participant_id\", 'dementia']].groupby(\"participant_id\").agg(func=max)\n",
    "\n",
    "#Join to main HY dataframe \n",
    "DEM_merged = merged.join(DEM_EV, on = \"participant_id\", rsuffix=\"_EV\").reset_index(drop = True)\n",
    "\n",
    "DEM_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 duplicated\n"
     ]
    }
   ],
   "source": [
    "#Now want to calculate the time to event for dementia\n",
    "\n",
    "#First filter for individuals who do not ever meet dementia, and take the last visit age where H&Y3+ was not met\n",
    "DEM_no = (DEM_merged[(DEM_merged.dementia_EV == 0) & (DEM_merged.dementia == 0)].drop_duplicates(subset=[\"participant_id\"],keep=\"last\")\n",
    "                               .rename({\"YEARS_FROM_DIAGNOSIS\":\"dementia_TIME_FROM_DIAGNOSIS\"}, axis=1))\n",
    "\n",
    "#Secondly filter for individuals who have meet dementia, and take the FIRST visit age where dementia was met\n",
    "DEM_yes = (DEM_merged[(DEM_merged.dementia_EV == 1) & (DEM_merged.dementia == 1)].drop_duplicates(subset=[\"participant_id\"],keep=\"first\")\n",
    "                               .rename({\"YEARS_FROM_DIAGNOSIS\":\"dementia_TIME_FROM_DIAGNOSIS\"}, axis=1))\n",
    "\n",
    "#Join dataframes - concatenate (add rows)\n",
    "DEMENTIA = pd.concat([DEM_no, DEM_yes]).reset_index(drop = True)\n",
    "\n",
    "#Check for duplicates\n",
    "DEMENTIA_dup = DEMENTIA[DEMENTIA.duplicated(['participant_id'], keep = False)]\n",
    "n_dup = len(DEMENTIA_dup)\n",
    "print(str(n_dup) + ' duplicated')\n",
    "\n",
    "#If a patient has dementia at baseline (visit_month == 0), they are left censored - code as 2 \n",
    "DEMENTIA.loc[:,\"dementia_EV\"] = DEMENTIA.dementia_EV.mask((DEMENTIA.visit_month == 0) & (DEMENTIA.dementia == 1), 2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Now select relevant columns - only one row per participant\n",
    "DEMENTIA_export = DEMENTIA.copy()\n",
    "DEMENTIA_export = DEMENTIA_export.loc[:, ['FID', 'IID', 'age_at_diagnosis', 'age_at_baseline', 'sex', 'dementia_EV', 'dementia_TIME_FROM_DIAGNOSIS']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    2427\n",
       "2     200\n",
       "1     182\n",
       "Name: dementia_EV, dtype: int64"
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DEMENTIA_export.dementia_EV.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FID</th>\n",
       "      <th>IID</th>\n",
       "      <th>age_at_diagnosis</th>\n",
       "      <th>age_at_baseline</th>\n",
       "      <th>sex</th>\n",
       "      <th>dementia_EV</th>\n",
       "      <th>dementia_TIME_FROM_DIAGNOSIS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BF-1002</td>\n",
       "      <td>BF-1002</td>\n",
       "      <td>61.0</td>\n",
       "      <td>66</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BF-1003</td>\n",
       "      <td>BF-1003</td>\n",
       "      <td>56.0</td>\n",
       "      <td>61</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BF-1004</td>\n",
       "      <td>BF-1004</td>\n",
       "      <td>55.0</td>\n",
       "      <td>62</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BF-1006</td>\n",
       "      <td>BF-1006</td>\n",
       "      <td>52.0</td>\n",
       "      <td>59</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BF-1008</td>\n",
       "      <td>BF-1008</td>\n",
       "      <td>65.0</td>\n",
       "      <td>69</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       FID      IID  age_at_diagnosis  age_at_baseline  sex  dementia_EV  \\\n",
       "0  BF-1002  BF-1002              61.0               66    2            0   \n",
       "1  BF-1003  BF-1003              56.0               61    1            0   \n",
       "2  BF-1004  BF-1004              55.0               62    1            0   \n",
       "3  BF-1006  BF-1006              52.0               59    1            0   \n",
       "4  BF-1008  BF-1008              65.0               69    2            0   \n",
       "\n",
       "   dementia_TIME_FROM_DIAGNOSIS  \n",
       "0                           5.0  \n",
       "1                           5.0  \n",
       "2                           7.0  \n",
       "3                           7.0  \n",
       "4                           4.0  "
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DEMENTIA_export.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Write as output\n",
    "#Note this is all the participants with clinical data, not just our list of PD only participants\n",
    "with open('/home/jupyter/clinical_data_v2-5/dementia_allparticipants.tab', 'w') as f:\n",
    "    f.write(DEMENTIA_export.to_csv(index=False, sep='\\t'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Now run survival analysis in R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "dementia_results_list <- list()\n",
    "\n",
    "#Loop over each pathway and cohort\n",
    "for (pathway in pathways) {\n",
    "    for (cohort in cohorts) {\n",
    "        i <- length(dementia_results_list)\n",
    "        dementia_results_list[[i+1]] <- analyse_surv(pathway, cohort, \"dementia_allparticipants\", \"dementia\")\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           pathway   study                Coeff                se\n",
      "1  adaptive_immune    PPMI   -0.113898994153847 0.110530123954694\n",
      "2  adaptive_immune    PDBP    0.158662380068633 0.138302942983449\n",
      "3  adaptive_immune     HBS                 <NA>              <NA>\n",
      "4  adaptive_immune     LCC                 <NA>              <NA>\n",
      "5  adaptive_immune  Steady                 <NA>              <NA>\n",
      "6  adaptive_immune    Sure                 <NA>              <NA>\n",
      "7  adaptive_immune BioFIND                 <NA>              <NA>\n",
      "8  alpha_synuclein    PPMI    0.080730518315116 0.124340273367051\n",
      "9  alpha_synuclein    PDBP -0.00310159809524902 0.144348455474514\n",
      "10 alpha_synuclein     HBS                 <NA>              <NA>\n",
      "11 alpha_synuclein     LCC                 <NA>              <NA>\n",
      "12 alpha_synuclein  Steady                 <NA>              <NA>\n",
      "13 alpha_synuclein    Sure                 <NA>              <NA>\n",
      "14 alpha_synuclein BioFIND                 <NA>              <NA>\n",
      "15   innate_immune    PPMI  -0.0622333349785923 0.116266508578009\n",
      "16   innate_immune    PDBP   -0.196039979893348 0.147846223181597\n",
      "17   innate_immune     HBS                 <NA>              <NA>\n",
      "18   innate_immune     LCC                 <NA>              <NA>\n",
      "19   innate_immune  Steady                 <NA>              <NA>\n",
      "20   innate_immune    Sure                 <NA>              <NA>\n",
      "21   innate_immune BioFIND                 <NA>              <NA>\n",
      "22        lysosome    PPMI    0.057211566285095 0.111640559670122\n",
      "23        lysosome    PDBP   0.0287703037755956 0.147004844654195\n",
      "24        lysosome     HBS                 <NA>              <NA>\n",
      "25        lysosome     LCC                 <NA>              <NA>\n",
      "26        lysosome  Steady                 <NA>              <NA>\n",
      "27        lysosome    Sure                 <NA>              <NA>\n",
      "28        lysosome BioFIND                 <NA>              <NA>\n",
      "29     endocytosis    PPMI  -0.0955959230679381  0.12623464691763\n",
      "30     endocytosis    PDBP   -0.064272263238099 0.147709447467169\n",
      "31     endocytosis     HBS                 <NA>              <NA>\n",
      "32     endocytosis     LCC                 <NA>              <NA>\n",
      "33     endocytosis  Steady                 <NA>              <NA>\n",
      "34     endocytosis    Sure                 <NA>              <NA>\n",
      "35     endocytosis BioFIND                 <NA>              <NA>\n",
      "36       microglia    PPMI   0.0639583999083892 0.111410611437674\n",
      "37       microglia    PDBP  -0.0315286025520065  0.14639716770987\n",
      "38       microglia     HBS                 <NA>              <NA>\n",
      "39       microglia     LCC                 <NA>              <NA>\n",
      "40       microglia  Steady                 <NA>              <NA>\n",
      "41       microglia    Sure                 <NA>              <NA>\n",
      "42       microglia BioFIND                 <NA>              <NA>\n",
      "43       monocytes    PPMI   0.0822210427856657 0.111401468009093\n",
      "44       monocytes    PDBP  -0.0166071616628791 0.156303143590182\n",
      "45       monocytes     HBS                 <NA>              <NA>\n",
      "46       monocytes     LCC                 <NA>              <NA>\n",
      "47       monocytes  Steady                 <NA>              <NA>\n",
      "48       monocytes    Sure                 <NA>              <NA>\n",
      "49       monocytes BioFIND                 <NA>              <NA>\n",
      "50    mitochondria    PPMI    0.166269288904327 0.111462440472969\n",
      "51    mitochondria    PDBP   -0.238992669549145 0.153199380366738\n",
      "52    mitochondria     HBS                 <NA>              <NA>\n",
      "53    mitochondria     LCC                 <NA>              <NA>\n",
      "54    mitochondria  Steady                 <NA>              <NA>\n",
      "55    mitochondria    Sure                 <NA>              <NA>\n",
      "56    mitochondria BioFIND                 <NA>              <NA>\n",
      "              Pvalue         Cox.zphPVal     ov.lik.ratio          logrank\n",
      "1  0.302785109530234   0.128516807857436 63.4303009863262 57.6111027463808\n",
      "2  0.251295255344991  0.0662523793010198 89.9130133132074 70.3087246947135\n",
      "3               <NA>                <NA>             <NA>             <NA>\n",
      "4               <NA>                <NA>             <NA>             <NA>\n",
      "5               <NA>                <NA>             <NA>             <NA>\n",
      "6               <NA>                <NA>             <NA>             <NA>\n",
      "7               <NA>                <NA>             <NA>             <NA>\n",
      "8  0.516163306449014  0.0328585289033493 62.7918401462903 57.7207491404181\n",
      "9  0.982857268438888   0.776342326091703 88.5912944599941 69.2032316163092\n",
      "10              <NA>                <NA>             <NA>             <NA>\n",
      "11              <NA>                <NA>             <NA>             <NA>\n",
      "12              <NA>                <NA>             <NA>             <NA>\n",
      "13              <NA>                <NA>             <NA>             <NA>\n",
      "14              <NA>                <NA>             <NA>             <NA>\n",
      "15 0.592466968255714   0.260861974838863 62.6564533841034 56.9747522071435\n",
      "16  0.18484891016667 0.00869118008783795 90.3515183335066 70.2169454805447\n",
      "17              <NA>                <NA>             <NA>             <NA>\n",
      "18              <NA>                <NA>             <NA>             <NA>\n",
      "19              <NA>                <NA>             <NA>             <NA>\n",
      "20              <NA>                <NA>             <NA>             <NA>\n",
      "21              <NA>                <NA>             <NA>             <NA>\n",
      "22 0.608327572140786   0.385011014521366   62.63147979353 57.1379315413853\n",
      "23 0.844837233431777    0.15939252314842 88.6292050514202 69.5032759854011\n",
      "24              <NA>                <NA>             <NA>             <NA>\n",
      "25              <NA>                <NA>             <NA>             <NA>\n",
      "26              <NA>                <NA>             <NA>             <NA>\n",
      "27              <NA>                <NA>             <NA>             <NA>\n",
      "28              <NA>                <NA>             <NA>             <NA>\n",
      "29 0.448877624360207   0.457720515211986 62.9432051279592 57.2278754676009\n",
      "30 0.663470764748635   0.104792633755664 88.7820597089431 69.5573779256126\n",
      "31              <NA>                <NA>             <NA>             <NA>\n",
      "32              <NA>                <NA>             <NA>             <NA>\n",
      "33              <NA>                <NA>             <NA>             <NA>\n",
      "34              <NA>                <NA>             <NA>             <NA>\n",
      "35              <NA>                <NA>             <NA>             <NA>\n",
      "36  0.56591490508137   0.330604341070013 62.7005784833709 57.0677507861828\n",
      "37 0.829483945713217   0.686970121765623  88.637072498924 69.1327844670373\n",
      "38              <NA>                <NA>             <NA>             <NA>\n",
      "39              <NA>                <NA>             <NA>             <NA>\n",
      "40              <NA>                <NA>             <NA>             <NA>\n",
      "41              <NA>                <NA>             <NA>             <NA>\n",
      "42              <NA>                <NA>             <NA>             <NA>\n",
      "43 0.460477578606539   0.205228359122279 62.9199164817521 57.1159573316412\n",
      "44 0.915384241479774   0.468818122784274 88.6021181748681 69.1292140179615\n",
      "45              <NA>                <NA>             <NA>             <NA>\n",
      "46              <NA>                <NA>             <NA>             <NA>\n",
      "47              <NA>                <NA>             <NA>             <NA>\n",
      "48              <NA>                <NA>             <NA>             <NA>\n",
      "49              <NA>                <NA>             <NA>             <NA>\n",
      "50 0.135776005674564   0.271005213125672 64.5750940957241 58.5321264944945\n",
      "51 0.118757357358987  0.0432785343692524 91.0291723797852 72.9671872096189\n",
      "52              <NA>                <NA>             <NA>             <NA>\n",
      "53              <NA>                <NA>             <NA>             <NA>\n",
      "54              <NA>                <NA>             <NA>             <NA>\n",
      "55              <NA>                <NA>             <NA>             <NA>\n",
      "56              <NA>                <NA>             <NA>             <NA>\n",
      "                  r2   N\n",
      "1  0.127514504779593 465\n",
      "2  0.121844944519105 692\n",
      "3               <NA> 538\n",
      "4               <NA>  55\n",
      "5               <NA>   0\n",
      "6               <NA> 244\n",
      "7               <NA>  90\n",
      "8  0.126315729682599 465\n",
      "9  0.120166067002174 692\n",
      "10              <NA> 538\n",
      "11              <NA>  55\n",
      "12              <NA>   0\n",
      "13              <NA> 244\n",
      "14              <NA>  90\n",
      "15 0.126061315691547 465\n",
      "16 0.122401235587508 692\n",
      "17              <NA> 538\n",
      "18              <NA>  55\n",
      "19              <NA>   0\n",
      "20              <NA> 244\n",
      "21              <NA>  90\n",
      "22  0.12601437811524 465\n",
      "23 0.120214266584705 692\n",
      "24              <NA> 538\n",
      "25              <NA>  55\n",
      "26              <NA>   0\n",
      "27              <NA> 244\n",
      "28              <NA>  90\n",
      "29  0.12660008168674 465\n",
      "30 0.120408579439641 692\n",
      "31              <NA> 538\n",
      "32              <NA>  55\n",
      "33              <NA>   0\n",
      "34              <NA> 244\n",
      "35              <NA>  90\n",
      "36 0.126144242146602 465\n",
      "37 0.120224268938355 692\n",
      "38              <NA> 538\n",
      "39              <NA>  55\n",
      "40              <NA>   0\n",
      "41              <NA> 244\n",
      "42              <NA>  90\n",
      "43  0.12655633800705 465\n",
      "44 0.120179828558752 692\n",
      "45              <NA> 538\n",
      "46              <NA>  55\n",
      "47              <NA>   0\n",
      "48              <NA> 244\n",
      "49              <NA>  90\n",
      "50 0.129659852928453 465\n",
      "51 0.123260220065914 692\n",
      "52              <NA> 538\n",
      "53              <NA>  55\n",
      "54              <NA>   0\n",
      "55              <NA> 244\n",
      "56              <NA>  90\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "#Combine lists into a single dataframe\n",
    "dementia_results <- do.call(rbind, dementia_results_list)\n",
    "\n",
    "#Set column names\n",
    "colnames(dementia_results) <- c(\"pathway\", \"study\", \"Coeff\", \"se\", \"Pvalue\", \"Cox.zphPVal\", \"ov.lik.ratio\",\"logrank\", \"r2\", \"N\")\n",
    "\n",
    "#Make into dataframe\n",
    "dementia_results <- as.data.frame(dementia_results)\n",
    "\n",
    "#Save output\n",
    "write.table(dementia_results, \"/home/jupyter/pathways/pathways_dementia_results.txt\",\n",
    "           sep = \"\\t\", row.names = F, col.names = T, quote = F)\n",
    "\n",
    "dementia_results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Analyse pathways vs PDQ8"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Query PDQ39 data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "pdq_query = f\"\"\"\n",
    "\n",
    "SELECT *\n",
    "FROM `{BQ_RELEASE_DATASET}.PDQ_39`\n",
    "\n",
    "\"\"\"\n",
    "pdq = bq_query(pdq_query)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>pdq39_07_getting_around_in_public</th>\n",
       "      <th>pdq39_12_dressing</th>\n",
       "      <th>pdq39_17_depressed</th>\n",
       "      <th>pdq39_27_close_personal_relations</th>\n",
       "      <th>pdq39_31_problem_with_concentration</th>\n",
       "      <th>pdq39_35_unable_to_communicate</th>\n",
       "      <th>pdq39_37_muscle_cramps</th>\n",
       "      <th>pdq39_25_embarassed_in_public</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HB-PD_INVCA430NAZ</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Never</td>\n",
       "      <td>Often</td>\n",
       "      <td>Occasionally</td>\n",
       "      <td>Occasionally</td>\n",
       "      <td>Never</td>\n",
       "      <td>Sometimes</td>\n",
       "      <td>Occasionally</td>\n",
       "      <td>Often</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PD-PDGR243KPT</td>\n",
       "      <td>M12</td>\n",
       "      <td>12.0</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Occasionally</td>\n",
       "      <td>Never</td>\n",
       "      <td>Often</td>\n",
       "      <td>Occasionally</td>\n",
       "      <td>Never</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PD-PDKG339XM0</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Occasionally</td>\n",
       "      <td>Often</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PD-PDMZ936CDD</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Occasionally</td>\n",
       "      <td>Often</td>\n",
       "      <td>Never</td>\n",
       "      <td>Always or cannot do at all</td>\n",
       "      <td>Occasionally</td>\n",
       "      <td>Occasionally</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PD-PDYE025CNY</td>\n",
       "      <td>M36</td>\n",
       "      <td>36.0</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "      <td>Occasionally</td>\n",
       "      <td>Always or cannot do at all</td>\n",
       "      <td>Never</td>\n",
       "      <td>Never</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      participant_id visit_name  visit_month  \\\n",
       "0  HB-PD_INVCA430NAZ         M0          0.0   \n",
       "1      PD-PDGR243KPT        M12         12.0   \n",
       "2      PD-PDKG339XM0         M0          0.0   \n",
       "3      PD-PDMZ936CDD         M0          0.0   \n",
       "4      PD-PDYE025CNY        M36         36.0   \n",
       "\n",
       "  pdq39_07_getting_around_in_public pdq39_12_dressing pdq39_17_depressed  \\\n",
       "0                             Never             Often       Occasionally   \n",
       "1                             Never             Never              Never   \n",
       "2                             Never             Never              Never   \n",
       "3                             Never             Never       Occasionally   \n",
       "4                             Never             Never              Never   \n",
       "\n",
       "  pdq39_27_close_personal_relations pdq39_31_problem_with_concentration  \\\n",
       "0                      Occasionally                               Never   \n",
       "1                      Occasionally                               Never   \n",
       "2                             Never                        Occasionally   \n",
       "3                             Often                               Never   \n",
       "4                             Never                        Occasionally   \n",
       "\n",
       "  pdq39_35_unable_to_communicate pdq39_37_muscle_cramps  \\\n",
       "0                      Sometimes           Occasionally   \n",
       "1                          Often           Occasionally   \n",
       "2                          Often                  Never   \n",
       "3     Always or cannot do at all           Occasionally   \n",
       "4     Always or cannot do at all                  Never   \n",
       "\n",
       "  pdq39_25_embarassed_in_public  \n",
       "0                         Often  \n",
       "1                         Never  \n",
       "2                         Never  \n",
       "3                  Occasionally  \n",
       "4                         Never  "
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Select just PDQ8 columns\n",
    "pdq8 = pdq.copy()\n",
    "pdq8 = pdq8.loc[:, ['participant_id', 'visit_name', 'visit_month', 'pdq39_07_getting_around_in_public',\n",
    "                   'pdq39_12_dressing', 'pdq39_17_depressed', 'pdq39_27_close_personal_relations',\n",
    "                   'pdq39_31_problem_with_concentration', 'pdq39_35_unable_to_communicate', 'pdq39_37_muscle_cramps',\n",
    "                   'pdq39_25_embarassed_in_public']]\n",
    "\n",
    "pdq8.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Recode variables as numeric (0 = Never, 1 = Occasionally, 2 = Sometimes, 3 = Often, 4 = Always)\n",
    "\n",
    "#Make dictionary to map values (text to numeric)\n",
    "Dict_PDQ = {'Never':0, 'Occasionally':1, 'Sometimes':2, 'Often':3, 'Always or cannot do at all':4}\n",
    "\n",
    "pdq8['pdq39_07_getting_around_in_public'] = pdq8.pdq39_07_getting_around_in_public.map(Dict_PDQ).astype(float)\n",
    "pdq8['pdq39_12_dressing'] = pdq8.pdq39_12_dressing.map(Dict_PDQ).astype(float)\n",
    "pdq8['pdq39_17_depressed'] = pdq8.pdq39_17_depressed.map(Dict_PDQ).astype(float)\n",
    "pdq8['pdq39_27_close_personal_relations'] = pdq8.pdq39_27_close_personal_relations.map(Dict_PDQ).astype(float)\n",
    "pdq8['pdq39_31_problem_with_concentration'] = pdq8.pdq39_31_problem_with_concentration.map(Dict_PDQ).astype(float)\n",
    "pdq8['pdq39_35_unable_to_communicate'] = pdq8.pdq39_35_unable_to_communicate.map(Dict_PDQ).astype(float)\n",
    "pdq8['pdq39_37_muscle_cramps'] = pdq8.pdq39_37_muscle_cramps.map(Dict_PDQ).astype(float)\n",
    "pdq8['pdq39_25_embarassed_in_public'] = pdq8.pdq39_25_embarassed_in_public.map(Dict_PDQ).astype(float)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Make total PDQ8 column\n",
    "#If there are any missing values, return NaN\n",
    "pdq8['PDQ8_TOTAL'] = pdq8.iloc[:, 3:].sum(min_count = 8, axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>pdq39_07_getting_around_in_public</th>\n",
       "      <th>pdq39_12_dressing</th>\n",
       "      <th>pdq39_17_depressed</th>\n",
       "      <th>pdq39_27_close_personal_relations</th>\n",
       "      <th>pdq39_31_problem_with_concentration</th>\n",
       "      <th>pdq39_35_unable_to_communicate</th>\n",
       "      <th>pdq39_37_muscle_cramps</th>\n",
       "      <th>pdq39_25_embarassed_in_public</th>\n",
       "      <th>PDQ8_TOTAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1554</th>\n",
       "      <td>SU-32242</td>\n",
       "      <td>M24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1555</th>\n",
       "      <td>SU-32242</td>\n",
       "      <td>M24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1906</th>\n",
       "      <td>SU-33113</td>\n",
       "      <td>M24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1907</th>\n",
       "      <td>SU-33113</td>\n",
       "      <td>M24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1908</th>\n",
       "      <td>SU-33114</td>\n",
       "      <td>M24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1909</th>\n",
       "      <td>SU-33114</td>\n",
       "      <td>M24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1980</th>\n",
       "      <td>SU-33368</td>\n",
       "      <td>M18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1981</th>\n",
       "      <td>SU-33368</td>\n",
       "      <td>M18</td>\n",
       "      <td>18.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     participant_id visit_name  visit_month  \\\n",
       "1554       SU-32242        M24         24.0   \n",
       "1555       SU-32242        M24         24.0   \n",
       "1906       SU-33113        M24         24.0   \n",
       "1907       SU-33113        M24         24.0   \n",
       "1908       SU-33114        M24         24.0   \n",
       "1909       SU-33114        M24         24.0   \n",
       "1980       SU-33368        M18         18.0   \n",
       "1981       SU-33368        M18         18.0   \n",
       "\n",
       "      pdq39_07_getting_around_in_public  pdq39_12_dressing  \\\n",
       "1554                                0.0                0.0   \n",
       "1555                                0.0                0.0   \n",
       "1906                                0.0                0.0   \n",
       "1907                                0.0                0.0   \n",
       "1908                                0.0                0.0   \n",
       "1909                                0.0                0.0   \n",
       "1980                                0.0                0.0   \n",
       "1981                                0.0                0.0   \n",
       "\n",
       "      pdq39_17_depressed  pdq39_27_close_personal_relations  \\\n",
       "1554                 0.0                                0.0   \n",
       "1555                 0.0                                0.0   \n",
       "1906                 0.0                                0.0   \n",
       "1907                 0.0                                0.0   \n",
       "1908                 1.0                                0.0   \n",
       "1909                 1.0                                0.0   \n",
       "1980                 0.0                                0.0   \n",
       "1981                 0.0                                0.0   \n",
       "\n",
       "      pdq39_31_problem_with_concentration  pdq39_35_unable_to_communicate  \\\n",
       "1554                                  0.0                             0.0   \n",
       "1555                                  0.0                             0.0   \n",
       "1906                                  0.0                             0.0   \n",
       "1907                                  0.0                             0.0   \n",
       "1908                                  0.0                             0.0   \n",
       "1909                                  1.0                             0.0   \n",
       "1980                                  1.0                             0.0   \n",
       "1981                                  1.0                             0.0   \n",
       "\n",
       "      pdq39_37_muscle_cramps  pdq39_25_embarassed_in_public  PDQ8_TOTAL  \n",
       "1554                     0.0                            0.0         0.0  \n",
       "1555                     2.0                            0.0         2.0  \n",
       "1906                     1.0                            0.0         1.0  \n",
       "1907                     1.0                            0.0         1.0  \n",
       "1908                     0.0                            0.0         1.0  \n",
       "1909                     3.0                            0.0         5.0  \n",
       "1980                     0.0                            0.0         1.0  \n",
       "1981                     0.0                            0.0         1.0  "
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pdq8 = pdq8.sort_values(by = ['participant_id', 'visit_month'])\n",
    "\n",
    "#Remove duplicated visit entries - rows with #\n",
    "pdq8_nodupli = pdq8.copy()\n",
    "pdq8_nodupli = pdq8_nodupli[~pdq8_nodupli.visit_name.str.contains('#')]\n",
    "\n",
    "#Check for any other duplicated entries by participant id and visit month\n",
    "pdq8_nodupli[pdq8_nodupli.duplicated(['participant_id', 'visit_month'], keep = False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#There are still a few individuals with duplicated entries by participant_id and visit_month with different MoCA scores\n",
    "#Just keep the first datapoint\n",
    "pdq8_nodupli_final = pdq8_nodupli.copy()\n",
    "pdq8_nodupli_final = pdq8_nodupli_final.drop_duplicates(subset = ['participant_id', 'visit_month'], keep = 'first')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>pdq39_07_getting_around_in_public</th>\n",
       "      <th>pdq39_12_dressing</th>\n",
       "      <th>pdq39_17_depressed</th>\n",
       "      <th>pdq39_27_close_personal_relations</th>\n",
       "      <th>pdq39_31_problem_with_concentration</th>\n",
       "      <th>pdq39_35_unable_to_communicate</th>\n",
       "      <th>pdq39_37_muscle_cramps</th>\n",
       "      <th>pdq39_25_embarassed_in_public</th>\n",
       "      <th>PDQ8_TOTAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HB-PD_INVCA430NAZ</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2328</th>\n",
       "      <td>HB-PD_INVCU917HHH</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>24.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>HB-PD_INVFP220NZU</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>HB-PD_INVGH702HGH</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>HB-PD_INVGT912XMQ</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         participant_id visit_name  visit_month  \\\n",
       "0     HB-PD_INVCA430NAZ         M0          0.0   \n",
       "2328  HB-PD_INVCU917HHH         M0          0.0   \n",
       "49    HB-PD_INVFP220NZU         M0          0.0   \n",
       "50    HB-PD_INVGH702HGH         M0          0.0   \n",
       "51    HB-PD_INVGT912XMQ         M0          0.0   \n",
       "\n",
       "      pdq39_07_getting_around_in_public  pdq39_12_dressing  \\\n",
       "0                                   0.0                3.0   \n",
       "2328                                3.0                3.0   \n",
       "49                                  1.0                1.0   \n",
       "50                                  1.0                3.0   \n",
       "51                                  1.0                1.0   \n",
       "\n",
       "      pdq39_17_depressed  pdq39_27_close_personal_relations  \\\n",
       "0                    1.0                                1.0   \n",
       "2328                 3.0                                3.0   \n",
       "49                   3.0                                NaN   \n",
       "50                   1.0                                3.0   \n",
       "51                   1.0                                3.0   \n",
       "\n",
       "      pdq39_31_problem_with_concentration  pdq39_35_unable_to_communicate  \\\n",
       "0                                     0.0                             2.0   \n",
       "2328                                  3.0                             3.0   \n",
       "49                                    3.0                             1.0   \n",
       "50                                    3.0                             1.0   \n",
       "51                                    3.0                             3.0   \n",
       "\n",
       "      pdq39_37_muscle_cramps  pdq39_25_embarassed_in_public  PDQ8_TOTAL  \n",
       "0                        1.0                            3.0        11.0  \n",
       "2328                     3.0                            3.0        24.0  \n",
       "49                       0.0                            1.0         NaN  \n",
       "50                       1.0                            3.0        16.0  \n",
       "51                       1.0                            1.0        14.0  "
      ]
     },
     "execution_count": 125,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pdq8_nodupli_final.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Write as output\n",
    "pdq8_export = pdq8_nodupli_final.copy()\n",
    "pdq8_export = pdq8_export.loc[:, ['participant_id', 'visit_name', 'visit_month', 'PDQ8_TOTAL']]\n",
    "\n",
    "with open('/home/jupyter/clinical_data_v2-5/PDQ8_allparticipants.tab', 'w') as f:\n",
    "    f.write(pdq8_export.to_csv(index=False, sep='\\t'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>PDQ8_TOTAL</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HB-PD_INVCA430NAZ</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2328</th>\n",
       "      <td>HB-PD_INVCU917HHH</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>24.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>HB-PD_INVFP220NZU</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>HB-PD_INVGH702HGH</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>16.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>HB-PD_INVGT912XMQ</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         participant_id visit_name  visit_month  PDQ8_TOTAL\n",
       "0     HB-PD_INVCA430NAZ         M0          0.0        11.0\n",
       "2328  HB-PD_INVCU917HHH         M0          0.0        24.0\n",
       "49    HB-PD_INVFP220NZU         M0          0.0         NaN\n",
       "50    HB-PD_INVGH702HGH         M0          0.0        16.0\n",
       "51    HB-PD_INVGT912XMQ         M0          0.0        14.0"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pdq8_export.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Now analyse in mixed model in R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"PPMI\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 5.747978 5.084956\n",
      "[1] \"HBS\"\n",
      "      mean       sd\n",
      "1 17.66667 4.830459\n",
      "[1] \"LCC\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 2.149798 2.706104\n",
      "[1] \"BioFIND\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"PPMI\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 5.747978 5.084956\n",
      "[1] \"HBS\"\n",
      "      mean       sd\n",
      "1 17.66667 4.830459\n",
      "[1] \"LCC\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 2.149798 2.706104\n",
      "[1] \"BioFIND\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"PPMI\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 5.747978 5.084956\n",
      "[1] \"HBS\"\n",
      "      mean       sd\n",
      "1 17.66667 4.830459\n",
      "[1] \"LCC\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 2.149798 2.706104\n",
      "[1] \"BioFIND\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"PPMI\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 5.747978 5.084956\n",
      "[1] \"HBS\"\n",
      "      mean       sd\n",
      "1 17.66667 4.830459\n",
      "[1] \"LCC\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 2.149798 2.706104\n",
      "[1] \"BioFIND\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"PPMI\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 5.747978 5.084956\n",
      "[1] \"HBS\"\n",
      "      mean       sd\n",
      "1 17.66667 4.830459\n",
      "[1] \"LCC\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 2.149798 2.706104\n",
      "[1] \"BioFIND\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"PPMI\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 5.747978 5.084956\n",
      "[1] \"HBS\"\n",
      "      mean       sd\n",
      "1 17.66667 4.830459\n",
      "[1] \"LCC\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 2.149798 2.706104\n",
      "[1] \"BioFIND\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"PPMI\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 5.747978 5.084956\n",
      "[1] \"HBS\"\n",
      "      mean       sd\n",
      "1 17.66667 4.830459\n",
      "[1] \"LCC\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 2.149798 2.706104\n",
      "[1] \"BioFIND\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"PPMI\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"PDBP\"\n",
      "      mean       sd\n",
      "1 5.747978 5.084956\n",
      "[1] \"HBS\"\n",
      "      mean       sd\n",
      "1 17.66667 4.830459\n",
      "[1] \"LCC\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Steady\"\n",
      "  mean sd\n",
      "1  NaN NA\n",
      "[1] \"Sure\"\n",
      "      mean       sd\n",
      "1 2.149798 2.706104\n",
      "[1] \"BioFIND\"\n",
      "  mean sd\n",
      "1  NaN NA\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "pdq8_results_list <- list()\n",
    "\n",
    "#Loop over each pathway and cohort\n",
    "for (pathway in pathways) {\n",
    "    for (cohort in cohorts) {\n",
    "        print(cohort)\n",
    "        i <- length(pdq8_results_list)\n",
    "        pdq8_results_list[[i+1]] <- analyse_cont(pathway, cohort, \"PDQ8_allparticipants\", \"PDQ8_TOTAL\")\n",
    "\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           pathway  cohort                Coeff                 se\n",
      "1  adaptive_immune    PPMI                 <NA>               <NA>\n",
      "2  adaptive_immune    PDBP   0.0635941979669745 0.0322937651133829\n",
      "3  adaptive_immune     HBS                 <NA>               <NA>\n",
      "4  adaptive_immune     LCC                 <NA>               <NA>\n",
      "5  adaptive_immune  Steady                 <NA>               <NA>\n",
      "6  adaptive_immune    Sure    -0.01818439043883 0.0586729153713862\n",
      "7  adaptive_immune BioFIND                 <NA>               <NA>\n",
      "8  alpha_synuclein    PPMI                 <NA>               <NA>\n",
      "9  alpha_synuclein    PDBP     0.06305876506887 0.0326873437208003\n",
      "10 alpha_synuclein     HBS                 <NA>               <NA>\n",
      "11 alpha_synuclein     LCC                 <NA>               <NA>\n",
      "12 alpha_synuclein  Steady                 <NA>               <NA>\n",
      "13 alpha_synuclein    Sure  -0.0319240324930557 0.0577936081145065\n",
      "14 alpha_synuclein BioFIND                 <NA>               <NA>\n",
      "15   innate_immune    PPMI                 <NA>               <NA>\n",
      "16   innate_immune    PDBP   0.0164079050188933 0.0304753498401251\n",
      "17   innate_immune     HBS                 <NA>               <NA>\n",
      "18   innate_immune     LCC                 <NA>               <NA>\n",
      "19   innate_immune  Steady                 <NA>               <NA>\n",
      "20   innate_immune    Sure  -0.0169443468133138 0.0586863965393975\n",
      "21   innate_immune BioFIND                 <NA>               <NA>\n",
      "22        lysosome    PPMI                 <NA>               <NA>\n",
      "23        lysosome    PDBP  0.00236687603382172 0.0322888133866094\n",
      "24        lysosome     HBS                 <NA>               <NA>\n",
      "25        lysosome     LCC                 <NA>               <NA>\n",
      "26        lysosome  Steady                 <NA>               <NA>\n",
      "27        lysosome    Sure -0.00337889200431338 0.0608809970106198\n",
      "28        lysosome BioFIND                 <NA>               <NA>\n",
      "29     endocytosis    PPMI                 <NA>               <NA>\n",
      "30     endocytosis    PDBP   0.0128861256077465 0.0322432200281139\n",
      "31     endocytosis     HBS                 <NA>               <NA>\n",
      "32     endocytosis     LCC                 <NA>               <NA>\n",
      "33     endocytosis  Steady                 <NA>               <NA>\n",
      "34     endocytosis    Sure  -0.0050504921006685 0.0598841759905535\n",
      "35     endocytosis BioFIND                 <NA>               <NA>\n",
      "36       microglia    PPMI                 <NA>               <NA>\n",
      "37       microglia    PDBP  -0.0169085634683457 0.0326787367162005\n",
      "38       microglia     HBS                 <NA>               <NA>\n",
      "39       microglia     LCC                 <NA>               <NA>\n",
      "40       microglia  Steady                 <NA>               <NA>\n",
      "41       microglia    Sure    0.184958870125412 0.0588040463334271\n",
      "42       microglia BioFIND                 <NA>               <NA>\n",
      "43       monocytes    PPMI                 <NA>               <NA>\n",
      "44       monocytes    PDBP  -0.0269567975639224 0.0321073512148839\n",
      "45       monocytes     HBS                 <NA>               <NA>\n",
      "46       monocytes     LCC                 <NA>               <NA>\n",
      "47       monocytes  Steady                 <NA>               <NA>\n",
      "48       monocytes    Sure    0.156095842167773 0.0581568692080896\n",
      "49       monocytes BioFIND                 <NA>               <NA>\n",
      "50    mitochondria    PPMI                 <NA>               <NA>\n",
      "51    mitochondria    PDBP  0.00297833884293328  0.030995037939447\n",
      "52    mitochondria     HBS                 <NA>               <NA>\n",
      "53    mitochondria     LCC                 <NA>               <NA>\n",
      "54    mitochondria  Steady                 <NA>               <NA>\n",
      "55    mitochondria    Sure    0.189384113600347 0.0586004836872612\n",
      "56    mitochondria BioFIND                 <NA>               <NA>\n",
      "                 df             t_value              Pvalue N_obs N_inds\n",
      "1              <NA>                <NA>                <NA>     0      0\n",
      "2  304.370591026619    1.96924074178703  0.0498322313179003  1768    747\n",
      "3              <NA>                <NA>                <NA>    21     21\n",
      "4              <NA>                <NA>                <NA>     0      0\n",
      "5              <NA>                <NA>                <NA>     0      0\n",
      "6  201.249424662852  -0.309928189586745    0.75693625775273   877    247\n",
      "7              <NA>                <NA>                <NA>     0      0\n",
      "8              <NA>                <NA>                <NA>     0      0\n",
      "9  356.458939713116    1.92914926362594   0.054505879361197  1768    747\n",
      "10             <NA>                <NA>                <NA>    21     21\n",
      "11             <NA>                <NA>                <NA>     0      0\n",
      "12             <NA>                <NA>                <NA>     0      0\n",
      "13 188.323819296543  -0.552379986897593   0.581343023941448   877    247\n",
      "14             <NA>                <NA>                <NA>     0      0\n",
      "15             <NA>                <NA>                <NA>     0      0\n",
      "16 242.075199828914   0.538399234298207    0.59079616116318  1768    747\n",
      "17             <NA>                <NA>                <NA>    21     21\n",
      "18             <NA>                <NA>                <NA>     0      0\n",
      "19             <NA>                <NA>                <NA>     0      0\n",
      "20 203.157458787162  -0.288726993178711    0.77308469256301   877    247\n",
      "21             <NA>                <NA>                <NA>     0      0\n",
      "22             <NA>                <NA>                <NA>     0      0\n",
      "23 361.937170383056  0.0733032832604277   0.941605272053018  1768    747\n",
      "24             <NA>                <NA>                <NA>    21     21\n",
      "25             <NA>                <NA>                <NA>     0      0\n",
      "26             <NA>                <NA>                <NA>     0      0\n",
      "27 223.211818962856 -0.0554999453068087   0.955789818706036   877    247\n",
      "28             <NA>                <NA>                <NA>     0      0\n",
      "29             <NA>                <NA>                <NA>     0      0\n",
      "30 336.062623876339   0.399653806180358   0.689665401027672  1768    747\n",
      "31             <NA>                <NA>                <NA>    21     21\n",
      "32             <NA>                <NA>                <NA>     0      0\n",
      "33             <NA>                <NA>                <NA>     0      0\n",
      "34 207.126113742084 -0.0843376738032632   0.932869418038489   877    247\n",
      "35             <NA>                <NA>                <NA>     0      0\n",
      "36             <NA>                <NA>                <NA>     0      0\n",
      "37 298.689024024699  -0.517417904345221     0.6052474313044  1768    747\n",
      "38             <NA>                <NA>                <NA>    21     21\n",
      "39             <NA>                <NA>                <NA>     0      0\n",
      "40             <NA>                <NA>                <NA>     0      0\n",
      "41 217.162572316455    3.14534256837819 0.00189116457983602   877    247\n",
      "42             <NA>                <NA>                <NA>     0      0\n",
      "43             <NA>                <NA>                <NA>     0      0\n",
      "44 304.504489959822  -0.839583352220787   0.401800709158889  1768    747\n",
      "45             <NA>                <NA>                <NA>    21     21\n",
      "46             <NA>                <NA>                <NA>     0      0\n",
      "47             <NA>                <NA>                <NA>     0      0\n",
      "48 208.432392936937    2.68404823528671 0.00785827715306779   877    247\n",
      "49             <NA>                <NA>                <NA>     0      0\n",
      "50             <NA>                <NA>                <NA>     0      0\n",
      "51 266.303203764569  0.0960908274657341   0.923520707556435  1768    747\n",
      "52             <NA>                <NA>                <NA>    21     21\n",
      "53             <NA>                <NA>                <NA>     0      0\n",
      "54             <NA>                <NA>                <NA>     0      0\n",
      "55 210.252197554678    3.23178413698855 0.00142807702294808   877    247\n",
      "56             <NA>                <NA>                <NA>     0      0\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "#Combine lists into a single dataframe\n",
    "pdq8_results <- do.call(rbind, pdq8_results_list)\n",
    "\n",
    "#Set column names\n",
    "colnames(pdq8_results) <- c(\"pathway\", \"cohort\", \"Coeff\", \"se\", \"df\", \"t_value\", \"Pvalue\", \"N_obs\", \"N_inds\")\n",
    "\n",
    "#Make into dataframe\n",
    "pdq8_results <- as.data.frame(pdq8_results)\n",
    "\n",
    "#Save output\n",
    "write.table(pdq8_results, \"/home/jupyter/pathways/pathways_pdq8_results.txt\",\n",
    "           sep = \"\\t\", row.names = F, col.names = T, quote = F)\n",
    "\n",
    "pdq8_results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Analyse pathways vs RBD (never vs. ever)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Query RBD Mayo questionnaire data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "rbd_Mayo_query = f\"\"\"\n",
    "\n",
    "SELECT *\n",
    "FROM `{BQ_RELEASE_DATASET}.REM_Sleep_Behavior_Disorder_Questionnaire_Mayo`\n",
    "\n",
    "\"\"\"\n",
    "rbd_Mayo = bq_query(rbd_Mayo_query)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>GUID</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>msq_info_source</th>\n",
       "      <th>msq_interviewee_live_with_subject</th>\n",
       "      <th>msq_interviewee_sleep_same_room</th>\n",
       "      <th>msq_distracting_sleep_behaviors</th>\n",
       "      <th>msq01_act_out_dreams</th>\n",
       "      <th>msq01a_act_out_years</th>\n",
       "      <th>...</th>\n",
       "      <th>msq02_legs_jerk</th>\n",
       "      <th>msq03_restless_legs</th>\n",
       "      <th>msq03a_leg_sensations_decrease</th>\n",
       "      <th>msq03b_time_leg_sensations_worst</th>\n",
       "      <th>msq04b_walked_asleep</th>\n",
       "      <th>msq05_snorted_awake</th>\n",
       "      <th>msq06_stop_breathing</th>\n",
       "      <th>msq06a_treated_for_stop_breathing</th>\n",
       "      <th>msq07_leg_cramps</th>\n",
       "      <th>msq08_rate_of_alertness</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PD-PDAX486GDU</td>\n",
       "      <td>PDAX486GDU</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Subject served as informant</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PD-PDBC294LL7</td>\n",
       "      <td>PDBC294LL7</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Informant</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PD-PDBK615WC6</td>\n",
       "      <td>PDBK615WC6</td>\n",
       "      <td>M24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PD-PDCZ531JKD</td>\n",
       "      <td>PDCZ531JKD</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Subject served as informant</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>None</td>\n",
       "      <td>Yes</td>\n",
       "      <td>No</td>\n",
       "      <td>No</td>\n",
       "      <td>None</td>\n",
       "      <td>Yes</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PD-PDDW009MK9</td>\n",
       "      <td>PDDW009MK9</td>\n",
       "      <td>M12</td>\n",
       "      <td>12.0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  participant_id        GUID visit_name  visit_month  \\\n",
       "0  PD-PDAX486GDU  PDAX486GDU         M0          0.0   \n",
       "1  PD-PDBC294LL7  PDBC294LL7         M0          0.0   \n",
       "2  PD-PDBK615WC6  PDBK615WC6        M24         24.0   \n",
       "3  PD-PDCZ531JKD  PDCZ531JKD         M0          0.0   \n",
       "4  PD-PDDW009MK9  PDDW009MK9        M12         12.0   \n",
       "\n",
       "               msq_info_source msq_interviewee_live_with_subject  \\\n",
       "0  Subject served as informant                              None   \n",
       "1                    Informant                              None   \n",
       "2                         None                              None   \n",
       "3  Subject served as informant                              None   \n",
       "4                         None                              None   \n",
       "\n",
       "  msq_interviewee_sleep_same_room msq_distracting_sleep_behaviors  \\\n",
       "0                            None                            None   \n",
       "1                            None                            None   \n",
       "2                            None                            None   \n",
       "3                            None                            None   \n",
       "4                            None                            None   \n",
       "\n",
       "  msq01_act_out_dreams  msq01a_act_out_years  ...  msq02_legs_jerk  \\\n",
       "0                 None                   NaN  ...             None   \n",
       "1                 None                   NaN  ...             None   \n",
       "2                 None                   NaN  ...             None   \n",
       "3                 None                   NaN  ...               No   \n",
       "4                 None                   NaN  ...             None   \n",
       "\n",
       "  msq03_restless_legs msq03a_leg_sensations_decrease  \\\n",
       "0                None                           None   \n",
       "1                None                           None   \n",
       "2                None                           None   \n",
       "3                 Yes                             No   \n",
       "4                None                           None   \n",
       "\n",
       "  msq03b_time_leg_sensations_worst msq04b_walked_asleep msq05_snorted_awake  \\\n",
       "0                             None                 None                None   \n",
       "1                             None                 None                None   \n",
       "2                             None                 None                None   \n",
       "3                             None                  Yes                  No   \n",
       "4                             None                 None                None   \n",
       "\n",
       "  msq06_stop_breathing msq06a_treated_for_stop_breathing msq07_leg_cramps  \\\n",
       "0                 None                              None             None   \n",
       "1                 None                              None             None   \n",
       "2                 None                              None             None   \n",
       "3                   No                              None              Yes   \n",
       "4                 None                              None             None   \n",
       "\n",
       "  msq08_rate_of_alertness  \n",
       "0                    <NA>  \n",
       "1                    <NA>  \n",
       "2                    <NA>  \n",
       "3                       8  \n",
       "4                    <NA>  \n",
       "\n",
       "[5 rows x 25 columns]"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rbd_Mayo.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "No     2489\n",
       "Yes    1285\n",
       "Name: msq01_act_out_dreams, dtype: int64"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rbd_Mayo.msq01_act_out_dreams.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Select just relevant columns - only question 1 is used to diagnose RBD\n",
    "rbd_Mayo = rbd_Mayo.loc[:, ['participant_id', 'visit_name', 'visit_month', 'msq01_act_out_dreams']]\n",
    "\n",
    "#Make binary variable for RBD\n",
    "rbd_Mayo['RBD_MAYO'] = [np.nan if pd.isna(i)\n",
    "                          else 1 if i == \"Yes\"\n",
    "                          else 0 for i in rbd_Mayo.msq01_act_out_dreams]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0    2489\n",
       "1.0    1285\n",
       "Name: RBD_MAYO, dtype: int64"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rbd_Mayo.RBD_MAYO.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>msq01_act_out_dreams</th>\n",
       "      <th>RBD_MAYO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1987</th>\n",
       "      <td>SU-33113</td>\n",
       "      <td>M24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>No</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2898</th>\n",
       "      <td>SU-33113</td>\n",
       "      <td>M24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>Yes</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     participant_id visit_name  visit_month msq01_act_out_dreams  RBD_MAYO\n",
       "1987       SU-33113        M24         24.0                   No       0.0\n",
       "2898       SU-33113        M24         24.0                  Yes       1.0"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Remove duplicated visit entries - rows with #\n",
    "rbd_Mayo_nodupli = rbd_Mayo.copy()\n",
    "rbd_Mayo_nodupli = rbd_Mayo_nodupli[~rbd_Mayo_nodupli.visit_name.str.contains('#')]\n",
    "\n",
    "#Check for any other duplicated entries by participant id and visit month\n",
    "rbd_Mayo_nodupli[rbd_Mayo_nodupli.duplicated(['participant_id', 'visit_month'], keep = False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#There are still a few individuals with duplicated entries by participant_id and visit_month with different MoCA scores\n",
    "#Just keep the first datapoint\n",
    "rbd_Mayo_nodupli_final = rbd_Mayo_nodupli.copy()\n",
    "rbd_Mayo_nodupli_final = rbd_Mayo_nodupli_final.drop_duplicates(subset = ['participant_id', 'visit_month'], keep = 'first')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>msq01_act_out_dreams</th>\n",
       "      <th>RBD_MAYO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PD-PDAX486GDU</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PD-PDBC294LL7</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PD-PDBK615WC6</td>\n",
       "      <td>M24</td>\n",
       "      <td>24.0</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PD-PDCZ531JKD</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PD-PDDW009MK9</td>\n",
       "      <td>M12</td>\n",
       "      <td>12.0</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  participant_id visit_name  visit_month msq01_act_out_dreams  RBD_MAYO\n",
       "0  PD-PDAX486GDU         M0          0.0                 None       NaN\n",
       "1  PD-PDBC294LL7         M0          0.0                 None       NaN\n",
       "2  PD-PDBK615WC6        M24         24.0                 None       NaN\n",
       "3  PD-PDCZ531JKD         M0          0.0                 None       NaN\n",
       "4  PD-PDDW009MK9        M12         12.0                 None       NaN"
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rbd_Mayo_nodupli_final.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Query RBDSQ (Stiasny-Kolster) data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "RBDSQ_query = f\"\"\"\n",
    "\n",
    "SELECT\n",
    "participant_id, visit_name, visit_month, rbd_summary_score\n",
    "FROM `{BQ_RELEASE_DATASET}.REM_Sleep_Behavior_Disorder_Questionnaire_Stiasny_Kolster`\n",
    "\n",
    "\"\"\"\n",
    "RBDSQ = bq_query(RBDSQ_query)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>rbd_summary_score</th>\n",
       "      <th>RBD_RBDSQ</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [participant_id, visit_name, visit_month, rbd_summary_score, RBD_RBDSQ]\n",
       "Index: []"
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Make new variable if summary score is >=6\n",
    "RBDSQ['RBD_RBDSQ'] = [np.nan if pd.isna(i)\n",
    "                          else 1 if i >=6\n",
    "                          else 0 for i in RBDSQ.rbd_summary_score]\n",
    "\n",
    "#Remove duplicated visit entries - rows with #\n",
    "RBDSQ_nodupli = RBDSQ.copy()\n",
    "RBDSQ_nodupli = RBDSQ_nodupli[~RBDSQ_nodupli.visit_name.str.contains('#')]\n",
    "\n",
    "#Check for any other duplicated entries by participant id and visit month\n",
    "RBDSQ_nodupli[RBDSQ_nodupli.duplicated(['participant_id', 'visit_month'], keep = False)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0    7683\n",
       "1.0    2527\n",
       "Name: RBD_RBDSQ, dtype: int64"
      ]
     },
     "execution_count": 139,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RBDSQ.RBD_RBDSQ.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>rbd_summary_score</th>\n",
       "      <th>RBD_RBDSQ</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HB-PD_INVAB465GYE</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HB-PD_INVAG058KWW</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HB-PD_INVAK106DV5</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HB-PD_INVAR135FG7</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HB-PD_INVBC055WLN</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      participant_id visit_name  visit_month  rbd_summary_score  RBD_RBDSQ\n",
       "0  HB-PD_INVAB465GYE         M0          0.0                  0        0.0\n",
       "1  HB-PD_INVAG058KWW         M0          0.0                  0        0.0\n",
       "2  HB-PD_INVAK106DV5         M0          0.0                  0        0.0\n",
       "3  HB-PD_INVAR135FG7         M0          0.0                  0        0.0\n",
       "4  HB-PD_INVBC055WLN         M0          0.0                  0        0.0"
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RBDSQ_nodupli.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>rbd_summary_score</th>\n",
       "      <th>RBD_RBDSQ</th>\n",
       "      <th>msq01_act_out_dreams</th>\n",
       "      <th>RBD_MAYO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [participant_id, visit_name, visit_month, rbd_summary_score, RBD_RBDSQ, msq01_act_out_dreams, RBD_MAYO]\n",
       "Index: []"
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Merge RBD datasets\n",
    "merged = RBDSQ_nodupli.merge(rbd_Mayo_nodupli_final, on = (['participant_id', 'visit_name', 'visit_month']), how = 'outer')\n",
    "\n",
    "#Check whether there are any individuals with multiple entires for visit_name and visit_month\n",
    "#Because we merged on participant_id + visit_name + visit_month, there may be some individuals with the same visit name but different visit months\n",
    "merged[merged.duplicated(['participant_id', 'visit_name'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>visit_name</th>\n",
       "      <th>visit_month</th>\n",
       "      <th>rbd_summary_score</th>\n",
       "      <th>RBD_RBDSQ</th>\n",
       "      <th>msq01_act_out_dreams</th>\n",
       "      <th>RBD_MAYO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PD-PDBY238FWE</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PD-PDED633MRU</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PD-PDEH654NL2</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PD-PDFW311WR6</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PD-PDNF237HNT</td>\n",
       "      <td>M0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  participant_id visit_name  visit_month  rbd_summary_score  RBD_RBDSQ  \\\n",
       "0  PD-PDBY238FWE         M0          0.0                  0        0.0   \n",
       "1  PD-PDED633MRU         M0          0.0                  0        0.0   \n",
       "2  PD-PDEH654NL2         M0          0.0                  0        0.0   \n",
       "3  PD-PDFW311WR6         M0          0.0                  0        0.0   \n",
       "4  PD-PDNF237HNT         M0          0.0                  0        0.0   \n",
       "\n",
       "  msq01_act_out_dreams  RBD_MAYO  \n",
       "0                   No       0.0  \n",
       "1                   No       0.0  \n",
       "2                   No       0.0  \n",
       "3                   No       0.0  \n",
       "4                   No       0.0  "
      ]
     },
     "execution_count": 142,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Check individuals who have data for both RBD questionnaires\n",
    "both = RBDSQ_nodupli.merge(rbd_Mayo_nodupli, on = (['participant_id', 'visit_name', 'visit_month']), how = 'inner')\n",
    "both.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Make a variable coding for RBD if EITHER the RBDSQ or Mayo criteria is met\n",
    "merged['RBD_sum'] = merged.fillna(0)['RBD_RBDSQ'] + merged.fillna(0)['RBD_MAYO']\n",
    "\n",
    "#Now make a combined variable which codes dementia if EITHER MOCA or MMSE are below cutoffs\n",
    "merged['RBD_final'] = [np.nan if pd.isna(i)\n",
    "                else 1 if i >= 1\n",
    "                else 0 for i in merged.RBD_sum]\n",
    "\n",
    "#However if both RBDSQ and MAYO criteria are NA, replace RBD_final as NA\n",
    "merged.loc[:, 'RBD_final'] = merged.RBD_final.mask((merged.RBD_RBDSQ.isnull()) & (merged.RBD_MAYO.isnull()), np.nan)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Make a variable for RBD EVER because we will analyse RBD as never vs. ever in a logistic regression\n",
    "RBD_export = merged.copy()\n",
    "RBD_export = RBD_export.loc[:, ['participant_id', 'RBD_final']].groupby('participant_id').agg(func = max)\n",
    "RBD_export = RBD_export.reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#Write as output\n",
    "with open('/home/jupyter/clinical_data_v2-5/RBD_allparticipants.tab', 'w') as f:\n",
    "    f.write(RBD_export.to_csv(index=False, sep='\\t'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>participant_id</th>\n",
       "      <th>RBD_final</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BF-1001</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BF-1002</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BF-1003</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BF-1004</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BF-1005</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  participant_id  RBD_final\n",
       "0        BF-1001        0.0\n",
       "1        BF-1002        0.0\n",
       "2        BF-1003        0.0\n",
       "3        BF-1004        0.0\n",
       "4        BF-1005        0.0"
      ]
     },
     "execution_count": 146,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RBD_export.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "hidden": true
   },
   "source": [
    "Analyse in R with logistic regression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "hidden": true
   },
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "#Make list of pathways\n",
    "pathways <- c(\"adaptive_immune\", \"alpha_synuclein\", \"innate_immune\", \"lysosome\", \"endocytosis\",\n",
    "              \"microglia\", \"monocytes\", \"mitochondria\")\n",
    "\n",
    "#Make list of cohorts - excluding LBD as these samples have been excluded from the genetic dataset\n",
    "cohorts <- c(\"PPMI\", \"PDBP\", \"HBS\", \"LCC\", \"Steady\", \"Sure\", \"BioFIND\")\n",
    "\n",
    "#Make function to analyse longitudinal outcomes using mixed effects models\n",
    "#df is the name of the .tab file in /home/jupyter/clinical_data_v2-5/ which has the data in long format\n",
    "#df = \"mdsupdrs3_allparticipants\"\n",
    "#outcome is the variable name of the outcome within df\n",
    "#outcome = \"mds_updrs_part_ii_summary_score\"\n",
    "\n",
    "analyse_binomial <- function(pathway, cohort, df, outcome) {\n",
    "    \n",
    "        #Read in data for PRS, age at diagnosis, cohort, and genetic PCs\n",
    "        PRS <- read.table(paste(\"/home/jupyter/pathways/\", cohort, \".\", pathway, \"_PRS_final.txt\", sep = \"\"), \n",
    "                          sep = \"\\t\", header = T)\n",
    "    \n",
    "        data <- read.table(paste(\"/home/jupyter/clinical_data_v2-5/\", df, \".tab\", sep = \"\"), \n",
    "                             sep = \"\\t\", header = T)\n",
    "    \n",
    "        aao <- read.table(\"/home/jupyter/clinical_data_v2-5/PD_cases_agedx.tab\", sep = \"\\t\", header = T)\n",
    "\n",
    "        demo <- read.table(\"/home/jupyter/clinical_data_v2-5/PD_cases_start_end_dx_sex.tab\", sep = \"\\t\", header = T)\n",
    "\n",
    "        #Read in genetic PCs\n",
    "        pcs <- read.table(paste(\"/home/jupyter/genetic_data_v2-5/\", cohort, \".pca.eigenvec\", sep = \"\"))\n",
    "        colnames(pcs) <- c(\"FID\", \"IID\", paste(\"PC\", 1:10, sep = \"\"))\n",
    "\n",
    "        #Merge - in long format, starting from the longitudinal outcome data\n",
    "        merged <- data %>%\n",
    "        inner_join(PRS, by = \"participant_id\") %>%\n",
    "        inner_join(aao, by = c(\"participant_id\" = \"FID\")) %>%\n",
    "        inner_join(demo, by = \"IID\") %>%\n",
    "        inner_join(pcs, by = c(\"FID\", \"IID\"))\n",
    "\n",
    "        #Select the outcome variable from the function argument\n",
    "        merged$y <- merged[[outcome]]\n",
    "    \n",
    "        #Select just relevant columns\n",
    "        #This cannot be done with dplyr::select\n",
    "        merged_selected <- merged[, c('y', 'participant_id', 'study', 'PRS_z', 'sex', \n",
    "                                      'age_at_baseline', 'age_at_diagnosis',\n",
    "                                      'PC1', 'PC2', 'PC3', 'PC4', 'PC5')]\n",
    "    \n",
    "        \n",
    "        #Remove individuals missing any rows\n",
    "        merged_selected_nomiss <- merged_selected %>%\n",
    "            drop_na()\n",
    "    \n",
    "        #Count how many individuals had RBD never vs. ever\n",
    "        print(merged_selected_nomiss %>%\n",
    "            group_by(y) %>%\n",
    "            summarise(count = n()))\n",
    "    \n",
    "        N_inds <- dim(merged_selected_nomiss)[1]\n",
    "\n",
    "        #For mixed effects models, we usually need 2 x number of individuals because we are estimating random effects for both the intercept and slope\n",
    "        #If the number of observations is more than 2 times the number of individuals (this is because we have N_inds * 2 random effects, one for the intercept and one for the slope)\n",
    "        if (N_inds > 0) {\n",
    "            \n",
    "            #Run GLM\n",
    "            glm_RBD_PRS <- glm(y ~ PRS_z + sex + age_at_diagnosis\n",
    "                       + PC1 + PC2 + PC3 + PC4 + PC5,\n",
    "                      data = merged_selected_nomiss,\n",
    "                      family = binomial())\n",
    "\n",
    "            \n",
    "            #Save model coefficients to results table\n",
    "            summary <-  c(pathway,\n",
    "                          cohort,\n",
    "                          summary(glm_RBD_PRS)$coefficients[2,1],\n",
    "                          summary(glm_RBD_PRS)$coefficients[2,2],\n",
    "                          summary(glm_RBD_PRS)$coefficients[2,4],\n",
    "                          nobs(glm_RBD_PRS))\n",
    "            \n",
    "            } else {\n",
    "            \n",
    "        \n",
    "            #If the dataframe is empty, print NA for summary stat\n",
    "            summary <- c(pathway,\n",
    "                        cohort,\n",
    "                        rep(NA, 3),\n",
    "                        N_inds)\n",
    "            \n",
    "            }\n",
    "        \n",
    "                     \n",
    "    return(summary)\n",
    "        \n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"PPMI\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   158\n",
      "2     1   271\n",
      "[1] \"PDBP\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   317\n",
      "2     1   422\n",
      "[1] \"HBS\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   256\n",
      "2     1   106\n",
      "[1] \"LCC\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0    36\n",
      "2     1    23\n",
      "[1] \"Steady\"\n",
      "# A tibble: 0 × 2\n",
      "# … with 2 variables: y <dbl>, count <int>\n",
      "[1] \"Sure\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   149\n",
      "2     1    93\n",
      "[1] \"BioFIND\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0    52\n",
      "2     1    40\n",
      "[1] \"PPMI\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   158\n",
      "2     1   271\n",
      "[1] \"PDBP\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   317\n",
      "2     1   422\n",
      "[1] \"HBS\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   256\n",
      "2     1   106\n",
      "[1] \"LCC\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0    36\n",
      "2     1    23\n",
      "[1] \"Steady\"\n",
      "# A tibble: 0 × 2\n",
      "# … with 2 variables: y <dbl>, count <int>\n",
      "[1] \"Sure\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   149\n",
      "2     1    93\n",
      "[1] \"BioFIND\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0    52\n",
      "2     1    40\n",
      "[1] \"PPMI\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   158\n",
      "2     1   271\n",
      "[1] \"PDBP\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   317\n",
      "2     1   422\n",
      "[1] \"HBS\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   256\n",
      "2     1   106\n",
      "[1] \"LCC\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0    36\n",
      "2     1    23\n",
      "[1] \"Steady\"\n",
      "# A tibble: 0 × 2\n",
      "# … with 2 variables: y <dbl>, count <int>\n",
      "[1] \"Sure\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   149\n",
      "2     1    93\n",
      "[1] \"BioFIND\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0    52\n",
      "2     1    40\n",
      "[1] \"PPMI\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   158\n",
      "2     1   271\n",
      "[1] \"PDBP\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   317\n",
      "2     1   422\n",
      "[1] \"HBS\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   256\n",
      "2     1   106\n",
      "[1] \"LCC\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0    36\n",
      "2     1    23\n",
      "[1] \"Steady\"\n",
      "# A tibble: 0 × 2\n",
      "# … with 2 variables: y <dbl>, count <int>\n",
      "[1] \"Sure\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   149\n",
      "2     1    93\n",
      "[1] \"BioFIND\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0    52\n",
      "2     1    40\n",
      "[1] \"PPMI\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   158\n",
      "2     1   271\n",
      "[1] \"PDBP\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   317\n",
      "2     1   422\n",
      "[1] \"HBS\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   256\n",
      "2     1   106\n",
      "[1] \"LCC\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0    36\n",
      "2     1    23\n",
      "[1] \"Steady\"\n",
      "# A tibble: 0 × 2\n",
      "# … with 2 variables: y <dbl>, count <int>\n",
      "[1] \"Sure\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   149\n",
      "2     1    93\n",
      "[1] \"BioFIND\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0    52\n",
      "2     1    40\n",
      "[1] \"PPMI\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   158\n",
      "2     1   271\n",
      "[1] \"PDBP\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   317\n",
      "2     1   422\n",
      "[1] \"HBS\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   256\n",
      "2     1   106\n",
      "[1] \"LCC\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0    36\n",
      "2     1    23\n",
      "[1] \"Steady\"\n",
      "# A tibble: 0 × 2\n",
      "# … with 2 variables: y <dbl>, count <int>\n",
      "[1] \"Sure\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   149\n",
      "2     1    93\n",
      "[1] \"BioFIND\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0    52\n",
      "2     1    40\n",
      "[1] \"PPMI\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   158\n",
      "2     1   271\n",
      "[1] \"PDBP\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   317\n",
      "2     1   422\n",
      "[1] \"HBS\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   256\n",
      "2     1   106\n",
      "[1] \"LCC\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0    36\n",
      "2     1    23\n",
      "[1] \"Steady\"\n",
      "# A tibble: 0 × 2\n",
      "# … with 2 variables: y <dbl>, count <int>\n",
      "[1] \"Sure\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   149\n",
      "2     1    93\n",
      "[1] \"BioFIND\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0    52\n",
      "2     1    40\n",
      "[1] \"PPMI\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   158\n",
      "2     1   271\n",
      "[1] \"PDBP\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   317\n",
      "2     1   422\n",
      "[1] \"HBS\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   256\n",
      "2     1   106\n",
      "[1] \"LCC\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0    36\n",
      "2     1    23\n",
      "[1] \"Steady\"\n",
      "# A tibble: 0 × 2\n",
      "# … with 2 variables: y <dbl>, count <int>\n",
      "[1] \"Sure\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0   149\n",
      "2     1    93\n",
      "[1] \"BioFIND\"\n",
      "# A tibble: 2 × 2\n",
      "      y count\n",
      "  <dbl> <int>\n",
      "1     0    52\n",
      "2     1    40\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "rbd_results_list <- list()\n",
    "\n",
    "#Loop over each pathway and cohort\n",
    "for (pathway in pathways) {\n",
    "    for (cohort in cohorts) {\n",
    "        print(cohort)\n",
    "        i <- length(rbd_results_list)\n",
    "        rbd_results_list[[i+1]] <- analyse_binomial(pathway, cohort, \"RBD_allparticipants\", \"RBD_final\")\n",
    "\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "           pathway  cohort                Coeff                 se\n",
      "1  adaptive_immune    PPMI  -0.0338571202839722  0.100979876312308\n",
      "2  adaptive_immune    PDBP    0.126202103234058 0.0758834375739297\n",
      "3  adaptive_immune     HBS    0.167901846139867  0.111436204786255\n",
      "4  adaptive_immune     LCC    0.333136332895856  0.300032347100603\n",
      "5  adaptive_immune  Steady                 <NA>               <NA>\n",
      "6  adaptive_immune    Sure -0.00624871776123015  0.145126304153982\n",
      "7  adaptive_immune BioFIND   -0.364072832263849  0.255099442421872\n",
      "8  alpha_synuclein    PPMI  -0.0857466846625837  0.103728735724328\n",
      "9  alpha_synuclein    PDBP  -0.0811883659301612 0.0759284161142755\n",
      "10 alpha_synuclein     HBS   0.0951744035004351  0.117969246595435\n",
      "11 alpha_synuclein     LCC    0.499183861767909   0.32806139337819\n",
      "12 alpha_synuclein  Steady                 <NA>               <NA>\n",
      "13 alpha_synuclein    Sure   -0.168850245384872  0.141660143752035\n",
      "14 alpha_synuclein BioFIND   -0.340629144818277  0.246395667267092\n",
      "15   innate_immune    PPMI    -0.29231591530225  0.107747417398996\n",
      "16   innate_immune    PDBP   0.0914050098405552 0.0761907051320133\n",
      "17   innate_immune     HBS     0.38674912206138  0.121691284461386\n",
      "18   innate_immune     LCC    0.172635472766492  0.320092590349607\n",
      "19   innate_immune  Steady                 <NA>               <NA>\n",
      "20   innate_immune    Sure    0.156204570150892  0.141080237803436\n",
      "21   innate_immune BioFIND   -0.139491234857059  0.236665050897972\n",
      "22        lysosome    PPMI   0.0258104286938687  0.101228909249764\n",
      "23        lysosome    PDBP   0.0872599603088942 0.0764350585902014\n",
      "24        lysosome     HBS   0.0855498608396937  0.118920318109004\n",
      "25        lysosome     LCC   -0.544926242673622  0.329767760275089\n",
      "26        lysosome  Steady                 <NA>               <NA>\n",
      "27        lysosome    Sure     0.11356979068211  0.141594073369266\n",
      "28        lysosome BioFIND   0.0821854494605959  0.232167779464766\n",
      "29     endocytosis    PPMI   0.0866165135472351  0.102858608916901\n",
      "30     endocytosis    PDBP  -0.0414856988679686 0.0756982527453729\n",
      "31     endocytosis     HBS   0.0760942393664008   0.11545573624755\n",
      "32     endocytosis     LCC  -0.0569285356417041  0.326370968426376\n",
      "33     endocytosis  Steady                 <NA>               <NA>\n",
      "34     endocytosis    Sure    0.208997176800747  0.146488461818522\n",
      "35     endocytosis BioFIND     0.27948664646779  0.240981026500399\n",
      "36       microglia    PPMI   0.0192786571936101  0.105078920876783\n",
      "37       microglia    PDBP    0.132559566233363 0.0764156926203159\n",
      "38       microglia     HBS   0.0803423536757292  0.116839698998443\n",
      "39       microglia     LCC    0.605863896590684  0.379983739326915\n",
      "40       microglia  Steady                 <NA>               <NA>\n",
      "41       microglia    Sure   -0.203912321498892  0.146888185375966\n",
      "42       microglia BioFIND    0.122176314067065   0.22591205602511\n",
      "43       monocytes    PPMI   -0.115129430497366    0.1045397015434\n",
      "44       monocytes    PDBP   0.0761622706538658 0.0757244101071633\n",
      "45       monocytes     HBS  0.00457255524669471  0.116411368605306\n",
      "46       monocytes     LCC    0.101470811481437  0.321949955936802\n",
      "47       monocytes  Steady                 <NA>               <NA>\n",
      "48       monocytes    Sure   -0.122188453987549  0.142446672617833\n",
      "49       monocytes BioFIND   0.0406469546074915  0.234962694534328\n",
      "50    mitochondria    PPMI -0.00235522739825261  0.101886150624034\n",
      "51    mitochondria    PDBP   0.0209073605791946 0.0758984453239703\n",
      "52    mitochondria     HBS    0.216125123358623  0.125760333959514\n",
      "53    mitochondria     LCC    0.124427248892543  0.318857722105084\n",
      "54    mitochondria  Steady                 <NA>               <NA>\n",
      "55    mitochondria    Sure    0.161059450473756   0.14723955863471\n",
      "56    mitochondria BioFIND   0.0514033479085391  0.269318771228832\n",
      "                Pvalue N_inds\n",
      "1    0.737409491914872    429\n",
      "2   0.0962914751636896    739\n",
      "3    0.131885462770063    362\n",
      "4     0.26685481583793     59\n",
      "5                 <NA>      0\n",
      "6    0.965656016963788    242\n",
      "7    0.153528006454232     92\n",
      "8     0.40843915669811    429\n",
      "9    0.284945720053237    739\n",
      "10   0.419797274912328    362\n",
      "11   0.128105023007663     59\n",
      "12                <NA>      0\n",
      "13    0.23328517760543    242\n",
      "14    0.16683425199021     92\n",
      "15  0.0066682388102398    429\n",
      "16   0.230260865750409    739\n",
      "17 0.00148234994733934    362\n",
      "18   0.589659310023424     59\n",
      "19                <NA>      0\n",
      "20    0.26820584107427    242\n",
      "21   0.555590556862169     92\n",
      "22   0.798745555194983    429\n",
      "23   0.253611089086399    739\n",
      "24   0.471901831345932    362\n",
      "25  0.0984419188896742     59\n",
      "26                <NA>      0\n",
      "27   0.422506616632679    242\n",
      "28   0.723345159886567     92\n",
      "29   0.399735904677513    429\n",
      "30   0.583664179579571    739\n",
      "31   0.509846218132627    362\n",
      "32   0.861528387575354     59\n",
      "33                <NA>      0\n",
      "34   0.153662274793208    242\n",
      "35   0.246135564107671     92\n",
      "36   0.854430543115836    429\n",
      "37  0.0827910244320423    739\n",
      "38   0.491686504304601    362\n",
      "39   0.110835974167682     59\n",
      "40                <NA>      0\n",
      "41    0.16507171194394    242\n",
      "42   0.588636062493818     92\n",
      "43   0.270766705554074    429\n",
      "44    0.31452030797702    739\n",
      "45   0.968667722578412    362\n",
      "46   0.752628233977385     59\n",
      "47                <NA>      0\n",
      "48     0.3910118340627    242\n",
      "49   0.862656751914467     92\n",
      "50   0.981557530458442    429\n",
      "51    0.78295905915402    739\n",
      "52  0.0856967753047158    362\n",
      "53    0.69636787115437     59\n",
      "54                <NA>      0\n",
      "55   0.274016423469059    242\n",
      "56   0.848631883769208     92\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "#Combine lists into a single dataframe\n",
    "rbd_results <- do.call(rbind, rbd_results_list)\n",
    "\n",
    "#Set column names\n",
    "colnames(rbd_results) <- c(\"pathway\", \"cohort\", \"Coeff\", \"se\", \"Pvalue\", \"N_inds\")\n",
    "\n",
    "#Make into dataframe\n",
    "rbd_results <- as.data.frame(rbd_results)\n",
    "\n",
    "#Save output\n",
    "write.table(rbd_results, \"/home/jupyter/pathways/pathways_rbd_results.txt\",\n",
    "           sep = \"\\t\", row.names = F, col.names = T, quote = F)\n",
    "\n",
    "rbd_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "adaptive_immune.bed\n",
      "adaptive_immune.bim\n",
      "adaptive_immune_chrbp.txt\n",
      "adaptive_immune_extract_hg38.txt\n",
      "adaptive_immune.fam\n",
      "adaptive_immune.log\n",
      "adaptive_immune_PD_PRS_Chang23andMe.all_score\n",
      "adaptive_immune_PD_PRS_Chang23andMe.log\n",
      "adaptive_immune_PD_PRS_Chang23andMe.mismatch\n",
      "adaptive_immune_PD_PRS_Chang23andMe.prsice\n",
      "adaptive_immune.updated_names.bed\n",
      "adaptive_immune.updated_names.bim\n",
      "adaptive_immune.updated_names.fam\n",
      "adaptive_immune.updated_names.log\n",
      "adaptive_immune.updated_names.no_dupli.bed\n",
      "adaptive_immune.updated_names.no_dupli.bim\n",
      "adaptive_immune.updated_names.no_dupli.fam\n",
      "adaptive_immune.updated_names.no_dupli.log\n",
      "alpha_synuclein.bed\n",
      "alpha_synuclein.bim\n",
      "alpha_synuclein_chrbp.txt\n",
      "alpha_synuclein_extract_hg38.txt\n",
      "alpha_synuclein.fam\n",
      "alpha_synuclein.log\n",
      "alpha_synuclein_PD_PRS_Chang23andMe.all_score\n",
      "alpha_synuclein_PD_PRS_Chang23andMe.log\n",
      "alpha_synuclein_PD_PRS_Chang23andMe.prsice\n",
      "alpha_synuclein.updated_names.bed\n",
      "alpha_synuclein.updated_names.bim\n",
      "alpha_synuclein.updated_names.fam\n",
      "alpha_synuclein.updated_names.log\n",
      "alpha_synuclein.updated_names.no_dupli.bed\n",
      "alpha_synuclein.updated_names.no_dupli.bim\n",
      "alpha_synuclein.updated_names.no_dupli.fam\n",
      "alpha_synuclein.updated_names.no_dupli.log\n",
      "BioFIND.adaptive_immune_PRS_final.txt\n",
      "BioFIND.alpha_synuclein_PRS_final.txt\n",
      "BioFIND.endocytosis_PRS_final.txt\n",
      "BioFIND.innate_immune_PRS_final.txt\n",
      "BioFIND.lysosome_PRS_final.txt\n",
      "BioFIND.microglia_PRS_final.txt\n",
      "BioFIND.mitochondria_PRS_final.txt\n",
      "BioFIND.monocytes_PRS_final.txt\n",
      "endocytosis.bed\n",
      "endocytosis.bim\n",
      "endocytosis_chrbp.txt\n",
      "endocytosis_extract_hg38.txt\n",
      "endocytosis.fam\n",
      "endocytosis.log\n",
      "endocytosis_PD_PRS_Chang23andMe.all_score\n",
      "endocytosis_PD_PRS_Chang23andMe.log\n",
      "endocytosis_PD_PRS_Chang23andMe.mismatch\n",
      "endocytosis_PD_PRS_Chang23andMe.prsice\n",
      "endocytosis.updated_names.bed\n",
      "endocytosis.updated_names.bim\n",
      "endocytosis.updated_names.fam\n",
      "endocytosis.updated_names.log\n",
      "endocytosis.updated_names.no_dupli.bed\n",
      "endocytosis.updated_names.no_dupli.bim\n",
      "endocytosis.updated_names.no_dupli.fam\n",
      "endocytosis.updated_names.no_dupli.log\n",
      "HBS.adaptive_immune_PRS_final.txt\n",
      "HBS.alpha_synuclein_PRS_final.txt\n",
      "HBS.endocytosis_PRS_final.txt\n",
      "HBS.innate_immune_PRS_final.txt\n",
      "HBS.lysosome_PRS_final.txt\n",
      "HBS.microglia_PRS_final.txt\n",
      "HBS.mitochondria_PRS_final.txt\n",
      "HBS.monocytes_PRS_final.txt\n",
      "innate_immune.bed\n",
      "innate_immune.bim\n",
      "innate_immune_chrbp.txt\n",
      "innate_immune_extract_hg38.txt\n",
      "innate_immune.fam\n",
      "innate_immune.log\n",
      "innate_immune_PD_PRS_Chang23andMe.all_score\n",
      "innate_immune_PD_PRS_Chang23andMe.log\n",
      "innate_immune_PD_PRS_Chang23andMe.mismatch\n",
      "innate_immune_PD_PRS_Chang23andMe.prsice\n",
      "innate_immune.updated_names.bed\n",
      "innate_immune.updated_names.bim\n",
      "innate_immune.updated_names.fam\n",
      "innate_immune.updated_names.log\n",
      "innate_immune.updated_names.no_dupli.bed\n",
      "innate_immune.updated_names.no_dupli.bim\n",
      "innate_immune.updated_names.no_dupli.fam\n",
      "innate_immune.updated_names.no_dupli.log\n",
      "LCC.adaptive_immune_PRS_final.txt\n",
      "LCC.alpha_synuclein_PRS_final.txt\n",
      "LCC.endocytosis_PRS_final.txt\n",
      "LCC.innate_immune_PRS_final.txt\n",
      "LCC.lysosome_PRS_final.txt\n",
      "LCC.microglia_PRS_final.txt\n",
      "LCC.mitochondria_PRS_final.txt\n",
      "LCC.monocytes_PRS_final.txt\n",
      "lysosome.bed\n",
      "lysosome.bim\n",
      "lysosome_chrbp.txt\n",
      "lysosome_extract_hg38.txt\n",
      "lysosome.fam\n",
      "lysosome.log\n",
      "lysosome_PD_PRS_Chang23andMe.all_score\n",
      "lysosome_PD_PRS_Chang23andMe.log\n",
      "lysosome_PD_PRS_Chang23andMe.mismatch\n",
      "lysosome_PD_PRS_Chang23andMe.prsice\n",
      "lysosome.updated_names.bed\n",
      "lysosome.updated_names.bim\n",
      "lysosome.updated_names.fam\n",
      "lysosome.updated_names.log\n",
      "lysosome.updated_names.no_dupli.bed\n",
      "lysosome.updated_names.no_dupli.bim\n",
      "lysosome.updated_names.no_dupli.fam\n",
      "lysosome.updated_names.no_dupli.log\n",
      "microglia.bed\n",
      "microglia.bim\n",
      "microglia_chrbp.txt\n",
      "microglia_extract_hg38.txt\n",
      "microglia.fam\n",
      "microglia.log\n",
      "microglia_PD_PRS_Chang23andMe.all_score\n",
      "microglia_PD_PRS_Chang23andMe.log\n",
      "microglia_PD_PRS_Chang23andMe.mismatch\n",
      "microglia_PD_PRS_Chang23andMe.prsice\n",
      "microglia.updated_names.bed\n",
      "microglia.updated_names.bim\n",
      "microglia.updated_names.fam\n",
      "microglia.updated_names.log\n",
      "microglia.updated_names.no_dupli.bed\n",
      "microglia.updated_names.no_dupli.bim\n",
      "microglia.updated_names.no_dupli.fam\n",
      "microglia.updated_names.no_dupli.log\n",
      "mitochondria.bed\n",
      "mitochondria.bim\n",
      "mitochondria_chrbp.txt\n",
      "mitochondria_extract_hg38.txt\n",
      "mitochondria.fam\n",
      "mitochondria.log\n",
      "mitochondria_PD_PRS_Chang23andMe.all_score\n",
      "mitochondria_PD_PRS_Chang23andMe.log\n",
      "mitochondria_PD_PRS_Chang23andMe.mismatch\n",
      "mitochondria_PD_PRS_Chang23andMe.prsice\n",
      "mitochondria.updated_names.bed\n",
      "mitochondria.updated_names.bim\n",
      "mitochondria.updated_names.fam\n",
      "mitochondria.updated_names.log\n",
      "mitochondria.updated_names.no_dupli.bed\n",
      "mitochondria.updated_names.no_dupli.bim\n",
      "mitochondria.updated_names.no_dupli.fam\n",
      "mitochondria.updated_names.no_dupli.log\n",
      "monocytes.bed\n",
      "monocytes.bim\n",
      "monocytes_chrbp.txt\n",
      "monocytes_extract_hg38.txt\n",
      "monocytes.fam\n",
      "monocytes.log\n",
      "monocytes_PD_PRS_Chang23andMe.all_score\n",
      "monocytes_PD_PRS_Chang23andMe.log\n",
      "monocytes_PD_PRS_Chang23andMe.mismatch\n",
      "monocytes_PD_PRS_Chang23andMe.prsice\n",
      "monocytes.updated_names.bed\n",
      "monocytes.updated_names.bim\n",
      "monocytes.updated_names.fam\n",
      "monocytes.updated_names.log\n",
      "monocytes.updated_names.no_dupli.bed\n",
      "monocytes.updated_names.no_dupli.bim\n",
      "monocytes.updated_names.no_dupli.fam\n",
      "monocytes.updated_names.no_dupli.log\n",
      "n_snps_in_pathwayPRS_summary.tab\n",
      "pathways_agediagnosis_results.txt\n",
      "pathways_dementia_results.txt\n",
      "pathways_HY3_results.txt\n",
      "pathways_mdsupdrs2_results.txt\n",
      "pathways_mdsupdrs3_results.txt\n",
      "pathways_moca_results.txt\n",
      "pathways_pdq8_results.txt\n",
      "pathways_rbd_results.txt\n",
      "PDBP.adaptive_immune_PRS_final.txt\n",
      "PDBP.alpha_synuclein_PRS_final.txt\n",
      "PDBP.endocytosis_PRS_final.txt\n",
      "PDBP.innate_immune_PRS_final.txt\n",
      "PDBP.lysosome_PRS_final.txt\n",
      "PDBP.microglia_PRS_final.txt\n",
      "PDBP.mitochondria_PRS_final.txt\n",
      "PDBP.monocytes_PRS_final.txt\n",
      "PPMI.adaptive_immune_PRS_final.txt\n",
      "PPMI.alpha_synuclein_PRS_final.txt\n",
      "PPMI.endocytosis_PRS_final.txt\n",
      "PPMI.innate_immune_PRS_final.txt\n",
      "PPMI.lysosome_PRS_final.txt\n",
      "PPMI.microglia_PRS_final.txt\n",
      "PPMI.mitochondria_PRS_final.txt\n",
      "PPMI.monocytes_PRS_final.txt\n",
      "Steady.adaptive_immune_PRS_final.txt\n",
      "Steady.alpha_synuclein_PRS_final.txt\n",
      "Steady.endocytosis_PRS_final.txt\n",
      "Steady.innate_immune_PRS_final.txt\n",
      "Steady.lysosome_PRS_final.txt\n",
      "Steady.microglia_PRS_final.txt\n",
      "Steady.mitochondria_PRS_final.txt\n",
      "Steady.monocytes_PRS_final.txt\n",
      "Sure.adaptive_immune_PRS_final.txt\n",
      "Sure.alpha_synuclein_PRS_final.txt\n",
      "Sure.endocytosis_PRS_final.txt\n",
      "Sure.innate_immune_PRS_final.txt\n",
      "Sure.lysosome_PRS_final.txt\n",
      "Sure.microglia_PRS_final.txt\n",
      "Sure.mitochondria_PRS_final.txt\n",
      "Sure.monocytes_PRS_final.txt\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /home/jupyter/pathways/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# Copy results files to workspace bucket"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u manuela-norway cp -r /home/jupyter/pathways/pathways_* gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181\n",
      "Copying file:///home/jupyter/pathways/pathways_agediagnosis_results.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/pathways/pathways_dementia_results.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/pathways/pathways_HY3_results.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/pathways/pathways_mdsupdrs2_results.txt [Content-Type=text/plain]...\n",
      "- [4 files][ 15.8 KiB/ 15.8 KiB]                                                \n",
      "==> NOTE: You are performing a sequence of gsutil operations that may\n",
      "run significantly faster if you instead use gsutil -m cp ... Please\n",
      "see the -m section under \"gsutil help options\" for further information\n",
      "about when gsutil -m can be advantageous.\n",
      "\n",
      "Copying file:///home/jupyter/pathways/pathways_mdsupdrs3_results.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/pathways/pathways_moca_results.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/pathways/pathways_pdq8_results.txt [Content-Type=text/plain]...\n",
      "Copying file:///home/jupyter/pathways/pathways_rbd_results.txt [Content-Type=text/plain]...\n",
      "\\ [8 files][ 32.2 KiB/ 32.2 KiB]                                                \n",
      "Operation completed over 8 objects/32.2 KiB.                                     \n",
      "Executing: gsutil -u manuela-norway cp -r /home/jupyter/pathways/n_snps_in_pathwayPRS_summary.tab gs://fc-2f037edd-3b8a-43cc-8283-b6704707b181\n",
      "Copying file:///home/jupyter/pathways/n_snps_in_pathwayPRS_summary.tab [Content-Type=application/octet-stream]...\n",
      "/ [1 files][  239.0 B/  239.0 B]                                                \n",
      "Operation completed over 1 objects/239.0 B.                                      \n"
     ]
    }
   ],
   "source": [
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} cp -r /home/jupyter/pathways/pathways_* {WORKSPACE_BUCKET}')\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} cp -r /home/jupyter/pathways/n_snps_in_pathwayPRS_summary.tab {WORKSPACE_BUCKET}')\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
